// @po-ui/ng-components/esm2020/lib/components/po-tabs/po-tab-dropdown/po-tab-dropdown.component.mjs
import { Component, ElementRef, EventEmitter, Input, Output, ViewChild } from '@angular/core';
import * as i0 from "@angular/core";
import * as i1 from "@angular/common";
import * as i2 from "../../po-popover/po-popover.component";
import * as i3 from "../po-tab-button/po-tab-button.component";
const _c0 = ["popover"];
const _c1 = ["tabDrop"];
function PoTabDropdownComponent_po_tab_button_8_Template(rf, ctx) { if (rf & 1) {
    const _r5 = i0.ɵɵgetCurrentView();
    i0.ɵɵelementStart(0, "po-tab-button", 8);
    i0.ɵɵlistener("p-activated", function PoTabDropdownComponent_po_tab_button_8_Template_po_tab_button_p_activated_0_listener() { const restoredCtx = i0.ɵɵrestoreView(_r5); const tab_r3 = restoredCtx.$implicit; const ctx_r4 = i0.ɵɵnextContext(); return i0.ɵɵresetView(ctx_r4.activated.emit(tab_r3)); })("p-change-state", function PoTabDropdownComponent_po_tab_button_8_Template_po_tab_button_p_change_state_0_listener() { const restoredCtx = i0.ɵɵrestoreView(_r5); const tab_r3 = restoredCtx.$implicit; const ctx_r6 = i0.ɵɵnextContext(); return i0.ɵɵresetView(ctx_r6.changeState.emit(tab_r3)); })("p-click", function PoTabDropdownComponent_po_tab_button_8_Template_po_tab_button_p_click_0_listener() { const restoredCtx = i0.ɵɵrestoreView(_r5); const tab_r3 = restoredCtx.$implicit; const ctx_r7 = i0.ɵɵnextContext(); const _r1 = i0.ɵɵreference(6); _r1.close(); return i0.ɵɵresetView(ctx_r7.click.emit(tab_r3)); });
    i0.ɵɵelementEnd();
} if (rf & 2) {
    const tab_r3 = ctx.$implicit;
    const ctx_r2 = i0.ɵɵnextContext();
    i0.ɵɵclassProp("po-tab-dropdown-button-active", tab_r3.active);
    i0.ɵɵproperty("p-active", tab_r3.active)("p-disabled", tab_r3.disabled)("p-hide", tab_r3.hide)("p-id", tab_r3.id)("p-label", tab_r3.label)("p-small", ctx_r2.small);
} }
/**
 * @docsPrivate
 *
 * @description
 *
 * Componente para agrupamento de componentes `po-tab-button` que os rendereiza em uma um caixa de diálogo
 * no formato de lista.
 */
export class PoTabDropdownComponent {
    constructor() {
        // Lista de abas
        this.tabs = [];
        // Evento que será emitido ao ativar uma aba
        this.activated = new EventEmitter();
        // Evento que será emitido a aba for desabilitada ou ocultada
        this.changeState = new EventEmitter();
        // Evento de click
        this.click = new EventEmitter();
    }
    getContainerClass() {
        return this.small ? 'po-tab-dropdown-container-sm' : 'po-tab-dropdown-container';
    }
    togglePopover() {
        this.popover.isHidden ? this.popover.open() : this.popover.close();
    }
}
PoTabDropdownComponent.ɵfac = function PoTabDropdownComponent_Factory(t) { return new (t || PoTabDropdownComponent)(); };
PoTabDropdownComponent.ɵcmp = /*@__PURE__*/ i0.ɵɵdefineComponent({ type: PoTabDropdownComponent, selectors: [["po-tab-dropdown"]], viewQuery: function PoTabDropdownComponent_Query(rf, ctx) { if (rf & 1) {
        i0.ɵɵviewQuery(_c0, 7);
        i0.ɵɵviewQuery(_c1, 7, ElementRef);
    } if (rf & 2) {
        let _t;
        i0.ɵɵqueryRefresh(_t = i0.ɵɵloadQuery()) && (ctx.popover = _t.first);
        i0.ɵɵqueryRefresh(_t = i0.ɵɵloadQuery()) && (ctx.tabDropdown = _t.first);
    } }, inputs: { label: ["p-label", "label"], small: ["p-small", "small"], tabs: ["p-tabs", "tabs"] }, outputs: { activated: "p-activated", changeState: "p-change-state", click: "p-click" }, decls: 9, vars: 8, consts: [[1, "po-tab-dropdown-content", "po-tab-button-md", 3, "keyup.enter"], ["tabDrop", ""], [1, "po-tab-button-label"], [1, "po-icon", 3, "ngClass"], ["p-hide-arrow", "", "p-position", "bottom", 3, "p-target"], ["popover", ""], [3, "ngClass"], ["class", "po-tab-dropdown-button", 3, "po-tab-dropdown-button-active", "p-active", "p-disabled", "p-hide", "p-id", "p-label", "p-small", "p-activated", "p-change-state", "p-click", 4, "ngFor", "ngForOf"], [1, "po-tab-dropdown-button", 3, "p-active", "p-disabled", "p-hide", "p-id", "p-label", "p-small", "p-activated", "p-change-state", "p-click"]], template: function PoTabDropdownComponent_Template(rf, ctx) { if (rf & 1) {
        i0.ɵɵelementStart(0, "div", 0, 1);
        i0.ɵɵlistener("keyup.enter", function PoTabDropdownComponent_Template_div_keyup_enter_0_listener() { return ctx.togglePopover(); });
        i0.ɵɵelementStart(2, "div", 2);
        i0.ɵɵtext(3);
        i0.ɵɵelement(4, "span", 3);
        i0.ɵɵelementEnd()();
        i0.ɵɵelementStart(5, "po-popover", 4, 5)(7, "div", 6);
        i0.ɵɵtemplate(8, PoTabDropdownComponent_po_tab_button_8_Template, 1, 8, "po-tab-button", 7);
        i0.ɵɵelementEnd()();
    } if (rf & 2) {
        const _r1 = i0.ɵɵreference(6);
        i0.ɵɵclassProp("po-tab-button-sm", ctx.small);
        i0.ɵɵattribute("tabindex", 0);
        i0.ɵɵadvance(3);
        i0.ɵɵtextInterpolate1(" ", ctx.label, " ");
        i0.ɵɵadvance(1);
        i0.ɵɵproperty("ngClass", _r1.isHidden ? "po-icon-arrow-down" : "po-icon-arrow-up");
        i0.ɵɵadvance(1);
        i0.ɵɵproperty("p-target", ctx.tabDropdown);
        i0.ɵɵadvance(2);
        i0.ɵɵproperty("ngClass", ctx.getContainerClass());
        i0.ɵɵadvance(1);
        i0.ɵɵproperty("ngForOf", ctx.tabs);
    } }, dependencies: [i1.NgClass, i1.NgForOf, i2.PoPopoverComponent, i3.PoTabButtonComponent], encapsulation: 2 });
(function () { (typeof ngDevMode === "undefined" || ngDevMode) && i0.ɵsetClassMetadata(PoTabDropdownComponent, [{
        type: Component,
        args: [{ selector: 'po-tab-dropdown', template: "<div\r\n  #tabDrop\r\n  class=\"po-tab-dropdown-content po-tab-button-md\"\r\n  [attr.tabindex]=\"0\"\r\n  [class.po-tab-button-sm]=\"small\"\r\n  (keyup.enter)=\"togglePopover()\"\r\n>\r\n  <div class=\"po-tab-button-label\">\r\n    {{ label }}\r\n    <span class=\"po-icon\" [ngClass]=\"popover.isHidden ? 'po-icon-arrow-down' : 'po-icon-arrow-up'\"> </span>\r\n  </div>\r\n</div>\r\n\r\n<po-popover #popover p-hide-arrow p-position=\"bottom\" [p-target]=\"tabDropdown\">\r\n  <div [ngClass]=\"getContainerClass()\">\r\n    <po-tab-button\r\n      *ngFor=\"let tab of tabs\"\r\n      class=\"po-tab-dropdown-button\"\r\n      [class.po-tab-dropdown-button-active]=\"tab.active\"\r\n      [p-active]=\"tab.active\"\r\n      [p-disabled]=\"tab.disabled\"\r\n      [p-hide]=\"tab.hide\"\r\n      [p-id]=\"tab.id\"\r\n      [p-label]=\"tab.label\"\r\n      [p-small]=\"small\"\r\n      (p-activated)=\"activated.emit(tab)\"\r\n      (p-change-state)=\"changeState.emit(tab)\"\r\n      (p-click)=\"popover.close(); click.emit(tab)\"\r\n    >\r\n    </po-tab-button>\r\n  </div>\r\n</po-popover>\r\n" }]
    }], null, { popover: [{
            type: ViewChild,
            args: ['popover', { static: true }]
        }], tabDropdown: [{
            type: ViewChild,
            args: ['tabDrop', { read: ElementRef, static: true }]
        }], label: [{
            type: Input,
            args: ['p-label']
        }], small: [{
            type: Input,
            args: ['p-small']
        }], tabs: [{
            type: Input,
            args: ['p-tabs']
        }], activated: [{
            type: Output,
            args: ['p-activated']
        }], changeState: [{
            type: Output,
            args: ['p-change-state']
        }], click: [{
            type: Output,
            args: ['p-click']
        }] }); })();
