// @po-ui/ng-components/esm2020/lib/components/po-dropdown/po-dropdown.module.mjs
import { CommonModule } from '@angular/common';
import { NgModule } from '@angular/core';
import { PoPopupModule } from '../po-popup/po-popup.module';
import { PoDropdownComponent } from './po-dropdown.component';
import * as i0 from "@angular/core";
/**
 * @description
 * Módulo do componente po-dropdown.
 */
export class PoDropdownModule {
}
PoDropdownModule.ɵfac = function PoDropdownModule_Factory(t) { return new (t || PoDropdownModule)(); };
PoDropdownModule.ɵmod = /*@__PURE__*/ i0.ɵɵdefineNgModule({ type: PoDropdownModule });
PoDropdownModule.ɵinj = /*@__PURE__*/ i0.ɵɵdefineInjector({ imports: [CommonModule, PoPopupModule] });
(function () { (typeof ngDevMode === "undefined" || ngDevMode) && i0.ɵsetClassMetadata(PoDropdownModule, [{
        type: NgModule,
        args: [{
                imports: [CommonModule, PoPopupModule],
                declarations: [PoDropdownComponent],
                exports: [PoDropdownComponent]
            }]
    }], null, null); })();
(function () { (typeof ngJitMode === "undefined" || ngJitMode) && i0.ɵɵsetNgModuleScope(PoDropdownModule, { declarations: [PoDropdownComponent], imports: [CommonModule, PoPopupModule], exports: [PoDropdownComponent] }); })();
