// @po-ui/ng-components/esm2020/lib/components/po-toolbar/po-toolbar.module.mjs
import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { PoAvatarModule } from './../po-avatar/po-avatar.module';
import { PoIconModule } from './../po-icon/po-icon.module';
import { PoPopupModule } from '../po-popup/po-popup.module';
import { PoToolbarActionsComponent } from './po-toolbar-actions/po-toolbar-actions.component';
import { PoToolbarComponent } from './po-toolbar.component';
import { PoToolbarNotificationComponent } from './po-toolbar-notification/po-toolbar-notification.component';
import { PoToolbarProfileComponent } from './po-toolbar-profile/po-toolbar-profile.component';
import * as i0 from "@angular/core";
/**
 * @description
 *
 * Módulo do componente po-toolbar
 *
 */
export class PoToolbarModule {
}
PoToolbarModule.ɵfac = function PoToolbarModule_Factory(t) { return new (t || PoToolbarModule)(); };
PoToolbarModule.ɵmod = /*@__PURE__*/ i0.ɵɵdefineNgModule({ type: PoToolbarModule });
PoToolbarModule.ɵinj = /*@__PURE__*/ i0.ɵɵdefineInjector({ imports: [CommonModule, PoAvatarModule, PoPopupModule, PoIconModule] });
(function () { (typeof ngDevMode === "undefined" || ngDevMode) && i0.ɵsetClassMetadata(PoToolbarModule, [{
        type: NgModule,
        args: [{
                imports: [CommonModule, PoAvatarModule, PoPopupModule, PoIconModule],
                declarations: [
                    PoToolbarActionsComponent,
                    PoToolbarComponent,
                    PoToolbarNotificationComponent,
                    PoToolbarProfileComponent
                ],
                exports: [PoToolbarComponent],
                providers: [],
                schemas: []
            }]
    }], null, null); })();
(function () { (typeof ngJitMode === "undefined" || ngJitMode) && i0.ɵɵsetNgModuleScope(PoToolbarModule, { declarations: [PoToolbarActionsComponent,
        PoToolbarComponent,
        PoToolbarNotificationComponent,
        PoToolbarProfileComponent], imports: [CommonModule, PoAvatarModule, PoPopupModule, PoIconModule], exports: [PoToolbarComponent] }); })();
