// @po-ui/ng-components/esm2020/lib/components/po-grid/po-grid-cell-action/po-grid-cell-action.component.mjs
import { Component, Input } from '@angular/core';
import * as i0 from "@angular/core";
export class PoGridCellActionComponent {
    constructor() {
        this.position = '';
    }
    onKeyDownContent(event) {
        // ENTER
        if (event.keyCode === 13) {
            event.preventDefault();
        }
    }
}
PoGridCellActionComponent.ɵfac = function PoGridCellActionComponent_Factory(t) { return new (t || PoGridCellActionComponent)(); };
PoGridCellActionComponent.ɵcmp = /*@__PURE__*/ i0.ɵɵdefineComponent({ type: PoGridCellActionComponent, selectors: [["po-grid-cell-action"]], inputs: { position: ["p-position", "position"], value: ["p-value", "value"] }, decls: 5, vars: 2, consts: [["tabindex", "0", 1, "po-grid-cell-action-content", 3, "id", "keydown"], ["contentElement", ""]], template: function PoGridCellActionComponent_Template(rf, ctx) { if (rf & 1) {
        i0.ɵɵelementStart(0, "div")(1, "div", 0, 1);
        i0.ɵɵlistener("keydown", function PoGridCellActionComponent_Template_div_keydown_1_listener($event) { return ctx.onKeyDownContent($event); });
        i0.ɵɵelementStart(3, "span");
        i0.ɵɵtext(4);
        i0.ɵɵelementEnd()()();
    } if (rf & 2) {
        i0.ɵɵadvance(1);
        i0.ɵɵproperty("id", ctx.position);
        i0.ɵɵadvance(3);
        i0.ɵɵtextInterpolate(ctx.value || "\u00A0");
    } }, encapsulation: 2 });
(function () { (typeof ngDevMode === "undefined" || ngDevMode) && i0.ɵsetClassMetadata(PoGridCellActionComponent, [{
        type: Component,
        args: [{ selector: 'po-grid-cell-action', template: "<div>\r\n  <div\r\n    #contentElement\r\n    [id]=\"position\"\r\n    class=\"po-grid-cell-action-content\"\r\n    tabindex=\"0\"\r\n    (keydown)=\"onKeyDownContent($event)\"\r\n  >\r\n    <span>{{ value || '&nbsp;' }}</span>\r\n  </div>\r\n</div>\r\n" }]
    }], function () { return []; }, { position: [{
            type: Input,
            args: ['p-position']
        }], value: [{
            type: Input,
            args: ['p-value']
        }] }); })();
