// @po-ui/ng-components/esm2020/lib/components/po-gauge/po-gauge.module.mjs
import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { PoTooltipModule } from '../../directives/po-tooltip/po-tooltip.module';
import { PoGaugeComponent } from './po-gauge.component';
import { PoGaugeDescriptionComponent } from './po-gauge-description/po-gauge-description.component';
import { PoGaugeLegendComponent } from './po-gauge-legend/po-gauge-legend.component';
import { PoGaugePathComponent } from './po-gauge-svg/po-gauge-path/po-gauge-path.component';
import { PoGaugeSvgComponent } from './po-gauge-svg/po-gauge-svg.component';
import { PoGaugeTitleComponent } from './po-gauge-title/po-gauge-title.component';
import { PoGaugePointerComponent } from './po-gauge-svg/po-gauge-pointer/po-gauge-pointer.component';
import * as i0 from "@angular/core";
/**
 * @description
 * Módulo do componente po-gauge.
 */
export class PoGaugeModule {
}
PoGaugeModule.ɵfac = function PoGaugeModule_Factory(t) { return new (t || PoGaugeModule)(); };
PoGaugeModule.ɵmod = /*@__PURE__*/ i0.ɵɵdefineNgModule({ type: PoGaugeModule });
PoGaugeModule.ɵinj = /*@__PURE__*/ i0.ɵɵdefineInjector({ imports: [CommonModule, PoTooltipModule] });
(function () { (typeof ngDevMode === "undefined" || ngDevMode) && i0.ɵsetClassMetadata(PoGaugeModule, [{
        type: NgModule,
        args: [{
                imports: [CommonModule, PoTooltipModule],
                declarations: [
                    PoGaugeComponent,
                    PoGaugeTitleComponent,
                    PoGaugeLegendComponent,
                    PoGaugeSvgComponent,
                    PoGaugePathComponent,
                    PoGaugeDescriptionComponent,
                    PoGaugePointerComponent
                ],
                exports: [PoGaugeComponent]
            }]
    }], null, null); })();
(function () { (typeof ngJitMode === "undefined" || ngJitMode) && i0.ɵɵsetNgModuleScope(PoGaugeModule, { declarations: [PoGaugeComponent,
        PoGaugeTitleComponent,
        PoGaugeLegendComponent,
        PoGaugeSvgComponent,
        PoGaugePathComponent,
        PoGaugeDescriptionComponent,
        PoGaugePointerComponent], imports: [CommonModule, PoTooltipModule], exports: [PoGaugeComponent] }); })();
