// @po-ui/ng-components/esm2020/lib/components/po-loading/po-loading.module.mjs
import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { PoLanguageModule } from './../../services/po-language/po-language.module';
import { PoLoadingComponent } from './po-loading.component';
import { PoLoadingIconComponent } from './po-loading-icon/po-loading-icon.component';
import { PoLoadingOverlayComponent } from './po-loading-overlay/po-loading-overlay.component';
import * as i0 from "@angular/core";
/**
 *
 * @description
 *
 * Módulo do componente po-loading-overlay.
 */
export class PoLoadingModule {
}
PoLoadingModule.ɵfac = function PoLoadingModule_Factory(t) { return new (t || PoLoadingModule)(); };
PoLoadingModule.ɵmod = /*@__PURE__*/ i0.ɵɵdefineNgModule({ type: PoLoadingModule });
PoLoadingModule.ɵinj = /*@__PURE__*/ i0.ɵɵdefineInjector({ imports: [CommonModule, PoLanguageModule] });
(function () { (typeof ngDevMode === "undefined" || ngDevMode) && i0.ɵsetClassMetadata(PoLoadingModule, [{
        type: NgModule,
        args: [{
                imports: [CommonModule, PoLanguageModule],
                declarations: [PoLoadingComponent, PoLoadingIconComponent, PoLoadingOverlayComponent],
                exports: [PoLoadingComponent, PoLoadingIconComponent, PoLoadingOverlayComponent]
            }]
    }], null, null); })();
(function () { (typeof ngJitMode === "undefined" || ngJitMode) && i0.ɵɵsetNgModuleScope(PoLoadingModule, { declarations: [PoLoadingComponent, PoLoadingIconComponent, PoLoadingOverlayComponent], imports: [CommonModule, PoLanguageModule], exports: [PoLoadingComponent, PoLoadingIconComponent, PoLoadingOverlayComponent] }); })();
