// @po-ui/ng-components/esm2020/lib/components/po-loading/po-loading.component.mjs
import { Component } from '@angular/core';
import { PoLoadingBaseComponent } from './po-loading-base.component';
import * as i0 from "@angular/core";
import * as i1 from "@angular/common";
import * as i2 from "./po-loading-icon/po-loading-icon.component";
function PoLoadingComponent_span_2_Template(rf, ctx) { if (rf & 1) {
    i0.ɵɵelementStart(0, "span", 2);
    i0.ɵɵtext(1);
    i0.ɵɵelementEnd();
} if (rf & 2) {
    const ctx_r0 = i0.ɵɵnextContext();
    i0.ɵɵadvance(1);
    i0.ɵɵtextInterpolate(ctx_r0.text);
} }
/**
 * @docsPrivate
 *
 * @docsExtends PoLoadingBaseComponent
 */
export class PoLoadingComponent extends PoLoadingBaseComponent {
}
PoLoadingComponent.ɵfac = /*@__PURE__*/ function () { let ɵPoLoadingComponent_BaseFactory; return function PoLoadingComponent_Factory(t) { return (ɵPoLoadingComponent_BaseFactory || (ɵPoLoadingComponent_BaseFactory = i0.ɵɵgetInheritedFactory(PoLoadingComponent)))(t || PoLoadingComponent); }; }();
PoLoadingComponent.ɵcmp = /*@__PURE__*/ i0.ɵɵdefineComponent({ type: PoLoadingComponent, selectors: [["po-loading"]], features: [i0.ɵɵInheritDefinitionFeature], decls: 3, vars: 1, consts: [[1, "po-loading"], ["class", "po-loading-label po-text-ellipsis", 4, "ngIf"], [1, "po-loading-label", "po-text-ellipsis"]], template: function PoLoadingComponent_Template(rf, ctx) { if (rf & 1) {
        i0.ɵɵelementStart(0, "div", 0);
        i0.ɵɵelement(1, "po-loading-icon");
        i0.ɵɵtemplate(2, PoLoadingComponent_span_2_Template, 2, 1, "span", 1);
        i0.ɵɵelementEnd();
    } if (rf & 2) {
        i0.ɵɵadvance(2);
        i0.ɵɵproperty("ngIf", ctx.text);
    } }, dependencies: [i1.NgIf, i2.PoLoadingIconComponent], encapsulation: 2 });
(function () { (typeof ngDevMode === "undefined" || ngDevMode) && i0.ɵsetClassMetadata(PoLoadingComponent, [{
        type: Component,
        args: [{ selector: 'po-loading', template: "<div class=\"po-loading\">\r\n  <po-loading-icon></po-loading-icon>\r\n  <span class=\"po-loading-label po-text-ellipsis\" *ngIf=\"text\">{{ text }}</span>\r\n</div>\r\n" }]
    }], null, null); })();
