// @po-ui/ng-components/esm2020/lib/components/po-navbar/po-navbar-actions/po-navbar-action/po-navbar-action.component.mjs
import { Component, Input } from '@angular/core';
import { isExternalLink, openExternalLink } from '../../../../utils/util';
import * as i0 from "@angular/core";
import * as i1 from "@angular/router";
import * as i2 from "../../../../directives/po-tooltip/po-tooltip.directive";
import * as i3 from "../../../po-icon/po-icon.component";
export class PoNavbarActionComponent {
    constructor(viewContainerRef, router) {
        this.router = router;
    }
    click() {
        if (this.action) {
            this.action();
            return;
        }
        if (this.link) {
            return this.openUrl(this.link);
        }
    }
    openUrl(url) {
        if (isExternalLink(url)) {
            return openExternalLink(url);
        }
        if (url) {
            return this.router.navigate([url]);
        }
    }
}
PoNavbarActionComponent.ɵfac = function PoNavbarActionComponent_Factory(t) { return new (t || PoNavbarActionComponent)(i0.ɵɵdirectiveInject(i0.ViewContainerRef), i0.ɵɵdirectiveInject(i1.Router)); };
PoNavbarActionComponent.ɵcmp = /*@__PURE__*/ i0.ɵɵdefineComponent({ type: PoNavbarActionComponent, selectors: [["po-navbar-action"]], inputs: { action: ["p-action", "action"], icon: ["p-icon", "icon"], label: ["p-label", "label"], link: ["p-link", "link"], tooltip: ["p-tooltip-text", "tooltip"] }, decls: 1, vars: 2, consts: [["tabindex", "0", 1, "po-navbar-action-content", "po-clickable", 3, "p-icon", "p-tooltip", "click"]], template: function PoNavbarActionComponent_Template(rf, ctx) { if (rf & 1) {
        i0.ɵɵelementStart(0, "po-icon", 0);
        i0.ɵɵlistener("click", function PoNavbarActionComponent_Template_po_icon_click_0_listener() { return ctx.click(); });
        i0.ɵɵelementEnd();
    } if (rf & 2) {
        i0.ɵɵproperty("p-icon", ctx.icon)("p-tooltip", ctx.tooltip);
    } }, dependencies: [i2.PoTooltipDirective, i3.PoIconComponent], encapsulation: 2 });
(function () { (typeof ngDevMode === "undefined" || ngDevMode) && i0.ɵsetClassMetadata(PoNavbarActionComponent, [{
        type: Component,
        args: [{ selector: 'po-navbar-action', template: "<po-icon\r\n  tabindex=\"0\"\r\n  class=\"po-navbar-action-content po-clickable\"\r\n  [p-icon]=\"icon\"\r\n  [p-tooltip]=\"tooltip\"\r\n  (click)=\"click()\"\r\n></po-icon>\r\n" }]
    }], function () { return [{ type: i0.ViewContainerRef }, { type: i1.Router }]; }, { action: [{
            type: Input,
            args: ['p-action']
        }], icon: [{
            type: Input,
            args: ['p-icon']
        }], label: [{
            type: Input,
            args: ['p-label']
        }], link: [{
            type: Input,
            args: ['p-link']
        }], tooltip: [{
            type: Input,
            args: ['p-tooltip-text']
        }] }); })();
