// @po-ui/ng-components/esm2020/lib/components/po-accordion/services/po-accordion.service.mjs
import { Injectable } from '@angular/core';
import { Subject } from 'rxjs';
import * as i0 from "@angular/core";
/**
 * @docsPrivate
 *
 * @description
 *
 * Serviço para poder notificar o componente `po-accordion` quando um `po-accordion-item` for
 * expandido/colapsado.
 */
export class PoAccordionService {
    constructor() {
        this.subjectChild = new Subject();
    }
    // Recebe o accordionItem
    receiveFromChildAccordionClicked() {
        return this.subjectChild.asObservable();
    }
    // Envia accordionItem colapsado/expadido do accordion
    sendToParentAccordionItemClicked(accordionItem) {
        this.subjectChild.next(accordionItem);
    }
}
PoAccordionService.ɵfac = function PoAccordionService_Factory(t) { return new (t || PoAccordionService)(); };
PoAccordionService.ɵprov = /*@__PURE__*/ i0.ɵɵdefineInjectable({ token: PoAccordionService, factory: PoAccordionService.ɵfac });
(function () { (typeof ngDevMode === "undefined" || ngDevMode) && i0.ɵsetClassMetadata(PoAccordionService, [{
        type: Injectable
    }], null, null); })();
