// @po-ui/ng-components/esm2020/lib/components/po-field/po-field-container/po-field-container.component.mjs
import { ChangeDetectionStrategy, Component, Input } from '@angular/core';
import { convertToBoolean } from '../../../utils/util';
import { poFieldContainerLiterals } from './po-field-container-literals';
import * as i0 from "@angular/core";
import * as i1 from "../../../services/po-language/po-language.service";
import * as i2 from "@angular/common";
import * as i3 from "../../po-label/po-label.component";
function PoFieldContainerComponent_po_label_2_Template(rf, ctx) { if (rf & 1) {
    i0.ɵɵelement(0, "po-label", 4);
} if (rf & 2) {
    const ctx_r0 = i0.ɵɵnextContext();
    i0.ɵɵproperty("p-disabled", ctx_r0.disabled)("p-for", ctx_r0.id)("p-label", ctx_r0.label)("p-requirement", ctx_r0.requirement);
} }
function PoFieldContainerComponent_div_3_Template(rf, ctx) { if (rf & 1) {
    i0.ɵɵelementStart(0, "div", 5);
    i0.ɵɵtext(1);
    i0.ɵɵelementEnd();
} if (rf & 2) {
    const ctx_r1 = i0.ɵɵnextContext();
    i0.ɵɵadvance(1);
    i0.ɵɵtextInterpolate(ctx_r1.help);
} }
const _c0 = ["*"];
/**
 * @docsPrivate
 *
 * Componente de uso interno, responsável por atribuir uma label para o campo
 */
export class PoFieldContainerComponent {
    constructor(languageService) {
        this._optional = false;
        this._required = false;
        /** Define se a indicação de campo obrigatório será exibida. */
        this.showRequired = false;
        const language = languageService.getShortLanguage();
        this.literals = {
            ...poFieldContainerLiterals[language]
        };
    }
    /** Indica se o campo será opcional. */
    set optional(value) {
        this._optional = convertToBoolean(value);
    }
    get optional() {
        return this._optional;
    }
    /** Indica se o campo será obrigatório. */
    set required(value) {
        this._required = convertToBoolean(value);
    }
    get required() {
        return this._required;
    }
    ngOnInit() {
        this.setRequirement();
    }
    ngOnChanges(changes) {
        if (changes.help || changes.label || changes.optional || changes.required || changes.showRequired) {
            this.setRequirement();
        }
    }
    setRequirement() {
        if (this.label || this.help) {
            if (!this.required && this.optional) {
                this.requirement = this.literals['optional'];
            }
            else if (this.required && this.showRequired) {
                this.requirement = this.literals['required'];
            }
            else {
                this.requirement = undefined;
            }
        }
    }
}
PoFieldContainerComponent.ɵfac = function PoFieldContainerComponent_Factory(t) { return new (t || PoFieldContainerComponent)(i0.ɵɵdirectiveInject(i1.PoLanguageService)); };
PoFieldContainerComponent.ɵcmp = /*@__PURE__*/ i0.ɵɵdefineComponent({ type: PoFieldContainerComponent, selectors: [["po-field-container"]], inputs: { disabled: ["p-disabled", "disabled"], id: ["p-id", "id"], label: ["p-label", "label"], help: ["p-help", "help"], optional: ["p-optional", "optional"], required: ["p-required", "required"], showRequired: ["p-show-required", "showRequired"] }, features: [i0.ɵɵNgOnChangesFeature], ngContentSelectors: _c0, decls: 5, vars: 2, consts: [[1, "po-field-container"], [1, "po-field-container-title"], ["p-field", "true", 3, "p-disabled", "p-for", "p-label", "p-requirement", 4, "ngIf"], ["class", "po-field-help", 4, "ngIf"], ["p-field", "true", 3, "p-disabled", "p-for", "p-label", "p-requirement"], [1, "po-field-help"]], template: function PoFieldContainerComponent_Template(rf, ctx) { if (rf & 1) {
        i0.ɵɵprojectionDef();
        i0.ɵɵelementStart(0, "div", 0)(1, "div", 1);
        i0.ɵɵtemplate(2, PoFieldContainerComponent_po_label_2_Template, 1, 4, "po-label", 2);
        i0.ɵɵelementEnd();
        i0.ɵɵtemplate(3, PoFieldContainerComponent_div_3_Template, 2, 1, "div", 3);
        i0.ɵɵprojection(4);
        i0.ɵɵelementEnd();
    } if (rf & 2) {
        i0.ɵɵadvance(2);
        i0.ɵɵproperty("ngIf", ctx.label || ctx.help);
        i0.ɵɵadvance(1);
        i0.ɵɵproperty("ngIf", ctx.help);
    } }, dependencies: [i2.NgIf, i3.PoLabelComponent], encapsulation: 2, changeDetection: 0 });
(function () { (typeof ngDevMode === "undefined" || ngDevMode) && i0.ɵsetClassMetadata(PoFieldContainerComponent, [{
        type: Component,
        args: [{ selector: 'po-field-container', changeDetection: ChangeDetectionStrategy.OnPush, template: "<div class=\"po-field-container\">\r\n  <div class=\"po-field-container-title\">\r\n    <po-label\r\n      *ngIf=\"label || help\"\r\n      [p-disabled]=\"disabled\"\r\n      p-field=\"true\"\r\n      [p-for]=\"id\"\r\n      [p-label]=\"label\"\r\n      [p-requirement]=\"requirement\"\r\n    ></po-label>\r\n  </div>\r\n\r\n  <div class=\"po-field-help\" *ngIf=\"help\">{{ help }}</div>\r\n\r\n  <ng-content></ng-content>\r\n</div>\r\n" }]
    }], function () { return [{ type: i1.PoLanguageService }]; }, { disabled: [{
            type: Input,
            args: ['p-disabled']
        }], id: [{
            type: Input,
            args: ['p-id']
        }], label: [{
            type: Input,
            args: ['p-label']
        }], help: [{
            type: Input,
            args: ['p-help']
        }], optional: [{
            type: Input,
            args: ['p-optional']
        }], required: [{
            type: Input,
            args: ['p-required']
        }], showRequired: [{
            type: Input,
            args: ['p-show-required']
        }] }); })();
