// @po-ui/ng-components/esm2020/lib/components/po-field/po-radio-group/po-radio-group.module.mjs
import { CommonModule } from '@angular/common';
import { FormsModule } from '@angular/forms';
import { NgModule } from '@angular/core';
import { PoFieldContainerModule } from '../po-field-container/po-field-container.module';
import { PoRadioGroupComponent } from './po-radio-group.component';
import { PoRadioModule } from '../po-radio/po-radio.module';
import * as i0 from "@angular/core";
/**
 * @description
 *
 * Módulo do componente `po-radio-group`.
 */
export class PoRadioGroupModule {
}
PoRadioGroupModule.ɵfac = function PoRadioGroupModule_Factory(t) { return new (t || PoRadioGroupModule)(); };
PoRadioGroupModule.ɵmod = /*@__PURE__*/ i0.ɵɵdefineNgModule({ type: PoRadioGroupModule });
PoRadioGroupModule.ɵinj = /*@__PURE__*/ i0.ɵɵdefineInjector({ imports: [CommonModule, FormsModule, PoFieldContainerModule, PoRadioModule] });
(function () { (typeof ngDevMode === "undefined" || ngDevMode) && i0.ɵsetClassMetadata(PoRadioGroupModule, [{
        type: NgModule,
        args: [{
                imports: [CommonModule, FormsModule, PoFieldContainerModule, PoRadioModule],
                exports: [PoRadioGroupComponent],
                declarations: [PoRadioGroupComponent]
            }]
    }], null, null); })();
(function () { (typeof ngJitMode === "undefined" || ngJitMode) && i0.ɵɵsetNgModuleScope(PoRadioGroupModule, { declarations: [PoRadioGroupComponent], imports: [CommonModule, FormsModule, PoFieldContainerModule, PoRadioModule], exports: [PoRadioGroupComponent] }); })();
