// @po-ui/ng-components/esm2020/lib/components/po-navbar/po-navbar-logo/po-navbar-logo.component.mjs
import { Component, Input } from '@angular/core';
import * as i0 from "@angular/core";
import * as i1 from "@angular/common";
import * as i2 from "../../po-logo/po-logo.component";
function PoNavbarLogoComponent_po_logo_0_Template(rf, ctx) { if (rf & 1) {
    i0.ɵɵelement(0, "po-logo", 1);
} if (rf & 2) {
    const ctx_r0 = i0.ɵɵnextContext();
    i0.ɵɵproperty("p-logo", ctx_r0.logo)("p-logo-alt", ctx_r0.logoAlt);
} }
export class PoNavbarLogoComponent {
}
PoNavbarLogoComponent.ɵfac = function PoNavbarLogoComponent_Factory(t) { return new (t || PoNavbarLogoComponent)(); };
PoNavbarLogoComponent.ɵcmp = /*@__PURE__*/ i0.ɵɵdefineComponent({ type: PoNavbarLogoComponent, selectors: [["po-navbar-logo"]], inputs: { logo: ["p-logo", "logo"], logoAlt: ["p-logo-alt", "logoAlt"] }, decls: 1, vars: 1, consts: [["p-class", "po-navbar-logo-image", 3, "p-logo", "p-logo-alt", 4, "ngIf"], ["p-class", "po-navbar-logo-image", 3, "p-logo", "p-logo-alt"]], template: function PoNavbarLogoComponent_Template(rf, ctx) { if (rf & 1) {
        i0.ɵɵtemplate(0, PoNavbarLogoComponent_po_logo_0_Template, 1, 2, "po-logo", 0);
    } if (rf & 2) {
        i0.ɵɵproperty("ngIf", ctx.logo);
    } }, dependencies: [i1.NgIf, i2.PoLogoComponent], encapsulation: 2 });
(function () { (typeof ngDevMode === "undefined" || ngDevMode) && i0.ɵsetClassMetadata(PoNavbarLogoComponent, [{
        type: Component,
        args: [{ selector: 'po-navbar-logo', template: "<po-logo *ngIf=\"logo\" p-class=\"po-navbar-logo-image\" [p-logo]=\"logo\" [p-logo-alt]=\"logoAlt\"></po-logo>\r\n" }]
    }], null, { logo: [{
            type: Input,
            args: ['p-logo']
        }], logoAlt: [{
            type: Input,
            args: ['p-logo-alt']
        }] }); })();
