// @po-ui/ng-components/esm2020/lib/components/po-stepper/po-stepper.module.mjs
import { CommonModule } from '@angular/common';
import { NgModule } from '@angular/core';
import { PoStepComponent } from './po-step/po-step.component';
import { PoStepperCircleComponent } from './po-stepper-circle/po-stepper-circle.component';
import { PoStepperComponent } from './po-stepper.component';
import { PoStepperLabelComponent } from './po-stepper-label/po-stepper-label.component';
import { PoStepperStepComponent } from './po-stepper-step/po-stepper-step.component';
import * as i0 from "@angular/core";
/**
 * @description
 * Módulo do componente po-stepper
 */
export class PoStepperModule {
}
PoStepperModule.ɵfac = function PoStepperModule_Factory(t) { return new (t || PoStepperModule)(); };
PoStepperModule.ɵmod = /*@__PURE__*/ i0.ɵɵdefineNgModule({ type: PoStepperModule });
PoStepperModule.ɵinj = /*@__PURE__*/ i0.ɵɵdefineInjector({ imports: [CommonModule] });
(function () { (typeof ngDevMode === "undefined" || ngDevMode) && i0.ɵsetClassMetadata(PoStepperModule, [{
        type: NgModule,
        args: [{
                imports: [CommonModule],
                declarations: [
                    PoStepComponent,
                    PoStepperCircleComponent,
                    PoStepperComponent,
                    PoStepperLabelComponent,
                    PoStepperStepComponent
                ],
                exports: [PoStepComponent, PoStepperComponent]
            }]
    }], null, null); })();
(function () { (typeof ngJitMode === "undefined" || ngJitMode) && i0.ɵɵsetNgModuleScope(PoStepperModule, { declarations: [PoStepComponent,
        PoStepperCircleComponent,
        PoStepperComponent,
        PoStepperLabelComponent,
        PoStepperStepComponent], imports: [CommonModule], exports: [PoStepComponent, PoStepperComponent] }); })();
