// @po-ui/ng-components/esm2020/lib/components/po-accordion/po-accordion-item-header/po-accordion-item-header.component.mjs
import { ChangeDetectionStrategy, Component, EventEmitter, Input, Output } from '@angular/core';
import * as i0 from "@angular/core";
export class PoAccordionItemHeaderComponent {
    constructor() {
        this.expanded = false;
        this.toggle = new EventEmitter();
    }
    onClick() {
        this.expanded = !this.expanded;
        this.toggle.emit(this.expanded);
    }
}
PoAccordionItemHeaderComponent.ɵfac = function PoAccordionItemHeaderComponent_Factory(t) { return new (t || PoAccordionItemHeaderComponent)(); };
PoAccordionItemHeaderComponent.ɵcmp = /*@__PURE__*/ i0.ɵɵdefineComponent({ type: PoAccordionItemHeaderComponent, selectors: [["po-accordion-item-header"]], inputs: { expanded: ["p-expanded", "expanded"], label: ["p-label", "label"] }, outputs: { toggle: "p-toggle" }, decls: 6, vars: 1, consts: [[1, "po-accordion-item-header"], [1, "po-accordion-item-header-button", "po-clickable", 3, "click"], [1, "po-text-ellipsis", "po-accordion-item-header-title"], [1, "po-icon", "po-accordion-item-header-icon", "po-icon-arrow-down"], ["icon", ""]], template: function PoAccordionItemHeaderComponent_Template(rf, ctx) { if (rf & 1) {
        i0.ɵɵelementStart(0, "header", 0)(1, "button", 1);
        i0.ɵɵlistener("click", function PoAccordionItemHeaderComponent_Template_button_click_1_listener() { return ctx.onClick(); });
        i0.ɵɵelementStart(2, "div", 2);
        i0.ɵɵtext(3);
        i0.ɵɵelementEnd();
        i0.ɵɵelement(4, "span", 3, 4);
        i0.ɵɵelementEnd()();
    } if (rf & 2) {
        i0.ɵɵadvance(3);
        i0.ɵɵtextInterpolate(ctx.label);
    } }, encapsulation: 2, changeDetection: 0 });
(function () { (typeof ngDevMode === "undefined" || ngDevMode) && i0.ɵsetClassMetadata(PoAccordionItemHeaderComponent, [{
        type: Component,
        args: [{ selector: 'po-accordion-item-header', changeDetection: ChangeDetectionStrategy.OnPush, template: "<header class=\"po-accordion-item-header\">\r\n  <button class=\"po-accordion-item-header-button po-clickable\" (click)=\"onClick()\">\r\n    <div class=\"po-text-ellipsis po-accordion-item-header-title\">{{ label }}</div>\r\n    <span #icon class=\"po-icon po-accordion-item-header-icon po-icon-arrow-down\"> </span>\r\n  </button>\r\n</header>\r\n" }]
    }], null, { expanded: [{
            type: Input,
            args: ['p-expanded']
        }], label: [{
            type: Input,
            args: ['p-label']
        }], toggle: [{
            type: Output,
            args: ['p-toggle']
        }] }); })();
