// @po-ui/ng-components/esm2020/lib/components/po-page/po-page.module.mjs
import { CommonModule } from '@angular/common';
import { NgModule } from '@angular/core';
import { FormsModule } from '@angular/forms';
import { RouterModule } from '@angular/router';
import { PoBreadcrumbModule } from '../po-breadcrumb/po-breadcrumb.module';
import { PoButtonModule } from '../po-button/po-button.module';
import { PoDisclaimerGroupModule } from '../po-disclaimer-group/po-disclaimer-group.module';
import { PoDropdownModule } from '../po-dropdown/po-dropdown.module';
import { PoFieldModule } from '../po-field/po-field.module';
import { PoLanguageModule } from './../../services/po-language/po-language.module';
import { PoModalModule } from './../po-modal/po-modal.module';
import { PoPageContentComponent } from './po-page-content/po-page-content.component';
import { PoPageDefaultComponent } from './po-page-default/po-page-default.component';
import { PoPageDetailComponent } from './po-page-detail/po-page-detail.component';
import { PoPageEditComponent } from './po-page-edit/po-page-edit.component';
import { PoPageHeaderComponent } from './po-page-header/po-page-header.component';
import { PoPageListComponent } from './po-page-list/po-page-list.component';
import { PoPageSlideComponent } from './po-page-slide/po-page-slide.component';
import { PoPageComponent } from './po-page.component';
import { PoPageSlideModule } from './po-page-slide/po-page-slide.module';
import * as i0 from "@angular/core";
/**
 * @description
 * Módulo dos componentes po-page-default, po-page-detail, po-page-edit,
 * po-page-list e po-page-slide.
 */
export class PoPageModule {
}
PoPageModule.ɵfac = function PoPageModule_Factory(t) { return new (t || PoPageModule)(); };
PoPageModule.ɵmod = /*@__PURE__*/ i0.ɵɵdefineNgModule({ type: PoPageModule });
PoPageModule.ɵinj = /*@__PURE__*/ i0.ɵɵdefineInjector({ imports: [CommonModule,
        FormsModule,
        RouterModule,
        PoBreadcrumbModule,
        PoButtonModule,
        PoDisclaimerGroupModule,
        PoDropdownModule,
        PoFieldModule,
        PoLanguageModule,
        PoModalModule,
        PoPageSlideModule] });
(function () { (typeof ngDevMode === "undefined" || ngDevMode) && i0.ɵsetClassMetadata(PoPageModule, [{
        type: NgModule,
        args: [{
                imports: [
                    CommonModule,
                    FormsModule,
                    RouterModule,
                    PoBreadcrumbModule,
                    PoButtonModule,
                    PoDisclaimerGroupModule,
                    PoDropdownModule,
                    PoFieldModule,
                    PoLanguageModule,
                    PoModalModule,
                    PoPageSlideModule
                ],
                declarations: [
                    PoPageComponent,
                    PoPageContentComponent,
                    PoPageDefaultComponent,
                    PoPageDetailComponent,
                    PoPageEditComponent,
                    PoPageHeaderComponent,
                    PoPageListComponent
                ],
                exports: [
                    PoPageDefaultComponent,
                    PoPageDetailComponent,
                    PoPageEditComponent,
                    PoPageListComponent,
                    PoPageSlideComponent
                ]
            }]
    }], null, null); })();
(function () { (typeof ngJitMode === "undefined" || ngJitMode) && i0.ɵɵsetNgModuleScope(PoPageModule, { declarations: [PoPageComponent,
        PoPageContentComponent,
        PoPageDefaultComponent,
        PoPageDetailComponent,
        PoPageEditComponent,
        PoPageHeaderComponent,
        PoPageListComponent], imports: [CommonModule,
        FormsModule,
        RouterModule,
        PoBreadcrumbModule,
        PoButtonModule,
        PoDisclaimerGroupModule,
        PoDropdownModule,
        PoFieldModule,
        PoLanguageModule,
        PoModalModule,
        PoPageSlideModule], exports: [PoPageDefaultComponent,
        PoPageDetailComponent,
        PoPageEditComponent,
        PoPageListComponent,
        PoPageSlideComponent] }); })();
