// @po-ui/ng-components/esm2020/lib/components/po-breadcrumb/po-breadcrumb-item/po-breadcrumb-item.component.mjs
import { Component, Input } from '@angular/core';
import * as i0 from "@angular/core";
import * as i1 from "@angular/common";
import * as i2 from "@angular/router";
function PoBreadcrumbItemComponent_li_0_a_1_Template(rf, ctx) { if (rf & 1) {
    i0.ɵɵelementStart(0, "a", 5)(1, "label", 6);
    i0.ɵɵtext(2);
    i0.ɵɵelementEnd();
    i0.ɵɵelement(3, "div", 7);
    i0.ɵɵelementEnd();
} if (rf & 2) {
    const ctx_r2 = i0.ɵɵnextContext(2);
    i0.ɵɵproperty("routerLink", ctx_r2.link);
    i0.ɵɵadvance(2);
    i0.ɵɵtextInterpolate(ctx_r2.label);
} }
function PoBreadcrumbItemComponent_li_0_a_2_Template(rf, ctx) { if (rf & 1) {
    const _r6 = i0.ɵɵgetCurrentView();
    i0.ɵɵelementStart(0, "a", 8);
    i0.ɵɵlistener("click", function PoBreadcrumbItemComponent_li_0_a_2_Template_a_click_0_listener() { i0.ɵɵrestoreView(_r6); const ctx_r5 = i0.ɵɵnextContext(2); return i0.ɵɵresetView(ctx_r5.action(ctx_r5.label)); });
    i0.ɵɵelementStart(1, "label", 6);
    i0.ɵɵtext(2);
    i0.ɵɵelementEnd();
    i0.ɵɵelement(3, "div", 7);
    i0.ɵɵelementEnd();
} if (rf & 2) {
    const ctx_r3 = i0.ɵɵnextContext(2);
    i0.ɵɵadvance(2);
    i0.ɵɵtextInterpolate(ctx_r3.label);
} }
function PoBreadcrumbItemComponent_li_0_a_3_Template(rf, ctx) { if (rf & 1) {
    i0.ɵɵelementStart(0, "a")(1, "label", 6);
    i0.ɵɵtext(2);
    i0.ɵɵelementEnd();
    i0.ɵɵelement(3, "div", 7);
    i0.ɵɵelementEnd();
} if (rf & 2) {
    const ctx_r4 = i0.ɵɵnextContext(2);
    i0.ɵɵadvance(2);
    i0.ɵɵtextInterpolate(ctx_r4.label);
} }
function PoBreadcrumbItemComponent_li_0_Template(rf, ctx) { if (rf & 1) {
    i0.ɵɵelementStart(0, "li");
    i0.ɵɵtemplate(1, PoBreadcrumbItemComponent_li_0_a_1_Template, 4, 2, "a", 2);
    i0.ɵɵtemplate(2, PoBreadcrumbItemComponent_li_0_a_2_Template, 4, 1, "a", 3);
    i0.ɵɵtemplate(3, PoBreadcrumbItemComponent_li_0_a_3_Template, 4, 1, "a", 4);
    i0.ɵɵelementEnd();
} if (rf & 2) {
    const ctx_r0 = i0.ɵɵnextContext();
    i0.ɵɵclassProp("po-breadcrumb-item", ctx_r0.link || ctx_r0.action)("po-breadcrumb-item-unclickable", !ctx_r0.link && !ctx_r0.action);
    i0.ɵɵadvance(1);
    i0.ɵɵproperty("ngIf", ctx_r0.link);
    i0.ɵɵadvance(1);
    i0.ɵɵproperty("ngIf", !ctx_r0.link && ctx_r0.action);
    i0.ɵɵadvance(1);
    i0.ɵɵproperty("ngIf", !ctx_r0.link && !ctx_r0.action);
} }
function PoBreadcrumbItemComponent_li_1_Template(rf, ctx) { if (rf & 1) {
    i0.ɵɵelementStart(0, "li", 9)(1, "label", 6);
    i0.ɵɵtext(2);
    i0.ɵɵelementEnd()();
} if (rf & 2) {
    const ctx_r1 = i0.ɵɵnextContext();
    i0.ɵɵadvance(2);
    i0.ɵɵtextInterpolate(ctx_r1.label);
} }
/**
 * @docsPrivate
 *
 * @description
 *
 * Componente que renderiza cada item do po-breadcrumb.
 */
export class PoBreadcrumbItemComponent {
    constructor() {
        // Especifica se item é o link ativo.
        this.itemActive = false;
    }
}
PoBreadcrumbItemComponent.ɵfac = function PoBreadcrumbItemComponent_Factory(t) { return new (t || PoBreadcrumbItemComponent)(); };
PoBreadcrumbItemComponent.ɵcmp = /*@__PURE__*/ i0.ɵɵdefineComponent({ type: PoBreadcrumbItemComponent, selectors: [["po-breadcrumb-item"]], inputs: { action: ["p-action", "action"], label: ["p-label", "label"], link: ["p-link", "link"], itemActive: ["p-item-active", "itemActive"] }, decls: 2, vars: 2, consts: [[3, "po-breadcrumb-item", "po-breadcrumb-item-unclickable", 4, "ngIf"], ["class", "po-breadcrumb-item-unclickable", 4, "ngIf"], [3, "routerLink", 4, "ngIf"], [3, "click", 4, "ngIf"], [4, "ngIf"], [3, "routerLink"], [1, "po-breadcrumb-label"], [1, "po-breadcrumb-arrow"], [3, "click"], [1, "po-breadcrumb-item-unclickable"]], template: function PoBreadcrumbItemComponent_Template(rf, ctx) { if (rf & 1) {
        i0.ɵɵtemplate(0, PoBreadcrumbItemComponent_li_0_Template, 4, 7, "li", 0);
        i0.ɵɵtemplate(1, PoBreadcrumbItemComponent_li_1_Template, 3, 1, "li", 1);
    } if (rf & 2) {
        i0.ɵɵproperty("ngIf", !ctx.itemActive);
        i0.ɵɵadvance(1);
        i0.ɵɵproperty("ngIf", ctx.itemActive);
    } }, dependencies: [i1.NgIf, i2.RouterLink], encapsulation: 2 });
(function () { (typeof ngDevMode === "undefined" || ngDevMode) && i0.ɵsetClassMetadata(PoBreadcrumbItemComponent, [{
        type: Component,
        args: [{ selector: 'po-breadcrumb-item', template: "<li\r\n  *ngIf=\"!itemActive\"\r\n  [class.po-breadcrumb-item]=\"link || action\"\r\n  [class.po-breadcrumb-item-unclickable]=\"!link && !action\"\r\n>\r\n  <a *ngIf=\"link\" [routerLink]=\"link\">\r\n    <label class=\"po-breadcrumb-label\">{{ label }}</label>\r\n    <div class=\"po-breadcrumb-arrow\"></div>\r\n  </a>\r\n\r\n  <a *ngIf=\"!link && action\" (click)=\"action(label)\">\r\n    <label class=\"po-breadcrumb-label\">{{ label }}</label>\r\n    <div class=\"po-breadcrumb-arrow\"></div>\r\n  </a>\r\n\r\n  <a *ngIf=\"!link && !action\">\r\n    <label class=\"po-breadcrumb-label\">{{ label }}</label>\r\n    <div class=\"po-breadcrumb-arrow\"></div>\r\n  </a>\r\n</li>\r\n\r\n<li *ngIf=\"itemActive\" class=\"po-breadcrumb-item-unclickable\">\r\n  <label class=\"po-breadcrumb-label\">{{ label }}</label>\r\n</li>\r\n" }]
    }], null, { action: [{
            type: Input,
            args: ['p-action']
        }], label: [{
            type: Input,
            args: ['p-label']
        }], link: [{
            type: Input,
            args: ['p-link']
        }], itemActive: [{
            type: Input,
            args: ['p-item-active']
        }] }); })();
