// @po-ui/ng-components/esm2020/lib/components/po-menu-panel/po-menu-panel.module.mjs
import { CommonModule } from '@angular/common';
import { NgModule } from '@angular/core';
import { RouterModule } from '@angular/router';
import { PoTooltipModule } from '../../directives/po-tooltip/index';
import { PoFieldModule } from '../po-field/po-field.module';
import { PoLogoModule } from '../po-logo/po-logo.module';
import { PoMenuPanelItemComponent } from './po-menu-panel-item/po-menu-panel-item.component';
import { PoMenuPanelComponent } from './po-menu-panel.component';
import { PoMenuPanelItemsService } from './services/po-menu-panel-items.service';
import * as i0 from "@angular/core";
/**
 * @description
 *
 * Módulo do componente po-menu-panel.
 */
export class PoMenuPanelModule {
}
PoMenuPanelModule.ɵfac = function PoMenuPanelModule_Factory(t) { return new (t || PoMenuPanelModule)(); };
PoMenuPanelModule.ɵmod = /*@__PURE__*/ i0.ɵɵdefineNgModule({ type: PoMenuPanelModule });
PoMenuPanelModule.ɵinj = /*@__PURE__*/ i0.ɵɵdefineInjector({ providers: [PoMenuPanelItemsService], imports: [CommonModule, RouterModule, PoFieldModule, PoLogoModule, PoTooltipModule] });
(function () { (typeof ngDevMode === "undefined" || ngDevMode) && i0.ɵsetClassMetadata(PoMenuPanelModule, [{
        type: NgModule,
        args: [{
                imports: [CommonModule, RouterModule, PoFieldModule, PoLogoModule, PoTooltipModule],
                declarations: [PoMenuPanelComponent, PoMenuPanelItemComponent],
                exports: [PoMenuPanelComponent],
                providers: [PoMenuPanelItemsService]
            }]
    }], null, null); })();
(function () { (typeof ngJitMode === "undefined" || ngJitMode) && i0.ɵɵsetNgModuleScope(PoMenuPanelModule, { declarations: [PoMenuPanelComponent, PoMenuPanelItemComponent], imports: [CommonModule, RouterModule, PoFieldModule, PoLogoModule, PoTooltipModule], exports: [PoMenuPanelComponent] }); })();
