// @po-ui/ng-components/esm2020/lib/components/po-table/po-table-show-subtitle/po-table-show-subtitle.component.mjs
import { Component, Input } from '@angular/core';
import * as i0 from "@angular/core";
import * as i1 from "@angular/common";
import * as i2 from "../../po-modal/po-modal.component";
import * as i3 from "../po-table-subtitle-circle/po-table-subtitle-circle.component";
function PoTableShowSubtitleComponent_div_6_Template(rf, ctx) { if (rf & 1) {
    i0.ɵɵelementStart(0, "div", 6);
    i0.ɵɵelement(1, "po-table-subtitle-circle", 7);
    i0.ɵɵelementStart(2, "span", 8);
    i0.ɵɵtext(3);
    i0.ɵɵelementEnd()();
} if (rf & 2) {
    const subtitle_r2 = ctx.$implicit;
    i0.ɵɵadvance(1);
    i0.ɵɵproperty("p-subtitle", subtitle_r2);
    i0.ɵɵadvance(2);
    i0.ɵɵtextInterpolate(subtitle_r2.label);
} }
/**
 * @docsPrivate
 *
 * @description
 *
 * Componente para a criação de um botão e modal para visualização de todas as legendas.
 */
export class PoTableShowSubtitleComponent {
}
PoTableShowSubtitleComponent.ɵfac = function PoTableShowSubtitleComponent_Factory(t) { return new (t || PoTableShowSubtitleComponent)(); };
PoTableShowSubtitleComponent.ɵcmp = /*@__PURE__*/ i0.ɵɵdefineComponent({ type: PoTableShowSubtitleComponent, selectors: [["po-table-show-subtitle"]], inputs: { subtitles: ["p-subtitles", "subtitles"], literals: ["p-literals", "literals"] }, decls: 7, vars: 3, consts: [[1, "po-table-footer-show-subtitle", 3, "click"], [1, "po-table-footer-show-subtitle-container"], [1, "po-table-show-subtitle"], ["p-size", "sm", 3, "p-title"], ["modal", ""], ["class", "po-table-subtitle-footer-modal po-row", 4, "ngFor", "ngForOf"], [1, "po-table-subtitle-footer-modal", "po-row"], ["p-hide-title", "true", 3, "p-subtitle"], [1, "po-table-subtitle-text"]], template: function PoTableShowSubtitleComponent_Template(rf, ctx) { if (rf & 1) {
        const _r3 = i0.ɵɵgetCurrentView();
        i0.ɵɵelementStart(0, "div", 0);
        i0.ɵɵlistener("click", function PoTableShowSubtitleComponent_Template_div_click_0_listener() { i0.ɵɵrestoreView(_r3); const _r0 = i0.ɵɵreference(5); return i0.ɵɵresetView(_r0.open()); });
        i0.ɵɵelementStart(1, "div", 1)(2, "span", 2);
        i0.ɵɵtext(3);
        i0.ɵɵelementEnd()()();
        i0.ɵɵelementStart(4, "po-modal", 3, 4);
        i0.ɵɵtemplate(6, PoTableShowSubtitleComponent_div_6_Template, 4, 2, "div", 5);
        i0.ɵɵelementEnd();
    } if (rf & 2) {
        i0.ɵɵadvance(3);
        i0.ɵɵtextInterpolate(ctx.literals.seeCompleteSubtitle);
        i0.ɵɵadvance(1);
        i0.ɵɵproperty("p-title", ctx.literals.completeSubtitle);
        i0.ɵɵadvance(2);
        i0.ɵɵproperty("ngForOf", ctx.subtitles);
    } }, dependencies: [i1.NgForOf, i2.PoModalComponent, i3.PoTableSubtitleCircleComponent], encapsulation: 2 });
(function () { (typeof ngDevMode === "undefined" || ngDevMode) && i0.ɵsetClassMetadata(PoTableShowSubtitleComponent, [{
        type: Component,
        args: [{ selector: 'po-table-show-subtitle', template: "<div class=\"po-table-footer-show-subtitle\" (click)=\"modal.open()\">\r\n  <div class=\"po-table-footer-show-subtitle-container\">\r\n    <span class=\"po-table-show-subtitle\">{{ literals.seeCompleteSubtitle }}</span>\r\n  </div>\r\n</div>\r\n\r\n<po-modal #modal p-size=\"sm\" [p-title]=\"literals.completeSubtitle\">\r\n  <div *ngFor=\"let subtitle of subtitles\" class=\"po-table-subtitle-footer-modal po-row\">\r\n    <po-table-subtitle-circle p-hide-title=\"true\" [p-subtitle]=\"subtitle\"> </po-table-subtitle-circle>\r\n    <span class=\"po-table-subtitle-text\">{{ subtitle.label }}</span>\r\n  </div>\r\n</po-modal>\r\n" }]
    }], null, { subtitles: [{
            type: Input,
            args: ['p-subtitles']
        }], literals: [{
            type: Input,
            args: ['p-literals']
        }] }); })();
