// @po-ui/ng-components/esm2020/lib/components/po-slide/po-slide-control/po-slide-control.component.mjs
import { Component, EventEmitter, Input, Output } from '@angular/core';
import * as i0 from "@angular/core";
/**
 * @docsPrivate
 *
 * @description
 *
 * Componente para a seta que controla a passagem de itens do `po-slide`.
 */
export class PoSlideControlComponent {
    constructor() {
        /** Evento emitido ao clicar em um controle. */
        this.click = new EventEmitter();
    }
}
PoSlideControlComponent.ɵfac = function PoSlideControlComponent_Factory(t) { return new (t || PoSlideControlComponent)(); };
PoSlideControlComponent.ɵcmp = /*@__PURE__*/ i0.ɵɵdefineComponent({ type: PoSlideControlComponent, selectors: [["po-slide-control"]], inputs: { control: ["p-control", "control"] }, outputs: { click: "p-click" }, decls: 3, vars: 6, consts: [[1, "po-slide-arrow-circle", 3, "click"]], template: function PoSlideControlComponent_Template(rf, ctx) { if (rf & 1) {
        i0.ɵɵelementStart(0, "div")(1, "div", 0);
        i0.ɵɵlistener("click", function PoSlideControlComponent_Template_div_click_1_listener() { return ctx.click.emit(); });
        i0.ɵɵelement(2, "div");
        i0.ɵɵelementEnd()();
    } if (rf & 2) {
        i0.ɵɵclassMapInterpolate1("po-slide-control-", ctx.control, "");
        i0.ɵɵadvance(2);
        i0.ɵɵclassMapInterpolate1("po-slide-arrow po-slide-arrow-", ctx.control, "");
    } }, encapsulation: 2 });
(function () { (typeof ngDevMode === "undefined" || ngDevMode) && i0.ɵsetClassMetadata(PoSlideControlComponent, [{
        type: Component,
        args: [{ selector: 'po-slide-control', template: "<div class=\"po-slide-control-{{ control }}\">\r\n  <div class=\"po-slide-arrow-circle\" (click)=\"click.emit()\">\r\n    <div class=\"po-slide-arrow po-slide-arrow-{{ control }}\"></div>\r\n  </div>\r\n</div>\r\n" }]
    }], null, { control: [{
            type: Input,
            args: ['p-control']
        }], click: [{
            type: Output,
            args: ['p-click']
        }] }); })();
