// @po-ui/ng-components/esm2020/lib/components/po-tag/po-tag.component.mjs
import { ChangeDetectionStrategy, Component, ViewChild } from '@angular/core';
import { PoTagBaseComponent } from './po-tag-base.component';
import { PoTagIcon } from './enums/po-tag-icon.enum';
import { PoTagType } from './enums/po-tag-type.enum';
import { PoTagLiterals } from './po-tag.literals';
import * as i0 from "@angular/core";
import * as i1 from "./../../services/po-language/po-language.service";
import * as i2 from "@angular/common";
import * as i3 from "../po-icon/po-icon.component";
import * as i4 from "../../directives/po-tooltip/po-tooltip.directive";
const _c0 = ["tagContainer"];
function PoTagComponent_div_2_Template(rf, ctx) { if (rf & 1) {
    i0.ɵɵelementStart(0, "div", 9)(1, "span", 10);
    i0.ɵɵtext(2);
    i0.ɵɵelementEnd()();
} if (rf & 2) {
    const ctx_r1 = i0.ɵɵnextContext();
    i0.ɵɵadvance(2);
    i0.ɵɵtextInterpolate(ctx_r1.tagOrientation ? ctx_r1.label + ":" : ctx_r1.label);
} }
const _c1 = function (a0) { return { "color": a0 }; };
function PoTagComponent_po_icon_5_Template(rf, ctx) { if (rf & 1) {
    i0.ɵɵelement(0, "po-icon", 11);
} if (rf & 2) {
    const ctx_r2 = i0.ɵɵnextContext();
    i0.ɵɵproperty("p-icon", !ctx_r2.type ? ctx_r2.icon : ctx_r2.iconFromType)("ngStyle", !ctx_r2.tagColor && ctx_r2.inverse && !ctx_r2.customTextColor ? i0.ɵɵpureFunction1(2, _c1, ctx_r2.customColor) : !ctx_r2.type && ctx_r2.customTextColor ? i0.ɵɵpureFunction1(4, _c1, ctx_r2.customTextColor) : "");
} }
function PoTagComponent_span_9_Template(rf, ctx) { if (rf & 1) {
    const _r5 = i0.ɵɵgetCurrentView();
    i0.ɵɵelementStart(0, "span", 12);
    i0.ɵɵlistener("click", function PoTagComponent_span_9_Template_span_click_0_listener() { i0.ɵɵrestoreView(_r5); const ctx_r4 = i0.ɵɵnextContext(); return i0.ɵɵresetView(ctx_r4.onClose()); })("keydown.enter", function PoTagComponent_span_9_Template_span_keydown_enter_0_listener() { i0.ɵɵrestoreView(_r5); const ctx_r6 = i0.ɵɵnextContext(); return i0.ɵɵresetView(ctx_r6.onClose()); });
    i0.ɵɵelementEnd();
} if (rf & 2) {
    const ctx_r3 = i0.ɵɵnextContext();
    i0.ɵɵclassProp("po-clickable", !ctx_r3.disabled);
    i0.ɵɵproperty("tabindex", !ctx_r3.disabled ? 0 : -1);
    i0.ɵɵattribute("aria-label", ctx_r3.setAriaLabel())("role", !ctx_r3.disabled ? "button" : "");
} }
const poTagTypeDefault = 'po-tag-' + PoTagType.Info;
/**
 * @docsExtends PoTagBaseComponent
 *
 * @example
 *
 * <example name="po-tag-basic" title="PO Tag Basic">
 *  <file name="sample-po-tag-basic/sample-po-tag-basic.component.html"> </file>
 *  <file name="sample-po-tag-basic/sample-po-tag-basic.component.ts"> </file>
 * </example>
 *
 * <example name="po-tag-labs" title="PO Tag Labs">
 *  <file name="sample-po-tag-labs/sample-po-tag-labs.component.html"> </file>
 *  <file name="sample-po-tag-labs/sample-po-tag-labs.component.ts"> </file>
 * </example>
 *
 * <example name="po-tag-bank-account" title="PO Tag - Bank Account">
 *  <file name="sample-po-tag-bank-account/sample-po-tag-bank-account.component.html"> </file>
 *  <file name="sample-po-tag-bank-account/sample-po-tag-bank-account.component.ts"> </file>
 * </example>
 */
export class PoTagComponent extends PoTagBaseComponent {
    constructor(el, languageService) {
        super();
        this.el = el;
        this.languageService = languageService;
        const language = this.languageService.getShortLanguage();
        this.literals = {
            ...PoTagLiterals[language]
        };
    }
    ngOnInit() {
        this.isClickable = this.click.observers.length > 0;
    }
    get iconFromType() {
        switch (this.type) {
            case PoTagType.Danger:
                return PoTagIcon.Danger;
            case PoTagType.Info:
                return PoTagIcon.Info;
            case PoTagType.Success:
                return PoTagIcon.Success;
            case PoTagType.Warning:
                return PoTagIcon.Warning;
        }
    }
    get tagColor() {
        if (this.type && !this.removable) {
            return this.inverse ? `po-tag-${this.type}-inverse` : `po-tag-${this.type}`;
        }
        if (this.color && !this.removable) {
            return this.inverse ? `po-text-${this.color}` : `po-${this.color}`;
        }
        if (!this.customColor && !this.removable) {
            return this.inverse ? `${poTagTypeDefault}-inverse` : poTagTypeDefault;
        }
    }
    get tagOrientation() {
        return this.orientation === this.poTagOrientation.Horizontal;
    }
    onClick() {
        if (!this.removable && !this.disabled) {
            const submittedTagItem = { value: this.value, type: this.type };
            this.click.emit(submittedTagItem);
        }
    }
    onClose() {
        if (!this.disabled) {
            this.click.emit(null);
            this.onRemove();
        }
    }
    onKeyPressed(event) {
        event.preventDefault();
        event.stopPropagation();
        this.onClick();
    }
    styleTag() {
        if (!this.tagColor && !this.inverse && !this.removable) {
            return { 'background-color': this.customColor, 'color': 'white' };
        }
        else if (!this.tagColor && this.inverse && !this.customTextColor) {
            return { 'border': '1px solid ' + this.customColor };
        }
        else if (!this.tagColor && this.inverse && this.customTextColor) {
            return { 'border': '1px solid ' + this.customTextColor, 'background-color': this.customColor };
        }
        else {
            return {};
        }
    }
    getWidthTag() {
        return this.tagContainer.nativeElement.offsetWidth > 155;
    }
    setAriaLabel() {
        return this.label ? this.label + ' ' + this.literals.remove : this.value + ' ' + this.literals.remove;
    }
    onRemove() {
        if (!this.disabled) {
            this.el.nativeElement.remove();
        }
    }
}
PoTagComponent.ɵfac = function PoTagComponent_Factory(t) { return new (t || PoTagComponent)(i0.ɵɵdirectiveInject(i0.ElementRef), i0.ɵɵdirectiveInject(i1.PoLanguageService)); };
PoTagComponent.ɵcmp = /*@__PURE__*/ i0.ɵɵdefineComponent({ type: PoTagComponent, selectors: [["po-tag"]], viewQuery: function PoTagComponent_Query(rf, ctx) { if (rf & 1) {
        i0.ɵɵviewQuery(_c0, 7);
    } if (rf & 2) {
        let _t;
        i0.ɵɵqueryRefresh(_t = i0.ɵɵloadQuery()) && (ctx.tagContainer = _t.first);
    } }, features: [i0.ɵɵInheritDefinitionFeature], decls: 10, vars: 24, consts: [["p-tooltip-position", "top", 1, "po-tag-container", 3, "p-tooltip"], ["tagContainer", ""], ["class", "po-tag-title po-text-nowrap", 4, "ngIf"], [1, "po-tag-sub-container"], [1, "po-tag", 3, "ngClass", "ngStyle", "tabindex", "click", "keydown.enter", "keydown.space", "keyup.space"], ["class", "po-tag-icon", 3, "p-icon", "ngStyle", 4, "ngIf"], [1, "po-tag-value"], [3, "ngStyle"], ["class", "po-tag-remove po-icon po-icon-close", 3, "po-clickable", "tabindex", "click", "keydown.enter", 4, "ngIf"], [1, "po-tag-title", "po-text-nowrap"], [1, "po-tag-label"], [1, "po-tag-icon", 3, "p-icon", "ngStyle"], [1, "po-tag-remove", "po-icon", "po-icon-close", 3, "tabindex", "click", "keydown.enter"]], template: function PoTagComponent_Template(rf, ctx) { if (rf & 1) {
        i0.ɵɵelementStart(0, "div", 0, 1);
        i0.ɵɵtemplate(2, PoTagComponent_div_2_Template, 3, 1, "div", 2);
        i0.ɵɵelementStart(3, "div", 3)(4, "div", 4);
        i0.ɵɵlistener("click", function PoTagComponent_Template_div_click_4_listener() { return ctx.onClick(); })("keydown.enter", function PoTagComponent_Template_div_keydown_enter_4_listener($event) { return ctx.onKeyPressed($event); })("keydown.space", function PoTagComponent_Template_div_keydown_space_4_listener($event) { return $event.preventDefault(); })("keyup.space", function PoTagComponent_Template_div_keyup_space_4_listener($event) { return ctx.onKeyPressed($event); });
        i0.ɵɵtemplate(5, PoTagComponent_po_icon_5_Template, 1, 6, "po-icon", 5);
        i0.ɵɵelementStart(6, "div", 6)(7, "span", 7);
        i0.ɵɵtext(8);
        i0.ɵɵelementEnd()();
        i0.ɵɵtemplate(9, PoTagComponent_span_9_Template, 1, 5, "span", 8);
        i0.ɵɵelementEnd()()();
    } if (rf & 2) {
        i0.ɵɵclassProp("po-tag-container-horizontal", ctx.tagOrientation);
        i0.ɵɵproperty("p-tooltip", ctx.getWidthTag() ? ctx.value : "");
        i0.ɵɵadvance(2);
        i0.ɵɵproperty("ngIf", ctx.label);
        i0.ɵɵadvance(2);
        i0.ɵɵclassProp("po-clickable", ctx.isClickable && !ctx.disabled && !ctx.removable)("po-tag-inverse", ctx.inverse && !ctx.type && !ctx.customTextColor && !ctx.removable)("po-tag-removable", ctx.removable)("po-tag-disabled", ctx.disabled && ctx.removable);
        i0.ɵɵproperty("ngClass", ctx.tagColor)("ngStyle", ctx.styleTag())("tabindex", ctx.isClickable && !ctx.removable ? 0 : -1);
        i0.ɵɵattribute("role", ctx.isClickable && !ctx.disabled ? "button" : "");
        i0.ɵɵadvance(1);
        i0.ɵɵproperty("ngIf", ctx.icon && !ctx.removable);
        i0.ɵɵadvance(2);
        i0.ɵɵproperty("ngStyle", !ctx.tagColor && ctx.inverse && !ctx.customTextColor ? i0.ɵɵpureFunction1(20, _c1, ctx.customColor) : !ctx.type && ctx.customTextColor && !ctx.removable ? i0.ɵɵpureFunction1(22, _c1, ctx.customTextColor) : "");
        i0.ɵɵadvance(1);
        i0.ɵɵtextInterpolate(ctx.value);
        i0.ɵɵadvance(1);
        i0.ɵɵproperty("ngIf", ctx.removable);
    } }, dependencies: [i2.NgClass, i2.NgIf, i2.NgStyle, i3.PoIconComponent, i4.PoTooltipDirective], encapsulation: 2, changeDetection: 0 });
(function () { (typeof ngDevMode === "undefined" || ngDevMode) && i0.ɵsetClassMetadata(PoTagComponent, [{
        type: Component,
        args: [{ selector: 'po-tag', changeDetection: ChangeDetectionStrategy.OnPush, template: "<div\r\n  #tagContainer\r\n  class=\"po-tag-container\"\r\n  [class.po-tag-container-horizontal]=\"tagOrientation\"\r\n  [p-tooltip]=\"getWidthTag() ? value : ''\"\r\n  p-tooltip-position=\"top\"\r\n>\r\n  <div *ngIf=\"label\" class=\"po-tag-title po-text-nowrap\">\r\n    <span class=\"po-tag-label\">{{ tagOrientation ? label + ':' : label }}</span>\r\n  </div>\r\n\r\n  <div class=\"po-tag-sub-container\">\r\n    <div\r\n      class=\"po-tag\"\r\n      [attr.role]=\"isClickable && !disabled ? 'button' : ''\"\r\n      [class.po-clickable]=\"isClickable && !disabled && !removable\"\r\n      [class.po-tag-inverse]=\"inverse && !type && !customTextColor && !removable\"\r\n      [class.po-tag-removable]=\"removable\"\r\n      [class.po-tag-disabled]=\"disabled && removable\"\r\n      [ngClass]=\"tagColor\"\r\n      [ngStyle]=\"styleTag()\"\r\n      [tabindex]=\"isClickable && !removable ? 0 : -1\"\r\n      (click)=\"onClick()\"\r\n      (keydown.enter)=\"onKeyPressed($event)\"\r\n      (keydown.space)=\"$event.preventDefault()\"\r\n      (keyup.space)=\"onKeyPressed($event)\"\r\n    >\r\n      <po-icon\r\n        *ngIf=\"icon && !removable\"\r\n        class=\"po-tag-icon\"\r\n        [p-icon]=\"!type ? icon : iconFromType\"\r\n        [ngStyle]=\"\r\n          !tagColor && inverse && !customTextColor\r\n            ? { 'color': customColor }\r\n            : !type && customTextColor\r\n            ? { 'color': customTextColor }\r\n            : ''\r\n        \"\r\n      >\r\n      </po-icon>\r\n\r\n      <div class=\"po-tag-value\">\r\n        <span\r\n          [ngStyle]=\"\r\n            !tagColor && inverse && !customTextColor\r\n              ? { 'color': customColor }\r\n              : !type && customTextColor && !removable\r\n              ? { 'color': customTextColor }\r\n              : ''\r\n          \"\r\n          >{{ value }}</span\r\n        >\r\n      </div>\r\n\r\n      <span\r\n        *ngIf=\"removable\"\r\n        [attr.aria-label]=\"setAriaLabel()\"\r\n        class=\"po-tag-remove po-icon po-icon-close\"\r\n        [class.po-clickable]=\"!disabled\"\r\n        [tabindex]=\"!disabled ? 0 : -1\"\r\n        [attr.role]=\"!disabled ? 'button' : ''\"\r\n        (click)=\"onClose()\"\r\n        (keydown.enter)=\"onClose()\"\r\n      >\r\n      </span>\r\n    </div>\r\n  </div>\r\n</div>\r\n" }]
    }], function () { return [{ type: i0.ElementRef }, { type: i1.PoLanguageService }]; }, { tagContainer: [{
            type: ViewChild,
            args: ['tagContainer', { static: true }]
        }] }); })();
