// @po-ui/ng-components/esm2020/lib/components/po-page/po-page-slide/po-page-slide.module.mjs
import { CommonModule } from '@angular/common';
import { NgModule } from '@angular/core';
import { FormsModule } from '@angular/forms';
import { PoPageSlideComponent } from './po-page-slide.component';
import { PoButtonModule } from './../../po-button/po-button.module';
import * as i0 from "@angular/core";
export class PoPageSlideModule {
}
PoPageSlideModule.ɵfac = function PoPageSlideModule_Factory(t) { return new (t || PoPageSlideModule)(); };
PoPageSlideModule.ɵmod = /*@__PURE__*/ i0.ɵɵdefineNgModule({ type: PoPageSlideModule });
PoPageSlideModule.ɵinj = /*@__PURE__*/ i0.ɵɵdefineInjector({ imports: [CommonModule, FormsModule, PoButtonModule] });
(function () { (typeof ngDevMode === "undefined" || ngDevMode) && i0.ɵsetClassMetadata(PoPageSlideModule, [{
        type: NgModule,
        args: [{
                declarations: [PoPageSlideComponent],
                exports: [PoPageSlideComponent],
                imports: [CommonModule, FormsModule, PoButtonModule]
            }]
    }], null, null); })();
(function () { (typeof ngJitMode === "undefined" || ngJitMode) && i0.ɵɵsetNgModuleScope(PoPageSlideModule, { declarations: [PoPageSlideComponent], imports: [CommonModule, FormsModule, PoButtonModule], exports: [PoPageSlideComponent] }); })();
