// @po-ui/ng-components/esm2020/lib/components/po-table/po-table.component.mjs
import { CdkVirtualScrollViewport } from '@angular/cdk/scrolling';
import { moveItemInArray } from '@angular/cdk/drag-drop';
import { Component, ContentChild, ContentChildren, ElementRef, ViewChild, ViewChildren, inject } from '@angular/core';
import { Subscription } from 'rxjs';
import { PoDateService } from '../../services/po-date/po-date.service';
import { PoNotificationService } from '../../services/po-notification/po-notification.service';
import { convertToBoolean } from '../../utils/util';
import { PoModalComponent } from '../po-modal';
import { uuid } from '../../utils/util';
import { PoTableRowTemplateArrowDirection } from './enums/po-table-row-template-arrow-direction.enum';
import { PoTableBaseComponent } from './po-table-base.component';
import { PoTableCellTemplateDirective } from './po-table-cell-template/po-table-cell-template.directive';
import { PoTableColumnTemplateDirective } from './po-table-column-template/po-table-column-template.directive';
import { PoTableRowTemplateDirective } from './po-table-row-template/po-table-row-template.directive';
import * as i0 from "@angular/core";
import * as i1 from "../../services/po-date/po-date.service";
import * as i2 from "../../services/po-language/po-language.service";
import * as i3 from "@angular/common";
import * as i4 from "@angular/router";
import * as i5 from "./services/po-table.service";
const _c0 = ["noColumnsHeader"];
const _c1 = ["popup"];
const _c2 = ["tableFooter"];
const _c3 = ["tableWrapper"];
const _c4 = ["tableTemplate"];
const _c5 = ["tableVirtualScroll"];
const _c6 = ["columnManager"];
const _c7 = ["columnBatchActions"];
const _c8 = ["columnActionLeft"];
const _c9 = ["filterInput"];
const _c10 = ["poSearchInput"];
const _c11 = ["columnManagerTarget"];
const _c12 = ["columnManagerTargetFixed"];
const _c13 = ["actionsIconElement"];
const _c14 = ["actionsElement"];
function PoTableComponent_div_1_ng_template_4_Template(rf, ctx) { if (rf & 1) {
    i0.ɵɵtext(0);
} if (rf & 2) {
    const ctx_r29 = i0.ɵɵnextContext(2);
    i0.ɵɵtextInterpolate(ctx_r29.literals.noItem);
} }
function PoTableComponent_div_1_ng_template_5_Template(rf, ctx) { if (rf & 1) {
    i0.ɵɵtext(0);
} if (rf & 2) {
    const ctx_r30 = i0.ɵɵnextContext(2);
    i0.ɵɵtextInterpolate(ctx_r30.literals.oneItem);
} }
function PoTableComponent_div_1_ng_template_6_Template(rf, ctx) { if (rf & 1) {
    i0.ɵɵtext(0);
} if (rf & 2) {
    const ctx_r31 = i0.ɵɵnextContext(2);
    i0.ɵɵtextInterpolate2("", ctx_r31.itemsSelected.length, " ", ctx_r31.literals.multipleItems, "");
} }
function PoTableComponent_div_1_Template(rf, ctx) { if (rf & 1) {
    const _r33 = i0.ɵɵgetCurrentView();
    i0.ɵɵelementStart(0, "div", 22, 23)(2, "div", 24)(3, "strong");
    i0.ɵɵtemplate(4, PoTableComponent_div_1_ng_template_4_Template, 1, 1, "ng-template", 25);
    i0.ɵɵtemplate(5, PoTableComponent_div_1_ng_template_5_Template, 1, 1, "ng-template", 26);
    i0.ɵɵtemplate(6, PoTableComponent_div_1_ng_template_6_Template, 1, 2, "ng-template", 27);
    i0.ɵɵelementEnd()();
    i0.ɵɵelementStart(7, "div", 28)(8, "po-button", 29);
    i0.ɵɵlistener("p-click", function PoTableComponent_div_1_Template_po_button_p_click_8_listener() { i0.ɵɵrestoreView(_r33); i0.ɵɵnextContext(); const _r27 = i0.ɵɵreference(30); return i0.ɵɵresetView(_r27.open()); });
    i0.ɵɵelementEnd()()();
} if (rf & 2) {
    const ctx_r0 = i0.ɵɵnextContext();
    i0.ɵɵadvance(2);
    i0.ɵɵproperty("ngPlural", ctx_r0.itemsSelected.length);
    i0.ɵɵadvance(6);
    i0.ɵɵproperty("p-danger", true)("p-disabled", ctx_r0.itemsSelected.length > 1 && ctx_r0.serviceDeleteApi !== undefined)("p-label", ctx_r0.literals.delete);
} }
function PoTableComponent_div_2_Template(rf, ctx) { if (rf & 1) {
    const _r37 = i0.ɵɵgetCurrentView();
    i0.ɵɵelementStart(0, "div", 30, 31)(2, "po-button", 32, 33);
    i0.ɵɵlistener("p-click", function PoTableComponent_div_2_Template_po_button_p_click_2_listener() { i0.ɵɵrestoreView(_r37); const ctx_r36 = i0.ɵɵnextContext(); return i0.ɵɵresetView(ctx_r36.onOpenColumnManager()); });
    i0.ɵɵelementEnd()();
} if (rf & 2) {
    const ctx_r1 = i0.ɵɵnextContext();
    i0.ɵɵadvance(2);
    i0.ɵɵproperty("p-aria-label", ctx_r1.literals.columnsManager)("p-tooltip", ctx_r1.literals.columnsManager);
} }
function PoTableComponent_div_3_Template(rf, ctx) { if (rf & 1) {
    const _r39 = i0.ɵɵgetCurrentView();
    i0.ɵɵelementStart(0, "div", 34)(1, "po-search", 35);
    i0.ɵɵlistener("p-filtered-items-change", function PoTableComponent_div_3_Template_po_search_p_filtered_items_change_1_listener($event) { i0.ɵɵrestoreView(_r39); const ctx_r38 = i0.ɵɵnextContext(); return i0.ɵɵresetView(ctx_r38.onFilteredItemsChange($event)); });
    i0.ɵɵelementEnd()();
} if (rf & 2) {
    const ctx_r2 = i0.ɵɵnextContext();
    i0.ɵɵadvance(1);
    i0.ɵɵproperty("p-items", ctx_r2.items)("p-filter-keys", ctx_r2.filteredColumns)("p-filter-type", ctx_r2.filterType);
} }
function PoTableComponent_po_container_4_ng_container_1_Template(rf, ctx) { if (rf & 1) {
    i0.ɵɵelementContainer(0);
} }
function PoTableComponent_po_container_4_Template(rf, ctx) { if (rf & 1) {
    i0.ɵɵelementStart(0, "po-container", 36);
    i0.ɵɵtemplate(1, PoTableComponent_po_container_4_ng_container_1_Template, 1, 0, "ng-container", 37);
    i0.ɵɵelementEnd();
} if (rf & 2) {
    i0.ɵɵnextContext();
    const _r4 = i0.ɵɵreference(6);
    i0.ɵɵadvance(1);
    i0.ɵɵproperty("ngTemplateOutlet", _r4);
} }
function PoTableComponent_ng_template_5_div_1_Template(rf, ctx) { if (rf & 1) {
    i0.ɵɵelementStart(0, "div", 45);
    i0.ɵɵelement(1, "po-loading", 46);
    i0.ɵɵelementEnd();
} if (rf & 2) {
    const ctx_r41 = i0.ɵɵnextContext(2);
    i0.ɵɵadvance(1);
    i0.ɵɵproperty("p-text", ctx_r41.literals.loadingData);
} }
function PoTableComponent_ng_template_5_div_5_ng_container_3_Template(rf, ctx) { if (rf & 1) {
    i0.ɵɵelementContainer(0);
} }
function PoTableComponent_ng_template_5_div_5_Template(rf, ctx) { if (rf & 1) {
    i0.ɵɵelementStart(0, "div", 47)(1, "div", 48, 49);
    i0.ɵɵtemplate(3, PoTableComponent_ng_template_5_div_5_ng_container_3_Template, 1, 0, "ng-container", 37);
    i0.ɵɵelementEnd()();
} if (rf & 2) {
    const ctx_r43 = i0.ɵɵnextContext(2);
    const _r9 = i0.ɵɵreference(11);
    i0.ɵɵstyleProp("height", ctx_r43.heightTableContainer, "px");
    i0.ɵɵadvance(3);
    i0.ɵɵproperty("ngTemplateOutlet", _r9);
} }
function PoTableComponent_ng_template_5_div_6_ng_container_1_Template(rf, ctx) { if (rf & 1) {
    i0.ɵɵelementContainer(0);
} }
function PoTableComponent_ng_template_5_div_6_Template(rf, ctx) { if (rf & 1) {
    i0.ɵɵelementStart(0, "div");
    i0.ɵɵtemplate(1, PoTableComponent_ng_template_5_div_6_ng_container_1_Template, 1, 0, "ng-container", 37);
    i0.ɵɵelementEnd();
} if (rf & 2) {
    i0.ɵɵnextContext(2);
    const _r7 = i0.ɵɵreference(9);
    i0.ɵɵadvance(1);
    i0.ɵɵproperty("ngTemplateOutlet", _r7);
} }
function PoTableComponent_ng_template_5_div_7_ng_container_1_Template(rf, ctx) { if (rf & 1) {
    i0.ɵɵelementContainerStart(0);
    i0.ɵɵelement(1, "po-table-subtitle-footer", 52);
    i0.ɵɵelementContainerEnd();
} if (rf & 2) {
    const column_r50 = ctx.$implicit;
    const ctx_r49 = i0.ɵɵnextContext(3);
    i0.ɵɵadvance(1);
    i0.ɵɵproperty("p-literals", ctx_r49.literals)("p-subtitles", column_r50.subtitles);
} }
function PoTableComponent_ng_template_5_div_7_Template(rf, ctx) { if (rf & 1) {
    i0.ɵɵelementStart(0, "div", 50);
    i0.ɵɵtemplate(1, PoTableComponent_ng_template_5_div_7_ng_container_1_Template, 2, 2, "ng-container", 51);
    i0.ɵɵelementEnd();
} if (rf & 2) {
    const ctx_r45 = i0.ɵɵnextContext(2);
    i0.ɵɵadvance(1);
    i0.ɵɵproperty("ngForOf", ctx_r45.subtitleColumns)("ngForTrackBy", ctx_r45.trackBy);
} }
function PoTableComponent_ng_template_5_Template(rf, ctx) { if (rf & 1) {
    i0.ɵɵelementStart(0, "div");
    i0.ɵɵtemplate(1, PoTableComponent_ng_template_5_div_1_Template, 2, 1, "div", 38);
    i0.ɵɵelementStart(2, "div", 39)(3, "div", 40, 41);
    i0.ɵɵtemplate(5, PoTableComponent_ng_template_5_div_5_Template, 4, 3, "div", 42);
    i0.ɵɵtemplate(6, PoTableComponent_ng_template_5_div_6_Template, 2, 1, "div", 43);
    i0.ɵɵelementEnd()()();
    i0.ɵɵtemplate(7, PoTableComponent_ng_template_5_div_7_Template, 2, 2, "div", 44);
} if (rf & 2) {
    const ctx_r5 = i0.ɵɵnextContext();
    i0.ɵɵclassProp("po-table-container-relative", ctx_r5.loading);
    i0.ɵɵadvance(1);
    i0.ɵɵproperty("ngIf", ctx_r5.loading);
    i0.ɵɵadvance(2);
    i0.ɵɵstyleProp("opacity", ctx_r5.tableOpacity);
    i0.ɵɵclassProp("po-table-header-fixed-columns-pixels", ctx_r5.allColumnsWidthPixels);
    i0.ɵɵadvance(2);
    i0.ɵɵproperty("ngIf", ctx_r5.height);
    i0.ɵɵadvance(1);
    i0.ɵɵproperty("ngIf", !ctx_r5.height);
    i0.ɵɵadvance(1);
    i0.ɵɵproperty("ngIf", ctx_r5.hasFooter);
} }
function PoTableComponent_div_7_Template(rf, ctx) { if (rf & 1) {
    const _r53 = i0.ɵɵgetCurrentView();
    i0.ɵɵelementStart(0, "div", 53, 54)(2, "po-button", 55);
    i0.ɵɵlistener("p-click", function PoTableComponent_div_7_Template_po_button_p_click_2_listener() { i0.ɵɵrestoreView(_r53); const ctx_r52 = i0.ɵɵnextContext(); return i0.ɵɵresetView(ctx_r52.onShowMore()); });
    i0.ɵɵelementEnd()();
} if (rf & 2) {
    const ctx_r6 = i0.ɵɵnextContext();
    i0.ɵɵclassProp("po-invisible", ctx_r6.showMore.observers.length === 0 && !ctx_r6.hasService);
    i0.ɵɵadvance(2);
    i0.ɵɵproperty("p-disabled", ctx_r6.showMoreDisabled)("p-label", ctx_r6.literals.loadMoreData)("p-loading", ctx_r6.loadingShowMore);
} }
function PoTableComponent_ng_template_8_th_3_po_checkbox_2_Template(rf, ctx) { if (rf & 1) {
    const _r69 = i0.ɵɵgetCurrentView();
    i0.ɵɵelementStart(0, "po-checkbox", 69);
    i0.ɵɵlistener("p-change", function PoTableComponent_ng_template_8_th_3_po_checkbox_2_Template_po_checkbox_p_change_0_listener() { i0.ɵɵrestoreView(_r69); const ctx_r68 = i0.ɵɵnextContext(3); return i0.ɵɵresetView(ctx_r68.selectAllRows()); });
    i0.ɵɵelementEnd();
} if (rf & 2) {
    const ctx_r67 = i0.ɵɵnextContext(3);
    i0.ɵɵproperty("p-checkboxValue", ctx_r67.selectAll === null ? "mixed" : ctx_r67.selectAll);
} }
function PoTableComponent_ng_template_8_th_3_Template(rf, ctx) { if (rf & 1) {
    i0.ɵɵelementStart(0, "th", 67)(1, "div");
    i0.ɵɵtemplate(2, PoTableComponent_ng_template_8_th_3_po_checkbox_2_Template, 1, 1, "po-checkbox", 68);
    i0.ɵɵelementEnd()();
} if (rf & 2) {
    const ctx_r54 = i0.ɵɵnextContext(2);
    i0.ɵɵstyleProp("pointer-events", ctx_r54.hideSelectAll ? "none" : "auto");
    i0.ɵɵadvance(1);
    i0.ɵɵclassProp("po-table-header-fixed-inner", ctx_r54.height);
    i0.ɵɵadvance(1);
    i0.ɵɵproperty("ngIf", !ctx_r54.hideSelectAll);
} }
function PoTableComponent_ng_template_8_th_4_Template(rf, ctx) { if (rf & 1) {
    i0.ɵɵelement(0, "th", 70);
} }
function PoTableComponent_ng_template_8_th_5_Template(rf, ctx) { if (rf & 1) {
    i0.ɵɵelement(0, "th", null, 71);
} if (rf & 2) {
    const ctx_r56 = i0.ɵɵnextContext(2);
    i0.ɵɵclassProp("po-table-header-master-detail", !ctx_r56.isSingleAction)("po-table-header-single-action", ctx_r56.isSingleAction);
} }
function PoTableComponent_ng_template_8_th_6_ng_container_2_Template(rf, ctx) { if (rf & 1) {
    i0.ɵɵelementContainer(0);
} }
function PoTableComponent_ng_template_8_th_6_Template(rf, ctx) { if (rf & 1) {
    i0.ɵɵelementStart(0, "th", 72, 73);
    i0.ɵɵtemplate(2, PoTableComponent_ng_template_8_th_6_ng_container_2_Template, 1, 0, "ng-container", 62);
    i0.ɵɵelementEnd();
} if (rf & 2) {
    const ctx_r57 = i0.ɵɵnextContext(2);
    const _r20 = i0.ɵɵreference(23);
    const _r22 = i0.ɵɵreference(25);
    i0.ɵɵadvance(2);
    i0.ɵɵproperty("ngIf", ctx_r57.height)("ngIfThen", _r20)("ngIfElse", _r22);
} }
function PoTableComponent_ng_template_8_ng_container_7_Template(rf, ctx) { if (rf & 1) {
    i0.ɵɵelementContainer(0);
} }
function PoTableComponent_ng_template_8_ng_template_8_th_0_ng_container_4_Template(rf, ctx) { if (rf & 1) {
    i0.ɵɵelementContainerStart(0);
    i0.ɵɵnamespaceSVG();
    i0.ɵɵelementStart(1, "svg", 78);
    i0.ɵɵelement(2, "circle", 79)(3, "circle", 80)(4, "circle", 81)(5, "circle", 82)(6, "circle", 83)(7, "circle", 84);
    i0.ɵɵelementEnd();
    i0.ɵɵelementContainerEnd();
} }
function PoTableComponent_ng_template_8_ng_template_8_th_0_ng_container_5_Template(rf, ctx) { if (rf & 1) {
    i0.ɵɵelementContainer(0);
} }
const _c15 = function (a0) { return { "po-table-header-sorted": a0 }; };
const _c16 = function (a0) { return { $implicit: a0 }; };
function PoTableComponent_ng_template_8_ng_template_8_th_0_Template(rf, ctx) { if (rf & 1) {
    const _r79 = i0.ɵɵgetCurrentView();
    i0.ɵɵelementStart(0, "th", 75);
    i0.ɵɵlistener("click", function PoTableComponent_ng_template_8_ng_template_8_th_0_Template_th_click_0_listener() { const restoredCtx = i0.ɵɵrestoreView(_r79); const column_r74 = restoredCtx.$implicit; const ctx_r78 = i0.ɵɵnextContext(3); return i0.ɵɵresetView(ctx_r78.sortColumn(column_r74)); });
    i0.ɵɵpipe(1, "lowercase");
    i0.ɵɵpipe(2, "titlecase");
    i0.ɵɵelementStart(3, "div", 76);
    i0.ɵɵtemplate(4, PoTableComponent_ng_template_8_ng_template_8_th_0_ng_container_4_Template, 8, 0, "ng-container", 43);
    i0.ɵɵtemplate(5, PoTableComponent_ng_template_8_ng_template_8_th_0_ng_container_5_Template, 1, 0, "ng-container", 77);
    i0.ɵɵelementEnd()();
} if (rf & 2) {
    const column_r74 = ctx.$implicit;
    const ctx_r73 = i0.ɵɵnextContext(3);
    const _r18 = i0.ɵɵreference(21);
    i0.ɵɵstyleProp("width", column_r74.width)("max-width", column_r74.width)("min-width", column_r74.width);
    i0.ɵɵclassProp("po-clickable", ctx_r73.sort && column_r74.sortable !== false || ctx_r73.hasService)("po-table-header-subtitle", column_r74.type === "subtitle")("po-table-column-drag-box", ctx_r73.isDraggable);
    i0.ɵɵproperty("ngClass", i0.ɵɵpureFunction1(29, _c15, ctx_r73.sort && ctx_r73.JSON.stringify(ctx_r73.sortedColumn == null ? null : ctx_r73.sortedColumn.property) === ctx_r73.JSON.stringify(column_r74) && (ctx_r73.sortedColumn.ascending || !ctx_r73.sortedColumn.ascending)))("cdkDragDisabled", column_r74.fixed ? "true" : "false")("pFrozenColumn", column_r74.fixed);
    i0.ɵɵattribute("data-po-table-column-name", i0.ɵɵpipeBind1(1, 25, column_r74.label || i0.ɵɵpipeBind1(2, 27, column_r74.property)));
    i0.ɵɵadvance(3);
    i0.ɵɵclassProp("po-table-header-fixed-inner", ctx_r73.height)("po-table-header-flex-right", column_r74.type === "currency" || column_r74.type === "number")("po-table-header-flex-center", column_r74.type === "subtitle");
    i0.ɵɵadvance(1);
    i0.ɵɵproperty("ngIf", ctx_r73.isDraggable && !column_r74.fixed);
    i0.ɵɵadvance(1);
    i0.ɵɵproperty("ngTemplateOutlet", _r18)("ngTemplateOutletContext", i0.ɵɵpureFunction1(31, _c16, column_r74));
} }
function PoTableComponent_ng_template_8_ng_template_8_Template(rf, ctx) { if (rf & 1) {
    i0.ɵɵtemplate(0, PoTableComponent_ng_template_8_ng_template_8_th_0_Template, 6, 33, "th", 74);
} if (rf & 2) {
    const ctx_r60 = i0.ɵɵnextContext(2);
    i0.ɵɵproperty("ngForOf", ctx_r60.mainColumns)("ngForTrackBy", ctx_r60.trackBy);
} }
function PoTableComponent_ng_template_8_ng_template_10_th_0_ng_container_4_Template(rf, ctx) { if (rf & 1) {
    i0.ɵɵelementContainer(0);
} }
function PoTableComponent_ng_template_8_ng_template_10_th_0_Template(rf, ctx) { if (rf & 1) {
    const _r85 = i0.ɵɵgetCurrentView();
    i0.ɵɵelementStart(0, "th", 86);
    i0.ɵɵlistener("click", function PoTableComponent_ng_template_8_ng_template_10_th_0_Template_th_click_0_listener() { const restoredCtx = i0.ɵɵrestoreView(_r85); const column_r81 = restoredCtx.$implicit; const ctx_r84 = i0.ɵɵnextContext(3); return i0.ɵɵresetView(ctx_r84.sortColumn(column_r81)); });
    i0.ɵɵpipe(1, "lowercase");
    i0.ɵɵpipe(2, "titlecase");
    i0.ɵɵelementStart(3, "div", 76);
    i0.ɵɵtemplate(4, PoTableComponent_ng_template_8_ng_template_10_th_0_ng_container_4_Template, 1, 0, "ng-container", 77);
    i0.ɵɵelementEnd()();
} if (rf & 2) {
    const column_r81 = ctx.$implicit;
    const ctx_r80 = i0.ɵɵnextContext(3);
    const _r18 = i0.ɵɵreference(21);
    i0.ɵɵstyleProp("width", column_r81.width)("max-width", column_r81.width)("min-width", column_r81.width);
    i0.ɵɵclassProp("po-clickable", ctx_r80.sort && column_r81.sortable !== false || ctx_r80.hasService)("po-table-header-subtitle", column_r81.type === "subtitle");
    i0.ɵɵproperty("ngClass", i0.ɵɵpureFunction1(25, _c15, ctx_r80.sort && ctx_r80.JSON.stringify(ctx_r80.sortedColumn == null ? null : ctx_r80.sortedColumn.property) === ctx_r80.JSON.stringify(column_r81) && (ctx_r80.sortedColumn.ascending || !ctx_r80.sortedColumn.ascending)))("pFrozenColumn", column_r81.fixed);
    i0.ɵɵattribute("data-po-table-column-name", i0.ɵɵpipeBind1(1, 21, column_r81.label || i0.ɵɵpipeBind1(2, 23, column_r81.property)));
    i0.ɵɵadvance(3);
    i0.ɵɵclassProp("po-table-header-fixed-inner", ctx_r80.height)("po-table-header-flex-right", column_r81.type === "currency" || column_r81.type === "number")("po-table-header-flex-center", column_r81.type === "subtitle");
    i0.ɵɵadvance(1);
    i0.ɵɵproperty("ngTemplateOutlet", _r18)("ngTemplateOutletContext", i0.ɵɵpureFunction1(27, _c16, column_r81));
} }
function PoTableComponent_ng_template_8_ng_template_10_Template(rf, ctx) { if (rf & 1) {
    i0.ɵɵtemplate(0, PoTableComponent_ng_template_8_ng_template_10_th_0_Template, 5, 29, "th", 85);
} if (rf & 2) {
    const ctx_r62 = i0.ɵɵnextContext(2);
    i0.ɵɵproperty("ngForOf", ctx_r62.mainColumns)("ngForTrackBy", ctx_r62.trackBy);
} }
function PoTableComponent_ng_template_8_th_12_Template(rf, ctx) { if (rf & 1) {
    i0.ɵɵelement(0, "th", 70);
} }
function PoTableComponent_ng_template_8_th_13_Template(rf, ctx) { if (rf & 1) {
    i0.ɵɵelement(0, "th");
} if (rf & 2) {
    const ctx_r64 = i0.ɵɵnextContext(2);
    i0.ɵɵclassProp("po-table-header-single-action", ctx_r64.isSingleAction)("po-table-header-actions", !ctx_r64.isSingleAction);
} }
function PoTableComponent_ng_template_8_tbody_14_Template(rf, ctx) { if (rf & 1) {
    i0.ɵɵelementStart(0, "tbody", 87)(1, "tr", 88)(2, "td", 89)(3, "span");
    i0.ɵɵtext(4);
    i0.ɵɵelementEnd()()()();
} if (rf & 2) {
    const ctx_r65 = i0.ɵɵnextContext(2);
    i0.ɵɵadvance(2);
    i0.ɵɵproperty("colSpan", ctx_r65.columnCount);
    i0.ɵɵadvance(2);
    i0.ɵɵtextInterpolate1(" ", ctx_r65.literals.noData, " ");
} }
function PoTableComponent_ng_template_8_ng_container_15_tbody_1_td_2_ng_container_1_Template(rf, ctx) { if (rf & 1) {
    i0.ɵɵelementContainer(0);
} }
function PoTableComponent_ng_template_8_ng_container_15_tbody_1_td_2_Template(rf, ctx) { if (rf & 1) {
    i0.ɵɵelementStart(0, "td", 67);
    i0.ɵɵtemplate(1, PoTableComponent_ng_template_8_ng_container_15_tbody_1_td_2_ng_container_1_Template, 1, 0, "ng-container", 77);
    i0.ɵɵelementEnd();
} if (rf & 2) {
    const row_r87 = i0.ɵɵnextContext().$implicit;
    const ctx_r89 = i0.ɵɵnextContext(3);
    const _r14 = i0.ɵɵreference(17);
    const _r16 = i0.ɵɵreference(19);
    i0.ɵɵadvance(1);
    i0.ɵɵproperty("ngTemplateOutlet", ctx_r89.singleSelect ? _r14 : _r16)("ngTemplateOutletContext", i0.ɵɵpureFunction1(2, _c16, row_r87));
} }
function PoTableComponent_ng_template_8_ng_container_15_tbody_1_td_3_ng_template_1_Template(rf, ctx) { }
const _c17 = function (a0, a1) { return { row: a0, rowIndex: a1 }; };
function PoTableComponent_ng_template_8_ng_container_15_tbody_1_td_3_Template(rf, ctx) { if (rf & 1) {
    const _r103 = i0.ɵɵgetCurrentView();
    i0.ɵɵelementStart(0, "td", 95);
    i0.ɵɵlistener("click", function PoTableComponent_ng_template_8_ng_container_15_tbody_1_td_3_Template_td_click_0_listener() { i0.ɵɵrestoreView(_r103); const row_r87 = i0.ɵɵnextContext().$implicit; const ctx_r101 = i0.ɵɵnextContext(3); return i0.ɵɵresetView(ctx_r101.toggleDetail(row_r87)); });
    i0.ɵɵtemplate(1, PoTableComponent_ng_template_8_ng_container_15_tbody_1_td_3_ng_template_1_Template, 0, 0, "ng-template", 96);
    i0.ɵɵelementEnd();
} if (rf & 2) {
    const ctx_r104 = i0.ɵɵnextContext();
    const row_r87 = ctx_r104.$implicit;
    const rowIndex_r88 = ctx_r104.index;
    i0.ɵɵnextContext(3);
    const _r12 = i0.ɵɵreference(15);
    i0.ɵɵadvance(1);
    i0.ɵɵproperty("ngTemplateOutlet", _r12)("ngTemplateOutletContext", i0.ɵɵpureFunction2(2, _c17, row_r87, rowIndex_r88));
} }
function PoTableComponent_ng_template_8_ng_container_15_tbody_1_4_ng_template_0_Template(rf, ctx) { }
function PoTableComponent_ng_template_8_ng_container_15_tbody_1_4_Template(rf, ctx) { if (rf & 1) {
    i0.ɵɵtemplate(0, PoTableComponent_ng_template_8_ng_container_15_tbody_1_4_ng_template_0_Template, 0, 0, "ng-template", 96);
} if (rf & 2) {
    const ctx_r106 = i0.ɵɵnextContext();
    const row_r87 = ctx_r106.$implicit;
    const rowIndex_r88 = ctx_r106.index;
    i0.ɵɵnextContext(3);
    const _r24 = i0.ɵɵreference(27);
    i0.ɵɵproperty("ngTemplateOutlet", _r24)("ngTemplateOutletContext", i0.ɵɵpureFunction2(2, _c17, row_r87, rowIndex_r88));
} }
function PoTableComponent_ng_template_8_ng_container_15_tbody_1_td_5_ng_template_1_Template(rf, ctx) { }
function PoTableComponent_ng_template_8_ng_container_15_tbody_1_td_5_Template(rf, ctx) { if (rf & 1) {
    const _r110 = i0.ɵɵgetCurrentView();
    i0.ɵɵelementStart(0, "td", 95);
    i0.ɵɵlistener("click", function PoTableComponent_ng_template_8_ng_container_15_tbody_1_td_5_Template_td_click_0_listener() { i0.ɵɵrestoreView(_r110); const row_r87 = i0.ɵɵnextContext().$implicit; const ctx_r108 = i0.ɵɵnextContext(3); return i0.ɵɵresetView(ctx_r108.toggleDetail(row_r87)); });
    i0.ɵɵtemplate(1, PoTableComponent_ng_template_8_ng_container_15_tbody_1_td_5_ng_template_1_Template, 0, 0, "ng-template", 96);
    i0.ɵɵelementEnd();
} if (rf & 2) {
    const ctx_r111 = i0.ɵɵnextContext();
    const row_r87 = ctx_r111.$implicit;
    const rowIndex_r88 = ctx_r111.index;
    i0.ɵɵnextContext(3);
    const _r12 = i0.ɵɵreference(15);
    i0.ɵɵadvance(1);
    i0.ɵɵproperty("ngTemplateOutlet", _r12)("ngTemplateOutletContext", i0.ɵɵpureFunction2(2, _c17, row_r87, rowIndex_r88));
} }
function PoTableComponent_ng_template_8_ng_container_15_tbody_1_td_6_span_2_ng_container_1_Template(rf, ctx) { if (rf & 1) {
    i0.ɵɵelementContainer(0);
} }
function PoTableComponent_ng_template_8_ng_container_15_tbody_1_td_6_span_2_Template(rf, ctx) { if (rf & 1) {
    i0.ɵɵelementStart(0, "span");
    i0.ɵɵtemplate(1, PoTableComponent_ng_template_8_ng_container_15_tbody_1_td_6_span_2_ng_container_1_Template, 1, 0, "ng-container", 77);
    i0.ɵɵelementEnd();
} if (rf & 2) {
    const column_r112 = i0.ɵɵnextContext().$implicit;
    const row_r87 = i0.ɵɵnextContext().$implicit;
    const ctx_r114 = i0.ɵɵnextContext(3);
    i0.ɵɵadvance(1);
    i0.ɵɵproperty("ngTemplateOutlet", ctx_r114.getTemplate(column_r112))("ngTemplateOutletContext", i0.ɵɵpureFunction1(2, _c16, ctx_r114.getCellData(row_r87, column_r112)));
} }
function PoTableComponent_ng_template_8_ng_container_15_tbody_1_td_6_span_3_ng_container_1_Template(rf, ctx) { if (rf & 1) {
    i0.ɵɵelementContainer(0);
} }
const _c18 = function (a0, a1) { return { row: a0, column: a1 }; };
function PoTableComponent_ng_template_8_ng_container_15_tbody_1_td_6_span_3_Template(rf, ctx) { if (rf & 1) {
    i0.ɵɵelementStart(0, "span");
    i0.ɵɵtemplate(1, PoTableComponent_ng_template_8_ng_container_15_tbody_1_td_6_span_3_ng_container_1_Template, 1, 0, "ng-container", 77);
    i0.ɵɵelementEnd();
} if (rf & 2) {
    const column_r112 = i0.ɵɵnextContext().$implicit;
    const row_r87 = i0.ɵɵnextContext().$implicit;
    const ctx_r115 = i0.ɵɵnextContext(3);
    i0.ɵɵadvance(1);
    i0.ɵɵproperty("ngTemplateOutlet", ctx_r115.tableCellTemplate == null ? null : ctx_r115.tableCellTemplate.templateRef)("ngTemplateOutletContext", i0.ɵɵpureFunction2(2, _c18, row_r87, column_r112));
} }
function PoTableComponent_ng_template_8_ng_container_15_tbody_1_td_6_span_4_Template(rf, ctx) { if (rf & 1) {
    i0.ɵɵelementStart(0, "span");
    i0.ɵɵtext(1);
    i0.ɵɵelementEnd();
} if (rf & 2) {
    const column_r112 = i0.ɵɵnextContext().$implicit;
    const row_r87 = i0.ɵɵnextContext().$implicit;
    const ctx_r116 = i0.ɵɵnextContext(3);
    i0.ɵɵadvance(1);
    i0.ɵɵtextInterpolate1(" ", ctx_r116.getBooleanLabel(ctx_r116.getCellData(row_r87, column_r112), column_r112), " ");
} }
function PoTableComponent_ng_template_8_ng_container_15_tbody_1_td_6_span_5_Template(rf, ctx) { if (rf & 1) {
    i0.ɵɵelementStart(0, "span");
    i0.ɵɵtext(1);
    i0.ɵɵpipe(2, "currency");
    i0.ɵɵelementEnd();
} if (rf & 2) {
    const column_r112 = i0.ɵɵnextContext().$implicit;
    const row_r87 = i0.ɵɵnextContext().$implicit;
    const ctx_r117 = i0.ɵɵnextContext(3);
    i0.ɵɵadvance(1);
    i0.ɵɵtextInterpolate1(" ", i0.ɵɵpipeBind4(2, 1, ctx_r117.getCellData(row_r87, column_r112), column_r112.format, "symbol", "1.2-2"), " ");
} }
function PoTableComponent_ng_template_8_ng_container_15_tbody_1_td_6_span_6_Template(rf, ctx) { if (rf & 1) {
    i0.ɵɵelementStart(0, "span");
    i0.ɵɵtext(1);
    i0.ɵɵpipe(2, "date");
    i0.ɵɵelementEnd();
} if (rf & 2) {
    const column_r112 = i0.ɵɵnextContext().$implicit;
    const row_r87 = i0.ɵɵnextContext().$implicit;
    const ctx_r118 = i0.ɵɵnextContext(3);
    i0.ɵɵadvance(1);
    i0.ɵɵtextInterpolate1(" ", i0.ɵɵpipeBind2(2, 1, ctx_r118.getCellData(row_r87, column_r112), column_r112.format || "dd/MM/yyyy"), " ");
} }
function PoTableComponent_ng_template_8_ng_container_15_tbody_1_td_6_span_7_Template(rf, ctx) { if (rf & 1) {
    i0.ɵɵelementStart(0, "span");
    i0.ɵɵtext(1);
    i0.ɵɵpipe(2, "po_time");
    i0.ɵɵelementEnd();
} if (rf & 2) {
    const column_r112 = i0.ɵɵnextContext().$implicit;
    const row_r87 = i0.ɵɵnextContext().$implicit;
    const ctx_r119 = i0.ɵɵnextContext(3);
    i0.ɵɵadvance(1);
    i0.ɵɵtextInterpolate1(" ", i0.ɵɵpipeBind2(2, 1, ctx_r119.getCellData(row_r87, column_r112), column_r112.format || "HH:mm:ss.ffffff"), " ");
} }
function PoTableComponent_ng_template_8_ng_container_15_tbody_1_td_6_span_8_Template(rf, ctx) { if (rf & 1) {
    i0.ɵɵelementStart(0, "span");
    i0.ɵɵtext(1);
    i0.ɵɵpipe(2, "date");
    i0.ɵɵelementEnd();
} if (rf & 2) {
    const column_r112 = i0.ɵɵnextContext().$implicit;
    const row_r87 = i0.ɵɵnextContext().$implicit;
    const ctx_r120 = i0.ɵɵnextContext(3);
    i0.ɵɵadvance(1);
    i0.ɵɵtextInterpolate1(" ", i0.ɵɵpipeBind2(2, 1, ctx_r120.getCellData(row_r87, column_r112), column_r112.format || "dd/MM/yyyy HH:mm:ss"), " ");
} }
function PoTableComponent_ng_template_8_ng_container_15_tbody_1_td_6_span_9_Template(rf, ctx) { if (rf & 1) {
    i0.ɵɵelementStart(0, "span");
    i0.ɵɵtext(1);
    i0.ɵɵelementEnd();
} if (rf & 2) {
    const column_r112 = i0.ɵɵnextContext().$implicit;
    const row_r87 = i0.ɵɵnextContext().$implicit;
    const ctx_r121 = i0.ɵɵnextContext(3);
    i0.ɵɵadvance(1);
    i0.ɵɵtextInterpolate1(" ", ctx_r121.formatNumber(ctx_r121.getCellData(row_r87, column_r112), column_r112.format), " ");
} }
function PoTableComponent_ng_template_8_ng_container_15_tbody_1_td_6_po_table_column_link_10_Template(rf, ctx) { if (rf & 1) {
    const _r147 = i0.ɵɵgetCurrentView();
    i0.ɵɵelementStart(0, "po-table-column-link", 103);
    i0.ɵɵlistener("click", function PoTableComponent_ng_template_8_ng_container_15_tbody_1_td_6_po_table_column_link_10_Template_po_table_column_link_click_0_listener($event) { i0.ɵɵrestoreView(_r147); const column_r112 = i0.ɵɵnextContext().$implicit; const row_r87 = i0.ɵɵnextContext().$implicit; const ctx_r145 = i0.ɵɵnextContext(3); return i0.ɵɵresetView(ctx_r145.onClickLink($event, row_r87, column_r112)); });
    i0.ɵɵelementEnd();
} if (rf & 2) {
    const column_r112 = i0.ɵɵnextContext().$implicit;
    const row_r87 = i0.ɵɵnextContext().$implicit;
    const ctx_r122 = i0.ɵɵnextContext(3);
    i0.ɵɵproperty("p-action", column_r112.action)("p-disabled", ctx_r122.checkDisabled(row_r87, column_r112))("p-link", row_r87[column_r112.link])("p-row", row_r87)("p-value", ctx_r122.getCellData(row_r87, column_r112));
} }
function PoTableComponent_ng_template_8_ng_container_15_tbody_1_td_6_po_table_column_icon_11_Template(rf, ctx) { if (rf & 1) {
    i0.ɵɵelement(0, "po-table-column-icon", 104);
} if (rf & 2) {
    const column_r112 = i0.ɵɵnextContext().$implicit;
    const row_r87 = i0.ɵɵnextContext().$implicit;
    const ctx_r123 = i0.ɵɵnextContext(3);
    i0.ɵɵproperty("p-column", column_r112)("p-icons", ctx_r123.getColumnIcons(row_r87, column_r112))("p-row", row_r87);
} }
function PoTableComponent_ng_template_8_ng_container_15_tbody_1_td_6_span_12_Template(rf, ctx) { if (rf & 1) {
    i0.ɵɵelementStart(0, "span");
    i0.ɵɵelement(1, "po-table-subtitle-circle", 105);
    i0.ɵɵelementEnd();
} if (rf & 2) {
    const column_r112 = i0.ɵɵnextContext().$implicit;
    const row_r87 = i0.ɵɵnextContext().$implicit;
    const ctx_r124 = i0.ɵɵnextContext(3);
    i0.ɵɵadvance(1);
    i0.ɵɵproperty("p-subtitle", ctx_r124.getSubtitleColumn(row_r87, column_r112));
} }
function PoTableComponent_ng_template_8_ng_container_15_tbody_1_td_6_span_13_Template(rf, ctx) { if (rf & 1) {
    i0.ɵɵelementStart(0, "span");
    i0.ɵɵelement(1, "po-table-column-label", 106);
    i0.ɵɵelementEnd();
} if (rf & 2) {
    const column_r112 = i0.ɵɵnextContext().$implicit;
    const row_r87 = i0.ɵɵnextContext().$implicit;
    const ctx_r125 = i0.ɵɵnextContext(3);
    i0.ɵɵadvance(1);
    i0.ɵɵproperty("p-value", ctx_r125.getColumnLabel(row_r87, column_r112));
} }
function PoTableComponent_ng_template_8_ng_container_15_tbody_1_td_6_span_14_Template(rf, ctx) { if (rf & 1) {
    i0.ɵɵelementStart(0, "span");
    i0.ɵɵtext(1);
    i0.ɵɵelementEnd();
} if (rf & 2) {
    const column_r112 = i0.ɵɵnextContext().$implicit;
    const row_r87 = i0.ɵɵnextContext().$implicit;
    const ctx_r126 = i0.ɵɵnextContext(3);
    i0.ɵɵadvance(1);
    i0.ɵɵtextInterpolate(ctx_r126.getCellData(row_r87, column_r112));
} }
function PoTableComponent_ng_template_8_ng_container_15_tbody_1_td_6_Template(rf, ctx) { if (rf & 1) {
    const _r160 = i0.ɵɵgetCurrentView();
    i0.ɵɵelementStart(0, "td", 97);
    i0.ɵɵlistener("click", function PoTableComponent_ng_template_8_ng_container_15_tbody_1_td_6_Template_td_click_0_listener() { i0.ɵɵrestoreView(_r160); const row_r87 = i0.ɵɵnextContext().$implicit; const ctx_r159 = i0.ɵɵnextContext(3); return i0.ɵɵresetView(ctx_r159.hasSelectableRow() ? ctx_r159.selectRow(row_r87) : "javascript:;"); });
    i0.ɵɵelementStart(1, "div", 98);
    i0.ɵɵlistener("mouseenter", function PoTableComponent_ng_template_8_ng_container_15_tbody_1_td_6_Template_div_mouseenter_1_listener($event) { const restoredCtx = i0.ɵɵrestoreView(_r160); const column_r112 = restoredCtx.$implicit; const row_r87 = i0.ɵɵnextContext().$implicit; const ctx_r162 = i0.ɵɵnextContext(3); return i0.ɵɵresetView(ctx_r162.tooltipMouseEnter($event, column_r112, row_r87)); })("mouseleave", function PoTableComponent_ng_template_8_ng_container_15_tbody_1_td_6_Template_div_mouseleave_1_listener() { i0.ɵɵrestoreView(_r160); const ctx_r164 = i0.ɵɵnextContext(4); return i0.ɵɵresetView(ctx_r164.tooltipMouseLeave()); });
    i0.ɵɵtemplate(2, PoTableComponent_ng_template_8_ng_container_15_tbody_1_td_6_span_2_Template, 2, 4, "span", 99);
    i0.ɵɵtemplate(3, PoTableComponent_ng_template_8_ng_container_15_tbody_1_td_6_span_3_Template, 2, 5, "span", 99);
    i0.ɵɵtemplate(4, PoTableComponent_ng_template_8_ng_container_15_tbody_1_td_6_span_4_Template, 2, 1, "span", 99);
    i0.ɵɵtemplate(5, PoTableComponent_ng_template_8_ng_container_15_tbody_1_td_6_span_5_Template, 3, 6, "span", 99);
    i0.ɵɵtemplate(6, PoTableComponent_ng_template_8_ng_container_15_tbody_1_td_6_span_6_Template, 3, 4, "span", 99);
    i0.ɵɵtemplate(7, PoTableComponent_ng_template_8_ng_container_15_tbody_1_td_6_span_7_Template, 3, 4, "span", 99);
    i0.ɵɵtemplate(8, PoTableComponent_ng_template_8_ng_container_15_tbody_1_td_6_span_8_Template, 3, 4, "span", 99);
    i0.ɵɵtemplate(9, PoTableComponent_ng_template_8_ng_container_15_tbody_1_td_6_span_9_Template, 2, 1, "span", 99);
    i0.ɵɵtemplate(10, PoTableComponent_ng_template_8_ng_container_15_tbody_1_td_6_po_table_column_link_10_Template, 1, 5, "po-table-column-link", 100);
    i0.ɵɵtemplate(11, PoTableComponent_ng_template_8_ng_container_15_tbody_1_td_6_po_table_column_icon_11_Template, 1, 3, "po-table-column-icon", 101);
    i0.ɵɵtemplate(12, PoTableComponent_ng_template_8_ng_container_15_tbody_1_td_6_span_12_Template, 2, 1, "span", 99);
    i0.ɵɵtemplate(13, PoTableComponent_ng_template_8_ng_container_15_tbody_1_td_6_span_13_Template, 2, 1, "span", 99);
    i0.ɵɵtemplate(14, PoTableComponent_ng_template_8_ng_container_15_tbody_1_td_6_span_14_Template, 2, 1, "span", 102);
    i0.ɵɵelementEnd()();
} if (rf & 2) {
    const column_r112 = ctx.$implicit;
    const row_r87 = i0.ɵɵnextContext().$implicit;
    const ctx_r93 = i0.ɵɵnextContext(3);
    i0.ɵɵstyleProp("width", column_r112.width)("max-width", column_r112.width)("min-width", column_r112.width);
    i0.ɵɵclassProp("po-table-column", column_r112.type !== "icon")("po-table-column-right", column_r112.type === "currency" || column_r112.type === "number")("po-table-column-center", column_r112.type === "subtitle")("po-table-column-icons", column_r112.type === "icon");
    i0.ɵɵproperty("pFrozenColumn", column_r112.fixed)("ngClass", ctx_r93.getClassColor(row_r87, column_r112));
    i0.ɵɵadvance(1);
    i0.ɵɵproperty("ngSwitch", column_r112.type)("p-tooltip", ctx_r93.tooltipText)("p-append-in-body", true);
    i0.ɵɵadvance(1);
    i0.ɵɵproperty("ngSwitchCase", "columnTemplate");
    i0.ɵɵadvance(1);
    i0.ɵɵproperty("ngSwitchCase", "cellTemplate");
    i0.ɵɵadvance(1);
    i0.ɵɵproperty("ngSwitchCase", "boolean");
    i0.ɵɵadvance(1);
    i0.ɵɵproperty("ngSwitchCase", "currency");
    i0.ɵɵadvance(1);
    i0.ɵɵproperty("ngSwitchCase", "date");
    i0.ɵɵadvance(1);
    i0.ɵɵproperty("ngSwitchCase", "time");
    i0.ɵɵadvance(1);
    i0.ɵɵproperty("ngSwitchCase", "dateTime");
    i0.ɵɵadvance(1);
    i0.ɵɵproperty("ngSwitchCase", "number");
    i0.ɵɵadvance(1);
    i0.ɵɵproperty("ngSwitchCase", "link");
    i0.ɵɵadvance(1);
    i0.ɵɵproperty("ngSwitchCase", "icon");
    i0.ɵɵadvance(1);
    i0.ɵɵproperty("ngSwitchCase", "subtitle");
    i0.ɵɵadvance(1);
    i0.ɵɵproperty("ngSwitchCase", "label");
} }
function PoTableComponent_ng_template_8_ng_container_15_tbody_1_td_7_ng_template_1_Template(rf, ctx) { }
function PoTableComponent_ng_template_8_ng_container_15_tbody_1_td_7_Template(rf, ctx) { if (rf & 1) {
    const _r169 = i0.ɵɵgetCurrentView();
    i0.ɵɵelementStart(0, "td", 95);
    i0.ɵɵlistener("click", function PoTableComponent_ng_template_8_ng_container_15_tbody_1_td_7_Template_td_click_0_listener() { i0.ɵɵrestoreView(_r169); const row_r87 = i0.ɵɵnextContext().$implicit; const ctx_r167 = i0.ɵɵnextContext(3); return i0.ɵɵresetView(ctx_r167.toggleDetail(row_r87)); });
    i0.ɵɵtemplate(1, PoTableComponent_ng_template_8_ng_container_15_tbody_1_td_7_ng_template_1_Template, 0, 0, "ng-template", 96);
    i0.ɵɵelementEnd();
} if (rf & 2) {
    const ctx_r170 = i0.ɵɵnextContext();
    const row_r87 = ctx_r170.$implicit;
    const rowIndex_r88 = ctx_r170.index;
    i0.ɵɵnextContext(3);
    const _r12 = i0.ɵɵreference(15);
    i0.ɵɵadvance(1);
    i0.ɵɵproperty("ngTemplateOutlet", _r12)("ngTemplateOutletContext", i0.ɵɵpureFunction2(2, _c17, row_r87, rowIndex_r88));
} }
function PoTableComponent_ng_template_8_ng_container_15_tbody_1_8_ng_template_0_Template(rf, ctx) { }
function PoTableComponent_ng_template_8_ng_container_15_tbody_1_8_Template(rf, ctx) { if (rf & 1) {
    i0.ɵɵtemplate(0, PoTableComponent_ng_template_8_ng_container_15_tbody_1_8_ng_template_0_Template, 0, 0, "ng-template", 96);
} if (rf & 2) {
    const ctx_r172 = i0.ɵɵnextContext();
    const row_r87 = ctx_r172.$implicit;
    const rowIndex_r88 = ctx_r172.index;
    i0.ɵɵnextContext(3);
    const _r24 = i0.ɵɵreference(27);
    i0.ɵɵproperty("ngTemplateOutlet", _r24)("ngTemplateOutletContext", i0.ɵɵpureFunction2(2, _c17, row_r87, rowIndex_r88));
} }
function PoTableComponent_ng_template_8_ng_container_15_tbody_1_tr_9_ng_template_2_Template(rf, ctx) { }
const _c19 = function (a0, a1) { return { $implicit: a0, rowIndex: a1 }; };
function PoTableComponent_ng_template_8_ng_container_15_tbody_1_tr_9_Template(rf, ctx) { if (rf & 1) {
    i0.ɵɵelementStart(0, "tr")(1, "td", 107);
    i0.ɵɵtemplate(2, PoTableComponent_ng_template_8_ng_container_15_tbody_1_tr_9_ng_template_2_Template, 0, 0, "ng-template", 96);
    i0.ɵɵelementEnd()();
} if (rf & 2) {
    const ctx_r174 = i0.ɵɵnextContext();
    const row_r87 = ctx_r174.$implicit;
    const rowIndex_r88 = ctx_r174.index;
    const ctx_r96 = i0.ɵɵnextContext(3);
    i0.ɵɵadvance(1);
    i0.ɵɵproperty("colSpan", ctx_r96.columnCountForMasterDetail);
    i0.ɵɵadvance(1);
    i0.ɵɵproperty("ngTemplateOutlet", ctx_r96.tableRowTemplate.templateRef)("ngTemplateOutletContext", i0.ɵɵpureFunction2(3, _c19, row_r87, rowIndex_r88));
} }
function PoTableComponent_ng_template_8_ng_container_15_tbody_1_tr_10_Template(rf, ctx) { if (rf & 1) {
    const _r176 = i0.ɵɵgetCurrentView();
    i0.ɵɵelementStart(0, "tr")(1, "td", 108)(2, "po-table-detail", 109);
    i0.ɵɵlistener("p-select-row", function PoTableComponent_ng_template_8_ng_container_15_tbody_1_tr_10_Template_po_table_detail_p_select_row_2_listener($event) { i0.ɵɵrestoreView(_r176); const ctx_r175 = i0.ɵɵnextContext(4); return i0.ɵɵresetView(ctx_r175.selectDetailRow($event)); });
    i0.ɵɵelementEnd()()();
} if (rf & 2) {
    const row_r87 = i0.ɵɵnextContext().$implicit;
    const ctx_r97 = i0.ɵɵnextContext(3);
    i0.ɵɵadvance(1);
    i0.ɵɵproperty("colSpan", ctx_r97.columnCountForMasterDetail);
    i0.ɵɵadvance(1);
    i0.ɵɵproperty("p-selectable", ctx_r97.selectable && !ctx_r97.detailHideSelect)("p-detail", ctx_r97.columnMasterDetail.detail)("p-items", row_r87[ctx_r97.nameColumnDetail]);
} }
function PoTableComponent_ng_template_8_ng_container_15_tbody_1_Template(rf, ctx) { if (rf & 1) {
    i0.ɵɵelementStart(0, "tbody", 87)(1, "tr", 91);
    i0.ɵɵtemplate(2, PoTableComponent_ng_template_8_ng_container_15_tbody_1_td_2_Template, 2, 4, "td", 92);
    i0.ɵɵtemplate(3, PoTableComponent_ng_template_8_ng_container_15_tbody_1_td_3_Template, 2, 5, "td", 93);
    i0.ɵɵtemplate(4, PoTableComponent_ng_template_8_ng_container_15_tbody_1_4_Template, 1, 5, null, 43);
    i0.ɵɵtemplate(5, PoTableComponent_ng_template_8_ng_container_15_tbody_1_td_5_Template, 2, 5, "td", 93);
    i0.ɵɵtemplate(6, PoTableComponent_ng_template_8_ng_container_15_tbody_1_td_6_Template, 15, 31, "td", 94);
    i0.ɵɵtemplate(7, PoTableComponent_ng_template_8_ng_container_15_tbody_1_td_7_Template, 2, 5, "td", 93);
    i0.ɵɵtemplate(8, PoTableComponent_ng_template_8_ng_container_15_tbody_1_8_Template, 1, 5, null, 43);
    i0.ɵɵelementEnd();
    i0.ɵɵtemplate(9, PoTableComponent_ng_template_8_ng_container_15_tbody_1_tr_9_Template, 3, 6, "tr", 43);
    i0.ɵɵtemplate(10, PoTableComponent_ng_template_8_ng_container_15_tbody_1_tr_10_Template, 3, 4, "tr", 43);
    i0.ɵɵelementEnd();
} if (rf & 2) {
    const row_r87 = ctx.$implicit;
    const rowIndex_r88 = ctx.index;
    const ctx_r86 = i0.ɵɵnextContext(3);
    i0.ɵɵadvance(1);
    i0.ɵɵclassProp("po-table-row-active", row_r87.$selected || row_r87.$selected === null && ctx_r86.selectable);
    i0.ɵɵadvance(1);
    i0.ɵɵproperty("ngIf", ctx_r86.selectable);
    i0.ɵɵadvance(1);
    i0.ɵɵproperty("ngIf", ctx_r86.columnMasterDetail && !ctx_r86.hideDetail && !ctx_r86.hasRowTemplate);
    i0.ɵɵadvance(1);
    i0.ɵɵproperty("ngIf", !ctx_r86.actionRight && (ctx_r86.visibleActions.length > 1 || ctx_r86.isSingleAction));
    i0.ɵɵadvance(1);
    i0.ɵɵproperty("ngIf", ctx_r86.hasRowTemplate && !ctx_r86.hasRowTemplateWithArrowDirectionRight);
    i0.ɵɵadvance(1);
    i0.ɵɵproperty("ngForOf", ctx_r86.mainColumns)("ngForTrackBy", ctx_r86.trackBy);
    i0.ɵɵadvance(1);
    i0.ɵɵproperty("ngIf", ctx_r86.hasRowTemplateWithArrowDirectionRight);
    i0.ɵɵadvance(1);
    i0.ɵɵproperty("ngIf", ctx_r86.actionRight);
    i0.ɵɵadvance(1);
    i0.ɵɵproperty("ngIf", ctx_r86.hasMainColumns && ctx_r86.hasRowTemplate && row_r87.$showDetail && ctx_r86.isShowRowTemplate(row_r87, rowIndex_r88));
    i0.ɵɵadvance(1);
    i0.ɵɵproperty("ngIf", ctx_r86.hasMainColumns && ctx_r86.isShowMasterDetail(row_r87));
} }
function PoTableComponent_ng_template_8_ng_container_15_Template(rf, ctx) { if (rf & 1) {
    i0.ɵɵelementContainerStart(0);
    i0.ɵɵtemplate(1, PoTableComponent_ng_template_8_ng_container_15_tbody_1_Template, 11, 12, "tbody", 90);
    i0.ɵɵelementContainerEnd();
} if (rf & 2) {
    const ctx_r66 = i0.ɵɵnextContext(2);
    i0.ɵɵadvance(1);
    i0.ɵɵproperty("ngForOf", ctx_r66.filteredItems)("ngForTrackBy", ctx_r66.trackBy);
} }
const _c20 = function (a0, a1, a2) { return { "po-table-interactive": a0, "po-table-selectable": a1, "po-table-striped": a2 }; };
const _c21 = function (a0, a1) { return { "no-hover": a0, "po-table-column-drag": a1 }; };
function PoTableComponent_ng_template_8_Template(rf, ctx) { if (rf & 1) {
    const _r179 = i0.ɵɵgetCurrentView();
    i0.ɵɵelementStart(0, "table", 56)(1, "thead")(2, "tr", 57);
    i0.ɵɵlistener("cdkDropListDropped", function PoTableComponent_ng_template_8_Template_tr_cdkDropListDropped_2_listener($event) { i0.ɵɵrestoreView(_r179); const ctx_r178 = i0.ɵɵnextContext(); return i0.ɵɵresetView(ctx_r178.drop($event)); });
    i0.ɵɵtemplate(3, PoTableComponent_ng_template_8_th_3_Template, 3, 5, "th", 58);
    i0.ɵɵtemplate(4, PoTableComponent_ng_template_8_th_4_Template, 1, 0, "th", 59);
    i0.ɵɵtemplate(5, PoTableComponent_ng_template_8_th_5_Template, 2, 4, "th", 60);
    i0.ɵɵtemplate(6, PoTableComponent_ng_template_8_th_6_Template, 3, 3, "th", 61);
    i0.ɵɵtemplate(7, PoTableComponent_ng_template_8_ng_container_7_Template, 1, 0, "ng-container", 62);
    i0.ɵɵtemplate(8, PoTableComponent_ng_template_8_ng_template_8_Template, 1, 2, "ng-template", null, 63, i0.ɵɵtemplateRefExtractor);
    i0.ɵɵtemplate(10, PoTableComponent_ng_template_8_ng_template_10_Template, 1, 2, "ng-template", null, 64, i0.ɵɵtemplateRefExtractor);
    i0.ɵɵtemplate(12, PoTableComponent_ng_template_8_th_12_Template, 1, 0, "th", 59);
    i0.ɵɵtemplate(13, PoTableComponent_ng_template_8_th_13_Template, 1, 4, "th", 65);
    i0.ɵɵelementEnd()();
    i0.ɵɵtemplate(14, PoTableComponent_ng_template_8_tbody_14_Template, 5, 2, "tbody", 66);
    i0.ɵɵtemplate(15, PoTableComponent_ng_template_8_ng_container_15_Template, 2, 2, "ng-container", 43);
    i0.ɵɵelementEnd();
} if (rf & 2) {
    const _r59 = i0.ɵɵreference(9);
    const _r61 = i0.ɵɵreference(11);
    const ctx_r8 = i0.ɵɵnextContext();
    i0.ɵɵproperty("ngClass", i0.ɵɵpureFunction3(16, _c20, ctx_r8.selectable || ctx_r8.sort, ctx_r8.selectable, ctx_r8.striped));
    i0.ɵɵattribute("p-spacing", ctx_r8.spacing);
    i0.ɵɵadvance(2);
    i0.ɵɵclassProp("po-table-header", !ctx_r8.height);
    i0.ɵɵproperty("ngClass", i0.ɵɵpureFunction2(20, _c21, ctx_r8.hideSelectAll, ctx_r8.isDraggable));
    i0.ɵɵadvance(1);
    i0.ɵɵproperty("ngIf", ctx_r8.hasSelectableColumn);
    i0.ɵɵadvance(1);
    i0.ɵɵproperty("ngIf", (ctx_r8.hasMasterDetailColumn || ctx_r8.hasRowTemplate) && !ctx_r8.hasRowTemplateWithArrowDirectionRight);
    i0.ɵɵadvance(1);
    i0.ɵɵproperty("ngIf", !ctx_r8.actionRight && (ctx_r8.visibleActions.length > 1 || ctx_r8.isSingleAction));
    i0.ɵɵadvance(1);
    i0.ɵɵproperty("ngIf", !ctx_r8.hasMainColumns);
    i0.ɵɵadvance(1);
    i0.ɵɵproperty("ngIf", ctx_r8.isDraggable || ctx_r8.hasSomeFixed())("ngIfThen", _r59)("ngIfElse", _r61);
    i0.ɵɵadvance(5);
    i0.ɵɵproperty("ngIf", ctx_r8.hasRowTemplateWithArrowDirectionRight && (ctx_r8.hasVisibleActions || ctx_r8.hideColumnsManager));
    i0.ɵɵadvance(1);
    i0.ɵɵproperty("ngIf", ctx_r8.hasVisibleActions && ctx_r8.actionRight && (ctx_r8.visibleActions.length > 1 || ctx_r8.isSingleAction));
    i0.ɵɵadvance(1);
    i0.ɵɵproperty("ngIf", !ctx_r8.hasItems || !ctx_r8.hasMainColumns);
    i0.ɵɵadvance(1);
    i0.ɵɵproperty("ngIf", ctx_r8.hasMainColumns);
} }
function PoTableComponent_ng_template_10_th_5_po_checkbox_2_Template(rf, ctx) { if (rf & 1) {
    const _r196 = i0.ɵɵgetCurrentView();
    i0.ɵɵelementStart(0, "po-checkbox", 69);
    i0.ɵɵlistener("p-change", function PoTableComponent_ng_template_10_th_5_po_checkbox_2_Template_po_checkbox_p_change_0_listener() { i0.ɵɵrestoreView(_r196); const ctx_r195 = i0.ɵɵnextContext(3); return i0.ɵɵresetView(ctx_r195.selectAllRows()); });
    i0.ɵɵelementEnd();
} if (rf & 2) {
    const ctx_r194 = i0.ɵɵnextContext(3);
    i0.ɵɵproperty("p-checkboxValue", ctx_r194.selectAll === null ? "mixed" : ctx_r194.selectAll);
} }
function PoTableComponent_ng_template_10_th_5_Template(rf, ctx) { if (rf & 1) {
    i0.ɵɵelementStart(0, "th", 67)(1, "div");
    i0.ɵɵtemplate(2, PoTableComponent_ng_template_10_th_5_po_checkbox_2_Template, 1, 1, "po-checkbox", 68);
    i0.ɵɵelementEnd()();
} if (rf & 2) {
    const ctx_r181 = i0.ɵɵnextContext(2);
    i0.ɵɵstyleProp("pointer-events", ctx_r181.hideSelectAll ? "none" : "auto");
    i0.ɵɵadvance(1);
    i0.ɵɵclassProp("po-table-header-fixed-inner", ctx_r181.height);
    i0.ɵɵadvance(1);
    i0.ɵɵproperty("ngIf", !ctx_r181.hideSelectAll);
} }
function PoTableComponent_ng_template_10_th_6_Template(rf, ctx) { if (rf & 1) {
    i0.ɵɵelement(0, "th", 70);
} }
function PoTableComponent_ng_template_10_th_7_Template(rf, ctx) { if (rf & 1) {
    i0.ɵɵelement(0, "th", null, 71);
} if (rf & 2) {
    const ctx_r183 = i0.ɵɵnextContext(2);
    i0.ɵɵclassProp("po-table-header-master-detail", !ctx_r183.isSingleAction)("po-table-header-single-action", ctx_r183.isSingleAction);
} }
function PoTableComponent_ng_template_10_th_8_ng_container_2_Template(rf, ctx) { if (rf & 1) {
    i0.ɵɵelementContainer(0);
} }
function PoTableComponent_ng_template_10_th_8_Template(rf, ctx) { if (rf & 1) {
    i0.ɵɵelementStart(0, "th", 72, 73);
    i0.ɵɵtemplate(2, PoTableComponent_ng_template_10_th_8_ng_container_2_Template, 1, 0, "ng-container", 62);
    i0.ɵɵelementEnd();
} if (rf & 2) {
    const ctx_r184 = i0.ɵɵnextContext(2);
    const _r20 = i0.ɵɵreference(23);
    const _r22 = i0.ɵɵreference(25);
    i0.ɵɵadvance(2);
    i0.ɵɵproperty("ngIf", ctx_r184.height)("ngIfThen", _r20)("ngIfElse", _r22);
} }
function PoTableComponent_ng_template_10_ng_container_9_Template(rf, ctx) { if (rf & 1) {
    i0.ɵɵelementContainer(0);
} }
function PoTableComponent_ng_template_10_ng_template_10_th_0_ng_container_4_Template(rf, ctx) { if (rf & 1) {
    i0.ɵɵelementContainerStart(0);
    i0.ɵɵnamespaceSVG();
    i0.ɵɵelementStart(1, "svg", 78);
    i0.ɵɵelement(2, "circle", 79)(3, "circle", 80)(4, "circle", 81)(5, "circle", 82)(6, "circle", 83)(7, "circle", 84);
    i0.ɵɵelementEnd();
    i0.ɵɵelementContainerEnd();
} }
function PoTableComponent_ng_template_10_ng_template_10_th_0_ng_container_5_Template(rf, ctx) { if (rf & 1) {
    i0.ɵɵelementContainer(0);
} }
const _c22 = function (a0) { return { "width": a0 }; };
function PoTableComponent_ng_template_10_ng_template_10_th_0_Template(rf, ctx) { if (rf & 1) {
    const _r206 = i0.ɵɵgetCurrentView();
    i0.ɵɵelementStart(0, "th", 118);
    i0.ɵɵlistener("click", function PoTableComponent_ng_template_10_ng_template_10_th_0_Template_th_click_0_listener() { const restoredCtx = i0.ɵɵrestoreView(_r206); const column_r201 = restoredCtx.$implicit; const ctx_r205 = i0.ɵɵnextContext(3); return i0.ɵɵresetView(ctx_r205.sortColumn(column_r201)); });
    i0.ɵɵpipe(1, "lowercase");
    i0.ɵɵpipe(2, "titlecase");
    i0.ɵɵelementStart(3, "div", 76);
    i0.ɵɵtemplate(4, PoTableComponent_ng_template_10_ng_template_10_th_0_ng_container_4_Template, 8, 0, "ng-container", 43);
    i0.ɵɵtemplate(5, PoTableComponent_ng_template_10_ng_template_10_th_0_ng_container_5_Template, 1, 0, "ng-container", 77);
    i0.ɵɵelementEnd()();
} if (rf & 2) {
    const column_r201 = ctx.$implicit;
    const ctx_r200 = i0.ɵɵnextContext(3);
    const _r18 = i0.ɵɵreference(21);
    i0.ɵɵstyleProp("width", column_r201.width)("max-width", column_r201.width)("min-width", column_r201.width);
    i0.ɵɵclassProp("po-clickable", ctx_r200.sort && column_r201.sortable !== false || ctx_r200.hasService)("po-table-header-subtitle", column_r201.type === "subtitle")("po-table-column-drag-box", ctx_r200.isDraggable);
    i0.ɵɵproperty("ngClass", i0.ɵɵpureFunction1(30, _c15, ctx_r200.sort && ctx_r200.JSON.stringify(ctx_r200.sortedColumn == null ? null : ctx_r200.sortedColumn.property) === ctx_r200.JSON.stringify(column_r201) && (ctx_r200.sortedColumn.ascending || !ctx_r200.sortedColumn.ascending)))("ngStyle", i0.ɵɵpureFunction1(32, _c22, !ctx_r200.hasItems ? "100%" : "auto"))("cdkDragDisabled", column_r201.fixed ? "true" : "false")("pFrozenColumn", column_r201.fixed);
    i0.ɵɵattribute("data-po-table-column-name", i0.ɵɵpipeBind1(1, 26, column_r201.label || i0.ɵɵpipeBind1(2, 28, column_r201.property)));
    i0.ɵɵadvance(3);
    i0.ɵɵclassProp("po-table-header-fixed-inner", ctx_r200.height)("po-table-header-flex-right", column_r201.type === "currency" || column_r201.type === "number")("po-table-header-flex-center", column_r201.type === "subtitle");
    i0.ɵɵadvance(1);
    i0.ɵɵproperty("ngIf", ctx_r200.isDraggable && !column_r201.fixed);
    i0.ɵɵadvance(1);
    i0.ɵɵproperty("ngTemplateOutlet", _r18)("ngTemplateOutletContext", i0.ɵɵpureFunction1(34, _c16, column_r201));
} }
function PoTableComponent_ng_template_10_ng_template_10_Template(rf, ctx) { if (rf & 1) {
    i0.ɵɵtemplate(0, PoTableComponent_ng_template_10_ng_template_10_th_0_Template, 6, 36, "th", 117);
} if (rf & 2) {
    const ctx_r187 = i0.ɵɵnextContext(2);
    i0.ɵɵproperty("ngForOf", ctx_r187.mainColumns)("ngForTrackBy", ctx_r187.trackBy);
} }
function PoTableComponent_ng_template_10_ng_template_12_th_0_ng_container_4_Template(rf, ctx) { if (rf & 1) {
    i0.ɵɵelementContainer(0);
} }
function PoTableComponent_ng_template_10_ng_template_12_th_0_Template(rf, ctx) { if (rf & 1) {
    const _r212 = i0.ɵɵgetCurrentView();
    i0.ɵɵelementStart(0, "th", 120);
    i0.ɵɵlistener("click", function PoTableComponent_ng_template_10_ng_template_12_th_0_Template_th_click_0_listener() { const restoredCtx = i0.ɵɵrestoreView(_r212); const column_r208 = restoredCtx.$implicit; const ctx_r211 = i0.ɵɵnextContext(3); return i0.ɵɵresetView(ctx_r211.sortColumn(column_r208)); });
    i0.ɵɵpipe(1, "lowercase");
    i0.ɵɵpipe(2, "titlecase");
    i0.ɵɵelementStart(3, "div", 76);
    i0.ɵɵtemplate(4, PoTableComponent_ng_template_10_ng_template_12_th_0_ng_container_4_Template, 1, 0, "ng-container", 77);
    i0.ɵɵelementEnd()();
} if (rf & 2) {
    const column_r208 = ctx.$implicit;
    const ctx_r207 = i0.ɵɵnextContext(3);
    const _r18 = i0.ɵɵreference(21);
    i0.ɵɵstyleProp("width", column_r208.width)("max-width", column_r208.width)("min-width", column_r208.width);
    i0.ɵɵclassProp("po-clickable", ctx_r207.sort && column_r208.sortable !== false || ctx_r207.hasService)("po-table-header-subtitle", column_r208.type === "subtitle");
    i0.ɵɵproperty("ngClass", i0.ɵɵpureFunction1(26, _c15, ctx_r207.sort && ctx_r207.JSON.stringify(ctx_r207.sortedColumn == null ? null : ctx_r207.sortedColumn.property) === ctx_r207.JSON.stringify(column_r208) && (ctx_r207.sortedColumn.ascending || !ctx_r207.sortedColumn.ascending)))("ngStyle", i0.ɵɵpureFunction1(28, _c22, !ctx_r207.hasItems ? "100%" : "auto"))("pFrozenColumn", column_r208.fixed);
    i0.ɵɵattribute("data-po-table-column-name", i0.ɵɵpipeBind1(1, 22, column_r208.label || i0.ɵɵpipeBind1(2, 24, column_r208.property)));
    i0.ɵɵadvance(3);
    i0.ɵɵclassProp("po-table-header-fixed-inner", ctx_r207.height)("po-table-header-flex-right", column_r208.type === "currency" || column_r208.type === "number")("po-table-header-flex-center", column_r208.type === "subtitle");
    i0.ɵɵadvance(1);
    i0.ɵɵproperty("ngTemplateOutlet", _r18)("ngTemplateOutletContext", i0.ɵɵpureFunction1(30, _c16, column_r208));
} }
function PoTableComponent_ng_template_10_ng_template_12_Template(rf, ctx) { if (rf & 1) {
    i0.ɵɵtemplate(0, PoTableComponent_ng_template_10_ng_template_12_th_0_Template, 5, 32, "th", 119);
} if (rf & 2) {
    const ctx_r189 = i0.ɵɵnextContext(2);
    i0.ɵɵproperty("ngForOf", ctx_r189.mainColumns)("ngForTrackBy", ctx_r189.trackBy);
} }
function PoTableComponent_ng_template_10_th_14_Template(rf, ctx) { if (rf & 1) {
    i0.ɵɵelement(0, "th", 70);
} }
function PoTableComponent_ng_template_10_th_15_Template(rf, ctx) { if (rf & 1) {
    i0.ɵɵelement(0, "th");
} if (rf & 2) {
    const ctx_r191 = i0.ɵɵnextContext(2);
    i0.ɵɵclassProp("po-table-header-single-action", ctx_r191.isSingleAction)("po-table-header-actions", !ctx_r191.isSingleAction);
} }
function PoTableComponent_ng_template_10_tbody_16_Template(rf, ctx) { if (rf & 1) {
    i0.ɵɵelementStart(0, "tbody", 87)(1, "tr", 88)(2, "td", 89)(3, "span");
    i0.ɵɵtext(4);
    i0.ɵɵelementEnd()()()();
} if (rf & 2) {
    const ctx_r192 = i0.ɵɵnextContext(2);
    i0.ɵɵadvance(2);
    i0.ɵɵproperty("colSpan", ctx_r192.columnCount);
    i0.ɵɵadvance(2);
    i0.ɵɵtextInterpolate1(" ", ctx_r192.literals.noData, " ");
} }
function PoTableComponent_ng_template_10_ng_container_17_tbody_1_td_2_ng_container_1_Template(rf, ctx) { if (rf & 1) {
    i0.ɵɵelementContainer(0);
} }
function PoTableComponent_ng_template_10_ng_container_17_tbody_1_td_2_Template(rf, ctx) { if (rf & 1) {
    i0.ɵɵelementStart(0, "td", 67);
    i0.ɵɵtemplate(1, PoTableComponent_ng_template_10_ng_container_17_tbody_1_td_2_ng_container_1_Template, 1, 0, "ng-container", 77);
    i0.ɵɵelementEnd();
} if (rf & 2) {
    const row_r214 = i0.ɵɵnextContext().$implicit;
    const ctx_r216 = i0.ɵɵnextContext(3);
    const _r14 = i0.ɵɵreference(17);
    const _r16 = i0.ɵɵreference(19);
    i0.ɵɵadvance(1);
    i0.ɵɵproperty("ngTemplateOutlet", ctx_r216.singleSelect ? _r14 : _r16)("ngTemplateOutletContext", i0.ɵɵpureFunction1(2, _c16, row_r214));
} }
function PoTableComponent_ng_template_10_ng_container_17_tbody_1_td_3_ng_template_1_Template(rf, ctx) { }
function PoTableComponent_ng_template_10_ng_container_17_tbody_1_td_3_Template(rf, ctx) { if (rf & 1) {
    const _r230 = i0.ɵɵgetCurrentView();
    i0.ɵɵelementStart(0, "td", 95);
    i0.ɵɵlistener("click", function PoTableComponent_ng_template_10_ng_container_17_tbody_1_td_3_Template_td_click_0_listener() { i0.ɵɵrestoreView(_r230); const row_r214 = i0.ɵɵnextContext().$implicit; const ctx_r228 = i0.ɵɵnextContext(3); return i0.ɵɵresetView(ctx_r228.toggleDetail(row_r214)); });
    i0.ɵɵtemplate(1, PoTableComponent_ng_template_10_ng_container_17_tbody_1_td_3_ng_template_1_Template, 0, 0, "ng-template", 96);
    i0.ɵɵelementEnd();
} if (rf & 2) {
    const ctx_r231 = i0.ɵɵnextContext();
    const row_r214 = ctx_r231.$implicit;
    const rowIndex_r215 = ctx_r231.index;
    i0.ɵɵnextContext(3);
    const _r12 = i0.ɵɵreference(15);
    i0.ɵɵadvance(1);
    i0.ɵɵproperty("ngTemplateOutlet", _r12)("ngTemplateOutletContext", i0.ɵɵpureFunction2(2, _c17, row_r214, rowIndex_r215));
} }
function PoTableComponent_ng_template_10_ng_container_17_tbody_1_4_ng_template_0_Template(rf, ctx) { }
function PoTableComponent_ng_template_10_ng_container_17_tbody_1_4_Template(rf, ctx) { if (rf & 1) {
    i0.ɵɵtemplate(0, PoTableComponent_ng_template_10_ng_container_17_tbody_1_4_ng_template_0_Template, 0, 0, "ng-template", 96);
} if (rf & 2) {
    const ctx_r233 = i0.ɵɵnextContext();
    const row_r214 = ctx_r233.$implicit;
    const rowIndex_r215 = ctx_r233.index;
    i0.ɵɵnextContext(3);
    const _r24 = i0.ɵɵreference(27);
    i0.ɵɵproperty("ngTemplateOutlet", _r24)("ngTemplateOutletContext", i0.ɵɵpureFunction2(2, _c17, row_r214, rowIndex_r215));
} }
function PoTableComponent_ng_template_10_ng_container_17_tbody_1_td_5_ng_template_1_Template(rf, ctx) { }
function PoTableComponent_ng_template_10_ng_container_17_tbody_1_td_5_Template(rf, ctx) { if (rf & 1) {
    const _r237 = i0.ɵɵgetCurrentView();
    i0.ɵɵelementStart(0, "td", 95);
    i0.ɵɵlistener("click", function PoTableComponent_ng_template_10_ng_container_17_tbody_1_td_5_Template_td_click_0_listener() { i0.ɵɵrestoreView(_r237); const row_r214 = i0.ɵɵnextContext().$implicit; const ctx_r235 = i0.ɵɵnextContext(3); return i0.ɵɵresetView(ctx_r235.toggleDetail(row_r214)); });
    i0.ɵɵtemplate(1, PoTableComponent_ng_template_10_ng_container_17_tbody_1_td_5_ng_template_1_Template, 0, 0, "ng-template", 96);
    i0.ɵɵelementEnd();
} if (rf & 2) {
    const ctx_r238 = i0.ɵɵnextContext();
    const row_r214 = ctx_r238.$implicit;
    const rowIndex_r215 = ctx_r238.index;
    i0.ɵɵnextContext(3);
    const _r12 = i0.ɵɵreference(15);
    i0.ɵɵadvance(1);
    i0.ɵɵproperty("ngTemplateOutlet", _r12)("ngTemplateOutletContext", i0.ɵɵpureFunction2(2, _c17, row_r214, rowIndex_r215));
} }
function PoTableComponent_ng_template_10_ng_container_17_tbody_1_td_6_span_2_ng_container_1_Template(rf, ctx) { if (rf & 1) {
    i0.ɵɵelementContainer(0);
} }
function PoTableComponent_ng_template_10_ng_container_17_tbody_1_td_6_span_2_Template(rf, ctx) { if (rf & 1) {
    i0.ɵɵelementStart(0, "span");
    i0.ɵɵtemplate(1, PoTableComponent_ng_template_10_ng_container_17_tbody_1_td_6_span_2_ng_container_1_Template, 1, 0, "ng-container", 77);
    i0.ɵɵelementEnd();
} if (rf & 2) {
    const column_r239 = i0.ɵɵnextContext().$implicit;
    const row_r214 = i0.ɵɵnextContext().$implicit;
    const ctx_r241 = i0.ɵɵnextContext(3);
    i0.ɵɵadvance(1);
    i0.ɵɵproperty("ngTemplateOutlet", ctx_r241.getTemplate(column_r239))("ngTemplateOutletContext", i0.ɵɵpureFunction1(2, _c16, ctx_r241.getCellData(row_r214, column_r239)));
} }
function PoTableComponent_ng_template_10_ng_container_17_tbody_1_td_6_span_3_ng_container_1_Template(rf, ctx) { if (rf & 1) {
    i0.ɵɵelementContainer(0);
} }
function PoTableComponent_ng_template_10_ng_container_17_tbody_1_td_6_span_3_Template(rf, ctx) { if (rf & 1) {
    i0.ɵɵelementStart(0, "span");
    i0.ɵɵtemplate(1, PoTableComponent_ng_template_10_ng_container_17_tbody_1_td_6_span_3_ng_container_1_Template, 1, 0, "ng-container", 77);
    i0.ɵɵelementEnd();
} if (rf & 2) {
    const column_r239 = i0.ɵɵnextContext().$implicit;
    const row_r214 = i0.ɵɵnextContext().$implicit;
    const ctx_r242 = i0.ɵɵnextContext(3);
    i0.ɵɵadvance(1);
    i0.ɵɵproperty("ngTemplateOutlet", ctx_r242.tableCellTemplate == null ? null : ctx_r242.tableCellTemplate.templateRef)("ngTemplateOutletContext", i0.ɵɵpureFunction2(2, _c18, row_r214, column_r239));
} }
function PoTableComponent_ng_template_10_ng_container_17_tbody_1_td_6_span_4_Template(rf, ctx) { if (rf & 1) {
    i0.ɵɵelementStart(0, "span");
    i0.ɵɵtext(1);
    i0.ɵɵelementEnd();
} if (rf & 2) {
    const column_r239 = i0.ɵɵnextContext().$implicit;
    const row_r214 = i0.ɵɵnextContext().$implicit;
    const ctx_r243 = i0.ɵɵnextContext(3);
    i0.ɵɵadvance(1);
    i0.ɵɵtextInterpolate1(" ", ctx_r243.getBooleanLabel(ctx_r243.getCellData(row_r214, column_r239), column_r239), " ");
} }
function PoTableComponent_ng_template_10_ng_container_17_tbody_1_td_6_span_5_Template(rf, ctx) { if (rf & 1) {
    i0.ɵɵelementStart(0, "span");
    i0.ɵɵtext(1);
    i0.ɵɵpipe(2, "currency");
    i0.ɵɵelementEnd();
} if (rf & 2) {
    const column_r239 = i0.ɵɵnextContext().$implicit;
    const row_r214 = i0.ɵɵnextContext().$implicit;
    const ctx_r244 = i0.ɵɵnextContext(3);
    i0.ɵɵadvance(1);
    i0.ɵɵtextInterpolate1(" ", i0.ɵɵpipeBind4(2, 1, ctx_r244.getCellData(row_r214, column_r239), column_r239.format, "symbol", "1.2-2"), " ");
} }
function PoTableComponent_ng_template_10_ng_container_17_tbody_1_td_6_span_6_Template(rf, ctx) { if (rf & 1) {
    i0.ɵɵelementStart(0, "span");
    i0.ɵɵtext(1);
    i0.ɵɵpipe(2, "date");
    i0.ɵɵelementEnd();
} if (rf & 2) {
    const column_r239 = i0.ɵɵnextContext().$implicit;
    const row_r214 = i0.ɵɵnextContext().$implicit;
    const ctx_r245 = i0.ɵɵnextContext(3);
    i0.ɵɵadvance(1);
    i0.ɵɵtextInterpolate1(" ", i0.ɵɵpipeBind2(2, 1, ctx_r245.getCellData(row_r214, column_r239), column_r239.format || "dd/MM/yyyy"), " ");
} }
function PoTableComponent_ng_template_10_ng_container_17_tbody_1_td_6_span_7_Template(rf, ctx) { if (rf & 1) {
    i0.ɵɵelementStart(0, "span");
    i0.ɵɵtext(1);
    i0.ɵɵpipe(2, "po_time");
    i0.ɵɵelementEnd();
} if (rf & 2) {
    const column_r239 = i0.ɵɵnextContext().$implicit;
    const row_r214 = i0.ɵɵnextContext().$implicit;
    const ctx_r246 = i0.ɵɵnextContext(3);
    i0.ɵɵadvance(1);
    i0.ɵɵtextInterpolate1(" ", i0.ɵɵpipeBind2(2, 1, ctx_r246.getCellData(row_r214, column_r239), column_r239.format || "HH:mm:ss.ffffff"), " ");
} }
function PoTableComponent_ng_template_10_ng_container_17_tbody_1_td_6_span_8_Template(rf, ctx) { if (rf & 1) {
    i0.ɵɵelementStart(0, "span");
    i0.ɵɵtext(1);
    i0.ɵɵpipe(2, "date");
    i0.ɵɵelementEnd();
} if (rf & 2) {
    const column_r239 = i0.ɵɵnextContext().$implicit;
    const row_r214 = i0.ɵɵnextContext().$implicit;
    const ctx_r247 = i0.ɵɵnextContext(3);
    i0.ɵɵadvance(1);
    i0.ɵɵtextInterpolate1(" ", i0.ɵɵpipeBind2(2, 1, ctx_r247.getCellData(row_r214, column_r239), column_r239.format || "dd/MM/yyyy HH:mm:ss"), " ");
} }
function PoTableComponent_ng_template_10_ng_container_17_tbody_1_td_6_span_9_Template(rf, ctx) { if (rf & 1) {
    i0.ɵɵelementStart(0, "span");
    i0.ɵɵtext(1);
    i0.ɵɵelementEnd();
} if (rf & 2) {
    const column_r239 = i0.ɵɵnextContext().$implicit;
    const row_r214 = i0.ɵɵnextContext().$implicit;
    const ctx_r248 = i0.ɵɵnextContext(3);
    i0.ɵɵadvance(1);
    i0.ɵɵtextInterpolate1(" ", ctx_r248.formatNumber(ctx_r248.getCellData(row_r214, column_r239), column_r239.format), " ");
} }
function PoTableComponent_ng_template_10_ng_container_17_tbody_1_td_6_po_table_column_link_10_Template(rf, ctx) { if (rf & 1) {
    const _r274 = i0.ɵɵgetCurrentView();
    i0.ɵɵelementStart(0, "po-table-column-link", 103);
    i0.ɵɵlistener("click", function PoTableComponent_ng_template_10_ng_container_17_tbody_1_td_6_po_table_column_link_10_Template_po_table_column_link_click_0_listener($event) { i0.ɵɵrestoreView(_r274); const column_r239 = i0.ɵɵnextContext().$implicit; const row_r214 = i0.ɵɵnextContext().$implicit; const ctx_r272 = i0.ɵɵnextContext(3); return i0.ɵɵresetView(ctx_r272.onClickLink($event, row_r214, column_r239)); });
    i0.ɵɵelementEnd();
} if (rf & 2) {
    const column_r239 = i0.ɵɵnextContext().$implicit;
    const row_r214 = i0.ɵɵnextContext().$implicit;
    const ctx_r249 = i0.ɵɵnextContext(3);
    i0.ɵɵproperty("p-action", column_r239.action)("p-disabled", ctx_r249.checkDisabled(row_r214, column_r239))("p-link", row_r214[column_r239.link])("p-row", row_r214)("p-value", ctx_r249.getCellData(row_r214, column_r239));
} }
function PoTableComponent_ng_template_10_ng_container_17_tbody_1_td_6_po_table_column_icon_11_Template(rf, ctx) { if (rf & 1) {
    i0.ɵɵelement(0, "po-table-column-icon", 104);
} if (rf & 2) {
    const column_r239 = i0.ɵɵnextContext().$implicit;
    const row_r214 = i0.ɵɵnextContext().$implicit;
    const ctx_r250 = i0.ɵɵnextContext(3);
    i0.ɵɵproperty("p-column", column_r239)("p-icons", ctx_r250.getColumnIcons(row_r214, column_r239))("p-row", row_r214);
} }
function PoTableComponent_ng_template_10_ng_container_17_tbody_1_td_6_span_12_Template(rf, ctx) { if (rf & 1) {
    i0.ɵɵelementStart(0, "span");
    i0.ɵɵelement(1, "po-table-subtitle-circle", 105);
    i0.ɵɵelementEnd();
} if (rf & 2) {
    const column_r239 = i0.ɵɵnextContext().$implicit;
    const row_r214 = i0.ɵɵnextContext().$implicit;
    const ctx_r251 = i0.ɵɵnextContext(3);
    i0.ɵɵadvance(1);
    i0.ɵɵproperty("p-subtitle", ctx_r251.getSubtitleColumn(row_r214, column_r239));
} }
function PoTableComponent_ng_template_10_ng_container_17_tbody_1_td_6_span_13_Template(rf, ctx) { if (rf & 1) {
    i0.ɵɵelementStart(0, "span");
    i0.ɵɵelement(1, "po-table-column-label", 106);
    i0.ɵɵelementEnd();
} if (rf & 2) {
    const column_r239 = i0.ɵɵnextContext().$implicit;
    const row_r214 = i0.ɵɵnextContext().$implicit;
    const ctx_r252 = i0.ɵɵnextContext(3);
    i0.ɵɵadvance(1);
    i0.ɵɵproperty("p-value", ctx_r252.getColumnLabel(row_r214, column_r239));
} }
function PoTableComponent_ng_template_10_ng_container_17_tbody_1_td_6_span_14_Template(rf, ctx) { if (rf & 1) {
    i0.ɵɵelementStart(0, "span");
    i0.ɵɵtext(1);
    i0.ɵɵelementEnd();
} if (rf & 2) {
    const column_r239 = i0.ɵɵnextContext().$implicit;
    const row_r214 = i0.ɵɵnextContext().$implicit;
    const ctx_r253 = i0.ɵɵnextContext(3);
    i0.ɵɵadvance(1);
    i0.ɵɵtextInterpolate(ctx_r253.getCellData(row_r214, column_r239));
} }
function PoTableComponent_ng_template_10_ng_container_17_tbody_1_td_6_Template(rf, ctx) { if (rf & 1) {
    const _r287 = i0.ɵɵgetCurrentView();
    i0.ɵɵelementStart(0, "td", 123);
    i0.ɵɵlistener("click", function PoTableComponent_ng_template_10_ng_container_17_tbody_1_td_6_Template_td_click_0_listener() { i0.ɵɵrestoreView(_r287); const row_r214 = i0.ɵɵnextContext().$implicit; const ctx_r286 = i0.ɵɵnextContext(3); return i0.ɵɵresetView(ctx_r286.hasSelectableRow() ? ctx_r286.selectRow(row_r214) : "javascript:;"); });
    i0.ɵɵelementStart(1, "div", 98);
    i0.ɵɵlistener("mouseenter", function PoTableComponent_ng_template_10_ng_container_17_tbody_1_td_6_Template_div_mouseenter_1_listener($event) { const restoredCtx = i0.ɵɵrestoreView(_r287); const column_r239 = restoredCtx.$implicit; const row_r214 = i0.ɵɵnextContext().$implicit; const ctx_r289 = i0.ɵɵnextContext(3); return i0.ɵɵresetView(ctx_r289.tooltipMouseEnter($event, column_r239, row_r214)); })("mouseleave", function PoTableComponent_ng_template_10_ng_container_17_tbody_1_td_6_Template_div_mouseleave_1_listener() { i0.ɵɵrestoreView(_r287); const ctx_r291 = i0.ɵɵnextContext(4); return i0.ɵɵresetView(ctx_r291.tooltipMouseLeave()); });
    i0.ɵɵtemplate(2, PoTableComponent_ng_template_10_ng_container_17_tbody_1_td_6_span_2_Template, 2, 4, "span", 99);
    i0.ɵɵtemplate(3, PoTableComponent_ng_template_10_ng_container_17_tbody_1_td_6_span_3_Template, 2, 5, "span", 99);
    i0.ɵɵtemplate(4, PoTableComponent_ng_template_10_ng_container_17_tbody_1_td_6_span_4_Template, 2, 1, "span", 99);
    i0.ɵɵtemplate(5, PoTableComponent_ng_template_10_ng_container_17_tbody_1_td_6_span_5_Template, 3, 6, "span", 99);
    i0.ɵɵtemplate(6, PoTableComponent_ng_template_10_ng_container_17_tbody_1_td_6_span_6_Template, 3, 4, "span", 99);
    i0.ɵɵtemplate(7, PoTableComponent_ng_template_10_ng_container_17_tbody_1_td_6_span_7_Template, 3, 4, "span", 99);
    i0.ɵɵtemplate(8, PoTableComponent_ng_template_10_ng_container_17_tbody_1_td_6_span_8_Template, 3, 4, "span", 99);
    i0.ɵɵtemplate(9, PoTableComponent_ng_template_10_ng_container_17_tbody_1_td_6_span_9_Template, 2, 1, "span", 99);
    i0.ɵɵtemplate(10, PoTableComponent_ng_template_10_ng_container_17_tbody_1_td_6_po_table_column_link_10_Template, 1, 5, "po-table-column-link", 100);
    i0.ɵɵtemplate(11, PoTableComponent_ng_template_10_ng_container_17_tbody_1_td_6_po_table_column_icon_11_Template, 1, 3, "po-table-column-icon", 101);
    i0.ɵɵtemplate(12, PoTableComponent_ng_template_10_ng_container_17_tbody_1_td_6_span_12_Template, 2, 1, "span", 99);
    i0.ɵɵtemplate(13, PoTableComponent_ng_template_10_ng_container_17_tbody_1_td_6_span_13_Template, 2, 1, "span", 99);
    i0.ɵɵtemplate(14, PoTableComponent_ng_template_10_ng_container_17_tbody_1_td_6_span_14_Template, 2, 1, "span", 102);
    i0.ɵɵelementEnd()();
} if (rf & 2) {
    const column_r239 = ctx.$implicit;
    const row_r214 = i0.ɵɵnextContext().$implicit;
    const ctx_r220 = i0.ɵɵnextContext(3);
    i0.ɵɵstyleProp("width", column_r239.width)("max-width", column_r239.width)("min-width", column_r239.width);
    i0.ɵɵclassProp("po-table-column", column_r239.type !== "icon")("po-table-column-right", column_r239.type === "currency" || column_r239.type === "number")("po-table-column-center", column_r239.type === "subtitle")("po-table-column-icons", column_r239.type === "icon");
    i0.ɵɵproperty("ngClass", ctx_r220.getClassColor(row_r214, column_r239))("pFrozenColumn", column_r239.fixed);
    i0.ɵɵadvance(1);
    i0.ɵɵproperty("ngSwitch", column_r239.type)("p-tooltip", ctx_r220.tooltipText)("p-append-in-body", true);
    i0.ɵɵadvance(1);
    i0.ɵɵproperty("ngSwitchCase", "columnTemplate");
    i0.ɵɵadvance(1);
    i0.ɵɵproperty("ngSwitchCase", "cellTemplate");
    i0.ɵɵadvance(1);
    i0.ɵɵproperty("ngSwitchCase", "boolean");
    i0.ɵɵadvance(1);
    i0.ɵɵproperty("ngSwitchCase", "currency");
    i0.ɵɵadvance(1);
    i0.ɵɵproperty("ngSwitchCase", "date");
    i0.ɵɵadvance(1);
    i0.ɵɵproperty("ngSwitchCase", "time");
    i0.ɵɵadvance(1);
    i0.ɵɵproperty("ngSwitchCase", "dateTime");
    i0.ɵɵadvance(1);
    i0.ɵɵproperty("ngSwitchCase", "number");
    i0.ɵɵadvance(1);
    i0.ɵɵproperty("ngSwitchCase", "link");
    i0.ɵɵadvance(1);
    i0.ɵɵproperty("ngSwitchCase", "icon");
    i0.ɵɵadvance(1);
    i0.ɵɵproperty("ngSwitchCase", "subtitle");
    i0.ɵɵadvance(1);
    i0.ɵɵproperty("ngSwitchCase", "label");
} }
function PoTableComponent_ng_template_10_ng_container_17_tbody_1_td_7_ng_template_1_Template(rf, ctx) { }
function PoTableComponent_ng_template_10_ng_container_17_tbody_1_td_7_Template(rf, ctx) { if (rf & 1) {
    const _r296 = i0.ɵɵgetCurrentView();
    i0.ɵɵelementStart(0, "td", 95);
    i0.ɵɵlistener("click", function PoTableComponent_ng_template_10_ng_container_17_tbody_1_td_7_Template_td_click_0_listener() { i0.ɵɵrestoreView(_r296); const row_r214 = i0.ɵɵnextContext().$implicit; const ctx_r294 = i0.ɵɵnextContext(3); return i0.ɵɵresetView(ctx_r294.toggleDetail(row_r214)); });
    i0.ɵɵtemplate(1, PoTableComponent_ng_template_10_ng_container_17_tbody_1_td_7_ng_template_1_Template, 0, 0, "ng-template", 96);
    i0.ɵɵelementEnd();
} if (rf & 2) {
    const ctx_r297 = i0.ɵɵnextContext();
    const row_r214 = ctx_r297.$implicit;
    const rowIndex_r215 = ctx_r297.index;
    i0.ɵɵnextContext(3);
    const _r12 = i0.ɵɵreference(15);
    i0.ɵɵadvance(1);
    i0.ɵɵproperty("ngTemplateOutlet", _r12)("ngTemplateOutletContext", i0.ɵɵpureFunction2(2, _c17, row_r214, rowIndex_r215));
} }
function PoTableComponent_ng_template_10_ng_container_17_tbody_1_8_ng_template_0_Template(rf, ctx) { }
function PoTableComponent_ng_template_10_ng_container_17_tbody_1_8_Template(rf, ctx) { if (rf & 1) {
    i0.ɵɵtemplate(0, PoTableComponent_ng_template_10_ng_container_17_tbody_1_8_ng_template_0_Template, 0, 0, "ng-template", 96);
} if (rf & 2) {
    const ctx_r299 = i0.ɵɵnextContext();
    const row_r214 = ctx_r299.$implicit;
    const rowIndex_r215 = ctx_r299.index;
    i0.ɵɵnextContext(3);
    const _r24 = i0.ɵɵreference(27);
    i0.ɵɵproperty("ngTemplateOutlet", _r24)("ngTemplateOutletContext", i0.ɵɵpureFunction2(2, _c17, row_r214, rowIndex_r215));
} }
function PoTableComponent_ng_template_10_ng_container_17_tbody_1_tr_9_ng_template_2_Template(rf, ctx) { }
function PoTableComponent_ng_template_10_ng_container_17_tbody_1_tr_9_Template(rf, ctx) { if (rf & 1) {
    i0.ɵɵelementStart(0, "tr")(1, "td", 107);
    i0.ɵɵtemplate(2, PoTableComponent_ng_template_10_ng_container_17_tbody_1_tr_9_ng_template_2_Template, 0, 0, "ng-template", 96);
    i0.ɵɵelementEnd()();
} if (rf & 2) {
    const ctx_r301 = i0.ɵɵnextContext();
    const row_r214 = ctx_r301.$implicit;
    const rowIndex_r215 = ctx_r301.index;
    const ctx_r223 = i0.ɵɵnextContext(3);
    i0.ɵɵadvance(1);
    i0.ɵɵproperty("colSpan", ctx_r223.columnCountForMasterDetail);
    i0.ɵɵadvance(1);
    i0.ɵɵproperty("ngTemplateOutlet", ctx_r223.tableRowTemplate.templateRef)("ngTemplateOutletContext", i0.ɵɵpureFunction2(3, _c19, row_r214, rowIndex_r215));
} }
function PoTableComponent_ng_template_10_ng_container_17_tbody_1_tr_10_Template(rf, ctx) { if (rf & 1) {
    const _r303 = i0.ɵɵgetCurrentView();
    i0.ɵɵelementStart(0, "tr")(1, "td", 108)(2, "po-table-detail", 109);
    i0.ɵɵlistener("p-select-row", function PoTableComponent_ng_template_10_ng_container_17_tbody_1_tr_10_Template_po_table_detail_p_select_row_2_listener($event) { i0.ɵɵrestoreView(_r303); const ctx_r302 = i0.ɵɵnextContext(4); return i0.ɵɵresetView(ctx_r302.selectDetailRow($event)); });
    i0.ɵɵelementEnd()()();
} if (rf & 2) {
    const row_r214 = i0.ɵɵnextContext().$implicit;
    const ctx_r224 = i0.ɵɵnextContext(3);
    i0.ɵɵadvance(1);
    i0.ɵɵproperty("colSpan", ctx_r224.columnCountForMasterDetail);
    i0.ɵɵadvance(1);
    i0.ɵɵproperty("p-selectable", ctx_r224.selectable && !ctx_r224.detailHideSelect)("p-detail", ctx_r224.columnMasterDetail.detail)("p-items", row_r214[ctx_r224.nameColumnDetail]);
} }
function PoTableComponent_ng_template_10_ng_container_17_tbody_1_Template(rf, ctx) { if (rf & 1) {
    i0.ɵɵelementStart(0, "tbody", 87)(1, "tr", 91);
    i0.ɵɵtemplate(2, PoTableComponent_ng_template_10_ng_container_17_tbody_1_td_2_Template, 2, 4, "td", 92);
    i0.ɵɵtemplate(3, PoTableComponent_ng_template_10_ng_container_17_tbody_1_td_3_Template, 2, 5, "td", 93);
    i0.ɵɵtemplate(4, PoTableComponent_ng_template_10_ng_container_17_tbody_1_4_Template, 1, 5, null, 43);
    i0.ɵɵtemplate(5, PoTableComponent_ng_template_10_ng_container_17_tbody_1_td_5_Template, 2, 5, "td", 93);
    i0.ɵɵtemplate(6, PoTableComponent_ng_template_10_ng_container_17_tbody_1_td_6_Template, 15, 31, "td", 122);
    i0.ɵɵtemplate(7, PoTableComponent_ng_template_10_ng_container_17_tbody_1_td_7_Template, 2, 5, "td", 93);
    i0.ɵɵtemplate(8, PoTableComponent_ng_template_10_ng_container_17_tbody_1_8_Template, 1, 5, null, 43);
    i0.ɵɵelementEnd();
    i0.ɵɵtemplate(9, PoTableComponent_ng_template_10_ng_container_17_tbody_1_tr_9_Template, 3, 6, "tr", 43);
    i0.ɵɵtemplate(10, PoTableComponent_ng_template_10_ng_container_17_tbody_1_tr_10_Template, 3, 4, "tr", 43);
    i0.ɵɵelementEnd();
} if (rf & 2) {
    const row_r214 = ctx.$implicit;
    const rowIndex_r215 = ctx.index;
    const ctx_r213 = i0.ɵɵnextContext(3);
    i0.ɵɵadvance(1);
    i0.ɵɵclassProp("po-table-row-active", row_r214.$selected || row_r214.$selected === null && ctx_r213.selectable);
    i0.ɵɵadvance(1);
    i0.ɵɵproperty("ngIf", ctx_r213.selectable);
    i0.ɵɵadvance(1);
    i0.ɵɵproperty("ngIf", ctx_r213.columnMasterDetail && !ctx_r213.hideDetail && !ctx_r213.hasRowTemplate);
    i0.ɵɵadvance(1);
    i0.ɵɵproperty("ngIf", !ctx_r213.actionRight && (ctx_r213.visibleActions.length > 1 || ctx_r213.isSingleAction));
    i0.ɵɵadvance(1);
    i0.ɵɵproperty("ngIf", ctx_r213.hasRowTemplate && !ctx_r213.hasRowTemplateWithArrowDirectionRight);
    i0.ɵɵadvance(1);
    i0.ɵɵproperty("ngForOf", ctx_r213.mainColumns)("ngForTrackBy", ctx_r213.trackBy);
    i0.ɵɵadvance(1);
    i0.ɵɵproperty("ngIf", ctx_r213.hasRowTemplateWithArrowDirectionRight);
    i0.ɵɵadvance(1);
    i0.ɵɵproperty("ngIf", ctx_r213.actionRight);
    i0.ɵɵadvance(1);
    i0.ɵɵproperty("ngIf", ctx_r213.hasMainColumns && ctx_r213.hasRowTemplate && row_r214.$showDetail && ctx_r213.isShowRowTemplate(row_r214, rowIndex_r215));
    i0.ɵɵadvance(1);
    i0.ɵɵproperty("ngIf", ctx_r213.hasMainColumns && ctx_r213.isShowMasterDetail(row_r214));
} }
function PoTableComponent_ng_template_10_ng_container_17_Template(rf, ctx) { if (rf & 1) {
    i0.ɵɵelementContainerStart(0);
    i0.ɵɵtemplate(1, PoTableComponent_ng_template_10_ng_container_17_tbody_1_Template, 11, 12, "tbody", 121);
    i0.ɵɵelementContainerEnd();
} if (rf & 2) {
    const ctx_r193 = i0.ɵɵnextContext(2);
    i0.ɵɵadvance(1);
    i0.ɵɵproperty("cdkVirtualForOf", ctx_r193.filteredItems)("cdkVirtualForTrackBy", ctx_r193.trackBy);
} }
const _c23 = function (a0) { return { "table-layout": a0 }; };
function PoTableComponent_ng_template_10_Template(rf, ctx) { if (rf & 1) {
    const _r306 = i0.ɵɵgetCurrentView();
    i0.ɵɵelementStart(0, "cdk-virtual-scroll-viewport", 110, 111)(2, "table", 112)(3, "thead", 113)(4, "tr", 114);
    i0.ɵɵlistener("cdkDropListDropped", function PoTableComponent_ng_template_10_Template_tr_cdkDropListDropped_4_listener($event) { i0.ɵɵrestoreView(_r306); const ctx_r305 = i0.ɵɵnextContext(); return i0.ɵɵresetView(ctx_r305.drop($event)); });
    i0.ɵɵtemplate(5, PoTableComponent_ng_template_10_th_5_Template, 3, 5, "th", 58);
    i0.ɵɵtemplate(6, PoTableComponent_ng_template_10_th_6_Template, 1, 0, "th", 59);
    i0.ɵɵtemplate(7, PoTableComponent_ng_template_10_th_7_Template, 2, 4, "th", 60);
    i0.ɵɵtemplate(8, PoTableComponent_ng_template_10_th_8_Template, 3, 3, "th", 61);
    i0.ɵɵtemplate(9, PoTableComponent_ng_template_10_ng_container_9_Template, 1, 0, "ng-container", 62);
    i0.ɵɵtemplate(10, PoTableComponent_ng_template_10_ng_template_10_Template, 1, 2, "ng-template", null, 115, i0.ɵɵtemplateRefExtractor);
    i0.ɵɵtemplate(12, PoTableComponent_ng_template_10_ng_template_12_Template, 1, 2, "ng-template", null, 116, i0.ɵɵtemplateRefExtractor);
    i0.ɵɵtemplate(14, PoTableComponent_ng_template_10_th_14_Template, 1, 0, "th", 59);
    i0.ɵɵtemplate(15, PoTableComponent_ng_template_10_th_15_Template, 1, 4, "th", 65);
    i0.ɵɵelementEnd()();
    i0.ɵɵtemplate(16, PoTableComponent_ng_template_10_tbody_16_Template, 5, 2, "tbody", 66);
    i0.ɵɵtemplate(17, PoTableComponent_ng_template_10_ng_container_17_Template, 2, 2, "ng-container", 43);
    i0.ɵɵelementEnd()();
} if (rf & 2) {
    const _r186 = i0.ɵɵreference(11);
    const _r188 = i0.ɵɵreference(13);
    const ctx_r10 = i0.ɵɵnextContext();
    i0.ɵɵstyleProp("height", ctx_r10.heightTableContainer, "px");
    i0.ɵɵproperty("itemSize", ctx_r10.itemSize)("minBufferPx", ctx_r10.heightTableContainer < 100 ? 100 : ctx_r10.heightTableContainer)("maxBufferPx", ctx_r10.heightTableContainer < 200 ? 200 : ctx_r10.heightTableContainer);
    i0.ɵɵadvance(2);
    i0.ɵɵproperty("ngClass", i0.ɵɵpureFunction3(23, _c20, ctx_r10.selectable || ctx_r10.sort, ctx_r10.selectable, ctx_r10.striped))("ngStyle", i0.ɵɵpureFunction1(27, _c23, !ctx_r10.hasItems ? "fixed" : "auto"));
    i0.ɵɵattribute("p-spacing", ctx_r10.spacing);
    i0.ɵɵadvance(1);
    i0.ɵɵstyleProp("top", ctx_r10.inverseOfTranslation);
    i0.ɵɵadvance(1);
    i0.ɵɵclassProp("po-table-header", !ctx_r10.height);
    i0.ɵɵadvance(1);
    i0.ɵɵproperty("ngIf", ctx_r10.hasSelectableColumn);
    i0.ɵɵadvance(1);
    i0.ɵɵproperty("ngIf", (ctx_r10.hasMasterDetailColumn || ctx_r10.hasRowTemplate) && !ctx_r10.hasRowTemplateWithArrowDirectionRight);
    i0.ɵɵadvance(1);
    i0.ɵɵproperty("ngIf", !ctx_r10.actionRight && (ctx_r10.visibleActions.length > 1 || ctx_r10.isSingleAction));
    i0.ɵɵadvance(1);
    i0.ɵɵproperty("ngIf", !ctx_r10.hasMainColumns);
    i0.ɵɵadvance(1);
    i0.ɵɵproperty("ngIf", ctx_r10.isDraggable || ctx_r10.hasSomeFixed())("ngIfThen", _r186)("ngIfElse", _r188);
    i0.ɵɵadvance(5);
    i0.ɵɵproperty("ngIf", ctx_r10.hasRowTemplateWithArrowDirectionRight && (ctx_r10.hasVisibleActions || ctx_r10.hideColumnsManager));
    i0.ɵɵadvance(1);
    i0.ɵɵproperty("ngIf", ctx_r10.hasVisibleActions && ctx_r10.actionRight && (ctx_r10.visibleActions.length > 1 || ctx_r10.isSingleAction));
    i0.ɵɵadvance(1);
    i0.ɵɵproperty("ngIf", !ctx_r10.hasItems || !ctx_r10.hasMainColumns);
    i0.ɵɵadvance(1);
    i0.ɵɵproperty("ngIf", ctx_r10.hasMainColumns);
} }
function PoTableComponent_ng_template_14_span_0_Template(rf, ctx) { if (rf & 1) {
    i0.ɵɵelement(0, "span", 125);
} if (rf & 2) {
    const row_r307 = i0.ɵɵnextContext().row;
    i0.ɵɵclassProp("po-icon-arrow-up", row_r307.$showDetail)("po-icon-arrow-down", !row_r307.$showDetail);
} }
function PoTableComponent_ng_template_14_Template(rf, ctx) { if (rf & 1) {
    i0.ɵɵtemplate(0, PoTableComponent_ng_template_14_span_0_Template, 1, 4, "span", 124);
} if (rf & 2) {
    const row_r307 = ctx.row;
    const rowIndex_r308 = ctx.rowIndex;
    const ctx_r13 = i0.ɵɵnextContext();
    i0.ɵɵproperty("ngIf", ctx_r13.containsMasterDetail(row_r307) && !ctx_r13.hasRowTemplate || ctx_r13.isShowRowTemplate(row_r307, rowIndex_r308) && ctx_r13.hasRowTemplate);
} }
function PoTableComponent_ng_template_16_Template(rf, ctx) { if (rf & 1) {
    const _r313 = i0.ɵɵgetCurrentView();
    i0.ɵɵelementStart(0, "po-radio", 126);
    i0.ɵɵlistener("p-change-selected", function PoTableComponent_ng_template_16_Template_po_radio_p_change_selected_0_listener() { const restoredCtx = i0.ɵɵrestoreView(_r313); const row_r311 = restoredCtx.$implicit; const ctx_r312 = i0.ɵɵnextContext(); return i0.ɵɵresetView(ctx_r312.selectRow(row_r311)); });
    i0.ɵɵelementEnd();
} if (rf & 2) {
    const row_r311 = ctx.$implicit;
    const ctx_r15 = i0.ɵɵnextContext();
    i0.ɵɵproperty("name", ctx_r15.idRadio)("p-checked", row_r311.$selected);
} }
function PoTableComponent_ng_template_18_Template(rf, ctx) { if (rf & 1) {
    const _r316 = i0.ɵɵgetCurrentView();
    i0.ɵɵelementStart(0, "po-checkbox", 127);
    i0.ɵɵlistener("p-change", function PoTableComponent_ng_template_18_Template_po_checkbox_p_change_0_listener() { const restoredCtx = i0.ɵɵrestoreView(_r316); const row_r314 = restoredCtx.$implicit; const ctx_r315 = i0.ɵɵnextContext(); return i0.ɵɵresetView(ctx_r315.selectable ? ctx_r315.selectRow(row_r314) : "javascript:;"); });
    i0.ɵɵelementEnd();
} if (rf & 2) {
    const row_r314 = ctx.$implicit;
    i0.ɵɵproperty("p-checkboxValue", row_r314.$selected);
} }
function PoTableComponent_ng_template_20_span_4_ng_container_1_Template(rf, ctx) { if (rf & 1) {
    i0.ɵɵelementContainerStart(0);
    i0.ɵɵnamespaceSVG();
    i0.ɵɵelementStart(1, "svg", 131);
    i0.ɵɵelement(2, "path", 132);
    i0.ɵɵelementEnd();
    i0.ɵɵelementContainerEnd();
} }
function PoTableComponent_ng_template_20_span_4_ng_container_2_Template(rf, ctx) { if (rf & 1) {
    i0.ɵɵelementContainerStart(0);
    i0.ɵɵnamespaceSVG();
    i0.ɵɵelementStart(1, "svg", 131);
    i0.ɵɵelement(2, "path", 133);
    i0.ɵɵelementEnd();
    i0.ɵɵelementContainerEnd();
} }
function PoTableComponent_ng_template_20_span_4_ng_container_3_Template(rf, ctx) { if (rf & 1) {
    i0.ɵɵelementContainerStart(0);
    i0.ɵɵnamespaceSVG();
    i0.ɵɵelementStart(1, "svg", 131);
    i0.ɵɵelement(2, "path", 134);
    i0.ɵɵelementEnd();
    i0.ɵɵelementContainerEnd();
} }
function PoTableComponent_ng_template_20_span_4_Template(rf, ctx) { if (rf & 1) {
    i0.ɵɵelementStart(0, "span");
    i0.ɵɵtemplate(1, PoTableComponent_ng_template_20_span_4_ng_container_1_Template, 3, 0, "ng-container", 43);
    i0.ɵɵtemplate(2, PoTableComponent_ng_template_20_span_4_ng_container_2_Template, 3, 0, "ng-container", 43);
    i0.ɵɵtemplate(3, PoTableComponent_ng_template_20_span_4_ng_container_3_Template, 3, 0, "ng-container", 43);
    i0.ɵɵelementEnd();
} if (rf & 2) {
    const column_r317 = i0.ɵɵnextContext().$implicit;
    const ctx_r319 = i0.ɵɵnextContext();
    i0.ɵɵclassProp("po-table-header-icon-unselected", ctx_r319.JSON.stringify(ctx_r319.sortedColumn == null ? null : ctx_r319.sortedColumn.property) !== ctx_r319.JSON.stringify(column_r317))("po-table-header-icon-descending", ctx_r319.JSON.stringify(ctx_r319.sortedColumn == null ? null : ctx_r319.sortedColumn.property) === ctx_r319.JSON.stringify(column_r317) && ctx_r319.sortedColumn.ascending)("po-table-header-icon-ascending", ctx_r319.JSON.stringify(ctx_r319.sortedColumn == null ? null : ctx_r319.sortedColumn.property) === ctx_r319.JSON.stringify(column_r317) && !ctx_r319.sortedColumn.ascending);
    i0.ɵɵadvance(1);
    i0.ɵɵproperty("ngIf", ctx_r319.JSON.stringify(ctx_r319.sortedColumn == null ? null : ctx_r319.sortedColumn.property) !== ctx_r319.JSON.stringify(column_r317));
    i0.ɵɵadvance(1);
    i0.ɵɵproperty("ngIf", ctx_r319.JSON.stringify(ctx_r319.sortedColumn == null ? null : ctx_r319.sortedColumn.property) === ctx_r319.JSON.stringify(column_r317) && ctx_r319.sortedColumn.ascending);
    i0.ɵɵadvance(1);
    i0.ɵɵproperty("ngIf", ctx_r319.JSON.stringify(ctx_r319.sortedColumn == null ? null : ctx_r319.sortedColumn.property) === ctx_r319.JSON.stringify(column_r317) && !ctx_r319.sortedColumn.ascending);
} }
function PoTableComponent_ng_template_20_Template(rf, ctx) { if (rf & 1) {
    const _r325 = i0.ɵɵgetCurrentView();
    i0.ɵɵelementStart(0, "span", 128, 129);
    i0.ɵɵlistener("mouseenter", function PoTableComponent_ng_template_20_Template_span_mouseenter_0_listener($event) { i0.ɵɵrestoreView(_r325); const ctx_r324 = i0.ɵɵnextContext(); return i0.ɵɵresetView(ctx_r324.tooltipMouseEnter($event)); })("mouseleave", function PoTableComponent_ng_template_20_Template_span_mouseleave_0_listener() { i0.ɵɵrestoreView(_r325); const ctx_r326 = i0.ɵɵnextContext(); return i0.ɵɵresetView(ctx_r326.tooltipMouseLeave()); });
    i0.ɵɵtext(2);
    i0.ɵɵpipe(3, "titlecase");
    i0.ɵɵelementEnd();
    i0.ɵɵtemplate(4, PoTableComponent_ng_template_20_span_4_Template, 4, 9, "span", 130);
} if (rf & 2) {
    const column_r317 = ctx.$implicit;
    const ctx_r19 = i0.ɵɵnextContext();
    i0.ɵɵproperty("p-tooltip", ctx_r19.tooltipText)("p-append-in-body", true);
    i0.ɵɵadvance(2);
    i0.ɵɵtextInterpolate1(" ", column_r317.label || i0.ɵɵpipeBind1(3, 4, column_r317.property), " ");
    i0.ɵɵadvance(2);
    i0.ɵɵproperty("ngIf", ctx_r19.sort && column_r317.sortable !== false);
} }
function PoTableComponent_ng_template_22_Template(rf, ctx) { if (rf & 1) {
    i0.ɵɵelementStart(0, "div", 135);
    i0.ɵɵtext(1);
    i0.ɵɵelementEnd();
} if (rf & 2) {
    const ctx_r21 = i0.ɵɵnextContext();
    i0.ɵɵstyleProp("width", ctx_r21.noColumnsHeader == null ? null : ctx_r21.noColumnsHeader.nativeElement.offsetWidth, "px");
    i0.ɵɵadvance(1);
    i0.ɵɵtextInterpolate1(" ", ctx_r21.hasValidColumns ? ctx_r21.literals.noVisibleColumn : ctx_r21.literals.noColumns, " ");
} }
function PoTableComponent_ng_template_24_Template(rf, ctx) { if (rf & 1) {
    i0.ɵɵtext(0);
} if (rf & 2) {
    const ctx_r23 = i0.ɵɵnextContext();
    i0.ɵɵtextInterpolate1(" ", ctx_r23.hasValidColumns ? ctx_r23.literals.noVisibleColumn : ctx_r23.literals.noColumns, "\n");
} }
function PoTableComponent_ng_template_26_td_0_div_1_po_icon_1_Template(rf, ctx) { if (rf & 1) {
    i0.ɵɵelement(0, "po-icon", 142);
} if (rf & 2) {
    const ctx_r332 = i0.ɵɵnextContext(4);
    i0.ɵɵproperty("p-icon", ctx_r332.firstAction.icon);
} }
function PoTableComponent_ng_template_26_td_0_div_1_Template(rf, ctx) { if (rf & 1) {
    const _r335 = i0.ɵɵgetCurrentView();
    i0.ɵɵelementStart(0, "div", 140);
    i0.ɵɵlistener("click", function PoTableComponent_ng_template_26_td_0_div_1_Template_div_click_0_listener() { i0.ɵɵrestoreView(_r335); const row_r327 = i0.ɵɵnextContext(2).row; const ctx_r333 = i0.ɵɵnextContext(); return i0.ɵɵresetView(ctx_r333.executeTableAction(row_r327, ctx_r333.firstAction)); });
    i0.ɵɵtemplate(1, PoTableComponent_ng_template_26_td_0_div_1_po_icon_1_Template, 1, 1, "po-icon", 141);
    i0.ɵɵtext(2);
    i0.ɵɵelementEnd();
} if (rf & 2) {
    const row_r327 = i0.ɵɵnextContext(2).row;
    const ctx_r331 = i0.ɵɵnextContext();
    i0.ɵɵclassProp("po-table-action-disabled", ctx_r331.firstAction.disabled ? ctx_r331.validateTableAction(row_r327, ctx_r331.firstAction) : false);
    i0.ɵɵadvance(1);
    i0.ɵɵproperty("ngIf", ctx_r331.firstAction.icon);
    i0.ɵɵadvance(1);
    i0.ɵɵtextInterpolate1(" ", ctx_r331.firstAction.label, " ");
} }
function PoTableComponent_ng_template_26_td_0_Template(rf, ctx) { if (rf & 1) {
    i0.ɵɵelementStart(0, "td", 138);
    i0.ɵɵtemplate(1, PoTableComponent_ng_template_26_td_0_div_1_Template, 3, 4, "div", 139);
    i0.ɵɵelementEnd();
} if (rf & 2) {
    const ctx_r329 = i0.ɵɵnextContext(2);
    i0.ɵɵstyleProp("width", ctx_r329.height && ctx_r329.actionRight ? ctx_r329.getWidthColumnManager() : "", "px")("max-width", ctx_r329.height && !ctx_r329.actionRight ? ctx_r329.getColumnWidthActionsLeft() : "", "px")("width", ctx_r329.height && !ctx_r329.actionRight ? ctx_r329.getColumnWidthActionsLeft() : "", "px");
    i0.ɵɵadvance(1);
    i0.ɵɵproperty("ngIf", ctx_r329.firstAction.visible !== false);
} }
function PoTableComponent_ng_template_26_td_1_Template(rf, ctx) { if (rf & 1) {
    const _r340 = i0.ɵɵgetCurrentView();
    i0.ɵɵelementStart(0, "td", 143)(1, "span", 144, 145);
    i0.ɵɵlistener("click", function PoTableComponent_ng_template_26_td_1_Template_span_click_1_listener() { i0.ɵɵrestoreView(_r340); const _r337 = i0.ɵɵreference(2); const row_r327 = i0.ɵɵnextContext().row; const ctx_r338 = i0.ɵɵnextContext(); return i0.ɵɵresetView(ctx_r338.togglePopup(row_r327, _r337)); });
    i0.ɵɵelementEnd()();
} }
function PoTableComponent_ng_template_26_Template(rf, ctx) { if (rf & 1) {
    i0.ɵɵtemplate(0, PoTableComponent_ng_template_26_td_0_Template, 2, 7, "td", 136);
    i0.ɵɵtemplate(1, PoTableComponent_ng_template_26_td_1_Template, 3, 0, "td", 137);
} if (rf & 2) {
    const ctx_r25 = i0.ɵɵnextContext();
    i0.ɵɵproperty("ngIf", ctx_r25.isSingleAction);
    i0.ɵɵadvance(1);
    i0.ɵɵproperty("ngIf", ctx_r25.visibleActions.length > 1);
} }
function PoTableComponent_po_table_column_manager_28_Template(rf, ctx) { if (rf & 1) {
    const _r342 = i0.ɵɵgetCurrentView();
    i0.ɵɵelementStart(0, "po-table-column-manager", 146);
    i0.ɵɵlistener("p-visible-columns-change", function PoTableComponent_po_table_column_manager_28_Template_po_table_column_manager_p_visible_columns_change_0_listener($event) { i0.ɵɵrestoreView(_r342); const ctx_r341 = i0.ɵɵnextContext(); return i0.ɵɵresetView(ctx_r341.onVisibleColumnsChange($event)); })("p-change-visible-columns", function PoTableComponent_po_table_column_manager_28_Template_po_table_column_manager_p_change_visible_columns_0_listener($event) { i0.ɵɵrestoreView(_r342); const ctx_r343 = i0.ɵɵnextContext(); return i0.ɵɵresetView(ctx_r343.onChangeVisibleColumns($event)); })("p-initial-columns", function PoTableComponent_po_table_column_manager_28_Template_po_table_column_manager_p_initial_columns_0_listener($event) { i0.ɵɵrestoreView(_r342); const ctx_r344 = i0.ɵɵnextContext(); return i0.ɵɵresetView(ctx_r344.onColumnRestoreManager($event)); });
    i0.ɵɵelementEnd();
} if (rf & 2) {
    const ctx_r26 = i0.ɵɵnextContext();
    i0.ɵɵproperty("p-columns", ctx_r26.columns)("p-max-columns", ctx_r26.maxColumns)("p-target", ctx_r26.columnManagerTarget)("p-last-visible-columns-selected", ctx_r26.lastVisibleColumnsSelected)("p-hide-action-fixed-columns", ctx_r26.hideActionFixedColumns)("p-columns-default", ctx_r26.initialColumns);
} }
/**
 * @docsExtends PoTableBaseComponent
 *
 * @example
 *
 * <example name="po-table-basic" title="PO Table Basic">
 *  <file name="sample-po-table-basic/sample-po-table-basic.component.ts"> </file>
 *  <file name="sample-po-table-basic/sample-po-table-basic.component.html"> </file>
 * </example>
 *
 * <example name="po-table-labs" title="PO Table Labs">
 *  <file name="sample-po-table-labs/sample-po-table-labs.component.ts"> </file>
 *  <file name="sample-po-table-labs/sample-po-table-labs.component.html"> </file>
 *  <file name="sample-po-table-labs/sample-po-table-labs.component.e2e-spec.ts"> </file>
 *  <file name="sample-po-table-labs/sample-po-table-labs.component.po.ts"> </file>
 *  <file name="sample-po-table-labs/sample-po-table-labs.service.ts"> </file>
 * </example>
 *
 * <example name="po-table-with-api" title="PO Table using API">
 *  <file name="sample-po-table-with-api/sample-po-table-with-api.component.ts"> </file>
 *  <file name="sample-po-table-with-api/sample-po-table-with-api.component.html"> </file>
 * </example>
 *
 * <example name="po-table-transport" title="PO Table - Transport">
 *  <file name="sample-po-table-transport/sample-po-table-transport.component.ts"> </file>
 *  <file name="sample-po-table-transport/sample-po-table-transport.component.html"> </file>
 *  <file name="sample-po-table-transport/sample-po-table-transport.service.ts"> </file>
 * </example>
 *
 * <example name="po-table-airfare" title="PO Table - Airfare">
 *  <file name="sample-po-table-airfare/sample-po-table-airfare.component.ts"> </file>
 *  <file name="sample-po-table-airfare/sample-po-table-airfare.component.html"> </file>
 *  <file name="sample-po-table-airfare/sample-po-table-airfare.service.ts"> </file>
 * </example>
 *
 * <example name="po-table-components" title="PO Table - Po Field Components">
 *  <file name="sample-po-table-components/sample-po-table-components.component.ts"> </file>
 *  <file name="sample-po-table-components/sample-po-table-components.enum.ts"> </file>
 *  <file name="sample-po-table-components/sample-po-table-components.component.html"> </file>
 *  <file name="sample-po-table-components/sample-po-table-components.service.ts"> </file>
 *  <file name="sample-po-table-components/sample-po-table-components.component.css"> </file>
 * </example>
 *
 * <example name="po-table-heroes" title="PO Table - Heroes">
 *  <file name="sample-po-table-heroes/sample-po-table-heroes.component.ts"> </file>
 *  <file name="sample-po-table-heroes/sample-po-table-heroes.component.html"> </file>
 *  <file name="sample-po-table-heroes/sample-po-table-heroes.service.ts"> </file>
 * </example>
 *
 * <example name="po-table-draggable" title="PO Table Drag and Drop">
 *  <file name="sample-po-table-draggable/sample-po-table-draggable.component.html"> </file>
 *  <file name="sample-po-table-draggable/sample-po-table-draggable.component.ts"> </file>
 * </example>
 */
export class PoTableComponent extends PoTableBaseComponent {
    constructor(poDate, differs, renderer, poLanguageService, changeDetector, decimalPipe, router, defaultService) {
        super(poDate, poLanguageService, defaultService);
        this.changeDetector = changeDetector;
        this.decimalPipe = decimalPipe;
        this.router = router;
        this.defaultService = defaultService;
        this.poNotification = inject(PoNotificationService);
        this.tableOpacity = 0;
        this.itemSize = 48;
        this.inputFieldValue = '';
        this.close = {
            action: () => {
                this.modalDelete.close();
            },
            label: this.literals.cancel,
            danger: true
        };
        this.confirm = {
            action: () => {
                this.deleteItems();
            },
            label: this.literals.delete
        };
        this.initialized = false;
        this.visibleElement = false;
        this.subscriptionService = new Subscription();
        this.JSON = JSON;
        this.differ = differs.find([]).create(null);
        // TODO: #5550 ao remover este listener, no portal, quando as colunas forem fixas não sofrem
        // alteração de largura, pois o ngDoCheck não é executado.
        this.clickListener = renderer.listen('document', 'click', () => { });
        this.resizeListener = renderer.listen('window', 'resize', (event) => {
            this.debounceResize();
        });
    }
    set columnManagerTarget(value) {
        this._columnManagerTarget = value;
        this.changeDetector.detectChanges();
    }
    get columnManagerTarget() {
        return this._columnManagerTarget;
    }
    set columnManagerTargetFixed(value) {
        this._columnManagerTargetFixed = value;
        this.changeDetector.detectChanges();
    }
    get columnManagerTargetFixed() {
        return this._columnManagerTargetFixed;
    }
    get hasRowTemplateWithArrowDirectionRight() {
        return this.tableRowTemplate?.tableRowTemplateArrowDirection === PoTableRowTemplateArrowDirection.Right;
    }
    get columnCount() {
        const columnCount = this.mainColumns.length +
            (this.actions.length > 0 ? 1 : 0) +
            (this.selectable ? 1 : 0) +
            (!this.hideDetail && this.columnMasterDetail !== undefined ? 1 : 0);
        return columnCount || 1;
    }
    get columnCountForMasterDetail() {
        // caso tiver ações será utilizado a sua coluna para exibir o columnManager
        const columnManager = this.actions.length ? 0 : 1;
        return this.mainColumns.length + 1 + (this.actions.length > 0 ? 1 : 0) + (this.selectable ? 1 : 0) + columnManager;
    }
    get detailHideSelect() {
        const masterDetail = this.columnMasterDetail;
        return masterDetail && masterDetail.detail ? masterDetail.detail.hideSelect : false;
    }
    get hasVisibleActions() {
        return !!this.visibleActions.length;
    }
    get firstAction() {
        return this.visibleActions && this.visibleActions[0];
    }
    get hasFooter() {
        return this.hasItems && this.hasVisibleSubtitleColumns;
    }
    get hasMasterDetailColumn() {
        return (this.hasMainColumns && this.hasItems && !this.hideDetail && !!(this.columnMasterDetail || this.hasRowTemplate));
    }
    get hasRowTemplate() {
        return !!this.tableRowTemplate;
    }
    get hasSelectableColumn() {
        return this.selectable && this.hasItems && this.hasMainColumns;
    }
    get hasValidColumns() {
        return !!this.validColumns.length;
    }
    get hasVisibleSubtitleColumns() {
        return this.subtitleColumns.some(column => column.visible !== false);
    }
    get isSingleAction() {
        return this.visibleActions.length === 1;
    }
    get isDraggable() {
        return this.draggable;
    }
    get inverseOfTranslation() {
        if (!this.viewPort || !this.viewPort['_renderedContentOffset']) {
            return '-0px';
        }
        const offset = this.viewPort['_renderedContentOffset'];
        return `-${offset}px`;
    }
    ngOnInit() {
        this.idRadio = `po-radio-${uuid()}`;
    }
    ngAfterViewInit() {
        this.initialized = true;
    }
    showMoreInfiniteScroll({ target }) {
        const scrollPosition = target.offsetHeight + target.scrollTop;
        if (!this.showMoreDisabled && scrollPosition >= target.scrollHeight * (this.infiniteScrollDistance / 110)) {
            this.onShowMore();
        }
    }
    ngDoCheck() {
        this.checkChangesItems();
        this.verifyCalculateHeightTableContainer();
        // Permite que os cabeçalhos sejam calculados na primeira vez que o componente torna-se visível,
        // evitando com isso, problemas com Tabs ou Divs que iniciem escondidas.
        if (this.tableWrapperElement?.nativeElement.offsetWidth && !this.visibleElement && this.initialized) {
            this.debounceResize();
            this.checkInfiniteScroll();
            this.visibleElement = true;
        }
    }
    ngOnDestroy() {
        this.removeListeners();
        this.subscriptionService?.unsubscribe();
    }
    /**
     * Método responsável por realizar busca no serviço de dados podendo informar filtros e com o retorno, atualiza a tabela.
     *
     * Caso não seja informado parâmetro, nada será adicionado ao GET, conforme abaixo:
     * ```
     * url + ?page=1&pageSize=10
     * ```
     * > Obs: os parâmetros `page` e `pageSize` sempre serão chamados independente de ser enviados outros parâmetros.
     *
     * Caso sejam informados os parâmetros `{ name: 'JOHN', age: '23' }`, todos serão adicionados ao GET, conforme abaixo:
     * ```
     * url + ?page=1&pageSize=10&name=JOHN&age=23
     * ```
     *
     * @param { { key: value } } queryParams Formato do objeto a ser enviado.
     * > Pode ser utilizada qualquer string como key, e qualquer string ou number como value.
     */
    applyFilters(queryParams) {
        this.page = 1;
        this.initializeData(queryParams);
    }
    /**
     * Método que colapsa uma linha com detalhe quando executada.
     *
     * @param { number } rowIndex Índice da linha que será colapsada.
     * > Ao reordenar os dados da tabela, o valor contido neste índice será alterado conforme a ordenação.
     */
    collapse(rowIndex) {
        this.setShowDetail(rowIndex, false);
    }
    /**
     * Método que expande uma linha com detalhe quando executada.
     *
     * @param { number } rowIndex Índice da linha que será expandida.
     * > Ao reordenar os dados da tabela, o valor contido neste índice será alterado conforme a ordenação.
     */
    expand(rowIndex) {
        this.setShowDetail(rowIndex, true);
    }
    /**
     * Retorna as linhas do `po-table` que estão selecionadas.
     */
    getSelectedRows() {
        return this.items.filter(item => item.$selected);
    }
    /**
     * Retorna as linhas do `po-table` que não estão selecionadas.
     */
    getUnselectedRows() {
        return this.items.filter(item => !item.$selected);
    }
    /**
     * Desmarca as linhas que estão selecionadas.
     */
    unselectRows() {
        const columnDetail = this.nameColumnDetail;
        this.items.forEach(item => {
            const detailItems = columnDetail ? item[columnDetail] : null;
            if (Array.isArray(detailItems)) {
                detailItems.forEach(detailItem => {
                    detailItem.$selected = false;
                });
            }
            item.$selected = false;
        });
        this.selectAll = false;
    }
    checkDisabled(row, column) {
        return column.disabled ? column.disabled(row) : false;
    }
    containsMasterDetail(row) {
        return row[this.nameColumnDetail] && row[this.nameColumnDetail].length;
    }
    executeTableAction(row, tableAction) {
        if (!row.disabled && !this.validateTableAction(row, tableAction)) {
            tableAction.action(row);
            this.toggleRowAction(row);
        }
    }
    /**
     * Desmarca uma linha que está selecionada.
     */
    unselectRowItem(itemfn) {
        this.toggleSelect(itemfn, false);
        if (this.items.every(item => !item.$selected)) {
            this.selectAll = false;
        }
        else {
            this.selectAll = null;
        }
    }
    /**
     * Seleciona uma linha do 'po-table'.
     */
    selectRowItem(itemfn) {
        this.toggleSelect(itemfn, true);
        if (this.items.every(item => item.$selected)) {
            this.selectAll = true;
        }
        else {
            this.selectAll = null;
        }
    }
    /**
     * Método responsável pela exclusão de itens em lote.
     * Caso a tabela esteja executando a propriedade `p-service-delete`, será necessário excluir 1 item por vez.
     *
     * Ao utilizar `p-service-delete` mas sem a propriedade `p-service-api`, será responsabilidade do usuário o tratamento
     * após a requisição DELETE ser executada.
     *
     * Caso a tabela utilize `p-height` e esteja sem serviço, é necessário a reatribuição dos itens utilizando o evento `(p-delete-items)`, por exemplo:
     *
     * ```
     *<po-table
     *  (p-delete-items)="items = $event"
     * >
     *</po-table>
     * ```
     */
    deleteItems() {
        const newItems = [...this.items];
        const newItemsFiltered = [...newItems].filter(item => !item.$selected);
        if (!this.serviceDeleteApi) {
            this.deleteItemsLocal(newItems, newItemsFiltered);
        }
        else {
            this.deleteItemsService(newItemsFiltered);
        }
    }
    formatNumber(value, format) {
        if (!format) {
            return value;
        }
        return this.decimalPipe.transform(value, format);
    }
    getCellData(row, column) {
        const arrayProperty = column.property.split('.');
        if (arrayProperty.length > 1) {
            const nestedProperties = arrayProperty;
            let value = row;
            for (const property of nestedProperties) {
                value = value[property] || value[property] === 0 ? value[property] : '';
            }
            return value;
        }
        else {
            return row[column.property];
        }
    }
    getBooleanLabel(rowValue, columnBoolean) {
        if (rowValue || rowValue === false || rowValue === 0) {
            rowValue = convertToBoolean(rowValue);
            if (columnBoolean.boolean) {
                return rowValue ? columnBoolean.boolean.trueLabel || 'Sim' : columnBoolean.boolean.falseLabel || 'Não';
            }
            else {
                return rowValue ? 'Sim' : 'Não';
            }
        }
        return rowValue;
    }
    getColumnIcons(row, column) {
        const rowIcons = this.getCellData(row, column);
        if (column.icons) {
            if (Array.isArray(rowIcons)) {
                return this.mergeCustomIcons(rowIcons, column.icons);
            }
            else {
                return this.findCustomIcon(rowIcons, column);
            }
        }
        return rowIcons;
    }
    getColumnLabel(row, columnLabel) {
        return columnLabel.labels.find(labelItem => this.getCellData(row, columnLabel) === labelItem.value);
    }
    getSubtitleColumn(row, subtitleColumn) {
        return subtitleColumn.subtitles.find(subtitleItem => this.getCellData(row, subtitleColumn) === subtitleItem.value);
    }
    isShowMasterDetail(row) {
        return (!this.hideDetail &&
            this.nameColumnDetail &&
            row.$showDetail &&
            this.containsMasterDetail(row) &&
            !this.hasRowTemplate);
    }
    isShowRowTemplate(row, index) {
        if (this.tableRowTemplate && this.tableRowTemplate.poTableRowTemplateShow) {
            return this.tableRowTemplate.poTableRowTemplateShow(row, index);
        }
        return true;
    }
    onClickLink(event, row, column) {
        if (!this.checkDisabled(row, column)) {
            event.stopPropagation();
        }
    }
    onChangeVisibleColumns(columns) {
        this.changeVisibleColumns.emit(columns);
    }
    onColumnRestoreManager(value) {
        this.columnRestoreManager.emit(value);
    }
    onVisibleColumnsChange(columns) {
        this.columns = columns;
        this.changeDetector.detectChanges();
    }
    tooltipMouseEnter(event, column, row) {
        this.tooltipText = undefined;
        if (event.target.offsetWidth < event.target.scrollWidth && event.target.innerText.trim()) {
            return (this.tooltipText = event.target.innerText);
        }
        if (column) {
            this.checkingIfColumnHasTooltip(column, row);
        }
    }
    tooltipMouseLeave() {
        this.tooltipText = undefined;
    }
    togglePopup(row, targetRef) {
        this.popupTarget = targetRef;
        this.changeDetector.detectChanges();
        this.poPopupComponent.toggle(row);
    }
    trackBy(index) {
        return index;
    }
    validateTableAction(row, tableAction) {
        if (typeof tableAction.disabled === 'function') {
            return tableAction.disabled(row);
        }
        else {
            return tableAction.disabled;
        }
    }
    onOpenColumnManager() {
        this.lastVisibleColumnsSelected = [...this.columns];
    }
    onFilteredItemsChange(items) {
        this.filteredItems = items;
    }
    /**
     * Método que remove um item da tabela.
     *
     * @param { number | { key: value } } item Índice da linha ou o item que será removido.
     * > Ao remover o item, a linha que o representa será excluída da tabela.
     */
    removeItem(item) {
        if (item instanceof Object) {
            this.items = this.items.filter(filterItem => filterItem !== item);
        }
        else if (typeof item === 'number') {
            const index = item;
            this.items.splice(index, 1);
        }
    }
    /**
     * Método que atualiza um item da tabela.
     *
     * @param { number | { key: value } } item Índice da linha ou o item que será atualizado.
     * @param { { key: value } } updatedItem Item que foi atualizado.
     * > Ao atualizar o item, a informação será alterada na tabela.
     */
    updateItem(item, updatedItem) {
        if (typeof item === 'number') {
            this.items.splice(item, 1, updatedItem);
        }
        else {
            const index = this.items.findIndex(indexItem => indexItem === item);
            this.items.splice(index, 1, updatedItem);
        }
    }
    drop(event) {
        if (!this.mainColumns[event.currentIndex].fixed) {
            moveItemInArray(this.mainColumns, event.previousIndex, event.currentIndex);
            if (this.hideColumnsManager === false) {
                this.newOrderColumns = this.mainColumns;
                const detail = this.columns.filter(item => item.property === 'detail')[0];
                if (detail !== undefined) {
                    this.newOrderColumns.push(detail);
                }
                this.columns.map((item, index) => {
                    if (!item.visible) {
                        this.newOrderColumns.splice(index, 0, item);
                    }
                });
                this.columns = this.newOrderColumns;
                this.onVisibleColumnsChange(this.newOrderColumns);
            }
        }
    }
    getTemplate(column) {
        const template = this.tableColumnTemplates.find(tableColumnTemplate => tableColumnTemplate.targetProperty === column.property);
        if (!template) {
            console.warn(`Não foi possível encontrar o template para a coluna: ${column.property}, por gentileza informe a propriedade [p-property]`);
            return null;
        }
        return template.templateRef;
    }
    getWidthColumnManager() {
        return this.columnManager?.nativeElement.offsetWidth;
    }
    getColumnWidthActionsLeft() {
        return this.columnActionLeft?.nativeElement.offsetWidth;
    }
    hasSomeFixed() {
        return this.columns.some(item => item.fixed === true);
    }
    calculateHeightTableContainer(height) {
        const value = parseFloat(height);
        this.heightTableContainer = value ? value - this.getHeightTableFooter() : undefined;
        this.heightTableVirtual = this.heightTableContainer ? this.heightTableContainer - this.itemSize : undefined;
        this.setTableOpacity(1);
        this.changeDetector.detectChanges();
    }
    verifyCalculateHeightTableContainer() {
        if (this.height && this.verifyChangeHeightInFooter()) {
            this.footerHeight = this.getHeightTableFooter();
            this.calculateHeightTableContainer(this.height);
        }
    }
    checkInfiniteScroll() {
        if (this.hasInfiniteScroll()) {
            if (this.tableVirtualScroll.nativeElement.scrollHeight >= this.height) {
                this.includeInfiniteScroll();
            }
            else {
                this.infiniteScroll = false;
            }
        }
        this.changeDetector.detectChanges();
    }
    changesAfterDelete(newItemsFiltered) {
        this.selectAll = false;
        this.setSelectedList();
        this.modalDelete.close();
        this.poNotification.success(this.literals.deleteSuccessful);
        this.eventDelete.emit(newItemsFiltered);
    }
    checkChangesItems() {
        const changesItems = this.differ.diff(this.items);
        if (changesItems && this.selectAll) {
            this.selectAll = null;
        }
        if (changesItems && !this.hasColumns && this.hasItems) {
            this.columns = this.getDefaultColumns(this.items[0]);
        }
    }
    checkingIfColumnHasTooltip(column, row) {
        if (column.type === 'link' && column.tooltip && !this.checkDisabled(row, column)) {
            return (this.tooltipText = column.tooltip);
        }
        if (column.type === 'label') {
            const columnLabel = this.getColumnLabel(row, column);
            return (this.tooltipText = columnLabel?.tooltip);
        }
    }
    debounceResize() {
        clearTimeout(this.timeoutResize);
        this.timeoutResize = setTimeout(() => {
            // show the table
            this.setTableOpacity(1);
        });
    }
    deleteItemsLocal(newItems, newItemsFiltered) {
        if (this.height) {
            this.items = newItemsFiltered;
        }
        else {
            let index = this.items.length - 1;
            newItems
                .slice()
                .reverse()
                .forEach(item => {
                if (item.$selected) {
                    this.removeItem(index);
                }
                index--;
            });
        }
        this.changesAfterDelete(newItemsFiltered);
        this.onFilteredItemsChange(newItemsFiltered);
    }
    deleteItemsService(newItemsFiltered) {
        this.subscriptionService.add(this.defaultService.deleteItem(this.paramDeleteApi, this.itemsSelected[0][this.paramDeleteApi]).subscribe({
            next: value => {
                if (this.hasService) {
                    const filteredParams = {
                        ...this.paramsFilter,
                        pageSize: newItemsFiltered.length + 1,
                        page: 1
                    };
                    this.loading = true;
                    this.subscriptionService.add(this.defaultService.getFilteredItems(filteredParams).subscribe(items => {
                        this.setTableResponseProperties(items);
                    }));
                }
                this.items = newItemsFiltered;
                this.changesAfterDelete(newItemsFiltered);
            },
            error: error => {
                this.poNotification.error(this.literals.deleteApiError);
                this.modalDelete.close();
                this.eventDelete.emit(this.items);
            }
        }));
    }
    findCustomIcon(rowIcons, column) {
        const customIcon = column.icons.find(icon => rowIcons === icon.value);
        return customIcon ? [customIcon] : undefined;
    }
    getHeightTableFooter() {
        return this.tableFooterElement ? this.tableFooterElement.nativeElement.offsetHeight : 0;
    }
    hasInfiniteScroll() {
        return (this.infiniteScroll &&
            this.hasItems &&
            !this.subscriptionScrollEvent &&
            this.height &&
            this.tableVirtualScroll.nativeElement.scrollHeight);
    }
    includeInfiniteScroll() {
        this.scrollEvent$ = this.defaultService.scrollListener(this.tableVirtualScroll.nativeElement);
        this.subscriptionScrollEvent = this.scrollEvent$.subscribe(event => this.showMoreInfiniteScroll(event));
        this.changeDetector.detectChanges();
    }
    mergeCustomIcons(rowIcons, customIcons) {
        const mergedIcons = [];
        rowIcons.forEach(columnValue => {
            const foundCustomIcon = customIcons.find(customIcon => columnValue === customIcon.icon || columnValue === customIcon.value);
            foundCustomIcon ? mergedIcons.push(foundCustomIcon) : mergedIcons.push(columnValue);
        });
        return mergedIcons;
    }
    removeListeners() {
        if (this.resizeListener) {
            this.resizeListener();
        }
        if (this.clickListener) {
            this.clickListener();
        }
        if (this.subscriptionScrollEvent) {
            this.subscriptionScrollEvent.unsubscribe();
        }
    }
    setTableOpacity(value) {
        this.tableOpacity = value;
    }
    verifyChangeHeightInFooter() {
        return this.footerHeight !== this.getHeightTableFooter();
    }
    toggleSelect(compare, selectValue) {
        if (typeof compare !== 'function') {
            this.items.forEach(item => {
                if (item === compare) {
                    item.$selected = selectValue;
                }
            });
        }
        else {
            this.items.forEach(item => {
                if (compare(item)) {
                    item.$selected = selectValue;
                }
            });
        }
    }
}
PoTableComponent.ɵfac = function PoTableComponent_Factory(t) { return new (t || PoTableComponent)(i0.ɵɵdirectiveInject(i1.PoDateService), i0.ɵɵdirectiveInject(i0.IterableDiffers), i0.ɵɵdirectiveInject(i0.Renderer2), i0.ɵɵdirectiveInject(i2.PoLanguageService), i0.ɵɵdirectiveInject(i0.ChangeDetectorRef), i0.ɵɵdirectiveInject(i3.DecimalPipe), i0.ɵɵdirectiveInject(i4.Router), i0.ɵɵdirectiveInject(i5.PoTableService)); };
PoTableComponent.ɵcmp = /*@__PURE__*/ i0.ɵɵdefineComponent({ type: PoTableComponent, selectors: [["po-table"]], contentQueries: function PoTableComponent_ContentQueries(rf, ctx, dirIndex) { if (rf & 1) {
        i0.ɵɵcontentQuery(dirIndex, PoTableRowTemplateDirective, 7);
        i0.ɵɵcontentQuery(dirIndex, PoTableCellTemplateDirective, 5);
        i0.ɵɵcontentQuery(dirIndex, PoTableColumnTemplateDirective, 4);
    } if (rf & 2) {
        let _t;
        i0.ɵɵqueryRefresh(_t = i0.ɵɵloadQuery()) && (ctx.tableRowTemplate = _t.first);
        i0.ɵɵqueryRefresh(_t = i0.ɵɵloadQuery()) && (ctx.tableCellTemplate = _t.first);
        i0.ɵɵqueryRefresh(_t = i0.ɵɵloadQuery()) && (ctx.tableColumnTemplates = _t);
    } }, viewQuery: function PoTableComponent_Query(rf, ctx) { if (rf & 1) {
        i0.ɵɵviewQuery(_c0, 5, ElementRef);
        i0.ɵɵviewQuery(_c1, 5);
        i0.ɵɵviewQuery(PoModalComponent, 7);
        i0.ɵɵviewQuery(_c2, 5, ElementRef);
        i0.ɵɵviewQuery(_c3, 5, ElementRef);
        i0.ɵɵviewQuery(_c4, 5, ElementRef);
        i0.ɵɵviewQuery(_c5, 5, ElementRef);
        i0.ɵɵviewQuery(_c6, 5, ElementRef);
        i0.ɵɵviewQuery(_c7, 5, ElementRef);
        i0.ɵɵviewQuery(_c8, 5, ElementRef);
        i0.ɵɵviewQuery(_c9, 5);
        i0.ɵɵviewQuery(_c10, 7, ElementRef);
        i0.ɵɵviewQuery(CdkVirtualScrollViewport, 5);
        i0.ɵɵviewQuery(_c11, 5);
        i0.ɵɵviewQuery(_c12, 5);
        i0.ɵɵviewQuery(_c13, 5, ElementRef);
        i0.ɵɵviewQuery(_c14, 5, ElementRef);
    } if (rf & 2) {
        let _t;
        i0.ɵɵqueryRefresh(_t = i0.ɵɵloadQuery()) && (ctx.noColumnsHeader = _t.first);
        i0.ɵɵqueryRefresh(_t = i0.ɵɵloadQuery()) && (ctx.poPopupComponent = _t.first);
        i0.ɵɵqueryRefresh(_t = i0.ɵɵloadQuery()) && (ctx.modalDelete = _t.first);
        i0.ɵɵqueryRefresh(_t = i0.ɵɵloadQuery()) && (ctx.tableFooterElement = _t.first);
        i0.ɵɵqueryRefresh(_t = i0.ɵɵloadQuery()) && (ctx.tableWrapperElement = _t.first);
        i0.ɵɵqueryRefresh(_t = i0.ɵɵloadQuery()) && (ctx.tableTemplate = _t.first);
        i0.ɵɵqueryRefresh(_t = i0.ɵɵloadQuery()) && (ctx.tableVirtualScroll = _t.first);
        i0.ɵɵqueryRefresh(_t = i0.ɵɵloadQuery()) && (ctx.columnManager = _t.first);
        i0.ɵɵqueryRefresh(_t = i0.ɵɵloadQuery()) && (ctx.columnBatchActions = _t.first);
        i0.ɵɵqueryRefresh(_t = i0.ɵɵloadQuery()) && (ctx.columnActionLeft = _t.first);
        i0.ɵɵqueryRefresh(_t = i0.ɵɵloadQuery()) && (ctx.filterInput = _t.first);
        i0.ɵɵqueryRefresh(_t = i0.ɵɵloadQuery()) && (ctx.poSearchInput = _t.first);
        i0.ɵɵqueryRefresh(_t = i0.ɵɵloadQuery()) && (ctx.viewPort = _t.first);
        i0.ɵɵqueryRefresh(_t = i0.ɵɵloadQuery()) && (ctx.columnManagerTarget = _t.first);
        i0.ɵɵqueryRefresh(_t = i0.ɵɵloadQuery()) && (ctx.columnManagerTargetFixed = _t.first);
        i0.ɵɵqueryRefresh(_t = i0.ɵɵloadQuery()) && (ctx.actionsIconElement = _t);
        i0.ɵɵqueryRefresh(_t = i0.ɵɵloadQuery()) && (ctx.actionsElement = _t);
    } }, features: [i0.ɵɵProvidersFeature([PoDateService]), i0.ɵɵInheritDefinitionFeature], decls: 33, vars: 14, consts: [[1, "po-table-actions"], ["class", "po-table-actions-batch-actions", 4, "ngIf"], ["class", "po-table-actions-column-manager", 4, "ngIf"], ["class", "po-table-search", 4, "ngIf"], ["p-no-padding", "", 4, "ngIf", "ngIfElse"], ["tableContainerTemplate", ""], ["class", "po-row po-table-footer-show-more", 3, "po-invisible", 4, "ngIf"], ["tableTemplate", ""], ["tableVirtualScrollTemplate", ""], [3, "p-actions", "p-target"], ["popup", ""], ["poTableColumnDetail", ""], ["inputRadio", ""], ["inputCheckbox", ""], ["contentHeaderTemplate", ""], ["noColumnsWithHeight", ""], ["noColumnsWithoutHeight", ""], ["ActionsColumnTemplate", ""], [3, "p-columns", "p-max-columns", "p-target", "p-last-visible-columns-selected", "p-hide-action-fixed-columns", "p-columns-default", "p-visible-columns-change", "p-change-visible-columns", "p-initial-columns", 4, "ngIf"], [3, "p-title", "p-primary-action", "p-secondary-action", "p-click-out"], ["modalDelete", ""], [1, "po-font-text-large"], [1, "po-table-actions-batch-actions"], ["columnBatchActions", ""], [1, "po-table-actions-batch-actions__label", 3, "ngPlural"], ["ngPluralCase", "=0"], ["ngPluralCase", "=1"], ["ngPluralCase", "other"], [1, "po-table-actions-batch-actions__buttons"], ["p-icon", "po-icon-delete", 3, "p-danger", "p-disabled", "p-label", "p-click"], [1, "po-table-actions-column-manager"], ["columnManager", ""], ["p-icon", "po-icon-settings", "p-kind", "tertiary", "p-tooltip-position", "left", 3, "p-aria-label", "p-tooltip", "p-click"], ["columnManagerTarget", ""], [1, "po-table-search"], [3, "p-items", "p-filter-keys", "p-filter-type", "p-filtered-items-change"], ["p-no-padding", ""], [4, "ngTemplateOutlet"], ["class", "po-table-overlay", 4, "ngIf"], [1, "po-table-main-container"], [1, "po-table-wrapper"], ["tableWrapper", ""], ["class", "po-table-container", 3, "height", 4, "ngIf"], [4, "ngIf"], ["class", "po-table-footer", 4, "ngIf"], [1, "po-table-overlay"], [1, "po-table-overlay-content", 3, "p-text"], [1, "po-table-container"], [1, "po-table-container-fixed-inner"], ["poTableTbody", ""], [1, "po-table-footer"], [4, "ngFor", "ngForOf", "ngForTrackBy"], [3, "p-literals", "p-subtitles"], [1, "po-row", "po-table-footer-show-more"], ["tableFooter", ""], [1, "po-offset-xl-4", "po-offset-lg-4", "po-offset-md-3", "po-lg-4", "po-md-6", 3, "p-disabled", "p-label", "p-loading", "p-click"], [1, "po-table", 3, "ngClass"], ["cdkDropList", "", "cdkDropListOrientation", "horizontal", 3, "ngClass", "cdkDropListDropped"], ["class", "po-table-column-selectable", 3, "pointer-events", 4, "ngIf"], ["class", "po-table-header-column po-table-header-master-detail", 4, "ngIf"], [3, "po-table-header-master-detail", "po-table-header-single-action", 4, "ngIf"], ["class", "po-table-header-column po-text-center", 4, "ngIf"], [4, "ngIf", "ngIfThen", "ngIfElse"], ["tableDefaultThDragDrop", ""], ["tableDefaultThDefault", ""], [3, "po-table-header-single-action", "po-table-header-actions", 4, "ngIf"], ["class", "po-table-group-row", 4, "ngIf"], [1, "po-table-column-selectable"], ["name", "selectAll", 3, "p-checkboxValue", "p-change", 4, "ngIf"], ["name", "selectAll", 3, "p-checkboxValue", "p-change"], [1, "po-table-header-column", "po-table-header-master-detail"], ["columnActionLeft", ""], [1, "po-table-header-column", "po-text-center"], ["noColumnsHeader", ""], ["class", "po-table-header-ellipsis p-element po-frozen-column", "cdkDrag", "", "cdkDragLockAxis", "x", 3, "width", "max-width", "min-width", "po-clickable", "ngClass", "po-table-header-subtitle", "po-table-column-drag-box", "cdkDragDisabled", "pFrozenColumn", "click", 4, "ngFor", "ngForOf", "ngForTrackBy"], ["cdkDrag", "", "cdkDragLockAxis", "x", 1, "po-table-header-ellipsis", "p-element", "po-frozen-column", 3, "ngClass", "cdkDragDisabled", "pFrozenColumn", "click"], [1, "po-table-header-flex"], [4, "ngTemplateOutlet", "ngTemplateOutletContext"], ["cdkDragHandle", "", "width", "24", "height", "24", "viewBox", "0 0 24 24", "fill", "none", "xmlns", "http://www.w3.org/2000/svg"], ["cx", "9", "cy", "6", "r", "2", "fill", "black"], ["cx", "15", "cy", "6", "r", "2", "fill", "black"], ["cx", "9", "cy", "12", "r", "2", "fill", "black"], ["cx", "15", "cy", "12", "r", "2", "fill", "black"], ["cx", "9", "cy", "18", "r", "2", "fill", "black"], ["cx", "15", "cy", "18", "r", "2", "fill", "black"], ["class", "po-table-header-ellipsis p-element po-frozen-column", 3, "width", "max-width", "min-width", "po-clickable", "ngClass", "po-table-header-subtitle", "pFrozenColumn", "click", 4, "ngFor", "ngForOf", "ngForTrackBy"], [1, "po-table-header-ellipsis", "p-element", "po-frozen-column", 3, "ngClass", "pFrozenColumn", "click"], [1, "po-table-group-row"], [1, "po-table-row", "po-table-row-no-data"], [1, "po-table-no-data", "po-text-center", 3, "colSpan"], ["class", "po-table-group-row", 4, "ngFor", "ngForOf", "ngForTrackBy"], [1, "po-table-row"], ["class", "po-table-column-selectable", 4, "ngIf"], ["class", "po-table-column-detail-toggle", 3, "click", 4, "ngIf"], ["class", "p-element po-frozen-column", 3, "width", "max-width", "min-width", "po-table-column", "po-table-column-right", "po-table-column-center", "po-table-column-icons", "pFrozenColumn", "ngClass", "click", 4, "ngFor", "ngForOf", "ngForTrackBy"], [1, "po-table-column-detail-toggle", 3, "click"], [3, "ngTemplateOutlet", "ngTemplateOutletContext"], [1, "p-element", "po-frozen-column", 3, "pFrozenColumn", "ngClass", "click"], [1, "po-table-column-cell", "po-table-body-ellipsis", "notranslate", 3, "ngSwitch", "p-tooltip", "p-append-in-body", "mouseenter", "mouseleave"], [4, "ngSwitchCase"], [3, "p-action", "p-disabled", "p-link", "p-row", "p-value", "click", 4, "ngSwitchCase"], [3, "p-column", "p-icons", "p-row", 4, "ngSwitchCase"], [4, "ngSwitchDefault"], [3, "p-action", "p-disabled", "p-link", "p-row", "p-value", "click"], [3, "p-column", "p-icons", "p-row"], [3, "p-subtitle"], [3, "p-value"], [1, "po-table-row-template-container", 3, "colSpan"], [1, "po-table-column-detail", 3, "colSpan"], [3, "p-selectable", "p-detail", "p-items", "p-select-row"], [3, "itemSize", "minBufferPx", "maxBufferPx"], ["tableVirtualScroll", ""], [1, "po-table", 3, "ngClass", "ngStyle"], [1, "po-table-header-sticky"], ["cdkDropList", "", "cdkDropListOrientation", "horizontal", 3, "cdkDropListDropped"], ["tableVirtualScrollThDragDrop", ""], ["tableVirtualScrollThDefault", ""], ["class", "po-table-header-ellipsis p-element po-frozen-column", "cdkDrag", "", "cdkDragLockAxis", "x", 3, "width", "max-width", "min-width", "po-clickable", "ngClass", "ngStyle", "po-table-header-subtitle", "po-table-column-drag-box", "cdkDragDisabled", "pFrozenColumn", "click", 4, "ngFor", "ngForOf", "ngForTrackBy"], ["cdkDrag", "", "cdkDragLockAxis", "x", 1, "po-table-header-ellipsis", "p-element", "po-frozen-column", 3, "ngClass", "ngStyle", "cdkDragDisabled", "pFrozenColumn", "click"], ["class", "po-table-header-ellipsis p-element po-frozen-column example-box", 3, "width", "max-width", "min-width", "po-clickable", "ngClass", "ngStyle", "po-table-header-subtitle", "pFrozenColumn", "click", 4, "ngFor", "ngForOf", "ngForTrackBy"], [1, "po-table-header-ellipsis", "p-element", "po-frozen-column", "example-box", 3, "ngClass", "ngStyle", "pFrozenColumn", "click"], ["class", "po-table-group-row", 4, "cdkVirtualFor", "cdkVirtualForOf", "cdkVirtualForTrackBy"], ["class", "p-element po-frozen-column", 3, "width", "max-width", "min-width", "po-table-column", "po-table-column-right", "po-table-column-center", "po-table-column-icons", "ngClass", "pFrozenColumn", "click", 4, "ngFor", "ngForOf", "ngForTrackBy"], [1, "p-element", "po-frozen-column", 3, "ngClass", "pFrozenColumn", "click"], ["class", "po-icon po-clickable", 3, "po-icon-arrow-up", "po-icon-arrow-down", 4, "ngIf"], [1, "po-icon", "po-clickable"], [3, "name", "p-checked", "p-change-selected"], ["name", "checkbox", 3, "p-checkboxValue", "p-change"], [1, "po-table-header-ellipsis", 3, "p-tooltip", "p-append-in-body", "mouseenter", "mouseleave"], ["columnHeader", ""], [3, "po-table-header-icon-unselected", "po-table-header-icon-descending", "po-table-header-icon-ascending", 4, "ngIf"], ["width", "24", "height", "24", "viewBox", "0 0 24 24", "fill", "none", "xmlns", "http://www.w3.org/2000/svg"], ["fill-rule", "evenodd", "clip-rule", "evenodd", "d", "M18.2929 15.2929L17 16.5858L17 10C17 9.44772 16.5523 9 16 9C15.4477 9 15 9.44772 15 10L15 16.5858L13.7071 15.2929C13.3166 14.9024 12.6834 14.9024 12.2929 15.2929C11.9024 15.6834 11.9024 16.3166 12.2929 16.7071L15.2929 19.7071C15.6834 20.0976 16.3166 20.0976 16.7071 19.7071L19.7071 16.7071C20.0976 16.3166 20.0976 15.6834 19.7071 15.2929C19.3166 14.9024 18.6834 14.9024 18.2929 15.2929ZM5.70716 8.7071L7.00006 7.4142L7.00003 14C7.00002 14.5523 7.44774 15 8.00002 15C8.55231 15 9.00002 14.5523 9.00003 14L9.00006 7.41418L10.2929 8.70707C10.6835 9.09759 11.3166 9.09758 11.7072 8.70706C12.0977 8.31653 12.0977 7.68336 11.7072 7.29284L8.70718 4.29286C8.31665 3.90234 7.68349 3.90235 7.29296 4.29287L4.29295 7.29289C3.90242 7.68342 3.90242 8.31658 4.29294 8.70711C4.68347 9.09763 5.31663 9.09762 5.70716 8.7071Z", "fill", "#1D1D30"], ["fill-rule", "evenodd", "clip-rule", "evenodd", "d", "M11 9.41421L9.70711 10.7071C9.31658 11.0976 8.68342 11.0976 8.29289 10.7071C7.90237 10.3166 7.90237 9.68342 8.29289 9.29289L11.2929 6.29289C11.6834 5.90237 12.3166 5.90237 12.7071 6.29289L15.7071 9.29289C16.0976 9.68342 16.0976 10.3166 15.7071 10.7071C15.3166 11.0976 14.6834 11.0976 14.2929 10.7071L13 9.41421V16C13 16.5523 12.5523 17 12 17C11.4477 17 11 16.5523 11 16V9.41421Z", "fill", "black"], ["fill-rule", "evenodd", "clip-rule", "evenodd", "d", "M9.70711 13.2929L11 14.5858V7C11 6.44772 11.4477 6 12 6C12.5523 6 13 6.44772 13 7V14.5858L14.2929 13.2929C14.6834 12.9024 15.3166 12.9024 15.7071 13.2929C16.0976 13.6834 16.0976 14.3166 15.7071 14.7071L12.7071 17.7071C12.3166 18.0976 11.6834 18.0976 11.2929 17.7071L8.29289 14.7071C7.90237 14.3166 7.90237 13.6834 8.29289 13.2929C8.68342 12.9024 9.31658 12.9024 9.70711 13.2929Z", "fill", "black"], [1, "po-table-header-fixed-inner"], ["class", "po-table-column po-table-column-single-action", 3, "width", "max-width", 4, "ngIf"], ["class", "po-table-column-actions", 4, "ngIf"], [1, "po-table-column", "po-table-column-single-action"], ["class", "po-table-single-action po-clickable", 3, "po-table-action-disabled", "click", 4, "ngIf"], [1, "po-table-single-action", "po-clickable", 3, "click"], ["class", "po-table-single-action-content", 3, "p-icon", 4, "ngIf"], [1, "po-table-single-action-content", 3, "p-icon"], [1, "po-table-column-actions"], [1, "po-icon", "po-icon-more", "po-clickable", 3, "click"], ["popupTarget", ""], [3, "p-columns", "p-max-columns", "p-target", "p-last-visible-columns-selected", "p-hide-action-fixed-columns", "p-columns-default", "p-visible-columns-change", "p-change-visible-columns", "p-initial-columns"]], template: function PoTableComponent_Template(rf, ctx) { if (rf & 1) {
        i0.ɵɵelementStart(0, "div", 0);
        i0.ɵɵtemplate(1, PoTableComponent_div_1_Template, 9, 4, "div", 1);
        i0.ɵɵtemplate(2, PoTableComponent_div_2_Template, 4, 2, "div", 2);
        i0.ɵɵtemplate(3, PoTableComponent_div_3_Template, 2, 3, "div", 3);
        i0.ɵɵelementEnd();
        i0.ɵɵtemplate(4, PoTableComponent_po_container_4_Template, 2, 1, "po-container", 4);
        i0.ɵɵtemplate(5, PoTableComponent_ng_template_5_Template, 8, 10, "ng-template", null, 5, i0.ɵɵtemplateRefExtractor);
        i0.ɵɵtemplate(7, PoTableComponent_div_7_Template, 3, 5, "div", 6);
        i0.ɵɵtemplate(8, PoTableComponent_ng_template_8_Template, 16, 23, "ng-template", null, 7, i0.ɵɵtemplateRefExtractor);
        i0.ɵɵtemplate(10, PoTableComponent_ng_template_10_Template, 18, 29, "ng-template", null, 8, i0.ɵɵtemplateRefExtractor);
        i0.ɵɵelement(12, "po-popup", 9, 10);
        i0.ɵɵtemplate(14, PoTableComponent_ng_template_14_Template, 1, 1, "ng-template", null, 11, i0.ɵɵtemplateRefExtractor);
        i0.ɵɵtemplate(16, PoTableComponent_ng_template_16_Template, 1, 2, "ng-template", null, 12, i0.ɵɵtemplateRefExtractor);
        i0.ɵɵtemplate(18, PoTableComponent_ng_template_18_Template, 1, 1, "ng-template", null, 13, i0.ɵɵtemplateRefExtractor);
        i0.ɵɵtemplate(20, PoTableComponent_ng_template_20_Template, 5, 6, "ng-template", null, 14, i0.ɵɵtemplateRefExtractor);
        i0.ɵɵtemplate(22, PoTableComponent_ng_template_22_Template, 2, 3, "ng-template", null, 15, i0.ɵɵtemplateRefExtractor);
        i0.ɵɵtemplate(24, PoTableComponent_ng_template_24_Template, 1, 1, "ng-template", null, 16, i0.ɵɵtemplateRefExtractor);
        i0.ɵɵtemplate(26, PoTableComponent_ng_template_26_Template, 2, 2, "ng-template", null, 17, i0.ɵɵtemplateRefExtractor);
        i0.ɵɵtemplate(28, PoTableComponent_po_table_column_manager_28_Template, 1, 6, "po-table-column-manager", 18);
        i0.ɵɵelementStart(29, "po-modal", 19, 20)(31, "p", 21);
        i0.ɵɵtext(32);
        i0.ɵɵelementEnd()();
    } if (rf & 2) {
        const _r4 = i0.ɵɵreference(6);
        i0.ɵɵadvance(1);
        i0.ɵɵproperty("ngIf", ctx.hasValidColumns && ctx.itemsSelected.length > 0 && !ctx.hideBatchActions);
        i0.ɵɵadvance(1);
        i0.ɵɵproperty("ngIf", ctx.hasValidColumns && !ctx.hideColumnsManager);
        i0.ɵɵadvance(1);
        i0.ɵɵproperty("ngIf", !ctx.hideTableSearch && ctx.hasValidColumns);
        i0.ɵɵadvance(1);
        i0.ɵɵproperty("ngIf", ctx.container)("ngIfElse", _r4);
        i0.ɵɵadvance(3);
        i0.ɵɵproperty("ngIf", !ctx.infiniteScroll);
        i0.ɵɵadvance(5);
        i0.ɵɵproperty("p-actions", ctx.actions)("p-target", ctx.popupTarget);
        i0.ɵɵadvance(16);
        i0.ɵɵproperty("ngIf", !ctx.hideColumnsManager);
        i0.ɵɵadvance(1);
        i0.ɵɵproperty("p-title", ctx.literals.delete)("p-primary-action", ctx.confirm)("p-secondary-action", ctx.close)("p-click-out", true);
        i0.ɵɵadvance(3);
        i0.ɵɵtextInterpolate(ctx.literals.bodyDelete);
    } }, encapsulation: 2 });
(function () { (typeof ngDevMode === "undefined" || ngDevMode) && i0.ɵsetClassMetadata(PoTableComponent, [{
        type: Component,
        args: [{ selector: 'po-table', providers: [PoDateService], template: "<div class=\"po-table-actions\">\r\n  <div\r\n    #columnBatchActions\r\n    *ngIf=\"hasValidColumns && itemsSelected.length > 0 && !hideBatchActions\"\r\n    class=\"po-table-actions-batch-actions\"\r\n  >\r\n    <div [ngPlural]=\"itemsSelected.length\" class=\"po-table-actions-batch-actions__label\">\r\n      <strong>\r\n        <ng-template ngPluralCase=\"=0\">{{ literals.noItem }}</ng-template>\r\n        <ng-template ngPluralCase=\"=1\">{{ literals.oneItem }}</ng-template>\r\n        <ng-template ngPluralCase=\"other\">{{ itemsSelected.length }} {{ literals.multipleItems }}</ng-template>\r\n      </strong>\r\n    </div>\r\n\r\n    <div class=\"po-table-actions-batch-actions__buttons\">\r\n      <po-button\r\n        p-icon=\"po-icon-delete\"\r\n        [p-danger]=\"true\"\r\n        [p-disabled]=\"itemsSelected.length > 1 && serviceDeleteApi !== undefined\"\r\n        [p-label]=\"literals.delete\"\r\n        (p-click)=\"modalDelete.open()\"\r\n      ></po-button>\r\n    </div>\r\n  </div>\r\n\r\n  <div #columnManager *ngIf=\"hasValidColumns && !hideColumnsManager\" class=\"po-table-actions-column-manager\">\r\n    <po-button\r\n      #columnManagerTarget\r\n      p-icon=\"po-icon-settings\"\r\n      p-kind=\"tertiary\"\r\n      p-tooltip-position=\"left\"\r\n      [p-aria-label]=\"literals.columnsManager\"\r\n      [p-tooltip]=\"literals.columnsManager\"\r\n      (p-click)=\"onOpenColumnManager()\"\r\n    ></po-button>\r\n  </div>\r\n\r\n  <div *ngIf=\"!hideTableSearch && hasValidColumns\" class=\"po-table-search\">\r\n    <po-search\r\n      [p-items]=\"items\"\r\n      [p-filter-keys]=\"filteredColumns\"\r\n      [p-filter-type]=\"filterType\"\r\n      (p-filtered-items-change)=\"onFilteredItemsChange($event)\"\r\n    >\r\n    </po-search>\r\n  </div>\r\n</div>\r\n<po-container *ngIf=\"container; else tableContainerTemplate\" p-no-padding>\r\n  <ng-container *ngTemplateOutlet=\"tableContainerTemplate\"></ng-container>\r\n</po-container>\r\n\r\n<ng-template #tableContainerTemplate>\r\n  <div [class.po-table-container-relative]=\"loading\">\r\n    <div *ngIf=\"loading\" class=\"po-table-overlay\">\r\n      <po-loading class=\"po-table-overlay-content\" [p-text]=\"literals.loadingData\"></po-loading>\r\n    </div>\r\n\r\n    <div class=\"po-table-main-container\">\r\n      <div\r\n        #tableWrapper\r\n        class=\"po-table-wrapper\"\r\n        [class.po-table-header-fixed-columns-pixels]=\"allColumnsWidthPixels\"\r\n        [style.opacity]=\"tableOpacity\"\r\n      >\r\n        <div *ngIf=\"height\" class=\"po-table-container\" [style.height.px]=\"heightTableContainer\">\r\n          <div #poTableTbody class=\"po-table-container-fixed-inner\">\r\n            <ng-container *ngTemplateOutlet=\"tableVirtualScrollTemplate\"></ng-container>\r\n          </div>\r\n        </div>\r\n\r\n        <div *ngIf=\"!height\">\r\n          <ng-container *ngTemplateOutlet=\"tableTemplate\"></ng-container>\r\n        </div>\r\n      </div>\r\n    </div>\r\n  </div>\r\n\r\n  <div class=\"po-table-footer\" *ngIf=\"hasFooter\">\r\n    <ng-container *ngFor=\"let column of subtitleColumns; trackBy: trackBy\">\r\n      <po-table-subtitle-footer [p-literals]=\"literals\" [p-subtitles]=\"column.subtitles\"> </po-table-subtitle-footer>\r\n    </ng-container>\r\n  </div>\r\n</ng-template>\r\n\r\n<!-- Show More Button -->\r\n<div\r\n  *ngIf=\"!infiniteScroll\"\r\n  class=\"po-row po-table-footer-show-more\"\r\n  [class.po-invisible]=\"showMore.observers.length === 0 && !hasService\"\r\n  #tableFooter\r\n>\r\n  <po-button\r\n    class=\"po-offset-xl-4 po-offset-lg-4 po-offset-md-3 po-lg-4 po-md-6\"\r\n    [p-disabled]=\"showMoreDisabled\"\r\n    [p-label]=\"literals.loadMoreData\"\r\n    [p-loading]=\"loadingShowMore\"\r\n    (p-click)=\"onShowMore()\"\r\n  >\r\n  </po-button>\r\n</div>\r\n\r\n<!-- Table default-->\r\n<ng-template #tableTemplate>\r\n  <table\r\n    class=\"po-table\"\r\n    [ngClass]=\"{\r\n      'po-table-interactive': selectable || sort,\r\n      'po-table-selectable': selectable,\r\n      'po-table-striped': striped\r\n    }\"\r\n    [attr.p-spacing]=\"spacing\"\r\n  >\r\n    <thead>\r\n      <tr\r\n        [ngClass]=\"{ 'no-hover': hideSelectAll, 'po-table-column-drag': this.isDraggable }\"\r\n        [class.po-table-header]=\"!height\"\r\n        cdkDropList\r\n        cdkDropListOrientation=\"horizontal\"\r\n        (cdkDropListDropped)=\"drop($event)\"\r\n      >\r\n        <th\r\n          *ngIf=\"hasSelectableColumn\"\r\n          [style.pointer-events]=\"hideSelectAll ? 'none' : 'auto'\"\r\n          class=\"po-table-column-selectable\"\r\n        >\r\n          <div [class.po-table-header-fixed-inner]=\"height\">\r\n            <po-checkbox\r\n              name=\"selectAll\"\r\n              *ngIf=\"!hideSelectAll\"\r\n              (p-change)=\"selectAllRows()\"\r\n              [p-checkboxValue]=\"selectAll === null ? 'mixed' : selectAll\"\r\n            ></po-checkbox>\r\n          </div>\r\n        </th>\r\n\r\n        <th\r\n          *ngIf=\"(hasMasterDetailColumn || hasRowTemplate) && !hasRowTemplateWithArrowDirectionRight\"\r\n          class=\"po-table-header-column po-table-header-master-detail\"\r\n        ></th>\r\n\r\n        <!-- Coluna criada para caso as a\u00E7\u00F5es fiquem no lado esquerdo -->\r\n        <th\r\n          #columnActionLeft\r\n          *ngIf=\"!actionRight && (visibleActions.length > 1 || isSingleAction)\"\r\n          [class.po-table-header-master-detail]=\"!isSingleAction\"\r\n          [class.po-table-header-single-action]=\"isSingleAction\"\r\n        ></th>\r\n\r\n        <th *ngIf=\"!hasMainColumns\" #noColumnsHeader class=\"po-table-header-column po-text-center\">\r\n          <ng-container *ngIf=\"height; then noColumnsWithHeight; else noColumnsWithoutHeight\"> </ng-container>\r\n        </th>\r\n\r\n        <ng-container\r\n          *ngIf=\"this.isDraggable || hasSomeFixed(); then tableDefaultThDragDrop; else tableDefaultThDefault\"\r\n        >\r\n        </ng-container>\r\n        <ng-template #tableDefaultThDragDrop>\r\n          <th\r\n            *ngFor=\"let column of mainColumns; let i = index; trackBy: trackBy\"\r\n            class=\"po-table-header-ellipsis p-element po-frozen-column\"\r\n            [style.width]=\"column.width\"\r\n            [style.max-width]=\"column.width\"\r\n            [style.min-width]=\"column.width\"\r\n            [attr.data-po-table-column-name]=\"column.label || (column.property | titlecase) | lowercase\"\r\n            [class.po-clickable]=\"(sort && column.sortable !== false) || hasService\"\r\n            [ngClass]=\"{\r\n              'po-table-header-sorted':\r\n                sort &&\r\n                JSON.stringify(sortedColumn?.property) === JSON.stringify(column) &&\r\n                (sortedColumn.ascending || !sortedColumn.ascending)\r\n            }\"\r\n            [class.po-table-header-subtitle]=\"column.type === 'subtitle'\"\r\n            [class.po-table-column-drag-box]=\"this.isDraggable\"\r\n            (click)=\"sortColumn(column)\"\r\n            cdkDrag\r\n            cdkDragLockAxis=\"x\"\r\n            [cdkDragDisabled]=\"column.fixed ? 'true' : 'false'\"\r\n            [pFrozenColumn]=\"column.fixed\"\r\n          >\r\n            <div\r\n              class=\"po-table-header-flex\"\r\n              [class.po-table-header-fixed-inner]=\"height\"\r\n              [class.po-table-header-flex-right]=\"column.type === 'currency' || column.type === 'number'\"\r\n              [class.po-table-header-flex-center]=\"column.type === 'subtitle'\"\r\n            >\r\n              <ng-container *ngIf=\"this.isDraggable && !column.fixed\">\r\n                <svg\r\n                  cdkDragHandle\r\n                  width=\"24\"\r\n                  height=\"24\"\r\n                  viewBox=\"0 0 24 24\"\r\n                  fill=\"none\"\r\n                  xmlns=\"http://www.w3.org/2000/svg\"\r\n                >\r\n                  <circle cx=\"9\" cy=\"6\" r=\"2\" fill=\"black\" />\r\n                  <circle cx=\"15\" cy=\"6\" r=\"2\" fill=\"black\" />\r\n                  <circle cx=\"9\" cy=\"12\" r=\"2\" fill=\"black\" />\r\n                  <circle cx=\"15\" cy=\"12\" r=\"2\" fill=\"black\" />\r\n                  <circle cx=\"9\" cy=\"18\" r=\"2\" fill=\"black\" />\r\n                  <circle cx=\"15\" cy=\"18\" r=\"2\" fill=\"black\" />\r\n                </svg>\r\n              </ng-container>\r\n              <ng-container *ngTemplateOutlet=\"contentHeaderTemplate; context: { $implicit: column }\"> </ng-container>\r\n            </div>\r\n          </th>\r\n        </ng-template>\r\n        <ng-template #tableDefaultThDefault>\r\n          <th\r\n            *ngFor=\"let column of mainColumns; let i = index; trackBy: trackBy\"\r\n            class=\"po-table-header-ellipsis p-element po-frozen-column\"\r\n            [style.width]=\"column.width\"\r\n            [style.max-width]=\"column.width\"\r\n            [style.min-width]=\"column.width\"\r\n            [attr.data-po-table-column-name]=\"column.label || (column.property | titlecase) | lowercase\"\r\n            [class.po-clickable]=\"(sort && column.sortable !== false) || hasService\"\r\n            [ngClass]=\"{\r\n              'po-table-header-sorted':\r\n                sort &&\r\n                JSON.stringify(sortedColumn?.property) === JSON.stringify(column) &&\r\n                (sortedColumn.ascending || !sortedColumn.ascending)\r\n            }\"\r\n            [class.po-table-header-subtitle]=\"column.type === 'subtitle'\"\r\n            (click)=\"sortColumn(column)\"\r\n            [pFrozenColumn]=\"column.fixed\"\r\n          >\r\n            <div\r\n              class=\"po-table-header-flex\"\r\n              [class.po-table-header-fixed-inner]=\"height\"\r\n              [class.po-table-header-flex-right]=\"column.type === 'currency' || column.type === 'number'\"\r\n              [class.po-table-header-flex-center]=\"column.type === 'subtitle'\"\r\n            >\r\n              <ng-container *ngTemplateOutlet=\"contentHeaderTemplate; context: { $implicit: column }\"> </ng-container>\r\n            </div>\r\n          </th>\r\n        </ng-template>\r\n\r\n        <th\r\n          *ngIf=\"hasRowTemplateWithArrowDirectionRight && (hasVisibleActions || hideColumnsManager)\"\r\n          class=\"po-table-header-column po-table-header-master-detail\"\r\n        ></th>\r\n\r\n        <th\r\n          *ngIf=\"hasVisibleActions && actionRight && (visibleActions.length > 1 || isSingleAction)\"\r\n          [class.po-table-header-single-action]=\"isSingleAction\"\r\n          [class.po-table-header-actions]=\"!isSingleAction\"\r\n        ></th>\r\n      </tr>\r\n    </thead>\r\n\r\n    <tbody class=\"po-table-group-row\" *ngIf=\"!hasItems || !hasMainColumns\">\r\n      <tr class=\"po-table-row po-table-row-no-data\">\r\n        <td [colSpan]=\"columnCount\" class=\"po-table-no-data po-text-center\">\r\n          <span> {{ literals.noData }} </span>\r\n        </td>\r\n      </tr>\r\n    </tbody>\r\n\r\n    <ng-container *ngIf=\"hasMainColumns\">\r\n      <tbody class=\"po-table-group-row\" *ngFor=\"let row of filteredItems; let rowIndex = index; trackBy: trackBy\">\r\n        <tr class=\"po-table-row\" [class.po-table-row-active]=\"row.$selected || (row.$selected === null && selectable)\">\r\n          <td *ngIf=\"selectable\" class=\"po-table-column-selectable\">\r\n            <ng-container *ngTemplateOutlet=\"singleSelect ? inputRadio : inputCheckbox; context: { $implicit: row }\">\r\n            </ng-container>\r\n          </td>\r\n\r\n          <!-- Valida se a origem do detail \u00E9 pelo input do po-table -->\r\n          <td\r\n            *ngIf=\"columnMasterDetail && !hideDetail && !hasRowTemplate\"\r\n            class=\"po-table-column-detail-toggle\"\r\n            (click)=\"toggleDetail(row)\"\r\n          >\r\n            <ng-template\r\n              [ngTemplateOutlet]=\"poTableColumnDetail\"\r\n              [ngTemplateOutletContext]=\"{ row: row, rowIndex: rowIndex }\"\r\n            >\r\n            </ng-template>\r\n          </td>\r\n\r\n          <!-- Coluna com as a\u00E7\u00F5es na esquerda (padr\u00E3o)-->\r\n          <ng-template\r\n            *ngIf=\"!actionRight && (visibleActions.length > 1 || isSingleAction)\"\r\n            [ngTemplateOutlet]=\"ActionsColumnTemplate\"\r\n            [ngTemplateOutletContext]=\"{ row: row, rowIndex: rowIndex }\"\r\n          >\r\n          </ng-template>\r\n\r\n          <!-- Valida se a origem do detail \u00E9 pela diretiva -->\r\n          <td\r\n            *ngIf=\"hasRowTemplate && !hasRowTemplateWithArrowDirectionRight\"\r\n            class=\"po-table-column-detail-toggle\"\r\n            (click)=\"toggleDetail(row)\"\r\n          >\r\n            <ng-template\r\n              [ngTemplateOutlet]=\"poTableColumnDetail\"\r\n              [ngTemplateOutletContext]=\"{ row: row, rowIndex: rowIndex }\"\r\n            >\r\n            </ng-template>\r\n          </td>\r\n\r\n          <td\r\n            *ngFor=\"let column of mainColumns; let columnIndex = index; trackBy: trackBy\"\r\n            [style.width]=\"column.width\"\r\n            [style.max-width]=\"column.width\"\r\n            [style.min-width]=\"column.width\"\r\n            [class.po-table-column]=\"column.type !== 'icon'\"\r\n            [class.po-table-column-right]=\"column.type === 'currency' || column.type === 'number'\"\r\n            [class.po-table-column-center]=\"column.type === 'subtitle'\"\r\n            [class.po-table-column-icons]=\"column.type === 'icon'\"\r\n            [pFrozenColumn]=\"column.fixed\"\r\n            class=\"p-element po-frozen-column\"\r\n            [ngClass]=\"getClassColor(row, column)\"\r\n            (click)=\"hasSelectableRow() ? selectRow(row) : 'javascript:;'\"\r\n          >\r\n            <div\r\n              class=\"po-table-column-cell po-table-body-ellipsis notranslate\"\r\n              [ngSwitch]=\"column.type\"\r\n              [p-tooltip]=\"tooltipText\"\r\n              [p-append-in-body]=\"true\"\r\n              (mouseenter)=\"tooltipMouseEnter($event, column, row)\"\r\n              (mouseleave)=\"tooltipMouseLeave()\"\r\n            >\r\n              <span *ngSwitchCase=\"'columnTemplate'\">\r\n                <ng-container *ngTemplateOutlet=\"getTemplate(column); context: { $implicit: getCellData(row, column) }\">\r\n                </ng-container>\r\n              </span>\r\n\r\n              <span *ngSwitchCase=\"'cellTemplate'\">\r\n                <ng-container *ngTemplateOutlet=\"tableCellTemplate?.templateRef; context: { row: row, column: column }\">\r\n                </ng-container>\r\n              </span>\r\n\r\n              <span *ngSwitchCase=\"'boolean'\">\r\n                {{ getBooleanLabel(getCellData(row, column), column) }}\r\n              </span>\r\n\r\n              <span *ngSwitchCase=\"'currency'\">\r\n                {{ getCellData(row, column) | currency: column.format:'symbol':'1.2-2' }}\r\n              </span>\r\n\r\n              <span *ngSwitchCase=\"'date'\">\r\n                {{ getCellData(row, column) | date: column.format || 'dd/MM/yyyy' }}\r\n              </span>\r\n\r\n              <span *ngSwitchCase=\"'time'\">\r\n                {{ getCellData(row, column) | po_time: column.format || 'HH:mm:ss.ffffff' }}\r\n              </span>\r\n\r\n              <span *ngSwitchCase=\"'dateTime'\">\r\n                {{ getCellData(row, column) | date: column.format || 'dd/MM/yyyy HH:mm:ss' }}\r\n              </span>\r\n\r\n              <span *ngSwitchCase=\"'number'\">\r\n                {{ formatNumber(getCellData(row, column), column.format) }}\r\n              </span>\r\n\r\n              <po-table-column-link\r\n                *ngSwitchCase=\"'link'\"\r\n                [p-action]=\"column.action\"\r\n                [p-disabled]=\"checkDisabled(row, column)\"\r\n                [p-link]=\"row[column.link]\"\r\n                [p-row]=\"row\"\r\n                [p-value]=\"getCellData(row, column)\"\r\n                (click)=\"onClickLink($event, row, column)\"\r\n              >\r\n              </po-table-column-link>\r\n\r\n              <po-table-column-icon\r\n                *ngSwitchCase=\"'icon'\"\r\n                [p-column]=\"column\"\r\n                [p-icons]=\"getColumnIcons(row, column)\"\r\n                [p-row]=\"row\"\r\n              >\r\n              </po-table-column-icon>\r\n\r\n              <span *ngSwitchCase=\"'subtitle'\">\r\n                <po-table-subtitle-circle [p-subtitle]=\"getSubtitleColumn(row, column)\"></po-table-subtitle-circle>\r\n              </span>\r\n              <span *ngSwitchCase=\"'label'\">\r\n                <po-table-column-label [p-value]=\"getColumnLabel(row, column)\"> </po-table-column-label>\r\n              </span>\r\n              <span *ngSwitchDefault>{{ getCellData(row, column) }}</span>\r\n            </div>\r\n          </td>\r\n\r\n          <td\r\n            *ngIf=\"hasRowTemplateWithArrowDirectionRight\"\r\n            class=\"po-table-column-detail-toggle\"\r\n            (click)=\"toggleDetail(row)\"\r\n          >\r\n            <ng-template\r\n              [ngTemplateOutlet]=\"poTableColumnDetail\"\r\n              [ngTemplateOutletContext]=\"{ row: row, rowIndex: rowIndex }\"\r\n            >\r\n            </ng-template>\r\n          </td>\r\n\r\n          <!-- Coluna de a\u00E7oes na direita -->\r\n          <ng-template\r\n            *ngIf=\"actionRight\"\r\n            [ngTemplateOutlet]=\"ActionsColumnTemplate\"\r\n            [ngTemplateOutletContext]=\"{ row: row, rowIndex: rowIndex }\"\r\n          >\r\n          </ng-template>\r\n        </tr>\r\n\r\n        <tr *ngIf=\"hasMainColumns && hasRowTemplate && row.$showDetail && isShowRowTemplate(row, rowIndex)\">\r\n          <td class=\"po-table-row-template-container\" [colSpan]=\"columnCountForMasterDetail\">\r\n            <ng-template\r\n              [ngTemplateOutlet]=\"tableRowTemplate.templateRef\"\r\n              [ngTemplateOutletContext]=\"{ $implicit: row, rowIndex: rowIndex }\"\r\n            >\r\n            </ng-template>\r\n          </td>\r\n        </tr>\r\n\r\n        <tr *ngIf=\"hasMainColumns && isShowMasterDetail(row)\">\r\n          <td class=\"po-table-column-detail\" [colSpan]=\"columnCountForMasterDetail\">\r\n            <po-table-detail\r\n              [p-selectable]=\"selectable && !detailHideSelect\"\r\n              [p-detail]=\"columnMasterDetail.detail\"\r\n              [p-items]=\"row[nameColumnDetail]\"\r\n              (p-select-row)=\"selectDetailRow($event)\"\r\n            >\r\n            </po-table-detail>\r\n          </td>\r\n        </tr>\r\n      </tbody>\r\n    </ng-container>\r\n  </table>\r\n</ng-template>\r\n\r\n<!-- Table with virtual scroll -->\r\n<ng-template #tableVirtualScrollTemplate>\r\n  <cdk-virtual-scroll-viewport\r\n    #tableVirtualScroll\r\n    [itemSize]=\"itemSize\"\r\n    [style.height.px]=\"heightTableContainer\"\r\n    [minBufferPx]=\"heightTableContainer < 100 ? 100 : heightTableContainer\"\r\n    [maxBufferPx]=\"heightTableContainer < 200 ? 200 : heightTableContainer\"\r\n  >\r\n    <table\r\n      class=\"po-table\"\r\n      [ngClass]=\"{\r\n        'po-table-interactive': selectable || sort,\r\n        'po-table-selectable': selectable,\r\n        'po-table-striped': striped\r\n      }\"\r\n      [ngStyle]=\"{ 'table-layout': !hasItems ? 'fixed' : 'auto' }\"\r\n      [attr.p-spacing]=\"spacing\"\r\n    >\r\n      <thead class=\"po-table-header-sticky\" [style.top]=\"inverseOfTranslation\">\r\n        <tr\r\n          [class.po-table-header]=\"!height\"\r\n          cdkDropList\r\n          cdkDropListOrientation=\"horizontal\"\r\n          (cdkDropListDropped)=\"drop($event)\"\r\n        >\r\n          <th\r\n            *ngIf=\"hasSelectableColumn\"\r\n            [style.pointer-events]=\"hideSelectAll ? 'none' : 'auto'\"\r\n            class=\"po-table-column-selectable\"\r\n          >\r\n            <div [class.po-table-header-fixed-inner]=\"height\">\r\n              <po-checkbox\r\n                name=\"selectAll\"\r\n                *ngIf=\"!hideSelectAll\"\r\n                (p-change)=\"selectAllRows()\"\r\n                [p-checkboxValue]=\"selectAll === null ? 'mixed' : selectAll\"\r\n              ></po-checkbox>\r\n            </div>\r\n          </th>\r\n\r\n          <th\r\n            *ngIf=\"(hasMasterDetailColumn || hasRowTemplate) && !hasRowTemplateWithArrowDirectionRight\"\r\n            class=\"po-table-header-column po-table-header-master-detail\"\r\n          ></th>\r\n\r\n          <!-- Coluna criada para caso as a\u00E7\u00F5es fiquem no lado esquerdo -->\r\n          <th\r\n            #columnActionLeft\r\n            *ngIf=\"!actionRight && (visibleActions.length > 1 || isSingleAction)\"\r\n            [class.po-table-header-master-detail]=\"!isSingleAction\"\r\n            [class.po-table-header-single-action]=\"isSingleAction\"\r\n          ></th>\r\n\r\n          <th *ngIf=\"!hasMainColumns\" #noColumnsHeader class=\"po-table-header-column po-text-center\">\r\n            <ng-container *ngIf=\"height; then noColumnsWithHeight; else noColumnsWithoutHeight\"> </ng-container>\r\n          </th>\r\n\r\n          <ng-container\r\n            *ngIf=\"\r\n              this.isDraggable || hasSomeFixed();\r\n              then tableVirtualScrollThDragDrop;\r\n              else tableVirtualScrollThDefault\r\n            \"\r\n          >\r\n          </ng-container>\r\n          <ng-template #tableVirtualScrollThDragDrop>\r\n            <th\r\n              *ngFor=\"let column of mainColumns; let i = index; trackBy: trackBy\"\r\n              class=\"po-table-header-ellipsis p-element po-frozen-column\"\r\n              [style.width]=\"column.width\"\r\n              [style.max-width]=\"column.width\"\r\n              [style.min-width]=\"column.width\"\r\n              [attr.data-po-table-column-name]=\"column.label || (column.property | titlecase) | lowercase\"\r\n              [class.po-clickable]=\"(sort && column.sortable !== false) || hasService\"\r\n              [ngClass]=\"{\r\n                'po-table-header-sorted':\r\n                  sort &&\r\n                  JSON.stringify(sortedColumn?.property) === JSON.stringify(column) &&\r\n                  (sortedColumn.ascending || !sortedColumn.ascending)\r\n              }\"\r\n              [ngStyle]=\"{ 'width': !hasItems ? '100%' : 'auto' }\"\r\n              [class.po-table-header-subtitle]=\"column.type === 'subtitle'\"\r\n              [class.po-table-column-drag-box]=\"this.isDraggable\"\r\n              (click)=\"sortColumn(column)\"\r\n              cdkDrag\r\n              cdkDragLockAxis=\"x\"\r\n              [cdkDragDisabled]=\"column.fixed ? 'true' : 'false'\"\r\n              [pFrozenColumn]=\"column.fixed\"\r\n            >\r\n              <div\r\n                class=\"po-table-header-flex\"\r\n                [class.po-table-header-fixed-inner]=\"height\"\r\n                [class.po-table-header-flex-right]=\"column.type === 'currency' || column.type === 'number'\"\r\n                [class.po-table-header-flex-center]=\"column.type === 'subtitle'\"\r\n              >\r\n                <ng-container *ngIf=\"this.isDraggable && !column.fixed\">\r\n                  <svg\r\n                    cdkDragHandle\r\n                    width=\"24\"\r\n                    height=\"24\"\r\n                    viewBox=\"0 0 24 24\"\r\n                    fill=\"none\"\r\n                    xmlns=\"http://www.w3.org/2000/svg\"\r\n                  >\r\n                    <circle cx=\"9\" cy=\"6\" r=\"2\" fill=\"black\" />\r\n                    <circle cx=\"15\" cy=\"6\" r=\"2\" fill=\"black\" />\r\n                    <circle cx=\"9\" cy=\"12\" r=\"2\" fill=\"black\" />\r\n                    <circle cx=\"15\" cy=\"12\" r=\"2\" fill=\"black\" />\r\n                    <circle cx=\"9\" cy=\"18\" r=\"2\" fill=\"black\" />\r\n                    <circle cx=\"15\" cy=\"18\" r=\"2\" fill=\"black\" />\r\n                  </svg>\r\n                </ng-container>\r\n                <ng-container *ngTemplateOutlet=\"contentHeaderTemplate; context: { $implicit: column }\"> </ng-container>\r\n              </div>\r\n            </th>\r\n          </ng-template>\r\n          <ng-template #tableVirtualScrollThDefault>\r\n            <th\r\n              *ngFor=\"let column of mainColumns; let i = index; trackBy: trackBy\"\r\n              class=\"po-table-header-ellipsis p-element po-frozen-column example-box\"\r\n              [style.width]=\"column.width\"\r\n              [style.max-width]=\"column.width\"\r\n              [style.min-width]=\"column.width\"\r\n              [attr.data-po-table-column-name]=\"column.label || (column.property | titlecase) | lowercase\"\r\n              [class.po-clickable]=\"(sort && column.sortable !== false) || hasService\"\r\n              [ngClass]=\"{\r\n                'po-table-header-sorted':\r\n                  sort &&\r\n                  JSON.stringify(sortedColumn?.property) === JSON.stringify(column) &&\r\n                  (sortedColumn.ascending || !sortedColumn.ascending)\r\n              }\"\r\n              [ngStyle]=\"{ 'width': !hasItems ? '100%' : 'auto' }\"\r\n              [class.po-table-header-subtitle]=\"column.type === 'subtitle'\"\r\n              (click)=\"sortColumn(column)\"\r\n              [pFrozenColumn]=\"column.fixed\"\r\n            >\r\n              <div\r\n                class=\"po-table-header-flex\"\r\n                [class.po-table-header-fixed-inner]=\"height\"\r\n                [class.po-table-header-flex-right]=\"column.type === 'currency' || column.type === 'number'\"\r\n                [class.po-table-header-flex-center]=\"column.type === 'subtitle'\"\r\n              >\r\n                <ng-container *ngTemplateOutlet=\"contentHeaderTemplate; context: { $implicit: column }\"> </ng-container>\r\n              </div>\r\n            </th>\r\n          </ng-template>\r\n\r\n          <th\r\n            *ngIf=\"hasRowTemplateWithArrowDirectionRight && (hasVisibleActions || hideColumnsManager)\"\r\n            class=\"po-table-header-column po-table-header-master-detail\"\r\n          ></th>\r\n\r\n          <th\r\n            *ngIf=\"hasVisibleActions && actionRight && (visibleActions.length > 1 || isSingleAction)\"\r\n            [class.po-table-header-single-action]=\"isSingleAction\"\r\n            [class.po-table-header-actions]=\"!isSingleAction\"\r\n          ></th>\r\n        </tr>\r\n      </thead>\r\n\r\n      <tbody class=\"po-table-group-row\" *ngIf=\"!hasItems || !hasMainColumns\">\r\n        <tr class=\"po-table-row po-table-row-no-data\">\r\n          <td [colSpan]=\"columnCount\" class=\"po-table-no-data po-text-center\">\r\n            <span> {{ literals.noData }} </span>\r\n          </td>\r\n        </tr>\r\n      </tbody>\r\n\r\n      <ng-container *ngIf=\"hasMainColumns\">\r\n        <tbody\r\n          class=\"po-table-group-row\"\r\n          *cdkVirtualFor=\"let row of filteredItems; let rowIndex = index; trackBy: trackBy\"\r\n        >\r\n          <tr\r\n            class=\"po-table-row\"\r\n            [class.po-table-row-active]=\"row.$selected || (row.$selected === null && selectable)\"\r\n          >\r\n            <td *ngIf=\"selectable\" class=\"po-table-column-selectable\">\r\n              <ng-container *ngTemplateOutlet=\"singleSelect ? inputRadio : inputCheckbox; context: { $implicit: row }\">\r\n              </ng-container>\r\n            </td>\r\n\r\n            <!-- Valida se a origem do detail \u00E9 pelo input do po-table -->\r\n            <td\r\n              *ngIf=\"columnMasterDetail && !hideDetail && !hasRowTemplate\"\r\n              class=\"po-table-column-detail-toggle\"\r\n              (click)=\"toggleDetail(row)\"\r\n            >\r\n              <ng-template\r\n                [ngTemplateOutlet]=\"poTableColumnDetail\"\r\n                [ngTemplateOutletContext]=\"{ row: row, rowIndex: rowIndex }\"\r\n              >\r\n              </ng-template>\r\n            </td>\r\n\r\n            <!-- Coluna com as a\u00E7\u00F5es na esquerda (padr\u00E3o)-->\r\n            <ng-template\r\n              *ngIf=\"!actionRight && (visibleActions.length > 1 || isSingleAction)\"\r\n              [ngTemplateOutlet]=\"ActionsColumnTemplate\"\r\n              [ngTemplateOutletContext]=\"{ row: row, rowIndex: rowIndex }\"\r\n            >\r\n            </ng-template>\r\n\r\n            <!-- Valida se a origem do detail \u00E9 pela diretiva -->\r\n            <td\r\n              *ngIf=\"hasRowTemplate && !hasRowTemplateWithArrowDirectionRight\"\r\n              class=\"po-table-column-detail-toggle\"\r\n              (click)=\"toggleDetail(row)\"\r\n            >\r\n              <ng-template\r\n                [ngTemplateOutlet]=\"poTableColumnDetail\"\r\n                [ngTemplateOutletContext]=\"{ row: row, rowIndex: rowIndex }\"\r\n              >\r\n              </ng-template>\r\n            </td>\r\n\r\n            <td\r\n              *ngFor=\"let column of mainColumns; let columnIndex = index; trackBy: trackBy\"\r\n              [style.width]=\"column.width\"\r\n              [style.max-width]=\"column.width\"\r\n              [style.min-width]=\"column.width\"\r\n              [class.po-table-column]=\"column.type !== 'icon'\"\r\n              [class.po-table-column-right]=\"column.type === 'currency' || column.type === 'number'\"\r\n              [class.po-table-column-center]=\"column.type === 'subtitle'\"\r\n              [class.po-table-column-icons]=\"column.type === 'icon'\"\r\n              [ngClass]=\"getClassColor(row, column)\"\r\n              [pFrozenColumn]=\"column.fixed\"\r\n              class=\"p-element po-frozen-column\"\r\n              (click)=\"hasSelectableRow() ? selectRow(row) : 'javascript:;'\"\r\n            >\r\n              <div\r\n                class=\"po-table-column-cell po-table-body-ellipsis notranslate\"\r\n                [ngSwitch]=\"column.type\"\r\n                [p-tooltip]=\"tooltipText\"\r\n                [p-append-in-body]=\"true\"\r\n                (mouseenter)=\"tooltipMouseEnter($event, column, row)\"\r\n                (mouseleave)=\"tooltipMouseLeave()\"\r\n              >\r\n                <span *ngSwitchCase=\"'columnTemplate'\">\r\n                  <ng-container\r\n                    *ngTemplateOutlet=\"getTemplate(column); context: { $implicit: getCellData(row, column) }\"\r\n                  >\r\n                  </ng-container>\r\n                </span>\r\n\r\n                <span *ngSwitchCase=\"'cellTemplate'\">\r\n                  <ng-container\r\n                    *ngTemplateOutlet=\"tableCellTemplate?.templateRef; context: { row: row, column: column }\"\r\n                  >\r\n                  </ng-container>\r\n                </span>\r\n\r\n                <span *ngSwitchCase=\"'boolean'\">\r\n                  {{ getBooleanLabel(getCellData(row, column), column) }}\r\n                </span>\r\n\r\n                <span *ngSwitchCase=\"'currency'\">\r\n                  {{ getCellData(row, column) | currency: column.format:'symbol':'1.2-2' }}\r\n                </span>\r\n\r\n                <span *ngSwitchCase=\"'date'\">\r\n                  {{ getCellData(row, column) | date: column.format || 'dd/MM/yyyy' }}\r\n                </span>\r\n\r\n                <span *ngSwitchCase=\"'time'\">\r\n                  {{ getCellData(row, column) | po_time: column.format || 'HH:mm:ss.ffffff' }}\r\n                </span>\r\n\r\n                <span *ngSwitchCase=\"'dateTime'\">\r\n                  {{ getCellData(row, column) | date: column.format || 'dd/MM/yyyy HH:mm:ss' }}\r\n                </span>\r\n\r\n                <span *ngSwitchCase=\"'number'\">\r\n                  {{ formatNumber(getCellData(row, column), column.format) }}\r\n                </span>\r\n\r\n                <po-table-column-link\r\n                  *ngSwitchCase=\"'link'\"\r\n                  [p-action]=\"column.action\"\r\n                  [p-disabled]=\"checkDisabled(row, column)\"\r\n                  [p-link]=\"row[column.link]\"\r\n                  [p-row]=\"row\"\r\n                  [p-value]=\"getCellData(row, column)\"\r\n                  (click)=\"onClickLink($event, row, column)\"\r\n                >\r\n                </po-table-column-link>\r\n\r\n                <po-table-column-icon\r\n                  *ngSwitchCase=\"'icon'\"\r\n                  [p-column]=\"column\"\r\n                  [p-icons]=\"getColumnIcons(row, column)\"\r\n                  [p-row]=\"row\"\r\n                >\r\n                </po-table-column-icon>\r\n\r\n                <span *ngSwitchCase=\"'subtitle'\">\r\n                  <po-table-subtitle-circle [p-subtitle]=\"getSubtitleColumn(row, column)\"></po-table-subtitle-circle>\r\n                </span>\r\n                <span *ngSwitchCase=\"'label'\">\r\n                  <po-table-column-label [p-value]=\"getColumnLabel(row, column)\"> </po-table-column-label>\r\n                </span>\r\n                <span *ngSwitchDefault>{{ getCellData(row, column) }}</span>\r\n              </div>\r\n            </td>\r\n\r\n            <td\r\n              *ngIf=\"hasRowTemplateWithArrowDirectionRight\"\r\n              class=\"po-table-column-detail-toggle\"\r\n              (click)=\"toggleDetail(row)\"\r\n            >\r\n              <ng-template\r\n                [ngTemplateOutlet]=\"poTableColumnDetail\"\r\n                [ngTemplateOutletContext]=\"{ row: row, rowIndex: rowIndex }\"\r\n              >\r\n              </ng-template>\r\n            </td>\r\n\r\n            <!-- Coluna de a\u00E7oes na direita -->\r\n            <ng-template\r\n              *ngIf=\"actionRight\"\r\n              [ngTemplateOutlet]=\"ActionsColumnTemplate\"\r\n              [ngTemplateOutletContext]=\"{ row: row, rowIndex: rowIndex }\"\r\n            >\r\n            </ng-template>\r\n          </tr>\r\n\r\n          <tr *ngIf=\"hasMainColumns && hasRowTemplate && row.$showDetail && isShowRowTemplate(row, rowIndex)\">\r\n            <td class=\"po-table-row-template-container\" [colSpan]=\"columnCountForMasterDetail\">\r\n              <ng-template\r\n                [ngTemplateOutlet]=\"tableRowTemplate.templateRef\"\r\n                [ngTemplateOutletContext]=\"{ $implicit: row, rowIndex: rowIndex }\"\r\n              >\r\n              </ng-template>\r\n            </td>\r\n          </tr>\r\n\r\n          <tr *ngIf=\"hasMainColumns && isShowMasterDetail(row)\">\r\n            <td class=\"po-table-column-detail\" [colSpan]=\"columnCountForMasterDetail\">\r\n              <po-table-detail\r\n                [p-selectable]=\"selectable && !detailHideSelect\"\r\n                [p-detail]=\"columnMasterDetail.detail\"\r\n                [p-items]=\"row[nameColumnDetail]\"\r\n                (p-select-row)=\"selectDetailRow($event)\"\r\n              >\r\n              </po-table-detail>\r\n            </td>\r\n          </tr>\r\n        </tbody>\r\n      </ng-container>\r\n    </table>\r\n  </cdk-virtual-scroll-viewport>\r\n</ng-template>\r\n\r\n<po-popup #popup [p-actions]=\"actions\" [p-target]=\"popupTarget\"> </po-popup>\r\n\r\n<ng-template #poTableColumnDetail let-row=\"row\" let-rowIndex=\"rowIndex\">\r\n  <span\r\n    *ngIf=\"(containsMasterDetail(row) && !hasRowTemplate) || (isShowRowTemplate(row, rowIndex) && hasRowTemplate)\"\r\n    class=\"po-icon po-clickable\"\r\n    [class.po-icon-arrow-up]=\"row.$showDetail\"\r\n    [class.po-icon-arrow-down]=\"!row.$showDetail\"\r\n  >\r\n  </span>\r\n</ng-template>\r\n\r\n<ng-template #inputRadio let-row>\r\n  <po-radio [name]=\"idRadio\" [p-checked]=\"row.$selected\" (p-change-selected)=\"selectRow(row)\"></po-radio>\r\n</ng-template>\r\n\r\n<ng-template #inputCheckbox let-row>\r\n  <po-checkbox\r\n    name=\"checkbox\"\r\n    (p-change)=\"selectable ? selectRow(row) : 'javascript:;'\"\r\n    [p-checkboxValue]=\"row.$selected\"\r\n  ></po-checkbox>\r\n</ng-template>\r\n\r\n<ng-template #contentHeaderTemplate let-column>\r\n  <span\r\n    #columnHeader\r\n    class=\"po-table-header-ellipsis\"\r\n    [p-tooltip]=\"tooltipText\"\r\n    [p-append-in-body]=\"true\"\r\n    (mouseenter)=\"tooltipMouseEnter($event)\"\r\n    (mouseleave)=\"tooltipMouseLeave()\"\r\n  >\r\n    {{ column.label || (column.property | titlecase) }}\r\n  </span>\r\n\r\n  <span\r\n    *ngIf=\"sort && column.sortable !== false\"\r\n    [class.po-table-header-icon-unselected]=\"JSON.stringify(sortedColumn?.property) !== JSON.stringify(column)\"\r\n    [class.po-table-header-icon-descending]=\"\r\n      JSON.stringify(sortedColumn?.property) === JSON.stringify(column) && sortedColumn.ascending\r\n    \"\r\n    [class.po-table-header-icon-ascending]=\"\r\n      JSON.stringify(sortedColumn?.property) === JSON.stringify(column) && !sortedColumn.ascending\r\n    \"\r\n  >\r\n    <ng-container *ngIf=\"JSON.stringify(sortedColumn?.property) !== JSON.stringify(column)\">\r\n      <svg width=\"24\" height=\"24\" viewBox=\"0 0 24 24\" fill=\"none\" xmlns=\"http://www.w3.org/2000/svg\">\r\n        <path\r\n          fill-rule=\"evenodd\"\r\n          clip-rule=\"evenodd\"\r\n          d=\"M18.2929 15.2929L17 16.5858L17 10C17 9.44772 16.5523 9 16 9C15.4477 9 15 9.44772 15 10L15 16.5858L13.7071 15.2929C13.3166 14.9024 12.6834 14.9024 12.2929 15.2929C11.9024 15.6834 11.9024 16.3166 12.2929 16.7071L15.2929 19.7071C15.6834 20.0976 16.3166 20.0976 16.7071 19.7071L19.7071 16.7071C20.0976 16.3166 20.0976 15.6834 19.7071 15.2929C19.3166 14.9024 18.6834 14.9024 18.2929 15.2929ZM5.70716 8.7071L7.00006 7.4142L7.00003 14C7.00002 14.5523 7.44774 15 8.00002 15C8.55231 15 9.00002 14.5523 9.00003 14L9.00006 7.41418L10.2929 8.70707C10.6835 9.09759 11.3166 9.09758 11.7072 8.70706C12.0977 8.31653 12.0977 7.68336 11.7072 7.29284L8.70718 4.29286C8.31665 3.90234 7.68349 3.90235 7.29296 4.29287L4.29295 7.29289C3.90242 7.68342 3.90242 8.31658 4.29294 8.70711C4.68347 9.09763 5.31663 9.09762 5.70716 8.7071Z\"\r\n          fill=\"#1D1D30\"\r\n        />\r\n      </svg>\r\n    </ng-container>\r\n\r\n    <ng-container *ngIf=\"JSON.stringify(sortedColumn?.property) === JSON.stringify(column) && sortedColumn.ascending\">\r\n      <svg width=\"24\" height=\"24\" viewBox=\"0 0 24 24\" fill=\"none\" xmlns=\"http://www.w3.org/2000/svg\">\r\n        <path\r\n          fill-rule=\"evenodd\"\r\n          clip-rule=\"evenodd\"\r\n          d=\"M11 9.41421L9.70711 10.7071C9.31658 11.0976 8.68342 11.0976 8.29289 10.7071C7.90237 10.3166 7.90237 9.68342 8.29289 9.29289L11.2929 6.29289C11.6834 5.90237 12.3166 5.90237 12.7071 6.29289L15.7071 9.29289C16.0976 9.68342 16.0976 10.3166 15.7071 10.7071C15.3166 11.0976 14.6834 11.0976 14.2929 10.7071L13 9.41421V16C13 16.5523 12.5523 17 12 17C11.4477 17 11 16.5523 11 16V9.41421Z\"\r\n          fill=\"black\"\r\n        />\r\n      </svg>\r\n    </ng-container>\r\n\r\n    <ng-container *ngIf=\"JSON.stringify(sortedColumn?.property) === JSON.stringify(column) && !sortedColumn.ascending\">\r\n      <svg width=\"24\" height=\"24\" viewBox=\"0 0 24 24\" fill=\"none\" xmlns=\"http://www.w3.org/2000/svg\">\r\n        <path\r\n          fill-rule=\"evenodd\"\r\n          clip-rule=\"evenodd\"\r\n          d=\"M9.70711 13.2929L11 14.5858V7C11 6.44772 11.4477 6 12 6C12.5523 6 13 6.44772 13 7V14.5858L14.2929 13.2929C14.6834 12.9024 15.3166 12.9024 15.7071 13.2929C16.0976 13.6834 16.0976 14.3166 15.7071 14.7071L12.7071 17.7071C12.3166 18.0976 11.6834 18.0976 11.2929 17.7071L8.29289 14.7071C7.90237 14.3166 7.90237 13.6834 8.29289 13.2929C8.68342 12.9024 9.31658 12.9024 9.70711 13.2929Z\"\r\n          fill=\"black\"\r\n        />\r\n      </svg>\r\n    </ng-container>\r\n  </span>\r\n</ng-template>\r\n\r\n<ng-template #noColumnsWithHeight>\r\n  <div class=\"po-table-header-fixed-inner\" [style.width.px]=\"noColumnsHeader?.nativeElement.offsetWidth\">\r\n    {{ hasValidColumns ? literals.noVisibleColumn : literals.noColumns }}\r\n  </div>\r\n</ng-template>\r\n\r\n<ng-template #noColumnsWithoutHeight>\r\n  {{ hasValidColumns ? literals.noVisibleColumn : literals.noColumns }}\r\n</ng-template>\r\n\r\n<!-- Template de a\u00E7\u00F5es -->\r\n<ng-template #ActionsColumnTemplate let-row=\"row\" let-rowIndex=\"rowIndex\">\r\n  <td\r\n    *ngIf=\"isSingleAction\"\r\n    class=\"po-table-column po-table-column-single-action\"\r\n    [style.width.px]=\"height && actionRight ? getWidthColumnManager() : ''\"\r\n    [style.max-width.px]=\"height && !actionRight ? getColumnWidthActionsLeft() : ''\"\r\n    [style.width.px]=\"height && !actionRight ? getColumnWidthActionsLeft() : ''\"\r\n  >\r\n    <div\r\n      *ngIf=\"firstAction.visible !== false\"\r\n      class=\"po-table-single-action po-clickable\"\r\n      [class.po-table-action-disabled]=\"firstAction.disabled ? validateTableAction(row, firstAction) : false\"\r\n      (click)=\"executeTableAction(row, firstAction)\"\r\n    >\r\n      <po-icon *ngIf=\"firstAction.icon\" class=\"po-table-single-action-content\" [p-icon]=\"firstAction.icon\"></po-icon>\r\n      {{ firstAction.label }}\r\n    </div>\r\n  </td>\r\n\r\n  <td *ngIf=\"visibleActions.length > 1\" class=\"po-table-column-actions\">\r\n    <span #popupTarget class=\"po-icon po-icon-more po-clickable\" (click)=\"togglePopup(row, popupTarget)\"></span>\r\n  </td>\r\n</ng-template>\r\n\r\n<po-table-column-manager\r\n  *ngIf=\"!hideColumnsManager\"\r\n  [p-columns]=\"columns\"\r\n  [p-max-columns]=\"maxColumns\"\r\n  [p-target]=\"columnManagerTarget\"\r\n  [p-last-visible-columns-selected]=\"lastVisibleColumnsSelected\"\r\n  [p-hide-action-fixed-columns]=\"hideActionFixedColumns\"\r\n  (p-visible-columns-change)=\"onVisibleColumnsChange($event)\"\r\n  (p-change-visible-columns)=\"onChangeVisibleColumns($event)\"\r\n  [p-columns-default]=\"initialColumns\"\r\n  (p-initial-columns)=\"onColumnRestoreManager($event)\"\r\n>\r\n</po-table-column-manager>\r\n\r\n<po-modal\r\n  #modalDelete\r\n  [p-title]=\"literals.delete\"\r\n  [p-primary-action]=\"confirm\"\r\n  [p-secondary-action]=\"close\"\r\n  [p-click-out]=\"true\"\r\n>\r\n  <p class=\"po-font-text-large\">{{ literals.bodyDelete }}</p>\r\n</po-modal>\r\n" }]
    }], function () { return [{ type: i1.PoDateService }, { type: i0.IterableDiffers }, { type: i0.Renderer2 }, { type: i2.PoLanguageService }, { type: i0.ChangeDetectorRef }, { type: i3.DecimalPipe }, { type: i4.Router }, { type: i5.PoTableService }]; }, { tableRowTemplate: [{
            type: ContentChild,
            args: [PoTableRowTemplateDirective, { static: true }]
        }], tableCellTemplate: [{
            type: ContentChild,
            args: [PoTableCellTemplateDirective]
        }], tableColumnTemplates: [{
            type: ContentChildren,
            args: [PoTableColumnTemplateDirective]
        }], noColumnsHeader: [{
            type: ViewChild,
            args: ['noColumnsHeader', { read: ElementRef }]
        }], poPopupComponent: [{
            type: ViewChild,
            args: ['popup']
        }], modalDelete: [{
            type: ViewChild,
            args: [PoModalComponent, { static: true }]
        }], tableFooterElement: [{
            type: ViewChild,
            args: ['tableFooter', { read: ElementRef, static: false }]
        }], tableWrapperElement: [{
            type: ViewChild,
            args: ['tableWrapper', { read: ElementRef, static: false }]
        }], tableTemplate: [{
            type: ViewChild,
            args: ['tableTemplate', { read: ElementRef, static: false }]
        }], tableVirtualScroll: [{
            type: ViewChild,
            args: ['tableVirtualScroll', { read: ElementRef, static: false }]
        }], columnManager: [{
            type: ViewChild,
            args: ['columnManager', { read: ElementRef, static: false }]
        }], columnBatchActions: [{
            type: ViewChild,
            args: ['columnBatchActions', { read: ElementRef, static: false }]
        }], columnActionLeft: [{
            type: ViewChild,
            args: ['columnActionLeft', { read: ElementRef, static: false }]
        }], actionsIconElement: [{
            type: ViewChildren,
            args: ['actionsIconElement', { read: ElementRef }]
        }], actionsElement: [{
            type: ViewChildren,
            args: ['actionsElement', { read: ElementRef }]
        }], filterInput: [{
            type: ViewChild,
            args: ['filterInput']
        }], poSearchInput: [{
            type: ViewChild,
            args: ['poSearchInput', { read: ElementRef, static: true }]
        }], viewPort: [{
            type: ViewChild,
            args: [CdkVirtualScrollViewport, { static: false }]
        }], columnManagerTarget: [{
            type: ViewChild,
            args: ['columnManagerTarget']
        }], columnManagerTargetFixed: [{
            type: ViewChild,
            args: ['columnManagerTargetFixed']
        }] }); })();
