// @po-ui/ng-components/esm2020/lib/components/po-menu/po-menu-item/po-menu-item.component.mjs
import { Component, Input, ViewChild } from '@angular/core';
import { convertToInt, convertToBoolean } from '../../../utils/util';
import * as i0 from "@angular/core";
import * as i1 from "../services/po-menu-items.service";
import * as i2 from "@angular/common";
import * as i3 from "@angular/router";
import * as i4 from "../../po-badge/po-badge.component";
import * as i5 from "../../po-icon/po-icon.component";
const _c0 = ["menuSubItems"];
function PoMenuItemComponent_a_0_ng_container_1_Template(rf, ctx) { if (rf & 1) {
    i0.ɵɵelementContainer(0);
} }
function PoMenuItemComponent_a_0_Template(rf, ctx) { if (rf & 1) {
    i0.ɵɵelementStart(0, "a", 6);
    i0.ɵɵtemplate(1, PoMenuItemComponent_a_0_ng_container_1_Template, 1, 0, "ng-container", 7);
    i0.ɵɵelementEnd();
} if (rf & 2) {
    const ctx_r0 = i0.ɵɵnextContext();
    const _r5 = i0.ɵɵreference(6);
    i0.ɵɵproperty("routerLink", ctx_r0.link);
    i0.ɵɵattribute("aria-label", ctx_r0.label);
    i0.ɵɵadvance(1);
    i0.ɵɵproperty("ngTemplateOutlet", _r5);
} }
function PoMenuItemComponent_a_1_ng_container_1_Template(rf, ctx) { if (rf & 1) {
    i0.ɵɵelementContainer(0);
} }
function PoMenuItemComponent_a_1_Template(rf, ctx) { if (rf & 1) {
    i0.ɵɵelementStart(0, "a", 8);
    i0.ɵɵtemplate(1, PoMenuItemComponent_a_1_ng_container_1_Template, 1, 0, "ng-container", 7);
    i0.ɵɵelementEnd();
} if (rf & 2) {
    const ctx_r1 = i0.ɵɵnextContext();
    const _r5 = i0.ɵɵreference(6);
    i0.ɵɵproperty("href", ctx_r1.link, i0.ɵɵsanitizeUrl);
    i0.ɵɵattribute("aria-label", ctx_r1.label);
    i0.ɵɵadvance(1);
    i0.ɵɵproperty("ngTemplateOutlet", _r5);
} }
function PoMenuItemComponent_a_2_ng_container_1_Template(rf, ctx) { if (rf & 1) {
    i0.ɵɵelementContainer(0);
} }
function PoMenuItemComponent_a_2_Template(rf, ctx) { if (rf & 1) {
    i0.ɵɵelementStart(0, "a", 9);
    i0.ɵɵtemplate(1, PoMenuItemComponent_a_2_ng_container_1_Template, 1, 0, "ng-container", 7);
    i0.ɵɵelementEnd();
} if (rf & 2) {
    const ctx_r2 = i0.ɵɵnextContext();
    const _r5 = i0.ɵɵreference(6);
    i0.ɵɵattribute("aria-label", ctx_r2.label);
    i0.ɵɵadvance(1);
    i0.ɵɵproperty("ngTemplateOutlet", _r5);
} }
function PoMenuItemComponent_div_3_ng_container_1_Template(rf, ctx) { if (rf & 1) {
    i0.ɵɵelementContainer(0);
} }
function PoMenuItemComponent_div_3_Template(rf, ctx) { if (rf & 1) {
    i0.ɵɵelementStart(0, "div", 10);
    i0.ɵɵtemplate(1, PoMenuItemComponent_div_3_ng_container_1_Template, 1, 0, "ng-container", 7);
    i0.ɵɵelementEnd();
} if (rf & 2) {
    i0.ɵɵnextContext();
    const _r5 = i0.ɵɵreference(6);
    i0.ɵɵadvance(1);
    i0.ɵɵproperty("ngTemplateOutlet", _r5);
} }
function PoMenuItemComponent_div_4_ng_container_1_Template(rf, ctx) { if (rf & 1) {
    i0.ɵɵelementContainer(0);
} }
function PoMenuItemComponent_div_4_li_5_Template(rf, ctx) { if (rf & 1) {
    i0.ɵɵelementStart(0, "li", 16);
    i0.ɵɵelement(1, "po-menu-item", 17);
    i0.ɵɵelementEnd();
} if (rf & 2) {
    const subItem_r14 = ctx.$implicit;
    i0.ɵɵattribute("aria-level", subItem_r14.level);
    i0.ɵɵadvance(1);
    i0.ɵɵproperty("p-action", subItem_r14.action)("p-badge-alert", subItem_r14.badgeAlert)("p-badge-color", subItem_r14.badge ? subItem_r14.badge.color : undefined)("p-badge-value", subItem_r14.badge ? subItem_r14.badge.value : undefined)("p-id", subItem_r14.id)("p-label", subItem_r14.label)("p-level", subItem_r14.level)("p-link", subItem_r14.link)("p-sub-items", subItem_r14.subItems)("p-type", subItem_r14.type);
} }
function PoMenuItemComponent_div_4_Template(rf, ctx) { if (rf & 1) {
    i0.ɵɵelementStart(0, "div", 11);
    i0.ɵɵtemplate(1, PoMenuItemComponent_div_4_ng_container_1_Template, 1, 0, "ng-container", 7);
    i0.ɵɵelementStart(2, "div", 12, 13)(4, "ul", 14);
    i0.ɵɵtemplate(5, PoMenuItemComponent_div_4_li_5_Template, 2, 11, "li", 15);
    i0.ɵɵelementEnd()()();
} if (rf & 2) {
    const ctx_r4 = i0.ɵɵnextContext();
    const _r5 = i0.ɵɵreference(6);
    i0.ɵɵattribute("aria-expanded", ctx_r4.isOpened ? true : false);
    i0.ɵɵadvance(1);
    i0.ɵɵproperty("ngTemplateOutlet", _r5);
    i0.ɵɵadvance(1);
    i0.ɵɵstyleProp("max-height", ctx_r4.maxHeight, "px");
    i0.ɵɵproperty("hidden", ctx_r4.collapsedMenu || !ctx_r4.isOpened);
    i0.ɵɵattribute("aria-expanded", ctx_r4.isOpened ? true : false)("aria-hidden", !ctx_r4.isOpened ? true : false);
    i0.ɵɵadvance(3);
    i0.ɵɵproperty("ngForOf", ctx_r4.subItems);
} }
function PoMenuItemComponent_ng_template_5_po_icon_1_Template(rf, ctx) { if (rf & 1) {
    i0.ɵɵelement(0, "po-icon", 23);
} if (rf & 2) {
    const ctx_r15 = i0.ɵɵnextContext(2);
    i0.ɵɵproperty("p-icon", ctx_r15.icon);
} }
function PoMenuItemComponent_ng_template_5_span_2_Template(rf, ctx) { if (rf & 1) {
    i0.ɵɵelement(0, "span", 24);
} if (rf & 2) {
    const ctx_r16 = i0.ɵɵnextContext(2);
    i0.ɵɵclassProp("po-icon-arrow-up", ctx_r16.isOpened)("po-icon-arrow-down", !ctx_r16.isOpened);
} }
function PoMenuItemComponent_ng_template_5_po_badge_5_Template(rf, ctx) { if (rf & 1) {
    i0.ɵɵelement(0, "po-badge", 25);
} if (rf & 2) {
    const ctx_r17 = i0.ɵɵnextContext(2);
    i0.ɵɵproperty("ngClass", !ctx_r17.collapsedMenu ? "po-menu-badge-align" : "po-menu-badge-align-collapsed")("p-color", ctx_r17.badgeColor)("p-value", ctx_r17.badgeValue);
} }
function PoMenuItemComponent_ng_template_5_span_6_Template(rf, ctx) { if (rf & 1) {
    i0.ɵɵelement(0, "span", 26);
} if (rf & 2) {
    const ctx_r18 = i0.ɵɵnextContext(2);
    i0.ɵɵproperty("ngClass", !ctx_r18.collapsedMenu ? "po-menu-badge-alert" : "po-menu-badge-alert-collapsed");
} }
function PoMenuItemComponent_ng_template_5_Template(rf, ctx) { if (rf & 1) {
    const _r20 = i0.ɵɵgetCurrentView();
    i0.ɵɵelementStart(0, "div", 18);
    i0.ɵɵlistener("click", function PoMenuItemComponent_ng_template_5_Template_div_click_0_listener($event) { i0.ɵɵrestoreView(_r20); const ctx_r19 = i0.ɵɵnextContext(); return i0.ɵɵresetView(ctx_r19.clickMenuItem($event)); })("keydown.enter", function PoMenuItemComponent_ng_template_5_Template_div_keydown_enter_0_listener($event) { i0.ɵɵrestoreView(_r20); const ctx_r21 = i0.ɵɵnextContext(); return i0.ɵɵresetView(ctx_r21.clickMenuItem($event)); })("keydown.space", function PoMenuItemComponent_ng_template_5_Template_div_keydown_space_0_listener($event) { i0.ɵɵrestoreView(_r20); const ctx_r22 = i0.ɵɵnextContext(); return i0.ɵɵresetView(ctx_r22.clickMenuItem($event)); });
    i0.ɵɵtemplate(1, PoMenuItemComponent_ng_template_5_po_icon_1_Template, 1, 1, "po-icon", 19);
    i0.ɵɵtemplate(2, PoMenuItemComponent_ng_template_5_span_2_Template, 1, 4, "span", 20);
    i0.ɵɵelementStart(3, "span");
    i0.ɵɵtext(4);
    i0.ɵɵelementEnd();
    i0.ɵɵtemplate(5, PoMenuItemComponent_ng_template_5_po_badge_5_Template, 1, 3, "po-badge", 21);
    i0.ɵɵtemplate(6, PoMenuItemComponent_ng_template_5_span_6_Template, 1, 1, "span", 22);
    i0.ɵɵelementEnd();
} if (rf & 2) {
    const ctx_r6 = i0.ɵɵnextContext();
    i0.ɵɵclassProp("po-menu-icon-container", ctx_r6.level === 1 && ctx_r6.icon)("po-menu-item-selected", ctx_r6.isSelected)("po-menu-item-level-two", ctx_r6.level === 2)("po-menu-item-level-three", ctx_r6.level === 3)("po-menu-item-level-four", ctx_r6.level === 4)("po-menu-item-grouper-up", ctx_r6.type === "subItems" && ctx_r6.isOpened)("po-menu-item-grouper-down", ctx_r6.type === "subItems" && !ctx_r6.isOpened)("po-menu-item-no-data", ctx_r6.type === "noData")("po-menu-sub-item-selected", ctx_r6.isSelectedSubItem);
    i0.ɵɵproperty("tabindex", ctx_r6.type === "subItems" ? 0 : -1);
    i0.ɵɵattribute("aria-label", ctx_r6.label)("aria-hidden", ctx_r6.type === "subItems" ? false : true)("role", ctx_r6.type === "subItems" ? "menuitem" : undefined);
    i0.ɵɵadvance(1);
    i0.ɵɵproperty("ngIf", ctx_r6.level === 1 && ctx_r6.icon);
    i0.ɵɵadvance(1);
    i0.ɵɵproperty("ngIf", ctx_r6.type === "subItems" && !ctx_r6.collapsedMenu);
    i0.ɵɵadvance(1);
    i0.ɵɵclassProp("po-menu-icon-label", ctx_r6.level === 1 && ctx_r6.icon);
    i0.ɵɵadvance(1);
    i0.ɵɵtextInterpolate1(" ", ctx_r6.label, " ");
    i0.ɵɵadvance(1);
    i0.ɵɵproperty("ngIf", ctx_r6.canShowBadge);
    i0.ɵɵadvance(1);
    i0.ɵɵproperty("ngIf", ctx_r6.badgeAlert);
} }
// valor para que caibam 3 linhas de `label`
const poMenuItemSubItemSize = 98;
/**
 * @docsPrivate
 *
 * @description
 *
 * Componente que implementa cada item do po-menu.
 */
export class PoMenuItemComponent {
    constructor(menuItemsService) {
        this.menuItemsService = menuItemsService;
        this.maxHeight = 0;
        this._isSelected = false;
        this._isSubItem = false;
    }
    // Valor do badge.
    set badgeValue(badgeValue) {
        this._badgeValue = convertToInt(badgeValue);
    }
    get badgeValue() {
        return this._badgeValue;
    }
    // Indica se o item está selecionado.
    set isSelected(value) {
        this._isSelected = convertToBoolean(value);
        this.isSelectedSubItem = this.isSelected && this.isSubItem;
    }
    get isSelected() {
        return this._isSelected;
    }
    // Indica se o item é um sub item
    set isSubItem(value) {
        this._isSubItem = convertToBoolean(value);
    }
    get isSubItem() {
        return this._isSubItem;
    }
    // Lista de sub-items.
    set subItems(subitems) {
        this._subItems = subitems;
        if (this.isOpened) {
            this.calcMenuSubItemsMaxHeight();
        }
    }
    get subItems() {
        return this._subItems;
    }
    get canShowBadge() {
        return this.type !== 'subItems' && (this.badgeValue || this.badgeValue === 0) && this.badgeValue >= 0;
    }
    ngOnDestroy() {
        this.itemSubscription.unsubscribe();
    }
    ngOnInit() {
        // subscribe to menu component messages
        this.itemSubscription = this.menuItemsService.receiveFromParentMenuClicked().subscribe(menu => {
            this.processMenuItem(menu);
        });
    }
    clickMenuItem(event) {
        if (!(event.ctrlKey || event.metaKey)) {
            event.preventDefault();
            // Emmit to parent
            this.menuItemsService.sendToParentMenuClicked({
                link: this.link,
                action: this.action,
                id: this.id,
                icon: this.icon,
                label: this.label,
                level: this.level,
                subItems: this.subItems,
                isSelected: this.isSelected,
                isOpened: this.isOpened,
                shortLabel: this.shortLabel,
                type: this.type
            });
        }
    }
    accordionAnimation(menuActive, menuOpened, hasSubItemOpened, activatedByRoute) {
        if (this.id === menuOpened['id']) {
            this.maxHeight = this.subItems.length * poMenuItemSubItemSize;
        }
        if (hasSubItemOpened) {
            this.maxHeight = menuOpened['isOpened']
                ? this.maxHeight + menuOpened.subItems.length * poMenuItemSubItemSize
                : this.maxHeight - menuOpened.subItems.length * poMenuItemSubItemSize;
            if (activatedByRoute) {
                this.maxHeight = this.getMinimumHeight(0, this, menuActive);
            }
        }
    }
    activateMenu(menu) {
        this.isSelected = menu && this.id === menu.id;
    }
    calcMenuSubItemsMaxHeight() {
        setTimeout(() => {
            const subItems = Array.from(this.menuSubItems.nativeElement.querySelectorAll('.po-menu-item'));
            subItems.forEach((menuItem) => (this.maxHeight += menuItem.offsetHeight));
        });
    }
    getMinimumHeight(minimumHeight, menuItem, menuActive) {
        minimumHeight += poMenuItemSubItemSize;
        if (menuItem.subItems && this.hasSubItem(menuItem.subItems, menuActive['id'])) {
            for (let index = 0; index < menuItem.subItems.length; index++) {
                minimumHeight = this.getMinimumHeight(minimumHeight, menuItem.subItems[index], menuActive);
            }
        }
        return minimumHeight;
    }
    groupedMenu(menuActive, menuOpened, activatedByRoute = false) {
        const hasSubItemOpened = menuOpened && this.id !== menuOpened['id'] ? this.hasSubItem(this.subItems, menuOpened['id']) : false;
        this.isOpened = this.isMenuOpened(menuOpened, hasSubItemOpened);
        this.isSelected = menuActive && !this.isOpened ? this.hasSubItem(this.subItems, menuActive['id']) : false;
        if (!this.isOpened) {
            this.maxHeight = 0;
            return;
        }
        this.accordionAnimation(menuActive, menuOpened, hasSubItemOpened, activatedByRoute);
    }
    hasSubItem(subItems, id) {
        if (subItems) {
            return subItems.some(item => (item['id'] === id ? true : this.hasSubItem(item.subItems, id)));
        }
    }
    isMenuOpened(menuOpened, hasSubItemOpened) {
        if (menuOpened) {
            return this.id === menuOpened['id'] ? menuOpened['isOpened'] : hasSubItemOpened;
        }
        return false;
    }
    processMenuItem(menu) {
        if (this.type === 'internalLink') {
            this.activateMenu(menu.active);
            return;
        }
        if (this.type === 'subItems') {
            this.groupedMenu(menu.active, menu.grouped, menu.activatedByRoute);
        }
    }
}
PoMenuItemComponent.ɵfac = function PoMenuItemComponent_Factory(t) { return new (t || PoMenuItemComponent)(i0.ɵɵdirectiveInject(i1.PoMenuItemsService)); };
PoMenuItemComponent.ɵcmp = /*@__PURE__*/ i0.ɵɵdefineComponent({ type: PoMenuItemComponent, selectors: [["po-menu-item"]], viewQuery: function PoMenuItemComponent_Query(rf, ctx) { if (rf & 1) {
        i0.ɵɵviewQuery(_c0, 5);
    } if (rf & 2) {
        let _t;
        i0.ɵɵqueryRefresh(_t = i0.ɵɵloadQuery()) && (ctx.menuSubItems = _t.first);
    } }, inputs: { action: ["p-action", "action"], badgeAlert: ["p-badge-alert", "badgeAlert"], badgeColor: ["p-badge-color", "badgeColor"], collapsedMenu: ["p-collapsed-menu", "collapsedMenu"], icon: ["p-icon", "icon"], id: ["p-id", "id"], isOpened: ["p-is-opened", "isOpened"], label: ["p-label", "label"], level: ["p-level", "level"], link: ["p-link", "link"], shortLabel: ["p-short-label", "shortLabel"], type: ["p-type", "type"], badgeValue: ["p-badge-value", "badgeValue"], isSelected: ["p-is-selected", "isSelected"], isSubItem: ["p-is-sub-item", "isSubItem"], subItems: ["p-sub-items", "subItems"] }, decls: 7, vars: 5, consts: [["class", "po-menu-item-link", "role", "menuitem", 3, "routerLink", 4, "ngIf"], ["class", "po-menu-item-link", "role", "menuitem", 3, "href", 4, "ngIf"], ["class", "po-menu-item-link", "href", "javascript:;", "role", "menuitem", 4, "ngIf"], ["class", "po-menu-item-link", "role", "none", 4, "ngIf"], ["class", "po-menu-item-link po-clickable", "role", "menuitem", 4, "ngIf"], ["menuItemTemplate", ""], ["role", "menuitem", 1, "po-menu-item-link", 3, "routerLink"], [4, "ngTemplateOutlet"], ["role", "menuitem", 1, "po-menu-item-link", 3, "href"], ["href", "javascript:;", "role", "menuitem", 1, "po-menu-item-link"], ["role", "none", 1, "po-menu-item-link"], ["role", "menuitem", 1, "po-menu-item-link", "po-clickable"], ["role", "group", 1, "po-menu-sub-items", 3, "hidden"], ["menuSubItems", ""], [1, "po-menu-sub-items-list"], ["class", "po-menu-subm-items-list-item", 4, "ngFor", "ngForOf"], [1, "po-menu-subm-items-list-item"], ["p-is-sub-item", "", 3, "p-action", "p-badge-alert", "p-badge-color", "p-badge-value", "p-id", "p-label", "p-level", "p-link", "p-sub-items", "p-type"], [1, "po-menu-item", 3, "tabindex", "click", "keydown.enter", "keydown.space"], ["aria-hidden", "true", "class", "po-menu-icon-item", 3, "p-icon", 4, "ngIf"], ["class", "po-icon po-menu-group-icon", 3, "po-icon-arrow-up", "po-icon-arrow-down", 4, "ngIf"], ["aria-hidden", "true", 3, "ngClass", "p-color", "p-value", 4, "ngIf"], ["aria-hidden", "true", "class", "po-color-07", 3, "ngClass", 4, "ngIf"], ["aria-hidden", "true", 1, "po-menu-icon-item", 3, "p-icon"], [1, "po-icon", "po-menu-group-icon"], ["aria-hidden", "true", 3, "ngClass", "p-color", "p-value"], ["aria-hidden", "true", 1, "po-color-07", 3, "ngClass"]], template: function PoMenuItemComponent_Template(rf, ctx) { if (rf & 1) {
        i0.ɵɵtemplate(0, PoMenuItemComponent_a_0_Template, 2, 3, "a", 0);
        i0.ɵɵtemplate(1, PoMenuItemComponent_a_1_Template, 2, 3, "a", 1);
        i0.ɵɵtemplate(2, PoMenuItemComponent_a_2_Template, 2, 2, "a", 2);
        i0.ɵɵtemplate(3, PoMenuItemComponent_div_3_Template, 2, 1, "div", 3);
        i0.ɵɵtemplate(4, PoMenuItemComponent_div_4_Template, 6, 8, "div", 4);
        i0.ɵɵtemplate(5, PoMenuItemComponent_ng_template_5_Template, 7, 29, "ng-template", null, 5, i0.ɵɵtemplateRefExtractor);
    } if (rf & 2) {
        i0.ɵɵproperty("ngIf", ctx.type === "internalLink");
        i0.ɵɵadvance(1);
        i0.ɵɵproperty("ngIf", ctx.type === "externalLink");
        i0.ɵɵadvance(1);
        i0.ɵɵproperty("ngIf", ctx.type === "noLink");
        i0.ɵɵadvance(1);
        i0.ɵɵproperty("ngIf", ctx.type === "noData");
        i0.ɵɵadvance(1);
        i0.ɵɵproperty("ngIf", ctx.type === "subItems");
    } }, dependencies: [i2.NgClass, i2.NgForOf, i2.NgIf, i2.NgTemplateOutlet, i3.RouterLink, i4.PoBadgeComponent, i5.PoIconComponent, PoMenuItemComponent], encapsulation: 2 });
(function () { (typeof ngDevMode === "undefined" || ngDevMode) && i0.ɵsetClassMetadata(PoMenuItemComponent, [{
        type: Component,
        args: [{ selector: 'po-menu-item', template: "<!-- menu com link interno -->\r\n<a\r\n  *ngIf=\"type === 'internalLink'\"\r\n  class=\"po-menu-item-link\"\r\n  role=\"menuitem\"\r\n  [attr.aria-label]=\"label\"\r\n  [routerLink]=\"link\"\r\n>\r\n  <ng-container *ngTemplateOutlet=\"menuItemTemplate\"></ng-container>\r\n</a>\r\n<!-- menu com link externo -->\r\n<a *ngIf=\"type === 'externalLink'\" class=\"po-menu-item-link\" role=\"menuitem\" [attr.aria-label]=\"label\" [href]=\"link\">\r\n  <ng-container *ngTemplateOutlet=\"menuItemTemplate\"></ng-container>\r\n</a>\r\n<!-- menu sem link -->\r\n<a *ngIf=\"type === 'noLink'\" class=\"po-menu-item-link\" href=\"javascript:;\" role=\"menuitem\" [attr.aria-label]=\"label\">\r\n  <ng-container *ngTemplateOutlet=\"menuItemTemplate\"></ng-container>\r\n</a>\r\n<!-- menu sem dados -->\r\n<div *ngIf=\"type === 'noData'\" class=\"po-menu-item-link\" role=\"none\">\r\n  <ng-container *ngTemplateOutlet=\"menuItemTemplate\"></ng-container>\r\n</div>\r\n<!-- menu com sub itens -->\r\n<div\r\n  *ngIf=\"type === 'subItems'\"\r\n  class=\"po-menu-item-link po-clickable\"\r\n  role=\"menuitem\"\r\n  [attr.aria-expanded]=\"isOpened ? true : false\"\r\n>\r\n  <ng-container *ngTemplateOutlet=\"menuItemTemplate\"></ng-container>\r\n\r\n  <div\r\n    #menuSubItems\r\n    class=\"po-menu-sub-items\"\r\n    [hidden]=\"collapsedMenu || !isOpened\"\r\n    [attr.aria-expanded]=\"isOpened ? true : false\"\r\n    [attr.aria-hidden]=\"!isOpened ? true : false\"\r\n    role=\"group\"\r\n    [style.maxHeight.px]=\"maxHeight\"\r\n  >\r\n    <ul class=\"po-menu-sub-items-list\">\r\n      <li class=\"po-menu-subm-items-list-item\" *ngFor=\"let subItem of subItems\" [attr.aria-level]=\"subItem.level\">\r\n        <po-menu-item\r\n          p-is-sub-item\r\n          [p-action]=\"subItem.action\"\r\n          [p-badge-alert]=\"subItem.badgeAlert\"\r\n          [p-badge-color]=\"subItem.badge ? subItem.badge.color : undefined\"\r\n          [p-badge-value]=\"subItem.badge ? subItem.badge.value : undefined\"\r\n          [p-id]=\"subItem.id\"\r\n          [p-label]=\"subItem.label\"\r\n          [p-level]=\"subItem.level\"\r\n          [p-link]=\"subItem.link\"\r\n          [p-sub-items]=\"subItem.subItems\"\r\n          [p-type]=\"subItem.type\"\r\n        ></po-menu-item>\r\n      </li>\r\n    </ul>\r\n  </div>\r\n</div>\r\n\r\n<ng-template #menuItemTemplate>\r\n  <div\r\n    class=\"po-menu-item\"\r\n    [tabindex]=\"type === 'subItems' ? 0 : -1\"\r\n    [attr.aria-label]=\"label\"\r\n    [attr.aria-hidden]=\"type === 'subItems' ? false : true\"\r\n    [attr.role]=\"type === 'subItems' ? 'menuitem' : undefined\"\r\n    [class.po-menu-icon-container]=\"level === 1 && icon\"\r\n    [class.po-menu-item-selected]=\"isSelected\"\r\n    [class.po-menu-item-level-two]=\"level === 2\"\r\n    [class.po-menu-item-level-three]=\"level === 3\"\r\n    [class.po-menu-item-level-four]=\"level === 4\"\r\n    [class.po-menu-item-grouper-up]=\"type === 'subItems' && isOpened\"\r\n    [class.po-menu-item-grouper-down]=\"type === 'subItems' && !isOpened\"\r\n    [class.po-menu-item-no-data]=\"type === 'noData'\"\r\n    [class.po-menu-sub-item-selected]=\"isSelectedSubItem\"\r\n    (click)=\"clickMenuItem($event)\"\r\n    (keydown.enter)=\"clickMenuItem($event)\"\r\n    (keydown.space)=\"clickMenuItem($event)\"\r\n  >\r\n    <po-icon *ngIf=\"level === 1 && icon\" aria-hidden=\"true\" class=\"po-menu-icon-item\" [p-icon]=\"icon\"></po-icon>\r\n\r\n    <span\r\n      *ngIf=\"type === 'subItems' && !collapsedMenu\"\r\n      class=\"po-icon po-menu-group-icon\"\r\n      [class.po-icon-arrow-up]=\"isOpened\"\r\n      [class.po-icon-arrow-down]=\"!isOpened\"\r\n    ></span>\r\n\r\n    <span [class.po-menu-icon-label]=\"level === 1 && icon\">\r\n      {{ label }}\r\n    </span>\r\n\r\n    <po-badge\r\n      *ngIf=\"canShowBadge\"\r\n      aria-hidden=\"true\"\r\n      [ngClass]=\"!collapsedMenu ? 'po-menu-badge-align' : 'po-menu-badge-align-collapsed'\"\r\n      [p-color]=\"badgeColor\"\r\n      [p-value]=\"badgeValue\"\r\n    ></po-badge>\r\n\r\n    <span\r\n      *ngIf=\"badgeAlert\"\r\n      aria-hidden=\"true\"\r\n      class=\"po-color-07\"\r\n      [ngClass]=\"!collapsedMenu ? 'po-menu-badge-alert' : 'po-menu-badge-alert-collapsed'\"\r\n    ></span>\r\n  </div>\r\n</ng-template>\r\n" }]
    }], function () { return [{ type: i1.PoMenuItemsService }]; }, { action: [{
            type: Input,
            args: ['p-action']
        }], badgeAlert: [{
            type: Input,
            args: ['p-badge-alert']
        }], badgeColor: [{
            type: Input,
            args: ['p-badge-color']
        }], collapsedMenu: [{
            type: Input,
            args: ['p-collapsed-menu']
        }], icon: [{
            type: Input,
            args: ['p-icon']
        }], id: [{
            type: Input,
            args: ['p-id']
        }], isOpened: [{
            type: Input,
            args: ['p-is-opened']
        }], label: [{
            type: Input,
            args: ['p-label']
        }], level: [{
            type: Input,
            args: ['p-level']
        }], link: [{
            type: Input,
            args: ['p-link']
        }], shortLabel: [{
            type: Input,
            args: ['p-short-label']
        }], type: [{
            type: Input,
            args: ['p-type']
        }], menuSubItems: [{
            type: ViewChild,
            args: ['menuSubItems']
        }], badgeValue: [{
            type: Input,
            args: ['p-badge-value']
        }], isSelected: [{
            type: Input,
            args: ['p-is-selected']
        }], isSubItem: [{
            type: Input,
            args: ['p-is-sub-item']
        }], subItems: [{
            type: Input,
            args: ['p-sub-items']
        }] }); })();
