// @po-ui/ng-components/esm2020/lib/components/po-disclaimer/po-disclaimer.component.mjs
import { Component, ViewChild } from '@angular/core';
import { isKeyCodeEnter } from '../../utils/util';
import { PoDisclaimerBaseComponent } from './po-disclaimer-base.component';
import * as i0 from "@angular/core";
import * as i1 from "@angular/common";
import * as i2 from "../../directives/po-tooltip/po-tooltip.directive";
const _c0 = ["disclaimerContainer"];
function PoDisclaimerComponent_div_2_span_4_Template(rf, ctx) { if (rf & 1) {
    const _r4 = i0.ɵɵgetCurrentView();
    i0.ɵɵelementStart(0, "span", 7);
    i0.ɵɵlistener("click", function PoDisclaimerComponent_div_2_span_4_Template_span_click_0_listener() { i0.ɵɵrestoreView(_r4); const ctx_r3 = i0.ɵɵnextContext(2); return i0.ɵɵresetView(ctx_r3.close()); })("keydown", function PoDisclaimerComponent_div_2_span_4_Template_span_keydown_0_listener($event) { i0.ɵɵrestoreView(_r4); const ctx_r5 = i0.ɵɵnextContext(2); return i0.ɵɵresetView(ctx_r5.onKeyPress($event)); });
    i0.ɵɵelementEnd();
} if (rf & 2) {
    const ctx_r2 = i0.ɵɵnextContext(2);
    i0.ɵɵclassProp("po-disclaimer-remove-danger", ctx_r2.type === "danger");
    i0.ɵɵattribute("aria-label", ctx_r2.setAriaLabel())("role", !ctx_r2.hideClose ? "button" : "");
} }
function PoDisclaimerComponent_div_2_Template(rf, ctx) { if (rf & 1) {
    i0.ɵɵelementStart(0, "div", 3)(1, "div", 4)(2, "span", 5);
    i0.ɵɵtext(3);
    i0.ɵɵelementEnd()();
    i0.ɵɵtemplate(4, PoDisclaimerComponent_div_2_span_4_Template, 1, 4, "span", 6);
    i0.ɵɵelementEnd();
} if (rf & 2) {
    const ctx_r1 = i0.ɵɵnextContext();
    i0.ɵɵclassProp("po-disclaimer-remove-all", ctx_r1.hideClose)("po-disclaimer-danger", ctx_r1.type === "danger");
    i0.ɵɵproperty("p-tooltip", ctx_r1.getWidthDisclaimer() ? ctx_r1.getLabel() : "")("tabindex", ctx_r1.hideClose ? "0" : "-1");
    i0.ɵɵadvance(1);
    i0.ɵɵclassProp("show-close", !ctx_r1.hideClose);
    i0.ɵɵadvance(2);
    i0.ɵɵtextInterpolate1(" ", ctx_r1.getLabel(), " ");
    i0.ɵɵadvance(1);
    i0.ɵɵproperty("ngIf", ctx_r1.hideClose === false);
} }
/**
 * @docsPrivate
 *
 * @docsExtends PoDisclaimerBaseComponent
 *
 * @examplePrivate
 *
 * <example-private name="po-disclaimer" title="PO Disclaimer">
 *   <file name="sample-po-disclaimer.component.html"> </file>
 *   <file name="sample-po-disclaimer.component.ts"> </file>
 * </example-private>
 */
export class PoDisclaimerComponent extends PoDisclaimerBaseComponent {
    onKeyPress(event) {
        if (isKeyCodeEnter(event)) {
            this.close();
        }
    }
    getWidthDisclaimer() {
        return this.disclaimerContainer.nativeElement.offsetWidth > 201;
    }
}
PoDisclaimerComponent.ɵfac = /*@__PURE__*/ function () { let ɵPoDisclaimerComponent_BaseFactory; return function PoDisclaimerComponent_Factory(t) { return (ɵPoDisclaimerComponent_BaseFactory || (ɵPoDisclaimerComponent_BaseFactory = i0.ɵɵgetInheritedFactory(PoDisclaimerComponent)))(t || PoDisclaimerComponent); }; }();
PoDisclaimerComponent.ɵcmp = /*@__PURE__*/ i0.ɵɵdefineComponent({ type: PoDisclaimerComponent, selectors: [["po-disclaimer"]], viewQuery: function PoDisclaimerComponent_Query(rf, ctx) { if (rf & 1) {
        i0.ɵɵviewQuery(_c0, 7);
    } if (rf & 2) {
        let _t;
        i0.ɵɵqueryRefresh(_t = i0.ɵɵloadQuery()) && (ctx.disclaimerContainer = _t.first);
    } }, features: [i0.ɵɵInheritDefinitionFeature], decls: 3, vars: 1, consts: [[1, "disclaimer-container"], ["disclaimerContainer", ""], ["class", "po-disclaimer", "p-tooltip-position", "bottom", 3, "po-disclaimer-remove-all", "po-disclaimer-danger", "p-tooltip", "tabindex", 4, "ngIf"], ["p-tooltip-position", "bottom", 1, "po-disclaimer", 3, "p-tooltip", "tabindex"], [1, "po-disclaimer-label"], [1, "label"], ["class", "po-disclaimer-remove po-icon po-icon-close po-clickable", "tabindex", "0", 3, "po-disclaimer-remove-danger", "click", "keydown", 4, "ngIf"], ["tabindex", "0", 1, "po-disclaimer-remove", "po-icon", "po-icon-close", "po-clickable", 3, "click", "keydown"]], template: function PoDisclaimerComponent_Template(rf, ctx) { if (rf & 1) {
        i0.ɵɵelementStart(0, "div", 0, 1);
        i0.ɵɵtemplate(2, PoDisclaimerComponent_div_2_Template, 5, 10, "div", 2);
        i0.ɵɵelementEnd();
    } if (rf & 2) {
        i0.ɵɵadvance(2);
        i0.ɵɵproperty("ngIf", ctx.showDisclaimer === true);
    } }, dependencies: [i1.NgIf, i2.PoTooltipDirective], encapsulation: 2 });
(function () { (typeof ngDevMode === "undefined" || ngDevMode) && i0.ɵsetClassMetadata(PoDisclaimerComponent, [{
        type: Component,
        args: [{ selector: 'po-disclaimer', template: "<div #disclaimerContainer class=\"disclaimer-container\">\r\n  <div\r\n    *ngIf=\"showDisclaimer === true\"\r\n    class=\"po-disclaimer\"\r\n    [class.po-disclaimer-remove-all]=\"hideClose\"\r\n    [class.po-disclaimer-danger]=\"type === 'danger'\"\r\n    [p-tooltip]=\"getWidthDisclaimer() ? getLabel() : ''\"\r\n    p-tooltip-position=\"bottom\"\r\n    [tabindex]=\"hideClose ? '0' : '-1'\"\r\n  >\r\n    <div class=\"po-disclaimer-label\" [class.show-close]=\"!hideClose\">\r\n      <span class=\"label\">\r\n        {{ getLabel() }}\r\n      </span>\r\n    </div>\r\n\r\n    <span\r\n      *ngIf=\"hideClose === false\"\r\n      [attr.aria-label]=\"setAriaLabel()\"\r\n      [attr.role]=\"!hideClose ? 'button' : ''\"\r\n      class=\"po-disclaimer-remove po-icon po-icon-close po-clickable\"\r\n      tabindex=\"0\"\r\n      [class.po-disclaimer-remove-danger]=\"type === 'danger'\"\r\n      (click)=\"close()\"\r\n      (keydown)=\"onKeyPress($event)\"\r\n    >\r\n    </span>\r\n  </div>\r\n</div>\r\n" }]
    }], null, { disclaimerContainer: [{
            type: ViewChild,
            args: ['disclaimerContainer', { static: true }]
        }] }); })();
