// @po-ui/ng-components/esm2020/lib/components/po-list-view/po-list-view.component.mjs
import { animate, state, style, transition, trigger } from '@angular/animations';
import { Component, ContentChild, ViewChild } from '@angular/core';
import { isTypeof } from '../../utils/util';
import { PoListViewBaseComponent } from './po-list-view-base.component';
import { PoListViewContentTemplateDirective } from './po-list-view-content-template/po-list-view-content-template.directive';
import { PoListViewDetailTemplateDirective } from './po-list-view-detail-template/po-list-view-detail-template.directive';
import * as i0 from "@angular/core";
import * as i1 from "../../services/po-language/po-language.service";
import * as i2 from "@angular/common";
import * as i3 from "@angular/forms";
import * as i4 from "@angular/router";
import * as i5 from "../po-button/po-button.component";
import * as i6 from "../po-popup/po-popup.component";
import * as i7 from "../po-field/po-checkbox/po-checkbox.component";
import * as i8 from "../po-container/po-container.component";
const _c0 = ["popup"];
function PoListViewComponent_div_1_Template(rf, ctx) { if (rf & 1) {
    const _r6 = i0.ɵɵgetCurrentView();
    i0.ɵɵelementStart(0, "div", 7)(1, "div", 8)(2, "div", 9)(3, "po-checkbox", 10);
    i0.ɵɵlistener("ngModelChange", function PoListViewComponent_div_1_Template_po_checkbox_ngModelChange_3_listener($event) { i0.ɵɵrestoreView(_r6); const ctx_r5 = i0.ɵɵnextContext(); return i0.ɵɵresetView(ctx_r5.selectAll = $event); })("p-change", function PoListViewComponent_div_1_Template_po_checkbox_p_change_3_listener() { i0.ɵɵrestoreView(_r6); const ctx_r7 = i0.ɵɵnextContext(); return i0.ɵɵresetView(ctx_r7.selectAllListItems()); })("click", function PoListViewComponent_div_1_Template_po_checkbox_click_3_listener() { i0.ɵɵrestoreView(_r6); const ctx_r8 = i0.ɵɵnextContext(); return i0.ɵɵresetView(ctx_r8.selectAllListItems()); });
    i0.ɵɵelementEnd()()()();
} if (rf & 2) {
    const ctx_r0 = i0.ɵɵnextContext();
    i0.ɵɵadvance(3);
    i0.ɵɵproperty("ngModel", ctx_r0.selectAll)("p-label", ctx_r0.literals.selectAll);
} }
function PoListViewComponent_po_container_3_Template(rf, ctx) { if (rf & 1) {
    i0.ɵɵelementStart(0, "po-container", 11)(1, "div", 12)(2, "span");
    i0.ɵɵtext(3);
    i0.ɵɵelementEnd()()();
} if (rf & 2) {
    const ctx_r1 = i0.ɵɵnextContext();
    i0.ɵɵproperty("p-no-padding", true);
    i0.ɵɵadvance(3);
    i0.ɵɵtextInterpolate1(" ", ctx_r1.literals.noData, " ");
} }
function PoListViewComponent_ng_container_4_div_5_Template(rf, ctx) { if (rf & 1) {
    const _r22 = i0.ɵɵgetCurrentView();
    i0.ɵɵelementStart(0, "div", 25)(1, "po-checkbox", 26);
    i0.ɵɵlistener("ngModelChange", function PoListViewComponent_ng_container_4_div_5_Template_po_checkbox_ngModelChange_1_listener($event) { i0.ɵɵrestoreView(_r22); const item_r9 = i0.ɵɵnextContext().$implicit; return i0.ɵɵresetView(item_r9.$selected = $event); })("p-change", function PoListViewComponent_ng_container_4_div_5_Template_po_checkbox_p_change_1_listener() { i0.ɵɵrestoreView(_r22); const item_r9 = i0.ɵɵnextContext().$implicit; const ctx_r23 = i0.ɵɵnextContext(); return i0.ɵɵresetView(ctx_r23.selectListItem(item_r9)); })("click", function PoListViewComponent_ng_container_4_div_5_Template_po_checkbox_click_1_listener() { i0.ɵɵrestoreView(_r22); const item_r9 = i0.ɵɵnextContext().$implicit; const ctx_r25 = i0.ɵɵnextContext(); return i0.ɵɵresetView(ctx_r25.selectListItem(item_r9)); });
    i0.ɵɵelementEnd()();
} if (rf & 2) {
    const item_r9 = i0.ɵɵnextContext().$implicit;
    i0.ɵɵadvance(1);
    i0.ɵɵproperty("ngModel", item_r9.$selected);
} }
function PoListViewComponent_ng_container_4_a_6_Template(rf, ctx) { if (rf & 1) {
    const _r30 = i0.ɵɵgetCurrentView();
    i0.ɵɵelementStart(0, "a", 27);
    i0.ɵɵlistener("click", function PoListViewComponent_ng_container_4_a_6_Template_a_click_0_listener() { i0.ɵɵrestoreView(_r30); const item_r9 = i0.ɵɵnextContext().$implicit; const ctx_r28 = i0.ɵɵnextContext(); return i0.ɵɵresetView(ctx_r28.runTitleAction(item_r9)); });
    i0.ɵɵtext(1);
    i0.ɵɵelementEnd();
} if (rf & 2) {
    const item_r9 = i0.ɵɵnextContext().$implicit;
    const ctx_r12 = i0.ɵɵnextContext();
    i0.ɵɵproperty("href", item_r9[ctx_r12.propertyLink], i0.ɵɵsanitizeUrl);
    i0.ɵɵadvance(1);
    i0.ɵɵtextInterpolate1(" ", ctx_r12.getItemTitle(item_r9), " ");
} }
function PoListViewComponent_ng_container_4_a_7_Template(rf, ctx) { if (rf & 1) {
    const _r34 = i0.ɵɵgetCurrentView();
    i0.ɵɵelementStart(0, "a", 28);
    i0.ɵɵlistener("click", function PoListViewComponent_ng_container_4_a_7_Template_a_click_0_listener() { i0.ɵɵrestoreView(_r34); const item_r9 = i0.ɵɵnextContext().$implicit; const ctx_r32 = i0.ɵɵnextContext(); return i0.ɵɵresetView(ctx_r32.runTitleAction(item_r9)); });
    i0.ɵɵtext(1);
    i0.ɵɵelementEnd();
} if (rf & 2) {
    const item_r9 = i0.ɵɵnextContext().$implicit;
    const ctx_r13 = i0.ɵɵnextContext();
    i0.ɵɵproperty("routerLink", item_r9[ctx_r13.propertyLink]);
    i0.ɵɵadvance(1);
    i0.ɵɵtextInterpolate1(" ", ctx_r13.getItemTitle(item_r9), " ");
} }
function PoListViewComponent_ng_container_4_span_8_Template(rf, ctx) { if (rf & 1) {
    const _r37 = i0.ɵɵgetCurrentView();
    i0.ɵɵelementStart(0, "span", 29);
    i0.ɵɵlistener("click", function PoListViewComponent_ng_container_4_span_8_Template_span_click_0_listener() { i0.ɵɵrestoreView(_r37); const item_r9 = i0.ɵɵnextContext().$implicit; const ctx_r36 = i0.ɵɵnextContext(); return i0.ɵɵresetView(ctx_r36.titleHasAction && ctx_r36.runTitleAction(item_r9)); });
    i0.ɵɵtext(1);
    i0.ɵɵelementEnd();
} if (rf & 2) {
    const item_r9 = i0.ɵɵnextContext().$implicit;
    const ctx_r14 = i0.ɵɵnextContext();
    i0.ɵɵclassProp("po-list-view-title-link", ctx_r14.titleHasAction);
    i0.ɵɵadvance(1);
    i0.ɵɵtextInterpolate1(" ", ctx_r14.getItemTitle(item_r9), " ");
} }
function PoListViewComponent_ng_container_4_div_9_po_button_1_Template(rf, ctx) { if (rf & 1) {
    const _r43 = i0.ɵɵgetCurrentView();
    i0.ɵɵelementStart(0, "po-button", 32);
    i0.ɵɵlistener("p-click", function PoListViewComponent_ng_container_4_div_9_po_button_1_Template_po_button_p_click_0_listener() { const restoredCtx = i0.ɵɵrestoreView(_r43); const action_r41 = restoredCtx.$implicit; const item_r9 = i0.ɵɵnextContext(2).$implicit; const ctx_r42 = i0.ɵɵnextContext(); return i0.ɵɵresetView(ctx_r42.onClickAction(action_r41, item_r9)); });
    i0.ɵɵelementEnd();
} if (rf & 2) {
    const action_r41 = ctx.$implicit;
    const item_r9 = i0.ɵɵnextContext(2).$implicit;
    const ctx_r40 = i0.ɵɵnextContext();
    i0.ɵɵproperty("p-disabled", ctx_r40.returnBooleanValue(action_r41, item_r9, "disabled"))("p-icon", action_r41.icon)("p-label", action_r41.label)("p-kind", action_r41.type);
} }
function PoListViewComponent_ng_container_4_div_9_Template(rf, ctx) { if (rf & 1) {
    i0.ɵɵelementStart(0, "div", 30);
    i0.ɵɵtemplate(1, PoListViewComponent_ng_container_4_div_9_po_button_1_Template, 1, 4, "po-button", 31);
    i0.ɵɵelementEnd();
} if (rf & 2) {
    const ctx_r15 = i0.ɵɵnextContext(2);
    i0.ɵɵadvance(1);
    i0.ɵɵproperty("ngForOf", ctx_r15.visibleActions);
} }
function PoListViewComponent_ng_container_4_div_10_Template(rf, ctx) { if (rf & 1) {
    const _r49 = i0.ɵɵgetCurrentView();
    i0.ɵɵelementStart(0, "div", 33)(1, "span", 34, 35);
    i0.ɵɵlistener("click", function PoListViewComponent_ng_container_4_div_10_Template_span_click_1_listener() { i0.ɵɵrestoreView(_r49); const _r46 = i0.ɵɵreference(2); const item_r9 = i0.ɵɵnextContext().$implicit; const ctx_r47 = i0.ɵɵnextContext(); return i0.ɵɵresetView(ctx_r47.togglePopup(item_r9, _r46)); });
    i0.ɵɵelementEnd()();
} }
function PoListViewComponent_ng_container_4_div_11_ng_template_1_Template(rf, ctx) { }
const _c1 = function (a0, a1) { return { $implicit: a0, index: a1 }; };
function PoListViewComponent_ng_container_4_div_11_Template(rf, ctx) { if (rf & 1) {
    i0.ɵɵelementStart(0, "div", 36);
    i0.ɵɵtemplate(1, PoListViewComponent_ng_container_4_div_11_ng_template_1_Template, 0, 0, "ng-template", 37);
    i0.ɵɵelementEnd();
} if (rf & 2) {
    const ctx_r51 = i0.ɵɵnextContext();
    const item_r9 = ctx_r51.$implicit;
    const index_r10 = ctx_r51.index;
    const ctx_r17 = i0.ɵɵnextContext();
    i0.ɵɵadvance(1);
    i0.ɵɵproperty("ngTemplateOutlet", ctx_r17.listViewContentTemplate.templateRef)("ngTemplateOutletContext", i0.ɵɵpureFunction2(2, _c1, item_r9, index_r10));
} }
function PoListViewComponent_ng_container_4_div_12_ng_template_1_Template(rf, ctx) { }
function PoListViewComponent_ng_container_4_div_12_Template(rf, ctx) { if (rf & 1) {
    const _r55 = i0.ɵɵgetCurrentView();
    i0.ɵɵelementStart(0, "div", 38);
    i0.ɵɵlistener("@showHideDetail.start", function PoListViewComponent_ng_container_4_div_12_Template_div_animation_showHideDetail_start_0_listener($event) { i0.ɵɵrestoreView(_r55); const item_r9 = i0.ɵɵnextContext().$implicit; const ctx_r53 = i0.ɵɵnextContext(); return i0.ɵɵresetView(item_r9.$showDetail ? ctx_r53.onAnimationEvent($event, item_r9) : "undefined"); });
    i0.ɵɵtemplate(1, PoListViewComponent_ng_container_4_div_12_ng_template_1_Template, 0, 0, "ng-template", 37);
    i0.ɵɵelementEnd();
} if (rf & 2) {
    const ctx_r56 = i0.ɵɵnextContext();
    const item_r9 = ctx_r56.$implicit;
    const index_r10 = ctx_r56.index;
    const ctx_r18 = i0.ɵɵnextContext();
    i0.ɵɵproperty("@showHideDetail", undefined);
    i0.ɵɵadvance(1);
    i0.ɵɵproperty("ngTemplateOutlet", ctx_r18.listViewDetailTemplate.templateRef)("ngTemplateOutletContext", i0.ɵɵpureFunction2(3, _c1, item_r9, index_r10));
} }
function PoListViewComponent_ng_container_4_div_13_Template(rf, ctx) { if (rf & 1) {
    const _r59 = i0.ɵɵgetCurrentView();
    i0.ɵɵelementStart(0, "div", 39)(1, "po-button", 40);
    i0.ɵɵlistener("p-click", function PoListViewComponent_ng_container_4_div_13_Template_po_button_p_click_1_listener() { i0.ɵɵrestoreView(_r59); const item_r9 = i0.ɵɵnextContext().$implicit; return i0.ɵɵresetView(item_r9.$showDetail = !item_r9.$showDetail); });
    i0.ɵɵelementEnd()();
} if (rf & 2) {
    const item_r9 = i0.ɵɵnextContext().$implicit;
    const ctx_r19 = i0.ɵɵnextContext();
    i0.ɵɵadvance(1);
    i0.ɵɵproperty("p-label", item_r9.$showDetail ? ctx_r19.literals.hideDetails : ctx_r19.literals.showDetails);
} }
function PoListViewComponent_ng_container_4_Template(rf, ctx) { if (rf & 1) {
    i0.ɵɵelementContainerStart(0);
    i0.ɵɵelementStart(1, "po-container", 11)(2, "div", 13)(3, "div", 14)(4, "div", 15);
    i0.ɵɵtemplate(5, PoListViewComponent_ng_container_4_div_5_Template, 2, 1, "div", 16);
    i0.ɵɵtemplate(6, PoListViewComponent_ng_container_4_a_6_Template, 2, 2, "a", 17);
    i0.ɵɵtemplate(7, PoListViewComponent_ng_container_4_a_7_Template, 2, 2, "a", 18);
    i0.ɵɵtemplate(8, PoListViewComponent_ng_container_4_span_8_Template, 2, 3, "span", 19);
    i0.ɵɵelementEnd();
    i0.ɵɵtemplate(9, PoListViewComponent_ng_container_4_div_9_Template, 2, 1, "div", 20);
    i0.ɵɵtemplate(10, PoListViewComponent_ng_container_4_div_10_Template, 3, 0, "div", 21);
    i0.ɵɵelementEnd();
    i0.ɵɵtemplate(11, PoListViewComponent_ng_container_4_div_11_Template, 2, 5, "div", 22);
    i0.ɵɵtemplate(12, PoListViewComponent_ng_container_4_div_12_Template, 2, 6, "div", 23);
    i0.ɵɵelementEnd();
    i0.ɵɵtemplate(13, PoListViewComponent_ng_container_4_div_13_Template, 2, 1, "div", 24);
    i0.ɵɵelementEnd();
    i0.ɵɵelementContainerEnd();
} if (rf & 2) {
    const item_r9 = ctx.$implicit;
    const ctx_r2 = i0.ɵɵnextContext();
    i0.ɵɵadvance(1);
    i0.ɵɵproperty("p-no-padding", true);
    i0.ɵɵadvance(3);
    i0.ɵɵproperty("ngSwitch", ctx_r2.checkTitleType(item_r9));
    i0.ɵɵadvance(1);
    i0.ɵɵproperty("ngIf", ctx_r2.select);
    i0.ɵɵadvance(1);
    i0.ɵɵproperty("ngSwitchCase", "externalLink");
    i0.ɵɵadvance(1);
    i0.ɵɵproperty("ngSwitchCase", "internalLink");
    i0.ɵɵadvance(1);
    i0.ɵɵproperty("ngSwitchCase", "noLink");
    i0.ɵɵadvance(1);
    i0.ɵɵproperty("ngIf", ctx_r2.showButtonsActions);
    i0.ɵɵadvance(1);
    i0.ɵɵproperty("ngIf", ctx_r2.showPopupActions);
    i0.ɵɵadvance(1);
    i0.ɵɵproperty("ngIf", ctx_r2.hasContentTemplate);
    i0.ɵɵadvance(1);
    i0.ɵɵproperty("ngIf", ctx_r2.hasDetailTemplate && item_r9.$showDetail);
    i0.ɵɵadvance(1);
    i0.ɵɵproperty("ngIf", ctx_r2.hasDetailTemplate);
} }
function PoListViewComponent_div_5_Template(rf, ctx) { if (rf & 1) {
    const _r62 = i0.ɵɵgetCurrentView();
    i0.ɵɵelementStart(0, "div", 41)(1, "po-button", 42);
    i0.ɵɵlistener("p-click", function PoListViewComponent_div_5_Template_po_button_p_click_1_listener() { i0.ɵɵrestoreView(_r62); const ctx_r61 = i0.ɵɵnextContext(); return i0.ɵɵresetView(ctx_r61.onShowMore()); });
    i0.ɵɵelementEnd()();
} if (rf & 2) {
    const ctx_r3 = i0.ɵɵnextContext();
    i0.ɵɵadvance(1);
    i0.ɵɵproperty("p-disabled", ctx_r3.showMoreDisabled)("p-label", ctx_r3.literals.loadMoreData);
} }
/**
 * @docsExtends PoListViewBaseComponent
 *
 * @example
 *
 * <example name="po-list-view-basic" title="PO List View Basic">
 *  <file name="sample-po-list-view-basic/sample-po-list-view-basic.component.html"> </file>
 *  <file name="sample-po-list-view-basic/sample-po-list-view-basic.component.ts"> </file>
 * </example>
 *
 * <example name="po-list-view-labs" title="PO List View Labs">
 *  <file name="sample-po-list-view-labs/sample-po-list-view-labs.component.html"> </file>
 *  <file name="sample-po-list-view-labs/sample-po-list-view-labs.component.ts"> </file>
 * </example>
 *
 * <example name="po-list-view-hiring-processes" title="PO List View - Hiring Processes">
 *  <file name="sample-po-list-view-hiring-processes/sample-po-list-view-hiring-processes.component.html"> </file>
 *  <file name="sample-po-list-view-hiring-processes/sample-po-list-view-hiring-processes.component.ts"> </file>
 *  <file name="sample-po-list-view-hiring-processes/sample-po-list-view-hiring-processes.service.ts"> </file>
 * </example>
 */
export class PoListViewComponent extends PoListViewBaseComponent {
    constructor(changeDetector, differs, languageService) {
        super(languageService);
        this.changeDetector = changeDetector;
        this.differ = differs.find([]).create(null);
    }
    get hasContentTemplate() {
        return !!this.listViewContentTemplate;
    }
    get hasDetailTemplate() {
        return !!this.listViewDetailTemplate;
    }
    get displayShowMoreButton() {
        return this.items && this.items.length > 0 && this.showMore.observers.length > 0;
    }
    get showButtonsActions() {
        return this.visibleActions && this.visibleActions.length > 0 && this.visibleActions.length <= 2;
    }
    get showPopupActions() {
        return this.visibleActions && this.visibleActions.length > 2;
    }
    get titleHasAction() {
        return this.titleAction.observers.length > 0;
    }
    get visibleActions() {
        return (this.actions &&
            this.actions.filter(action => this.returnBooleanValue(action, action.visible, 'visible') !== false));
    }
    ngAfterContentInit() {
        this.initShowDetail();
    }
    ngDoCheck() {
        this.checkItemsChange();
    }
    checkTitleType(item) {
        if (this.propertyLink && item[this.propertyLink]) {
            return item[this.propertyLink].startsWith('http') ? 'externalLink' : 'internalLink';
        }
        return 'noLink';
    }
    getItemTitle(item) {
        return this.hasContentTemplate && this.listViewContentTemplate.title
            ? this.listViewContentTemplate.title(item)
            : item[this.propertyTitle];
    }
    hasItems() {
        return this.items && this.items.length > 0;
    }
    returnBooleanValue(listViewAction, item, property) {
        return isTypeof(listViewAction[property], 'function')
            ? listViewAction[property](item)
            : listViewAction[property];
    }
    trackBy(index) {
        return index;
    }
    togglePopup(item, targetRef) {
        this.popupTarget = targetRef;
        this.changeDetector.detectChanges();
        this.poPopupComponent.toggle(item);
    }
    onAnimationEvent(event, detail) {
        this.showDetail.emit(detail);
    }
    checkItemsChange() {
        const changesItems = this.differ.diff(this.items);
        if (changesItems && this.selectAll) {
            this.selectAll = null;
        }
        if (changesItems && this.items && this.items.length && this.select && !this.hideSelectAll) {
            this.showHeader = true;
        }
    }
    initShowDetail() {
        if (this.items && this.items.length > 0 && this.hasDetailTemplate && this.listViewDetailTemplate.showDetail) {
            this.items.forEach(item => (item.$showDetail = this.listViewDetailTemplate.showDetail(item)));
        }
    }
}
PoListViewComponent.ɵfac = function PoListViewComponent_Factory(t) { return new (t || PoListViewComponent)(i0.ɵɵdirectiveInject(i0.ChangeDetectorRef), i0.ɵɵdirectiveInject(i0.IterableDiffers), i0.ɵɵdirectiveInject(i1.PoLanguageService)); };
PoListViewComponent.ɵcmp = /*@__PURE__*/ i0.ɵɵdefineComponent({ type: PoListViewComponent, selectors: [["po-list-view"]], contentQueries: function PoListViewComponent_ContentQueries(rf, ctx, dirIndex) { if (rf & 1) {
        i0.ɵɵcontentQuery(dirIndex, PoListViewContentTemplateDirective, 7);
        i0.ɵɵcontentQuery(dirIndex, PoListViewDetailTemplateDirective, 7);
    } if (rf & 2) {
        let _t;
        i0.ɵɵqueryRefresh(_t = i0.ɵɵloadQuery()) && (ctx.listViewContentTemplate = _t.first);
        i0.ɵɵqueryRefresh(_t = i0.ɵɵloadQuery()) && (ctx.listViewDetailTemplate = _t.first);
    } }, viewQuery: function PoListViewComponent_Query(rf, ctx) { if (rf & 1) {
        i0.ɵɵviewQuery(_c0, 7);
    } if (rf & 2) {
        let _t;
        i0.ɵɵqueryRefresh(_t = i0.ɵɵloadQuery()) && (ctx.poPopupComponent = _t.first);
    } }, features: [i0.ɵɵInheritDefinitionFeature], decls: 8, vars: 13, consts: [["class", "po-list-view-main-header", 4, "ngIf"], [1, "po-list-view-main-content"], [3, "p-no-padding", 4, "ngIf"], [4, "ngFor", "ngForOf", "ngForTrackBy"], ["class", "po-mb-2 po-text-center", 4, "ngIf"], [3, "p-actions", "p-target"], ["popup", ""], [1, "po-list-view-main-header"], [1, "po-checkbox-group-item"], [1, "po-list-view-main-select"], ["name", "selectAll", 3, "ngModel", "p-label", "ngModelChange", "p-change", "click"], [3, "p-no-padding"], [1, "po-list-view-container-no-data", "po-text-center"], [1, "po-list-view-container"], [1, "po-list-view-header"], [1, "po-list-view-title", 3, "ngSwitch"], ["class", "po-list-view-select", 4, "ngIf"], ["class", "po-list-view-title-link", "target", "_blank", 3, "href", "click", 4, "ngSwitchCase"], ["class", "po-list-view-title-link", 3, "routerLink", "click", 4, "ngSwitchCase"], ["class", "po-list-view-title-no-link", 3, "po-list-view-title-link", "click", 4, "ngSwitchCase"], ["class", "po-list-view-actions", 4, "ngIf"], ["class", "po-list-view-more-actions", 4, "ngIf"], ["class", "po-list-view-content", 4, "ngIf"], ["class", "po-list-view-detail", 4, "ngIf"], ["class", "po-list-view-detail-button", 4, "ngIf"], [1, "po-list-view-select"], ["name", "checkbox", 3, "ngModel", "ngModelChange", "p-change", "click"], ["target", "_blank", 1, "po-list-view-title-link", 3, "href", "click"], [1, "po-list-view-title-link", 3, "routerLink", "click"], [1, "po-list-view-title-no-link", 3, "click"], [1, "po-list-view-actions"], [3, "p-disabled", "p-icon", "p-label", "p-kind", "p-click", 4, "ngFor", "ngForOf"], [3, "p-disabled", "p-icon", "p-label", "p-kind", "p-click"], [1, "po-list-view-more-actions"], [1, "po-icon", "po-icon-more", "po-list-view-more-icon", "po-clickable", 3, "click"], ["popupTarget", ""], [1, "po-list-view-content"], [3, "ngTemplateOutlet", "ngTemplateOutletContext"], [1, "po-list-view-detail"], [1, "po-list-view-detail-button"], ["p-kind", "tertiary", 3, "p-label", "p-click"], [1, "po-mb-2", "po-text-center"], [3, "p-disabled", "p-label", "p-click"]], template: function PoListViewComponent_Template(rf, ctx) { if (rf & 1) {
        i0.ɵɵelementStart(0, "div");
        i0.ɵɵtemplate(1, PoListViewComponent_div_1_Template, 4, 2, "div", 0);
        i0.ɵɵelementStart(2, "div", 1);
        i0.ɵɵtemplate(3, PoListViewComponent_po_container_3_Template, 4, 2, "po-container", 2);
        i0.ɵɵtemplate(4, PoListViewComponent_ng_container_4_Template, 14, 11, "ng-container", 3);
        i0.ɵɵelementEnd()();
        i0.ɵɵtemplate(5, PoListViewComponent_div_5_Template, 2, 2, "div", 4);
        i0.ɵɵelement(6, "po-popup", 5, 6);
    } if (rf & 2) {
        i0.ɵɵstyleProp("height", ctx.height, "px");
        i0.ɵɵclassProp("po-list-view-main-container-header", ctx.showHeader)("po-list-view-main-container", !ctx.showHeader);
        i0.ɵɵadvance(1);
        i0.ɵɵproperty("ngIf", ctx.showHeader);
        i0.ɵɵadvance(2);
        i0.ɵɵproperty("ngIf", !ctx.hasItems());
        i0.ɵɵadvance(1);
        i0.ɵɵproperty("ngForOf", ctx.items)("ngForTrackBy", ctx.trackBy);
        i0.ɵɵadvance(1);
        i0.ɵɵproperty("ngIf", ctx.displayShowMoreButton);
        i0.ɵɵadvance(1);
        i0.ɵɵproperty("p-actions", ctx.actions)("p-target", ctx.popupTarget);
    } }, dependencies: [i2.NgForOf, i2.NgIf, i2.NgTemplateOutlet, i2.NgSwitch, i2.NgSwitchCase, i3.NgControlStatus, i3.NgModel, i4.RouterLink, i5.PoButtonComponent, i6.PoPopupComponent, i7.PoCheckboxComponent, i8.PoContainerComponent], encapsulation: 2, data: { animation: [
            trigger('showHideDetail', [
                state('*', style({ 'overflow-y': 'visible' })),
                state('void', style({ 'overflow-y': 'hidden' })),
                transition('* => void', [style({ height: '*', 'overflow-y': 'hidden' }), animate(100, style({ height: 0 }))]),
                transition('void => *', [style({ height: '0' }), animate(100, style({ height: '*' }))])
            ])
        ] } });
(function () { (typeof ngDevMode === "undefined" || ngDevMode) && i0.ɵsetClassMetadata(PoListViewComponent, [{
        type: Component,
        args: [{ selector: 'po-list-view', animations: [
                    trigger('showHideDetail', [
                        state('*', style({ 'overflow-y': 'visible' })),
                        state('void', style({ 'overflow-y': 'hidden' })),
                        transition('* => void', [style({ height: '*', 'overflow-y': 'hidden' }), animate(100, style({ height: 0 }))]),
                        transition('void => *', [style({ height: '0' }), animate(100, style({ height: '*' }))])
                    ])
                ], template: "<div\r\n  [class.po-list-view-main-container-header]=\"showHeader\"\r\n  [class.po-list-view-main-container]=\"!showHeader\"\r\n  [style.height.px]=\"height\"\r\n>\r\n  <div *ngIf=\"showHeader\" class=\"po-list-view-main-header\">\r\n    <div class=\"po-checkbox-group-item\">\r\n      <div class=\"po-list-view-main-select\">\r\n        <po-checkbox\r\n          name=\"selectAll\"\r\n          [(ngModel)]=\"selectAll\"\r\n          [p-label]=\"literals.selectAll\"\r\n          (p-change)=\"selectAllListItems()\"\r\n          (click)=\"selectAllListItems()\"\r\n        ></po-checkbox>\r\n      </div>\r\n    </div>\r\n  </div>\r\n\r\n  <div class=\"po-list-view-main-content\">\r\n    <po-container [p-no-padding]=\"true\" *ngIf=\"!hasItems()\">\r\n      <div class=\"po-list-view-container-no-data po-text-center\">\r\n        <span> {{ literals.noData }} </span>\r\n      </div>\r\n    </po-container>\r\n\r\n    <ng-container *ngFor=\"let item of items; let index = index; trackBy: trackBy\">\r\n      <po-container [p-no-padding]=\"true\">\r\n        <div class=\"po-list-view-container\">\r\n          <div class=\"po-list-view-header\">\r\n            <div class=\"po-list-view-title\" [ngSwitch]=\"checkTitleType(item)\">\r\n              <div *ngIf=\"select\" class=\"po-list-view-select\">\r\n                <po-checkbox\r\n                  [(ngModel)]=\"item.$selected\"\r\n                  name=\"checkbox\"\r\n                  (p-change)=\"selectListItem(item)\"\r\n                  (click)=\"selectListItem(item)\"\r\n                ></po-checkbox>\r\n              </div>\r\n              <a\r\n                *ngSwitchCase=\"'externalLink'\"\r\n                class=\"po-list-view-title-link\"\r\n                target=\"_blank\"\r\n                [href]=\"item[propertyLink]\"\r\n                (click)=\"runTitleAction(item)\"\r\n              >\r\n                {{ getItemTitle(item) }}\r\n              </a>\r\n              <a\r\n                *ngSwitchCase=\"'internalLink'\"\r\n                class=\"po-list-view-title-link\"\r\n                [routerLink]=\"item[propertyLink]\"\r\n                (click)=\"runTitleAction(item)\"\r\n              >\r\n                {{ getItemTitle(item) }}\r\n              </a>\r\n              <span\r\n                *ngSwitchCase=\"'noLink'\"\r\n                class=\"po-list-view-title-no-link\"\r\n                [class.po-list-view-title-link]=\"titleHasAction\"\r\n                (click)=\"titleHasAction && runTitleAction(item)\"\r\n              >\r\n                {{ getItemTitle(item) }}\r\n              </span>\r\n            </div>\r\n\r\n            <div *ngIf=\"showButtonsActions\" class=\"po-list-view-actions\">\r\n              <po-button\r\n                *ngFor=\"let action of visibleActions\"\r\n                [p-disabled]=\"returnBooleanValue(action, item, 'disabled')\"\r\n                [p-icon]=\"action.icon\"\r\n                [p-label]=\"action.label\"\r\n                [p-kind]=\"action.type\"\r\n                (p-click)=\"onClickAction(action, item)\"\r\n              >\r\n              </po-button>\r\n            </div>\r\n\r\n            <div *ngIf=\"showPopupActions\" class=\"po-list-view-more-actions\">\r\n              <span\r\n                #popupTarget\r\n                class=\"po-icon po-icon-more po-list-view-more-icon po-clickable\"\r\n                (click)=\"togglePopup(item, popupTarget)\"\r\n              >\r\n              </span>\r\n            </div>\r\n          </div>\r\n\r\n          <div *ngIf=\"hasContentTemplate\" class=\"po-list-view-content\">\r\n            <ng-template\r\n              [ngTemplateOutlet]=\"listViewContentTemplate.templateRef\"\r\n              [ngTemplateOutletContext]=\"{ $implicit: item, index: index }\"\r\n            >\r\n            </ng-template>\r\n          </div>\r\n\r\n          <div\r\n            @showHideDetail\r\n            (@showHideDetail.start)=\"item.$showDetail ? onAnimationEvent($event, item) : 'undefined'\"\r\n            *ngIf=\"hasDetailTemplate && item.$showDetail\"\r\n            class=\"po-list-view-detail\"\r\n          >\r\n            <ng-template\r\n              [ngTemplateOutlet]=\"listViewDetailTemplate.templateRef\"\r\n              [ngTemplateOutletContext]=\"{ $implicit: item, index: index }\"\r\n            >\r\n            </ng-template>\r\n          </div>\r\n        </div>\r\n\r\n        <div *ngIf=\"hasDetailTemplate\" class=\"po-list-view-detail-button\">\r\n          <po-button\r\n            p-kind=\"tertiary\"\r\n            [p-label]=\"item.$showDetail ? literals.hideDetails : literals.showDetails\"\r\n            (p-click)=\"item.$showDetail = !item.$showDetail\"\r\n          >\r\n          </po-button>\r\n        </div>\r\n      </po-container>\r\n    </ng-container>\r\n  </div>\r\n</div>\r\n\r\n<div *ngIf=\"displayShowMoreButton\" class=\"po-mb-2 po-text-center\">\r\n  <po-button [p-disabled]=\"showMoreDisabled\" [p-label]=\"literals.loadMoreData\" (p-click)=\"onShowMore()\"> </po-button>\r\n</div>\r\n\r\n<po-popup #popup [p-actions]=\"actions\" [p-target]=\"popupTarget\"> </po-popup>\r\n" }]
    }], function () { return [{ type: i0.ChangeDetectorRef }, { type: i0.IterableDiffers }, { type: i1.PoLanguageService }]; }, { listViewContentTemplate: [{
            type: ContentChild,
            args: [PoListViewContentTemplateDirective, { static: true }]
        }], listViewDetailTemplate: [{
            type: ContentChild,
            args: [PoListViewDetailTemplateDirective, { static: true }]
        }], poPopupComponent: [{
            type: ViewChild,
            args: ['popup', { static: true }]
        }] }); })();
