// @po-ui/ng-components/esm2020/lib/components/po-tabs/po-tabs.module.mjs
import { CommonModule } from '@angular/common';
import { NgModule } from '@angular/core';
import { PoPopoverModule } from '../po-popover/po-popover.module';
import { PoTabButtonComponent } from './po-tab-button/po-tab-button.component';
import { PoTabComponent } from './po-tab/po-tab.component';
import { PoTabDropdownComponent } from './po-tab-dropdown/po-tab-dropdown.component';
import { PoTabsComponent } from './po-tabs.component';
import * as i0 from "@angular/core";
/**
 * @description
 * Módulo do componente po-tabs
 */
export class PoTabsModule {
}
PoTabsModule.ɵfac = function PoTabsModule_Factory(t) { return new (t || PoTabsModule)(); };
PoTabsModule.ɵmod = /*@__PURE__*/ i0.ɵɵdefineNgModule({ type: PoTabsModule });
PoTabsModule.ɵinj = /*@__PURE__*/ i0.ɵɵdefineInjector({ imports: [CommonModule, PoPopoverModule] });
(function () { (typeof ngDevMode === "undefined" || ngDevMode) && i0.ɵsetClassMetadata(PoTabsModule, [{
        type: NgModule,
        args: [{
                imports: [CommonModule, PoPopoverModule],
                declarations: [PoTabButtonComponent, PoTabComponent, PoTabDropdownComponent, PoTabsComponent],
                exports: [PoTabComponent, PoTabsComponent]
            }]
    }], null, null); })();
(function () { (typeof ngJitMode === "undefined" || ngJitMode) && i0.ɵɵsetNgModuleScope(PoTabsModule, { declarations: [PoTabButtonComponent, PoTabComponent, PoTabDropdownComponent, PoTabsComponent], imports: [CommonModule, PoPopoverModule], exports: [PoTabComponent, PoTabsComponent] }); })();
