// @po-ui/ng-components/esm2020/lib/components/po-listbox/po-item-list/po-item-list-base.component.mjs
import { __decorate } from "tslib";
import { Directive, EventEmitter, HostBinding, Input, Output } from '@angular/core';
import { InputBoolean } from '../../../decorators';
import { PoItemListType } from '../enums/po-item-list-type.enum';
import * as i0 from "@angular/core";
/**
 * @description
 *
 * O componente `po-item-list` é a menor parte da lista de ação que compõem o componente [**PO Listbox**](/documentation/po-listbox).
 */
export class PoItemListBaseComponent {
    constructor() {
        this._visible = true;
        this._disabled = false;
        this.danger = false;
        /**
         * @optional
         *
         * @description
         *
         * Define se a ação está selecionada.
         *
         * @default `false`
         */
        this.selected = false;
        /**
         * @optional
         *
         * @description
         *
         * Atribui uma linha separadora acima do item.
         *
         * @default `false`
         */
        this.separator = false;
        /**
         * @optional
         *
         * @description
         *
         * Ação a ser realizada ao clicar no item do tipo `option`.
         */
        this.clickItem = new EventEmitter();
        /**
         * @optional
         *
         * @description
         *
         * Ação a ser realizada ao selecionar no item do type `check`.
         */
        this.selectItem = new EventEmitter();
        // MULTISELECT PROPERTIES
        //emissao de evento do checkbox
        this.checkboxItem = new EventEmitter();
        this.fieldValue = 'value';
        this.fieldLabel = 'label';
    }
    set type(value) {
        this._type = PoItemListType[value] ?? 'action';
    }
    get type() {
        return this._type;
    }
    /**
     * @optional
     *
     * @description
     *
     * Define o estado como visível.
     *
     * @default `true`
     */
    set visible(value) {
        if (value === true || value === null || value === undefined) {
            this._visible = true;
        }
        else {
            this._visible = false;
        }
    }
    get visible() {
        return this._visible;
    }
    /**
     * @optional
     *
     * @description
     *
     * Define o estado como desabilitado.
     *
     * @default `false`
     */
    set disabled(value) {
        if (value === false || value === null || value === undefined) {
            this._disabled = false;
        }
        else {
            this._disabled = true;
        }
    }
    get disabled() {
        return this._disabled;
    }
}
PoItemListBaseComponent.ɵfac = function PoItemListBaseComponent_Factory(t) { return new (t || PoItemListBaseComponent)(); };
PoItemListBaseComponent.ɵdir = /*@__PURE__*/ i0.ɵɵdefineDirective({ type: PoItemListBaseComponent, hostVars: 1, hostBindings: function PoItemListBaseComponent_HostBindings(rf, ctx) { if (rf & 2) {
        i0.ɵɵattribute("p-type", ctx.type);
    } }, inputs: { type: ["p-type", "type"], visible: ["p-visible", "visible"], item: ["p-item", "item"], label: ["p-label", "label"], value: ["p-value", "value"], danger: ["p-danger", "danger"], disabled: ["p-disabled", "disabled"], selected: ["p-selected", "selected"], separator: ["p-separator", "separator"], icon: ["p-icon", "icon"], checkboxValue: ["p-checkbox-value", "checkboxValue"], fieldValue: ["p-field-value", "fieldValue"], fieldLabel: ["p-field-label", "fieldLabel"], multiselectTemplate: ["p-multiselect-template", "multiselectTemplate"], templateContext: ["p-template-context", "templateContext"] }, outputs: { clickItem: "p-click-item", selectItem: "p-select-item", checkboxItem: "p-selectcheckbox-item" } });
__decorate([
    InputBoolean()
], PoItemListBaseComponent.prototype, "danger", void 0);
__decorate([
    InputBoolean()
], PoItemListBaseComponent.prototype, "selected", void 0);
__decorate([
    InputBoolean()
], PoItemListBaseComponent.prototype, "separator", void 0);
(function () { (typeof ngDevMode === "undefined" || ngDevMode) && i0.ɵsetClassMetadata(PoItemListBaseComponent, [{
        type: Directive
    }], function () { return []; }, { type: [{
            type: HostBinding,
            args: ['attr.p-type']
        }, {
            type: Input,
            args: ['p-type']
        }], visible: [{
            type: Input,
            args: ['p-visible']
        }], item: [{
            type: Input,
            args: ['p-item']
        }], label: [{
            type: Input,
            args: ['p-label']
        }], value: [{
            type: Input,
            args: ['p-value']
        }], danger: [{
            type: Input,
            args: ['p-danger']
        }], disabled: [{
            type: Input,
            args: ['p-disabled']
        }], selected: [{
            type: Input,
            args: ['p-selected']
        }], separator: [{
            type: Input,
            args: ['p-separator']
        }], icon: [{
            type: Input,
            args: ['p-icon']
        }], clickItem: [{
            type: Output,
            args: ['p-click-item']
        }], selectItem: [{
            type: Output,
            args: ['p-select-item']
        }], checkboxItem: [{
            type: Output,
            args: ['p-selectcheckbox-item']
        }], checkboxValue: [{
            type: Input,
            args: ['p-checkbox-value']
        }], fieldValue: [{
            type: Input,
            args: ['p-field-value']
        }], fieldLabel: [{
            type: Input,
            args: ['p-field-label']
        }], multiselectTemplate: [{
            type: Input,
            args: ['p-multiselect-template']
        }], templateContext: [{
            type: Input,
            args: ['p-template-context']
        }] }); })();
