// @po-ui/ng-components/esm2020/lib/components/po-listbox/po-item-list/po-item-list.component.mjs
import { Component, ViewChild } from '@angular/core';
import { PoItemListBaseComponent } from './po-item-list-base.component';
import * as i0 from "@angular/core";
import * as i1 from "@angular/router";
import * as i2 from "@angular/common";
import * as i3 from "../../po-field/po-checkbox/po-checkbox.component";
import * as i4 from "../../po-icon/po-icon.component";
const _c0 = ["itemList"];
function PoItemListComponent_div_1_po_icon_2_Template(rf, ctx) { if (rf & 1) {
    i0.ɵɵelement(0, "po-icon", 8);
} if (rf & 2) {
    const ctx_r4 = i0.ɵɵnextContext(2);
    i0.ɵɵproperty("p-icon", ctx_r4.icon);
} }
function PoItemListComponent_div_1_Template(rf, ctx) { if (rf & 1) {
    i0.ɵɵelementStart(0, "div", 4, 5);
    i0.ɵɵtemplate(2, PoItemListComponent_div_1_po_icon_2_Template, 1, 1, "po-icon", 6);
    i0.ɵɵelementStart(3, "span", 7);
    i0.ɵɵtext(4);
    i0.ɵɵelementEnd()();
} if (rf & 2) {
    const ctx_r0 = i0.ɵɵnextContext();
    i0.ɵɵclassProp("po-item-list__separator", ctx_r0.separator)("po-item-list__disabled", ctx_r0.disabled || !ctx_r0.visible)("po-item-list__selected", ctx_r0.selected && !ctx_r0.disabled)("po-item-list__danger", ctx_r0.danger);
    i0.ɵɵattribute("aria-disabled", ctx_r0.disabled || !ctx_r0.visible)("aria-label", ctx_r0.label);
    i0.ɵɵadvance(2);
    i0.ɵɵproperty("ngIf", ctx_r0.icon);
    i0.ɵɵadvance(2);
    i0.ɵɵtextInterpolate(ctx_r0.label);
} }
function PoItemListComponent_div_2_Template(rf, ctx) { if (rf & 1) {
    const _r7 = i0.ɵɵgetCurrentView();
    i0.ɵɵelementStart(0, "div", 9, 5);
    i0.ɵɵlistener("click", function PoItemListComponent_div_2_Template_div_click_0_listener() { i0.ɵɵrestoreView(_r7); const ctx_r6 = i0.ɵɵnextContext(); return i0.ɵɵresetView(ctx_r6.onSelectItem({ label: ctx_r6.label })); });
    i0.ɵɵelementStart(2, "span", 7);
    i0.ɵɵtext(3);
    i0.ɵɵelementEnd()();
} if (rf & 2) {
    const ctx_r1 = i0.ɵɵnextContext();
    i0.ɵɵclassProp("po-item-list__separator", ctx_r1.separator)("po-item-list__selected", ctx_r1.selected && !ctx_r1.disabled)("po-item-list__disabled", ctx_r1.disabled);
    i0.ɵɵattribute("aria-disabled", ctx_r1.disabled)("aria-label", ctx_r1.label);
    i0.ɵɵadvance(3);
    i0.ɵɵtextInterpolate(ctx_r1.label);
} }
function PoItemListComponent_div_3_span_2_Template(rf, ctx) { if (rf & 1) {
    i0.ɵɵelementStart(0, "span", 14);
    i0.ɵɵtext(1);
    i0.ɵɵelementEnd();
} if (rf & 2) {
    const ctx_r8 = i0.ɵɵnextContext(2);
    i0.ɵɵadvance(1);
    i0.ɵɵtextInterpolate(ctx_r8.label);
} }
function PoItemListComponent_div_3_ng_template_3_Template(rf, ctx) { }
const _c1 = function (a0) { return { $implicit: a0 }; };
function PoItemListComponent_div_3_Template(rf, ctx) { if (rf & 1) {
    const _r11 = i0.ɵɵgetCurrentView();
    i0.ɵɵelementStart(0, "div", 10);
    i0.ɵɵlistener("click", function PoItemListComponent_div_3_Template_div_click_0_listener() { i0.ɵɵrestoreView(_r11); const ctx_r10 = i0.ɵɵnextContext(); return i0.ɵɵresetView(ctx_r10.onCheckboxItem()); })("keydown", function PoItemListComponent_div_3_Template_div_keydown_0_listener($event) { i0.ɵɵrestoreView(_r11); const ctx_r12 = i0.ɵɵnextContext(); return i0.ɵɵresetView(ctx_r12.onCheckboxItemEmit($event)); })("keyup", function PoItemListComponent_div_3_Template_div_keyup_0_listener($event) { i0.ɵɵrestoreView(_r11); const ctx_r13 = i0.ɵɵnextContext(); return i0.ɵɵresetView(ctx_r13.onCheckboxItemEmit($event)); });
    i0.ɵɵelementStart(1, "po-checkbox", 11);
    i0.ɵɵlistener("p-change", function PoItemListComponent_div_3_Template_po_checkbox_p_change_1_listener() { i0.ɵɵrestoreView(_r11); const ctx_r14 = i0.ɵɵnextContext(); return i0.ɵɵresetView(ctx_r14.onSelectItem({ label: ctx_r14.label })); });
    i0.ɵɵelementEnd();
    i0.ɵɵtemplate(2, PoItemListComponent_div_3_span_2_Template, 2, 1, "span", 12);
    i0.ɵɵtemplate(3, PoItemListComponent_div_3_ng_template_3_Template, 0, 0, "ng-template", 13);
    i0.ɵɵelementEnd();
} if (rf & 2) {
    const ctx_r2 = i0.ɵɵnextContext();
    i0.ɵɵclassProp("po-item-list__separator", ctx_r2.separator)("po-item-list__selected", ctx_r2.selected && !ctx_r2.disabled)("po-item-list__disabled", ctx_r2.disabled);
    i0.ɵɵattribute("aria-disabled", ctx_r2.disabled)("aria-label", ctx_r2.label);
    i0.ɵɵadvance(1);
    i0.ɵɵproperty("p-disabled-tabindex", true)("p-checkboxValue", ctx_r2.checkboxValue === null ? "mixed" : ctx_r2.checkboxValue)("p-disabled", ctx_r2.disabled);
    i0.ɵɵadvance(1);
    i0.ɵɵproperty("ngIf", !ctx_r2.multiselectTemplate);
    i0.ɵɵadvance(1);
    i0.ɵɵproperty("ngTemplateOutlet", ctx_r2.multiselectTemplate == null ? null : ctx_r2.multiselectTemplate.templateRef)("ngTemplateOutletContext", i0.ɵɵpureFunction1(14, _c1, ctx_r2.templateContext));
} }
export class PoItemListComponent extends PoItemListBaseComponent {
    constructor(router) {
        super();
        this.router = router;
    }
    onSelectItem(itemListOption) {
        this.selectedView = itemListOption;
        this.selectItem.emit(itemListOption);
    }
    onCheckboxItem() {
        const option = { [this.fieldValue]: this.value, [this.fieldLabel]: this.label };
        const selected = !this.checkboxValue;
        this.checkboxItem.emit({ option, selected });
    }
    onCheckboxItemEmit(event) {
        if ((event && event.code === 'Enter') || event.code === 'Space') {
            this.onCheckboxItem();
        }
    }
}
PoItemListComponent.ɵfac = function PoItemListComponent_Factory(t) { return new (t || PoItemListComponent)(i0.ɵɵdirectiveInject(i1.Router)); };
PoItemListComponent.ɵcmp = /*@__PURE__*/ i0.ɵɵdefineComponent({ type: PoItemListComponent, selectors: [["po-item-list"]], viewQuery: function PoItemListComponent_Query(rf, ctx) { if (rf & 1) {
        i0.ɵɵviewQuery(_c0, 7);
    } if (rf & 2) {
        let _t;
        i0.ɵɵqueryRefresh(_t = i0.ɵɵloadQuery()) && (ctx.itemList = _t.first);
    } }, features: [i0.ɵɵInheritDefinitionFeature], decls: 4, vars: 3, consts: [[3, "ngSwitch", "keydown"], ["class", "po-item-list po-item-list__action", 3, "po-item-list__separator", "po-item-list__disabled", "po-item-list__selected", "po-item-list__danger", 4, "ngSwitchDefault"], ["class", "po-item-list po-item-list__option", 3, "po-item-list__separator", "po-item-list__selected", "po-item-list__disabled", "click", 4, "ngSwitchCase"], ["class", "po-item-list po-item-list__check", 3, "po-item-list__separator", "po-item-list__selected", "po-item-list__disabled", "click", "keydown", "keyup", 4, "ngSwitchCase"], [1, "po-item-list", "po-item-list__action"], ["itemList", ""], ["class", "po-popup-icon-item po-icon", 3, "p-icon", 4, "ngIf"], [1, "po-item-list-label"], [1, "po-popup-icon-item", "po-icon", 3, "p-icon"], [1, "po-item-list", "po-item-list__option", 3, "click"], [1, "po-item-list", "po-item-list__check", 3, "click", "keydown", "keyup"], [3, "p-disabled-tabindex", "p-checkboxValue", "p-disabled", "p-change"], ["class", "po-item-list__label", 4, "ngIf"], [3, "ngTemplateOutlet", "ngTemplateOutletContext"], [1, "po-item-list__label"]], template: function PoItemListComponent_Template(rf, ctx) { if (rf & 1) {
        i0.ɵɵelementContainerStart(0, 0);
        i0.ɵɵlistener("keydown", function PoItemListComponent_Template_ng_container_keydown_0_listener($event) { return ctx.onCheckboxItemEmit($event); });
        i0.ɵɵtemplate(1, PoItemListComponent_div_1_Template, 5, 12, "div", 1);
        i0.ɵɵtemplate(2, PoItemListComponent_div_2_Template, 4, 9, "div", 2);
        i0.ɵɵtemplate(3, PoItemListComponent_div_3_Template, 4, 16, "div", 3);
        i0.ɵɵelementContainerEnd();
    } if (rf & 2) {
        i0.ɵɵproperty("ngSwitch", ctx.type);
        i0.ɵɵadvance(2);
        i0.ɵɵproperty("ngSwitchCase", "option");
        i0.ɵɵadvance(1);
        i0.ɵɵproperty("ngSwitchCase", "check");
    } }, dependencies: [i2.NgIf, i2.NgTemplateOutlet, i2.NgSwitch, i2.NgSwitchCase, i2.NgSwitchDefault, i3.PoCheckboxComponent, i4.PoIconComponent], encapsulation: 2 });
(function () { (typeof ngDevMode === "undefined" || ngDevMode) && i0.ɵsetClassMetadata(PoItemListComponent, [{
        type: Component,
        args: [{ selector: 'po-item-list', template: "<ng-container [ngSwitch]=\"type\" (keydown)=\"onCheckboxItemEmit($event)\">\r\n  <div\r\n    *ngSwitchDefault\r\n    #itemList\r\n    [class.po-item-list__separator]=\"separator\"\r\n    [class.po-item-list__disabled]=\"disabled || !visible\"\r\n    [class.po-item-list__selected]=\"selected && !disabled\"\r\n    [attr.aria-disabled]=\"disabled || !visible\"\r\n    [attr.aria-label]=\"label\"\r\n    [class.po-item-list__danger]=\"danger\"\r\n    class=\"po-item-list po-item-list__action\"\r\n  >\r\n    <po-icon *ngIf=\"icon\" class=\"po-popup-icon-item po-icon\" [p-icon]=\"icon\"></po-icon>\r\n    <span class=\"po-item-list-label\">{{ label }}</span>\r\n  </div>\r\n  <div\r\n    *ngSwitchCase=\"'option'\"\r\n    #itemList\r\n    [class.po-item-list__separator]=\"separator\"\r\n    [class.po-item-list__selected]=\"selected && !disabled\"\r\n    [class.po-item-list__disabled]=\"disabled\"\r\n    [attr.aria-disabled]=\"disabled\"\r\n    [attr.aria-label]=\"label\"\r\n    class=\"po-item-list po-item-list__option\"\r\n    (click)=\"onSelectItem({label})\"\r\n  >\r\n    <span class=\"po-item-list-label\">{{ label }}</span>\r\n  </div>\r\n\r\n  <div\r\n    *ngSwitchCase=\"'check'\"\r\n    class=\"po-item-list po-item-list__check\"\r\n    [class.po-item-list__separator]=\"separator\"\r\n    [class.po-item-list__selected]=\"selected && !disabled\"\r\n    [class.po-item-list__disabled]=\"disabled\"\r\n    [attr.aria-disabled]=\"disabled\"\r\n    [attr.aria-label]=\"label\"\r\n    (click)=\"onCheckboxItem()\"\r\n    (keydown)=\"onCheckboxItemEmit($event)\"\r\n    (keyup)=\"onCheckboxItemEmit($event)\"\r\n  >\r\n    <po-checkbox\r\n      [p-disabled-tabindex]=\"true\"\r\n      [p-checkboxValue]=\"checkboxValue === null ? 'mixed' : checkboxValue\"\r\n      [p-disabled]=\"disabled\"\r\n      (p-change)=\"onSelectItem({label})\"\r\n    ></po-checkbox>\r\n\r\n    <span class=\"po-item-list__label\" *ngIf=\"!multiselectTemplate\">{{ label }}</span>\r\n\r\n    <ng-template\r\n      [ngTemplateOutlet]=\"multiselectTemplate?.templateRef\"\r\n      [ngTemplateOutletContext]=\"{ $implicit: templateContext }\"\r\n    >\r\n    </ng-template>\r\n  </div>\r\n</ng-container>\r\n" }]
    }], function () { return [{ type: i1.Router }]; }, { itemList: [{
            type: ViewChild,
            args: ['itemList', { static: true }]
        }] }); })();
