// @po-ui/ng-components/esm2020/lib/components/po-image/po-image.component.mjs
import { Component } from '@angular/core';
import { PoImageBaseComponent } from './po-image-base.component';
import * as i0 from "@angular/core";
import * as i1 from "@angular/common";
import * as i2 from "../po-container/po-container.component";
function PoImageComponent_po_container_0_img_1_Template(rf, ctx) { if (rf & 1) {
    i0.ɵɵelement(0, "img", 4);
} if (rf & 2) {
    const ctx_r3 = i0.ɵɵnextContext(2);
    i0.ɵɵproperty("priority", ctx_r3.priority)("loading", ctx_r3.loading)("fill", true)("ngSrc", ctx_r3.source);
    i0.ɵɵattribute("alt", ctx_r3.alternate);
} }
function PoImageComponent_po_container_0_Template(rf, ctx) { if (rf & 1) {
    i0.ɵɵelementStart(0, "po-container", 2);
    i0.ɵɵtemplate(1, PoImageComponent_po_container_0_img_1_Template, 1, 5, "img", 3);
    i0.ɵɵelementEnd();
} if (rf & 2) {
    const ctx_r0 = i0.ɵɵnextContext();
    const _r1 = i0.ɵɵreference(2);
    i0.ɵɵproperty("p-height", ctx_r0.height);
    i0.ɵɵadvance(1);
    i0.ɵɵproperty("ngIf", !ctx_r0.isBase64)("ngIfElse", _r1);
} }
function PoImageComponent_ng_template_1_Template(rf, ctx) { if (rf & 1) {
    i0.ɵɵelement(0, "img", 5);
} if (rf & 2) {
    const ctx_r2 = i0.ɵɵnextContext();
    i0.ɵɵproperty("loading", ctx_r2.loading)("src", ctx_r2.source, i0.ɵɵsanitizeUrl);
    i0.ɵɵattribute("alt", ctx_r2.alternate);
} }
/**
 * @docsExtends PoImageBaseComponent
 *
 * @example
 *
 * <example name="po-image-basic" title="PO Image Basic" >
 *  <file name="sample-po-image-basic/sample-po-image-basic.component.html"> </file>
 *  <file name="sample-po-image-basic/sample-po-image-basic.component.ts"> </file>
 * </example>
 *
 * <example name="po-image-labs" title="PO Image Labs" >
 *  <file name="sample-po-image-labs/sample-po-image-labs.component.html"> </file>
 *  <file name="sample-po-image-labs/sample-po-image-labs.component.ts"> </file>
 * </example>
 *
 * <example name="po-image-travel" title="PO Image Travel" >
 *  <file name="sample-po-image-travel/sample-po-image-travel.component.html"> </file>
 *  <file name="sample-po-image-travel/sample-po-image-travel.component.ts"> </file>
 * </example>
 *
 */
export class PoImageComponent extends PoImageBaseComponent {
}
PoImageComponent.ɵfac = /*@__PURE__*/ function () { let ɵPoImageComponent_BaseFactory; return function PoImageComponent_Factory(t) { return (ɵPoImageComponent_BaseFactory || (ɵPoImageComponent_BaseFactory = i0.ɵɵgetInheritedFactory(PoImageComponent)))(t || PoImageComponent); }; }();
PoImageComponent.ɵcmp = /*@__PURE__*/ i0.ɵɵdefineComponent({ type: PoImageComponent, selectors: [["po-image"]], features: [i0.ɵɵInheritDefinitionFeature], decls: 3, vars: 1, consts: [["class", "po-container-image", 3, "p-height", 4, "ngIf"], ["imageBase64", ""], [1, "po-container-image", 3, "p-height"], ["class", "po-image", 3, "priority", "loading", "fill", "ngSrc", 4, "ngIf", "ngIfElse"], [1, "po-image", 3, "priority", "loading", "fill", "ngSrc"], [1, "po-image", 3, "loading", "src"]], template: function PoImageComponent_Template(rf, ctx) { if (rf & 1) {
        i0.ɵɵtemplate(0, PoImageComponent_po_container_0_Template, 2, 3, "po-container", 0);
        i0.ɵɵtemplate(1, PoImageComponent_ng_template_1_Template, 1, 3, "ng-template", null, 1, i0.ɵɵtemplateRefExtractor);
    } if (rf & 2) {
        i0.ɵɵproperty("ngIf", ctx.source);
    } }, dependencies: [i1.NgIf, i2.PoContainerComponent, i1.NgOptimizedImage], encapsulation: 2 });
(function () { (typeof ngDevMode === "undefined" || ngDevMode) && i0.ɵsetClassMetadata(PoImageComponent, [{
        type: Component,
        args: [{ selector: 'po-image', template: "<po-container class=\"po-container-image\" *ngIf=\"source\" [p-height]=\"height\">\r\n  <img\r\n    *ngIf=\"!isBase64; else imageBase64\"\r\n    class=\"po-image\"\r\n    [attr.alt]=\"alternate\"\r\n    [priority]=\"priority\"\r\n    [loading]=\"loading\"\r\n    [fill]=\"true\"\r\n    [ngSrc]=\"source\"\r\n  />\r\n</po-container>\r\n\r\n<ng-template #imageBase64>\r\n  <img class=\"po-image\" [attr.alt]=\"alternate\" [loading]=\"loading\" [src]=\"source\" />\r\n</ng-template>\r\n" }]
    }], null, null); })();
