// @po-ui/ng-components/esm2020/lib/components/po-navbar/po-navbar-actions/po-navbar-actions.module.mjs
import { CommonModule } from '@angular/common';
import { NgModule } from '@angular/core';
import { PoNavbarActionComponent } from './po-navbar-action/po-navbar-action.component';
import { PoNavbarActionPopupComponent } from './po-navbar-action-popup/po-navbar-action-popup.component';
import { PoNavbarActionsComponent } from './po-navbar-actions.component';
import { PoPopupModule } from '../../po-popup/po-popup.module';
import { PoTooltipModule } from '../../../directives/po-tooltip/po-tooltip.module';
import { PoIconModule } from './../../po-icon/po-icon.module';
import * as i0 from "@angular/core";
export class PoNavbarActionsModule {
}
PoNavbarActionsModule.ɵfac = function PoNavbarActionsModule_Factory(t) { return new (t || PoNavbarActionsModule)(); };
PoNavbarActionsModule.ɵmod = /*@__PURE__*/ i0.ɵɵdefineNgModule({ type: PoNavbarActionsModule });
PoNavbarActionsModule.ɵinj = /*@__PURE__*/ i0.ɵɵdefineInjector({ imports: [CommonModule, PoPopupModule, PoTooltipModule, PoIconModule] });
(function () { (typeof ngDevMode === "undefined" || ngDevMode) && i0.ɵsetClassMetadata(PoNavbarActionsModule, [{
        type: NgModule,
        args: [{
                imports: [CommonModule, PoPopupModule, PoTooltipModule, PoIconModule],
                declarations: [PoNavbarActionComponent, PoNavbarActionPopupComponent, PoNavbarActionsComponent],
                exports: [PoNavbarActionsComponent]
            }]
    }], null, null); })();
(function () { (typeof ngJitMode === "undefined" || ngJitMode) && i0.ɵɵsetNgModuleScope(PoNavbarActionsModule, { declarations: [PoNavbarActionComponent, PoNavbarActionPopupComponent, PoNavbarActionsComponent], imports: [CommonModule, PoPopupModule, PoTooltipModule, PoIconModule], exports: [PoNavbarActionsComponent] }); })();
