// @po-ui/ng-components/esm2020/lib/components/po-page/po-page-header/po-page-header.component.mjs
import { Component } from '@angular/core';
import { PoPageHeaderBaseComponent } from './po-page-header-base.component';
import * as i0 from "@angular/core";
import * as i1 from "@angular/common";
import * as i2 from "../../po-breadcrumb/po-breadcrumb.component";
function PoPageHeaderComponent_div_1_Template(rf, ctx) { if (rf & 1) {
    i0.ɵɵelementStart(0, "div", 4);
    i0.ɵɵelement(1, "po-breadcrumb", 5);
    i0.ɵɵelementEnd();
} if (rf & 2) {
    const ctx_r0 = i0.ɵɵnextContext();
    i0.ɵɵadvance(1);
    i0.ɵɵproperty("p-favorite-service", ctx_r0.breadcrumb.favorite)("p-items", ctx_r0.breadcrumb.items)("p-params-service", ctx_r0.breadcrumb.params);
} }
function PoPageHeaderComponent_h1_3_Template(rf, ctx) { if (rf & 1) {
    i0.ɵɵelementStart(0, "h1", 6);
    i0.ɵɵtext(1);
    i0.ɵɵelementEnd();
} if (rf & 2) {
    const ctx_r1 = i0.ɵɵnextContext();
    i0.ɵɵadvance(1);
    i0.ɵɵtextInterpolate1(" ", ctx_r1.title, " ");
} }
const _c0 = ["*"];
/**
 * @docsPrivate
 * @docsExtends PoPageHeaderBaseComponent
 */
export class PoPageHeaderComponent extends PoPageHeaderBaseComponent {
}
PoPageHeaderComponent.ɵfac = /*@__PURE__*/ function () { let ɵPoPageHeaderComponent_BaseFactory; return function PoPageHeaderComponent_Factory(t) { return (ɵPoPageHeaderComponent_BaseFactory || (ɵPoPageHeaderComponent_BaseFactory = i0.ɵɵgetInheritedFactory(PoPageHeaderComponent)))(t || PoPageHeaderComponent); }; }();
PoPageHeaderComponent.ɵcmp = /*@__PURE__*/ i0.ɵɵdefineComponent({ type: PoPageHeaderComponent, selectors: [["po-page-header"]], features: [i0.ɵɵInheritDefinitionFeature], ngContentSelectors: _c0, decls: 5, vars: 2, consts: [[1, "po-page-header"], ["class", "po-page-header-breadcrumb", 4, "ngIf"], [1, "po-pt-1"], ["class", "po-page-header-title", 4, "ngIf"], [1, "po-page-header-breadcrumb"], [3, "p-favorite-service", "p-items", "p-params-service"], [1, "po-page-header-title"]], template: function PoPageHeaderComponent_Template(rf, ctx) { if (rf & 1) {
        i0.ɵɵprojectionDef();
        i0.ɵɵelementStart(0, "div", 0);
        i0.ɵɵtemplate(1, PoPageHeaderComponent_div_1_Template, 2, 3, "div", 1);
        i0.ɵɵelementStart(2, "div", 2);
        i0.ɵɵtemplate(3, PoPageHeaderComponent_h1_3_Template, 2, 1, "h1", 3);
        i0.ɵɵprojection(4);
        i0.ɵɵelementEnd()();
    } if (rf & 2) {
        i0.ɵɵadvance(1);
        i0.ɵɵproperty("ngIf", ctx.breadcrumb && ctx.breadcrumb.items.length);
        i0.ɵɵadvance(2);
        i0.ɵɵproperty("ngIf", ctx.title);
    } }, dependencies: [i1.NgIf, i2.PoBreadcrumbComponent], encapsulation: 2 });
(function () { (typeof ngDevMode === "undefined" || ngDevMode) && i0.ɵsetClassMetadata(PoPageHeaderComponent, [{
        type: Component,
        args: [{ selector: 'po-page-header', template: "<div class=\"po-page-header\">\r\n  <!-- Breadcrumb -->\r\n  <div class=\"po-page-header-breadcrumb\" *ngIf=\"breadcrumb && breadcrumb.items.length\">\r\n    <po-breadcrumb\r\n      [p-favorite-service]=\"breadcrumb.favorite\"\r\n      [p-items]=\"breadcrumb.items\"\r\n      [p-params-service]=\"breadcrumb.params\"\r\n    >\r\n    </po-breadcrumb>\r\n  </div>\r\n\r\n  <div class=\"po-pt-1\">\r\n    <!-- Titulo -->\r\n    <h1 class=\"po-page-header-title\" *ngIf=\"title\">\r\n      {{ title }}\r\n    </h1>\r\n\r\n    <!-- Opera\u00E7\u00F5es -->\r\n    <ng-content></ng-content>\r\n  </div>\r\n</div>\r\n" }]
    }], null, null); })();
