// @po-ui/ng-components/esm2020/lib/components/po-field/po-field-validate.model.mjs
import { __decorate } from "tslib";
import { Directive, Input } from '@angular/core';
import { InputBoolean } from '../../decorators';
import { PoFieldModel } from './po-field.model';
import { requiredFailed } from './validators';
import * as i0 from "@angular/core";
/**
 * @docsExtends PoFieldModel
 */
export class PoFieldValidateModel extends PoFieldModel {
    constructor() {
        super(...arguments);
        /**
         * @optional
         *
         * @description
         *
         * Define se a indicação de campo opcional será exibida.
         *
         * > Não será exibida a indicação se:
         * - O campo conter `p-required`;
         * - Não possuir `p-help` e/ou `p-label`.
         *
         * @default `false`
         */
        this.optional = false;
        /**
         * @optional
         *
         * @description
         *
         * Define que o campo será obrigatório.
         *
         * @default `false`
         */
        this.required = false;
        /**
         *  Define se a indicação de campo obrigatório será exibida.
         *
         * > Não será exibida a indicação se:
         * - Não possuir `p-help` e/ou `p-label`.
         */
        this.showRequired = false;
    }
    validate(abstractControl) {
        if (requiredFailed(this.required, this.disabled, abstractControl.value)) {
            return {
                required: {
                    valid: false
                }
            };
        }
        return this.extraValidation(abstractControl);
    }
    registerOnValidatorChange(fn) {
        this.onValidatorChange = fn;
    }
    validateModel() {
        if (this.onValidatorChange) {
            this.onValidatorChange();
        }
    }
}
PoFieldValidateModel.ɵfac = /*@__PURE__*/ function () { let ɵPoFieldValidateModel_BaseFactory; return function PoFieldValidateModel_Factory(t) { return (ɵPoFieldValidateModel_BaseFactory || (ɵPoFieldValidateModel_BaseFactory = i0.ɵɵgetInheritedFactory(PoFieldValidateModel)))(t || PoFieldValidateModel); }; }();
PoFieldValidateModel.ɵdir = /*@__PURE__*/ i0.ɵɵdefineDirective({ type: PoFieldValidateModel, inputs: { optional: ["p-optional", "optional"], required: ["p-required", "required"], showRequired: ["p-show-required", "showRequired"] }, features: [i0.ɵɵInheritDefinitionFeature] });
__decorate([
    InputBoolean()
], PoFieldValidateModel.prototype, "optional", void 0);
__decorate([
    InputBoolean()
], PoFieldValidateModel.prototype, "required", void 0);
(function () { (typeof ngDevMode === "undefined" || ngDevMode) && i0.ɵsetClassMetadata(PoFieldValidateModel, [{
        type: Directive
    }], null, { optional: [{
            type: Input,
            args: ['p-optional']
        }], required: [{
            type: Input,
            args: ['p-required']
        }], showRequired: [{
            type: Input,
            args: ['p-show-required']
        }] }); })();
