// @po-ui/ng-components/esm2020/lib/components/po-modal/po-modal.component.mjs
import { Component, ContentChild, ElementRef, ViewChild } from '@angular/core';
import { PoModalBaseComponent } from './po-modal-base.component';
import { PoModalFooterComponent } from './po-modal-footer/po-modal-footer.component';
import { uuid } from '../../utils/util';
import * as i0 from "@angular/core";
import * as i1 from "../../services/po-active-overlay/po-active-overlay.service";
import * as i2 from "../../services/po-language/po-language.service";
import * as i3 from "@angular/common";
import * as i4 from "../po-button/po-button.component";
import * as i5 from "../po-icon/po-icon.component";
import * as i6 from "./po-modal-footer/po-modal-footer.component";
const _c0 = ["modalContent"];
const _c1 = ["modalContainer"];
function PoModalComponent_div_0_po_icon_9_Template(rf, ctx) { if (rf & 1) {
    i0.ɵɵelement(0, "po-icon", 16);
} if (rf & 2) {
    const ctx_r3 = i0.ɵɵnextContext(2);
    i0.ɵɵproperty("p-icon", ctx_r3.icon);
} }
function PoModalComponent_div_0_po_button_12_Template(rf, ctx) { if (rf & 1) {
    const _r9 = i0.ɵɵgetCurrentView();
    i0.ɵɵelementStart(0, "po-button", 17);
    i0.ɵɵlistener("p-click", function PoModalComponent_div_0_po_button_12_Template_po_button_p_click_0_listener() { i0.ɵɵrestoreView(_r9); const ctx_r8 = i0.ɵɵnextContext(2); return i0.ɵɵresetView(ctx_r8.close(true)); });
    i0.ɵɵelementEnd();
} if (rf & 2) {
    const ctx_r4 = i0.ɵɵnextContext(2);
    i0.ɵɵproperty("p-aria-label", ctx_r4.literals.close);
} }
function PoModalComponent_div_0_ng_container_15_Template(rf, ctx) { if (rf & 1) {
    i0.ɵɵelementContainerStart(0);
    i0.ɵɵprojection(1, 1);
    i0.ɵɵelementContainerEnd();
} }
function PoModalComponent_div_0_ng_template_16_po_button_2_Template(rf, ctx) { if (rf & 1) {
    const _r12 = i0.ɵɵgetCurrentView();
    i0.ɵɵelementStart(0, "po-button", 21);
    i0.ɵɵlistener("p-click", function PoModalComponent_div_0_ng_template_16_po_button_2_Template_po_button_p_click_0_listener() { i0.ɵɵrestoreView(_r12); const ctx_r11 = i0.ɵɵnextContext(3); return i0.ɵɵresetView(ctx_r11.secondaryAction.action()); });
    i0.ɵɵelementEnd();
} if (rf & 2) {
    const ctx_r10 = i0.ɵɵnextContext(3);
    i0.ɵɵproperty("p-danger", ctx_r10.getSecondaryActionButtonDanger())("p-disabled", ctx_r10.secondaryAction.disabled)("p-label", ctx_r10.secondaryAction.label)("p-loading", ctx_r10.secondaryAction.loading);
} }
function PoModalComponent_div_0_ng_template_16_Template(rf, ctx) { if (rf & 1) {
    const _r14 = i0.ɵɵgetCurrentView();
    i0.ɵɵelementStart(0, "po-modal-footer")(1, "div", 18);
    i0.ɵɵtemplate(2, PoModalComponent_div_0_ng_template_16_po_button_2_Template, 1, 4, "po-button", 19);
    i0.ɵɵelementStart(3, "po-button", 20);
    i0.ɵɵlistener("p-click", function PoModalComponent_div_0_ng_template_16_Template_po_button_p_click_3_listener() { i0.ɵɵrestoreView(_r14); const ctx_r13 = i0.ɵɵnextContext(2); return i0.ɵɵresetView(ctx_r13.primaryAction.action()); });
    i0.ɵɵelementEnd()()();
} if (rf & 2) {
    const ctx_r7 = i0.ɵɵnextContext(2);
    i0.ɵɵadvance(2);
    i0.ɵɵproperty("ngIf", ctx_r7.secondaryAction);
    i0.ɵɵadvance(1);
    i0.ɵɵproperty("p-danger", ctx_r7.primaryAction.danger)("p-disabled", ctx_r7.primaryAction.disabled)("p-label", ctx_r7.primaryAction.label)("p-loading", ctx_r7.primaryAction.loading);
} }
function PoModalComponent_div_0_Template(rf, ctx) { if (rf & 1) {
    const _r16 = i0.ɵɵgetCurrentView();
    i0.ɵɵelementStart(0, "div", 1, 2);
    i0.ɵɵlistener("keydown.esc", function PoModalComponent_div_0_Template_div_keydown_esc_0_listener($event) { i0.ɵɵrestoreView(_r16); const ctx_r15 = i0.ɵɵnextContext(); return i0.ɵɵresetView(ctx_r15.closeModalOnEscapeKey($event)); });
    i0.ɵɵelementStart(2, "div", 3);
    i0.ɵɵlistener("mousedown", function PoModalComponent_div_0_Template_div_mousedown_2_listener($event) { i0.ɵɵrestoreView(_r16); const ctx_r17 = i0.ɵɵnextContext(); return i0.ɵɵresetView(ctx_r17.onClickOut($event)); });
    i0.ɵɵelementEnd();
    i0.ɵɵelementStart(3, "div", 4)(4, "div", 5)(5, "div", 6, 7)(7, "div", 8)(8, "div", 9);
    i0.ɵɵtemplate(9, PoModalComponent_div_0_po_icon_9_Template, 1, 1, "po-icon", 10);
    i0.ɵɵelementStart(10, "div", 11);
    i0.ɵɵtext(11);
    i0.ɵɵelementEnd()();
    i0.ɵɵtemplate(12, PoModalComponent_div_0_po_button_12_Template, 1, 1, "po-button", 12);
    i0.ɵɵelementEnd();
    i0.ɵɵelementStart(13, "div", 13);
    i0.ɵɵprojection(14);
    i0.ɵɵelementEnd();
    i0.ɵɵtemplate(15, PoModalComponent_div_0_ng_container_15_Template, 2, 0, "ng-container", 14);
    i0.ɵɵtemplate(16, PoModalComponent_div_0_ng_template_16_Template, 4, 5, "ng-template", null, 15, i0.ɵɵtemplateRefExtractor);
    i0.ɵɵelementEnd()()()();
} if (rf & 2) {
    const _r6 = i0.ɵɵreference(17);
    const ctx_r0 = i0.ɵɵnextContext();
    i0.ɵɵadvance(5);
    i0.ɵɵclassMapInterpolate1("po-modal-content po-modal-", ctx_r0.size, "");
    i0.ɵɵadvance(4);
    i0.ɵɵproperty("ngIf", ctx_r0.icon);
    i0.ɵɵadvance(2);
    i0.ɵɵtextInterpolate1(" ", ctx_r0.title, " ");
    i0.ɵɵadvance(1);
    i0.ɵɵproperty("ngIf", !ctx_r0.hideClose);
    i0.ɵɵadvance(3);
    i0.ɵɵproperty("ngIf", ctx_r0.modalFooter)("ngIfElse", _r6);
} }
const _c2 = ["*", [["po-modal-footer"]]];
const _c3 = ["*", "po-modal-footer"];
/**
 * @docsExtends PoModalBaseComponent
 *
 * @example
 *
 * <example name="po-modal-basic" title="PO Modal Basic">
 *  <file name="sample-po-modal-basic/sample-po-modal-basic.component.html"> </file>
 *  <file name="sample-po-modal-basic/sample-po-modal-basic.component.ts"> </file>
 * </example>
 *
 * <example name="po-modal-labs" title="PO Modal Labs">
 *  <file name="sample-po-modal-labs/sample-po-modal-labs.component.html"> </file>
 *  <file name="sample-po-modal-labs/sample-po-modal-labs.component.ts"> </file>
 * </example>
 *
 * <example name="po-modal-fruits-salad" title="PO Modal - Fruits Salad">
 *  <file name="sample-po-modal-fruits-salad/sample-po-modal-fruits-salad.component.html"> </file>
 *  <file name="sample-po-modal-fruits-salad/sample-po-modal-fruits-salad.component.ts"> </file>
 * </example>
 */
export class PoModalComponent extends PoModalBaseComponent {
    constructor(poActiveOverlayService, poLanguageService) {
        super(poLanguageService);
        this.poActiveOverlayService = poActiveOverlayService;
        this.focusableElements = 'input, select, textarea, button:not([disabled]), a';
        this.id = uuid();
    }
    close(xClosed = false) {
        this.poActiveOverlayService.activeOverlay.pop();
        super.close(xClosed);
        this.removeEventListeners();
        if (this.sourceElement) {
            this.sourceElement.focus();
        }
    }
    closeModalOnEscapeKey(event) {
        if (!this.hideClose) {
            event.preventDefault();
            event.stopPropagation();
            this.close();
        }
    }
    getSecondaryActionButtonDanger() {
        return this.secondaryAction && this.secondaryAction.danger && !this.primaryAction.danger ? 'true' : 'false';
    }
    onClickOut(event) {
        if (this.clickOut && !this.modalContent.nativeElement.contains(event.target)) {
            this.close();
        }
    }
    open() {
        this.sourceElement = document.activeElement;
        super.open();
        this.handleFocus();
    }
    handleFocus() {
        this.poActiveOverlayService.activeOverlay.push(this.id);
        setTimeout(() => {
            if (this.modalContent) {
                this.initFocus();
                document.addEventListener('focus', this.focusFunction, true);
            }
        });
    }
    initFocus() {
        this.focusFunction = (event) => {
            const modalElement = this.modalContent.nativeElement;
            if (!modalElement.contains(event.target) &&
                this.poActiveOverlayService.activeOverlay[this.poActiveOverlayService.activeOverlay.length - 1] === this.id) {
                event.stopPropagation();
                this.firstElement.focus();
            }
        };
        this.setFirstElement();
        if (this.hideClose) {
            this.firstElement.focus();
        }
        else {
            const firstFieldElement = this.modalContent.nativeElement.querySelectorAll(this.focusableElements)[1] || this.modalContent.nativeElement;
            firstFieldElement.focus();
        }
    }
    removeEventListeners() {
        document.removeEventListener('focus', this.focusFunction, true);
    }
    setFirstElement() {
        this.firstElement =
            this.modalContent.nativeElement.querySelector(this.focusableElements) || this.modalContent.nativeElement;
    }
}
PoModalComponent.ɵfac = function PoModalComponent_Factory(t) { return new (t || PoModalComponent)(i0.ɵɵdirectiveInject(i1.PoActiveOverlayService), i0.ɵɵdirectiveInject(i2.PoLanguageService)); };
PoModalComponent.ɵcmp = /*@__PURE__*/ i0.ɵɵdefineComponent({ type: PoModalComponent, selectors: [["po-modal"]], contentQueries: function PoModalComponent_ContentQueries(rf, ctx, dirIndex) { if (rf & 1) {
        i0.ɵɵcontentQuery(dirIndex, PoModalFooterComponent, 5);
    } if (rf & 2) {
        let _t;
        i0.ɵɵqueryRefresh(_t = i0.ɵɵloadQuery()) && (ctx.modalFooter = _t.first);
    } }, viewQuery: function PoModalComponent_Query(rf, ctx) { if (rf & 1) {
        i0.ɵɵviewQuery(_c0, 5, ElementRef);
        i0.ɵɵviewQuery(_c1, 5, ElementRef);
    } if (rf & 2) {
        let _t;
        i0.ɵɵqueryRefresh(_t = i0.ɵɵloadQuery()) && (ctx.modalContent = _t.first);
        i0.ɵɵqueryRefresh(_t = i0.ɵɵloadQuery()) && (ctx.modalContainer = _t.first);
    } }, features: [i0.ɵɵInheritDefinitionFeature], ngContentSelectors: _c3, decls: 1, vars: 1, consts: [["class", "po-modal", "tabindex", "0", 3, "keydown.esc", 4, "ngIf"], ["tabindex", "0", 1, "po-modal", 3, "keydown.esc"], ["modalContainer", ""], [1, "po-modal-overlay", 3, "mousedown"], [1, "po-modal-container", "po-pb-2", "po-pt-2"], [1, "po-modal-vertical-align"], ["tabindex", "-1"], ["modalContent", ""], [1, "po-modal-header"], [1, "po-modal-title", "po-text-ellipsis"], [3, "p-icon", 4, "ngIf"], [1, "po-text-ellipsis"], ["p-icon", "po-icon po-icon-close", "p-kind", "tertiary", 3, "p-aria-label", "p-click", 4, "ngIf"], [1, "po-modal-body"], [4, "ngIf", "ngIfElse"], ["defaultModalFooterTemplate", ""], [3, "p-icon"], ["p-icon", "po-icon po-icon-close", "p-kind", "tertiary", 3, "p-aria-label", "p-click"], [1, "po-modal-footer-basic"], ["p-kind", "secondary", 3, "p-danger", "p-disabled", "p-label", "p-loading", "p-click", 4, "ngIf"], ["p-kind", "primary", 1, "po-button-modal-first-action", 3, "p-danger", "p-disabled", "p-label", "p-loading", "p-click"], ["p-kind", "secondary", 3, "p-danger", "p-disabled", "p-label", "p-loading", "p-click"]], template: function PoModalComponent_Template(rf, ctx) { if (rf & 1) {
        i0.ɵɵprojectionDef(_c2);
        i0.ɵɵtemplate(0, PoModalComponent_div_0_Template, 18, 8, "div", 0);
    } if (rf & 2) {
        i0.ɵɵproperty("ngIf", !ctx.isHidden);
    } }, dependencies: [i3.NgIf, i4.PoButtonComponent, i5.PoIconComponent, i6.PoModalFooterComponent], encapsulation: 2 });
(function () { (typeof ngDevMode === "undefined" || ngDevMode) && i0.ɵsetClassMetadata(PoModalComponent, [{
        type: Component,
        args: [{ selector: 'po-modal', template: "<div #modalContainer *ngIf=\"!isHidden\" class=\"po-modal\" tabindex=\"0\" (keydown.esc)=\"closeModalOnEscapeKey($event)\">\r\n  <div class=\"po-modal-overlay\" (mousedown)=\"onClickOut($event)\"></div>\r\n  <div class=\"po-modal-container po-pb-2 po-pt-2\">\r\n    <div class=\"po-modal-vertical-align\">\r\n      <div #modalContent class=\"po-modal-content po-modal-{{ size }}\" tabindex=\"-1\">\r\n        <div class=\"po-modal-header\">\r\n          <div class=\"po-modal-title po-text-ellipsis\">\r\n            <po-icon *ngIf=\"icon\" [p-icon]=\"icon\"></po-icon>\r\n            <div class=\"po-text-ellipsis\">\r\n              {{ title }}\r\n            </div>\r\n          </div>\r\n          <po-button\r\n            *ngIf=\"!hideClose\"\r\n            [p-aria-label]=\"literals.close\"\r\n            p-icon=\"po-icon po-icon-close\"\r\n            (p-click)=\"close(true)\"\r\n            p-kind=\"tertiary\"\r\n          >\r\n          </po-button>\r\n        </div>\r\n\r\n        <div class=\"po-modal-body\">\r\n          <ng-content></ng-content>\r\n        </div>\r\n\r\n        <ng-container *ngIf=\"modalFooter; else defaultModalFooterTemplate\">\r\n          <ng-content select=\"po-modal-footer\"></ng-content>\r\n        </ng-container>\r\n\r\n        <ng-template #defaultModalFooterTemplate>\r\n          <po-modal-footer>\r\n            <div class=\"po-modal-footer-basic\">\r\n              <po-button\r\n                *ngIf=\"secondaryAction\"\r\n                [p-danger]=\"getSecondaryActionButtonDanger()\"\r\n                [p-disabled]=\"secondaryAction.disabled\"\r\n                [p-label]=\"secondaryAction.label\"\r\n                [p-loading]=\"secondaryAction.loading\"\r\n                p-kind=\"secondary\"\r\n                (p-click)=\"secondaryAction.action()\"\r\n              >\r\n              </po-button>\r\n\r\n              <po-button\r\n                class=\"po-button-modal-first-action\"\r\n                [p-danger]=\"primaryAction.danger\"\r\n                [p-disabled]=\"primaryAction.disabled\"\r\n                [p-label]=\"primaryAction.label\"\r\n                [p-loading]=\"primaryAction.loading\"\r\n                p-kind=\"primary\"\r\n                (p-click)=\"primaryAction.action()\"\r\n              >\r\n              </po-button>\r\n            </div>\r\n          </po-modal-footer>\r\n        </ng-template>\r\n      </div>\r\n    </div>\r\n  </div>\r\n</div>\r\n" }]
    }], function () { return [{ type: i1.PoActiveOverlayService }, { type: i2.PoLanguageService }]; }, { modalContent: [{
            type: ViewChild,
            args: ['modalContent', { read: ElementRef }]
        }], modalContainer: [{
            type: ViewChild,
            args: ['modalContainer', { read: ElementRef }]
        }], modalFooter: [{
            type: ContentChild,
            args: [PoModalFooterComponent]
        }] }); })();
