// @po-ui/ng-components/esm2020/lib/components/po-modal/po-modal.module.mjs
import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { PoButtonModule } from './../po-button/po-button.module';
import { PoIconModule } from './../po-icon/po-icon.module';
import { PoModalComponent } from './po-modal.component';
import { PoModalFooterComponent } from './po-modal-footer/po-modal-footer.component';
import * as i0 from "@angular/core";
/**
 * @description
 * Módulo do componente po-modal
 */
export class PoModalModule {
}
PoModalModule.ɵfac = function PoModalModule_Factory(t) { return new (t || PoModalModule)(); };
PoModalModule.ɵmod = /*@__PURE__*/ i0.ɵɵdefineNgModule({ type: PoModalModule });
PoModalModule.ɵinj = /*@__PURE__*/ i0.ɵɵdefineInjector({ imports: [CommonModule, PoButtonModule, PoIconModule] });
(function () { (typeof ngDevMode === "undefined" || ngDevMode) && i0.ɵsetClassMetadata(PoModalModule, [{
        type: NgModule,
        args: [{
                imports: [CommonModule, PoButtonModule, PoIconModule],
                declarations: [PoModalComponent, PoModalFooterComponent],
                exports: [PoModalComponent, PoModalFooterComponent]
            }]
    }], null, null); })();
(function () { (typeof ngJitMode === "undefined" || ngJitMode) && i0.ɵɵsetNgModuleScope(PoModalModule, { declarations: [PoModalComponent, PoModalFooterComponent], imports: [CommonModule, PoButtonModule, PoIconModule], exports: [PoModalComponent, PoModalFooterComponent] }); })();
