// @po-ui/ng-components/esm2020/lib/components/po-disclaimer-group/po-disclaimer-group.component.mjs
import { Component, ChangeDetectionStrategy } from '@angular/core';
import { PoDisclaimerGroupBaseComponent } from './po-disclaimer-group-base.component';
import * as i0 from "@angular/core";
import * as i1 from "../../services/po-language/po-language.service";
import * as i2 from "@angular/common";
import * as i3 from "../po-disclaimer/po-disclaimer.component";
import * as i4 from "./po-disclaimer-remove/po-disclaimer-remove.component";
function PoDisclaimerGroupComponent_div_0_po_disclaimer_remove_3_Template(rf, ctx) { if (rf & 1) {
    const _r4 = i0.ɵɵgetCurrentView();
    i0.ɵɵelementStart(0, "po-disclaimer-remove", 5);
    i0.ɵɵlistener("keydown", function PoDisclaimerGroupComponent_div_0_po_disclaimer_remove_3_Template_po_disclaimer_remove_keydown_0_listener($event) { i0.ɵɵrestoreView(_r4); const ctx_r3 = i0.ɵɵnextContext(2); return i0.ɵɵresetView(ctx_r3.onKeyPress($event)); })("p-remove-all-action", function PoDisclaimerGroupComponent_div_0_po_disclaimer_remove_3_Template_po_disclaimer_remove_p_remove_all_action_0_listener() { i0.ɵɵrestoreView(_r4); const ctx_r5 = i0.ɵɵnextContext(2); return i0.ɵɵresetView(ctx_r5.removeAllItems()); });
    i0.ɵɵelementEnd();
} if (rf & 2) {
    const ctx_r1 = i0.ɵɵnextContext(2);
    i0.ɵɵproperty("p-label", ctx_r1.literals.removeAll);
} }
function PoDisclaimerGroupComponent_div_0_po_disclaimer_4_Template(rf, ctx) { if (rf & 1) {
    const _r8 = i0.ɵɵgetCurrentView();
    i0.ɵɵelementStart(0, "po-disclaimer", 6);
    i0.ɵɵlistener("p-close-action", function PoDisclaimerGroupComponent_div_0_po_disclaimer_4_Template_po_disclaimer_p_close_action_0_listener() { const restoredCtx = i0.ɵɵrestoreView(_r8); const disclaimer_r6 = restoredCtx.$implicit; const ctx_r7 = i0.ɵɵnextContext(2); return i0.ɵɵresetView(ctx_r7.onCloseAction(disclaimer_r6)); });
    i0.ɵɵelementEnd();
} if (rf & 2) {
    const disclaimer_r6 = ctx.$implicit;
    i0.ɵɵproperty("p-hide-close", disclaimer_r6.hideClose)("p-label", disclaimer_r6.label)("p-property", disclaimer_r6.property)("p-value", disclaimer_r6.value);
} }
function PoDisclaimerGroupComponent_div_0_Template(rf, ctx) { if (rf & 1) {
    i0.ɵɵelementStart(0, "div", 1)(1, "div", 2);
    i0.ɵɵtext(2);
    i0.ɵɵelementEnd();
    i0.ɵɵtemplate(3, PoDisclaimerGroupComponent_div_0_po_disclaimer_remove_3_Template, 1, 1, "po-disclaimer-remove", 3);
    i0.ɵɵtemplate(4, PoDisclaimerGroupComponent_div_0_po_disclaimer_4_Template, 1, 4, "po-disclaimer", 4);
    i0.ɵɵelementEnd();
} if (rf & 2) {
    const ctx_r0 = i0.ɵɵnextContext();
    i0.ɵɵadvance(2);
    i0.ɵɵtextInterpolate(ctx_r0.title);
    i0.ɵɵadvance(1);
    i0.ɵɵproperty("ngIf", ctx_r0.isRemoveAll());
    i0.ɵɵadvance(1);
    i0.ɵɵproperty("ngForOf", ctx_r0.disclaimers);
} }
/**
 * @docsExtends PoDisclaimerGroupBaseComponent
 *
 * @example
 *
 * <example name="po-disclaimer-group-basic" title="PO Disclaimer Group Basic">
 *   <file name="sample-po-disclaimer-group-basic/sample-po-disclaimer-group-basic.component.html"> </file>
 *   <file name="sample-po-disclaimer-group-basic/sample-po-disclaimer-group-basic.component.ts"> </file>
 * </example>
 *
 * <example name="po-disclaimer-group-labs" title="PO Disclaimer Group Labs">
 *   <file name="sample-po-disclaimer-group-labs/sample-po-disclaimer-group-labs.component.html"> </file>
 *   <file name="sample-po-disclaimer-group-labs/sample-po-disclaimer-group-labs.component.ts"> </file>
 * </example>
 *
 * <example name="po-disclaimer-group-sw-planets" title="PO Disclaimer Group - Star Wars Planets">
 *   <file name="sample-po-disclaimer-group-sw-planets/sample-po-disclaimer-group-sw-planets.component.html"> </file>
 *   <file name="sample-po-disclaimer-group-sw-planets/sample-po-disclaimer-group-sw-planets.component.ts"> </file>
 *   <file name="sample-po-disclaimer-group-sw-planets/sample-po-disclaimer-group-sw-planets.service.ts"> </file>
 * </example>
 *
 */
export class PoDisclaimerGroupComponent extends PoDisclaimerGroupBaseComponent {
    constructor(differs, languageService, changeDetector) {
        super(differs, languageService, changeDetector);
        this.changeDetector = changeDetector;
    }
}
PoDisclaimerGroupComponent.ɵfac = function PoDisclaimerGroupComponent_Factory(t) { return new (t || PoDisclaimerGroupComponent)(i0.ɵɵdirectiveInject(i0.IterableDiffers), i0.ɵɵdirectiveInject(i1.PoLanguageService), i0.ɵɵdirectiveInject(i0.ChangeDetectorRef)); };
PoDisclaimerGroupComponent.ɵcmp = /*@__PURE__*/ i0.ɵɵdefineComponent({ type: PoDisclaimerGroupComponent, selectors: [["po-disclaimer-group"]], features: [i0.ɵɵInheritDefinitionFeature], decls: 1, vars: 1, consts: [["class", "po-disclaimer-group", 4, "ngIf"], [1, "po-disclaimer-group"], [1, "po-disclaimer-group-title"], ["class", "po-disclaimer-group-disclaimer-align", 3, "p-label", "keydown", "p-remove-all-action", 4, "ngIf"], ["class", "po-disclaimer-group-disclaimer-align", 3, "p-hide-close", "p-label", "p-property", "p-value", "p-close-action", 4, "ngFor", "ngForOf"], [1, "po-disclaimer-group-disclaimer-align", 3, "p-label", "keydown", "p-remove-all-action"], [1, "po-disclaimer-group-disclaimer-align", 3, "p-hide-close", "p-label", "p-property", "p-value", "p-close-action"]], template: function PoDisclaimerGroupComponent_Template(rf, ctx) { if (rf & 1) {
        i0.ɵɵtemplate(0, PoDisclaimerGroupComponent_div_0_Template, 5, 3, "div", 0);
    } if (rf & 2) {
        i0.ɵɵproperty("ngIf", ctx.disclaimers.length > 0);
    } }, dependencies: [i2.NgForOf, i2.NgIf, i3.PoDisclaimerComponent, i4.PoDisclaimerRemoveComponent], encapsulation: 2, changeDetection: 0 });
(function () { (typeof ngDevMode === "undefined" || ngDevMode) && i0.ɵsetClassMetadata(PoDisclaimerGroupComponent, [{
        type: Component,
        args: [{ selector: 'po-disclaimer-group', changeDetection: ChangeDetectionStrategy.OnPush, template: "<div class=\"po-disclaimer-group\" *ngIf=\"disclaimers.length > 0\">\r\n  <div class=\"po-disclaimer-group-title\">{{ title }}</div>\r\n\r\n  <po-disclaimer-remove\r\n    *ngIf=\"isRemoveAll()\"\r\n    class=\"po-disclaimer-group-disclaimer-align\"\r\n    [p-label]=\"literals.removeAll\"\r\n    (keydown)=\"onKeyPress($event)\"\r\n    (p-remove-all-action)=\"removeAllItems()\"\r\n  >\r\n  </po-disclaimer-remove>\r\n\r\n  <po-disclaimer\r\n    *ngFor=\"let disclaimer of disclaimers\"\r\n    class=\"po-disclaimer-group-disclaimer-align\"\r\n    [p-hide-close]=\"disclaimer.hideClose\"\r\n    [p-label]=\"disclaimer.label\"\r\n    [p-property]=\"disclaimer.property\"\r\n    [p-value]=\"disclaimer.value\"\r\n    (p-close-action)=\"onCloseAction(disclaimer)\"\r\n  >\r\n  </po-disclaimer>\r\n</div>\r\n" }]
    }], function () { return [{ type: i0.IterableDiffers }, { type: i1.PoLanguageService }, { type: i0.ChangeDetectorRef }]; }, null); })();
