// @po-ui/ng-components/esm2020/lib/components/po-widget/po-widget.module.mjs
import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { PoContainerModule } from '../po-container/index';
import { PoWidgetComponent } from './po-widget.component';
import * as i0 from "@angular/core";
/**
 * @description
 *
 * Módulo do componente po-widget
 */
export class PoWidgetModule {
}
PoWidgetModule.ɵfac = function PoWidgetModule_Factory(t) { return new (t || PoWidgetModule)(); };
PoWidgetModule.ɵmod = /*@__PURE__*/ i0.ɵɵdefineNgModule({ type: PoWidgetModule });
PoWidgetModule.ɵinj = /*@__PURE__*/ i0.ɵɵdefineInjector({ imports: [CommonModule, PoContainerModule] });
(function () { (typeof ngDevMode === "undefined" || ngDevMode) && i0.ɵsetClassMetadata(PoWidgetModule, [{
        type: NgModule,
        args: [{
                imports: [CommonModule, PoContainerModule],
                exports: [PoWidgetComponent],
                declarations: [PoWidgetComponent]
            }]
    }], null, null); })();
(function () { (typeof ngJitMode === "undefined" || ngJitMode) && i0.ɵɵsetNgModuleScope(PoWidgetModule, { declarations: [PoWidgetComponent], imports: [CommonModule, PoContainerModule], exports: [PoWidgetComponent] }); })();
