// @po-ui/ng-components/esm2020/lib/components/po-search/po-search.component.mjs
import { PoFilterMode } from './po-search-filter-mode.enum';
import { Component, ElementRef, ViewChild } from '@angular/core';
import { PoSearchBaseComponent } from './po-search-base.component';
import * as i0 from "@angular/core";
import * as i1 from "../../services/po-language/po-language.service";
import * as i2 from "@angular/common";
import * as i3 from "../po-field/po-clean/po-clean.component";
import * as i4 from "../po-icon/po-icon.component";
import * as i5 from "../po-loading/po-loading-icon/po-loading-icon.component";
const _c0 = ["poSearchInput"];
function PoSearchComponent_po_icon_2_Template(rf, ctx) { if (rf & 1) {
    i0.ɵɵelement(0, "po-icon", 7);
} }
function PoSearchComponent_po_loading_icon_3_Template(rf, ctx) { if (rf & 1) {
    i0.ɵɵelement(0, "po-loading-icon");
} }
function PoSearchComponent_div_6_Template(rf, ctx) { if (rf & 1) {
    const _r5 = i0.ɵɵgetCurrentView();
    i0.ɵɵelementStart(0, "div", 8)(1, "po-clean", 9);
    i0.ɵɵlistener("p-change-event", function PoSearchComponent_div_6_Template_po_clean_p_change_event_1_listener() { i0.ɵɵrestoreView(_r5); const ctx_r4 = i0.ɵɵnextContext(); return i0.ɵɵresetView(ctx_r4.clearSearch()); });
    i0.ɵɵelementEnd()();
} if (rf & 2) {
    const ctx_r3 = i0.ɵɵnextContext();
    const _r2 = i0.ɵɵreference(5);
    i0.ɵɵproperty("tabindex", !ctx_r3.disabled ? 0 : -1);
    i0.ɵɵadvance(1);
    i0.ɵɵproperty("p-element-ref", _r2);
} }
export const poSearchLiteralsDefault = {
    en: { search: 'Search' },
    es: { search: 'Buscar' },
    pt: { search: 'Pesquisar' },
    ru: { search: 'Поиск' }
};
export class PoSearchComponent extends PoSearchBaseComponent {
    constructor(languageService, renderer) {
        super();
        this.languageService = languageService;
        this.renderer = renderer;
        this.filteredItems = [];
        this.literals = {
            ...poSearchLiteralsDefault[this.languageService.getLanguageDefault()],
            ...poSearchLiteralsDefault[this.languageService.getShortLanguage()]
        };
    }
    ngOnInit() {
        this.filteredItems = this.items;
    }
    clearSearch() {
        this.poSearchInput.nativeElement.value = '';
        this.onSearchChange('');
        this.filteredItemsChange.emit(this.items);
    }
    onBlur() {
        this.renderer.removeClass(this.poSearchInput.nativeElement.parentElement, 'po-search-focused');
    }
    onFocus() {
        this.renderer.addClass(this.poSearchInput.nativeElement.parentElement, 'po-search-focused');
    }
    onSearchChange(searchText) {
        searchText = searchText.toLowerCase();
        if (this.items && this.items.length > 0) {
            this.filteredItems = this.items.filter(item => this.filterKeys.some(key => {
                let value = item[key];
                if (typeof value !== 'string') {
                    value = value != null ? value.toString() : null;
                }
                value = value != null ? value.toLowerCase() : null;
                if (this.filterType === PoFilterMode.startsWith) {
                    return value != null && value.startsWith(searchText);
                }
                else if (this.filterType === PoFilterMode.contains) {
                    return value != null && value.includes(searchText);
                }
                else if (this.filterType === PoFilterMode.endsWith) {
                    return value != null && value.endsWith(searchText);
                }
                return false;
            }));
            this.filteredItemsChange.emit(this.filteredItems);
        }
        else {
            this.filteredItems = [];
            this.filteredItemsChange.emit(this.filteredItems);
        }
    }
}
PoSearchComponent.ɵfac = function PoSearchComponent_Factory(t) { return new (t || PoSearchComponent)(i0.ɵɵdirectiveInject(i1.PoLanguageService), i0.ɵɵdirectiveInject(i0.Renderer2)); };
PoSearchComponent.ɵcmp = /*@__PURE__*/ i0.ɵɵdefineComponent({ type: PoSearchComponent, selectors: [["po-search"]], viewQuery: function PoSearchComponent_Query(rf, ctx) { if (rf & 1) {
        i0.ɵɵviewQuery(_c0, 7, ElementRef);
    } if (rf & 2) {
        let _t;
        i0.ɵɵqueryRefresh(_t = i0.ɵɵloadQuery()) && (ctx.poSearchInput = _t.first);
    } }, features: [i0.ɵɵInheritDefinitionFeature], decls: 7, vars: 5, consts: [[1, "po-search"], [1, "po-search-icon"], ["p-icon", "po-icon-search", 4, "ngIf"], [4, "ngIf"], ["type", "text", 3, "disabled", "placeholder", "blur", "focus", "input"], ["poSearchInput", ""], ["class", "po-search-clean", 3, "tabindex", 4, "ngIf"], ["p-icon", "po-icon-search"], [1, "po-search-clean", 3, "tabindex"], [3, "p-element-ref", "p-change-event"]], template: function PoSearchComponent_Template(rf, ctx) { if (rf & 1) {
        i0.ɵɵelementStart(0, "div", 0)(1, "div", 1);
        i0.ɵɵtemplate(2, PoSearchComponent_po_icon_2_Template, 1, 0, "po-icon", 2);
        i0.ɵɵtemplate(3, PoSearchComponent_po_loading_icon_3_Template, 1, 0, "po-loading-icon", 3);
        i0.ɵɵelementEnd();
        i0.ɵɵelementStart(4, "input", 4, 5);
        i0.ɵɵlistener("blur", function PoSearchComponent_Template_input_blur_4_listener() { return ctx.onBlur(); })("focus", function PoSearchComponent_Template_input_focus_4_listener() { return ctx.onFocus(); })("input", function PoSearchComponent_Template_input_input_4_listener($event) { return ctx.onSearchChange($event.target.value); });
        i0.ɵɵelementEnd();
        i0.ɵɵtemplate(6, PoSearchComponent_div_6_Template, 2, 2, "div", 6);
        i0.ɵɵelementEnd();
    } if (rf & 2) {
        const _r2 = i0.ɵɵreference(5);
        i0.ɵɵadvance(2);
        i0.ɵɵproperty("ngIf", !ctx.loading);
        i0.ɵɵadvance(1);
        i0.ɵɵproperty("ngIf", ctx.loading);
        i0.ɵɵadvance(1);
        i0.ɵɵproperty("disabled", ctx.disabled)("placeholder", ctx.literals.search);
        i0.ɵɵadvance(2);
        i0.ɵɵproperty("ngIf", !!_r2.value && !ctx.disabled);
    } }, dependencies: [i2.NgIf, i3.PoCleanComponent, i4.PoIconComponent, i5.PoLoadingIconComponent], encapsulation: 2 });
(function () { (typeof ngDevMode === "undefined" || ngDevMode) && i0.ɵsetClassMetadata(PoSearchComponent, [{
        type: Component,
        args: [{ selector: 'po-search', template: "<div class=\"po-search\">\r\n  <div class=\"po-search-icon\">\r\n    <po-icon *ngIf=\"!loading\" p-icon=\"po-icon-search\"></po-icon>\r\n    <po-loading-icon *ngIf=\"loading\"></po-loading-icon>\r\n  </div>\r\n\r\n  <input\r\n    #poSearchInput\r\n    type=\"text\"\r\n    [disabled]=\"disabled\"\r\n    [placeholder]=\"literals.search\"\r\n    (blur)=\"onBlur()\"\r\n    (focus)=\"onFocus()\"\r\n    (input)=\"onSearchChange($event.target.value)\"\r\n  />\r\n\r\n  <div *ngIf=\"!!poSearchInput.value && !disabled\" class=\"po-search-clean\" [tabindex]=\"!disabled ? 0 : -1\">\r\n    <po-clean [p-element-ref]=\"poSearchInput\" (p-change-event)=\"clearSearch()\"></po-clean>\r\n  </div>\r\n</div>\r\n" }]
    }], function () { return [{ type: i1.PoLanguageService }, { type: i0.Renderer2 }]; }, { poSearchInput: [{
            type: ViewChild,
            args: ['poSearchInput', { read: ElementRef, static: true }]
        }] }); })();
