// @po-ui/ng-components/esm2020/lib/components/po-search/po-search.module.mjs
import { CommonModule } from '@angular/common';
import { NgModule } from '@angular/core';
import { PoCleanModule } from '../po-field';
import { PoIconModule } from '../po-icon';
import { PoLoadingModule } from '../po-loading';
import { PoSearchComponent } from './po-search.component';
import { FormsModule } from '@angular/forms';
import * as i0 from "@angular/core";
/**
 * @description
 *
 * Módulo do componente po-search.
 */
export class PoSearchModule {
}
PoSearchModule.ɵfac = function PoSearchModule_Factory(t) { return new (t || PoSearchModule)(); };
PoSearchModule.ɵmod = /*@__PURE__*/ i0.ɵɵdefineNgModule({ type: PoSearchModule });
PoSearchModule.ɵinj = /*@__PURE__*/ i0.ɵɵdefineInjector({ imports: [CommonModule, PoCleanModule, PoIconModule, PoLoadingModule, FormsModule] });
(function () { (typeof ngDevMode === "undefined" || ngDevMode) && i0.ɵsetClassMetadata(PoSearchModule, [{
        type: NgModule,
        args: [{
                imports: [CommonModule, PoCleanModule, PoIconModule, PoLoadingModule, FormsModule],
                declarations: [PoSearchComponent],
                exports: [PoSearchComponent]
            }]
    }], null, null); })();
(function () { (typeof ngJitMode === "undefined" || ngJitMode) && i0.ɵɵsetNgModuleScope(PoSearchModule, { declarations: [PoSearchComponent], imports: [CommonModule, PoCleanModule, PoIconModule, PoLoadingModule, FormsModule], exports: [PoSearchComponent] }); })();
