// @po-ui/ng-components/esm2020/lib/components/po-search/po-search-base.component.mjs
import { __decorate } from "tslib";
import { PoFilterMode } from './po-search-filter-mode.enum';
import { Directive, EventEmitter, Input, Output } from '@angular/core';
import { InputBoolean } from '../../decorators';
import * as i0 from "@angular/core";
export class PoSearchBaseComponent {
    constructor() {
        this.items = [];
        this.filterKeys = [];
        this.filterType = PoFilterMode.startsWith;
        this.filteredItemsChange = new EventEmitter();
        this.filter = new EventEmitter();
    }
}
PoSearchBaseComponent.ɵfac = function PoSearchBaseComponent_Factory(t) { return new (t || PoSearchBaseComponent)(); };
PoSearchBaseComponent.ɵdir = /*@__PURE__*/ i0.ɵɵdefineDirective({ type: PoSearchBaseComponent, inputs: { disabled: ["p-disabled", "disabled"], loading: ["p-loading", "loading"], items: ["p-items", "items"], filterKeys: ["p-filter-keys", "filterKeys"], filterType: ["p-filter-type", "filterType"] }, outputs: { filteredItemsChange: "p-filtered-items-change", filter: "p-filter" } });
__decorate([
    InputBoolean()
], PoSearchBaseComponent.prototype, "disabled", void 0);
(function () { (typeof ngDevMode === "undefined" || ngDevMode) && i0.ɵsetClassMetadata(PoSearchBaseComponent, [{
        type: Directive
    }], null, { disabled: [{
            type: Input,
            args: ['p-disabled']
        }], loading: [{
            type: Input,
            args: ['p-loading']
        }], items: [{
            type: Input,
            args: ['p-items']
        }], filterKeys: [{
            type: Input,
            args: ['p-filter-keys']
        }], filterType: [{
            type: Input,
            args: ['p-filter-type']
        }], filteredItemsChange: [{
            type: Output,
            args: ['p-filtered-items-change']
        }], filter: [{
            type: Output,
            args: ['p-filter']
        }] }); })();
