// @totvs/protheus-lib-core/esm2020/services/pro-app-config/services/pro-app-config.service.mjs
import { EventEmitter, Injectable } from '@angular/core';
import { valueIsNull } from '../../../util';
import * as i0 from "@angular/core";
import * as i1 from "@angular/common/http";
import * as i2 from "./../../pro-js-to-advpl/services/pro-js-to-advpl.service";
const CACHE_KEY = 'ERPAPPCONFIG';
export class ProAppConfigService {
    constructor(http, advplService) {
        this.http = http;
        this.advplService = advplService;
        this.readyEmitter = new EventEmitter();
    }
    loadAppConfig() {
        return this.http
            .get('assets/data/appConfig.json')
            .toPromise()
            .then(data => {
            const proAppConfig = data;
            // A função FwCallApp do Protheus pode ter subido a AppConfig com valores manipulados e não podemos sobreescrever
            if (sessionStorage[CACHE_KEY] === undefined) {
                sessionStorage[CACHE_KEY] = JSON.stringify(proAppConfig);
            }
            this.readyEmitter.emit();
            return this.proAppConfig;
        });
    }
    /**
     * @description Método para encerrar o aplicativo
     * @param ask Boolean que indica se o sistema deve perguntar antes de fechar o app
     */
    callAppClose(ask = true) {
        if (ask) {
            this.advplService.AdvplCloseApp('force');
        }
        else {
            this.advplService.AdvplCloseApp();
        }
    }
    insideProtheus() {
        return this.advplService.protheusConnected();
    }
    get proAppConfig() {
        if (valueIsNull(this._ProAppConfig)) {
            this._ProAppConfig = JSON.parse(sessionStorage[CACHE_KEY]);
        }
        return this._ProAppConfig;
    }
    get nameApp() {
        return this.proAppConfig.name;
    }
    get serverWithApiUrl() {
        return valueIsNull(this.proAppConfig.api_baseUrl)
            ? this.proAppConfig.serverBackend + this.proAppConfig.restEntryPoint
            : this.proAppConfig.api_baseUrl;
    }
    get productLine() {
        return this.proAppConfig.productLine;
    }
    get isProtheusRender() {
        return this.productLine.toLowerCase() === 'protheus';
    }
    freeAppConfig() {
        this._ProAppConfig = null;
    }
}
ProAppConfigService.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "15.2.9", ngImport: i0, type: ProAppConfigService, deps: [{ token: i1.HttpClient }, { token: i2.ProJsToAdvplService }], target: i0.ɵɵFactoryTarget.Injectable });
ProAppConfigService.ɵprov = i0.ɵɵngDeclareInjectable({ minVersion: "12.0.0", version: "15.2.9", ngImport: i0, type: ProAppConfigService, providedIn: 'root' });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "15.2.9", ngImport: i0, type: ProAppConfigService, decorators: [{
            type: Injectable,
            args: [{
                    providedIn: 'root'
                }]
        }], ctorParameters: function () { return [{ type: i1.HttpClient }, { type: i2.ProJsToAdvplService }]; } });
