// @po-ui/ng-components/esm2020/lib/components/po-table/po-table-column-link/po-table-column-link.component.mjs
import { ChangeDetectionStrategy, Component, Input } from '@angular/core';
import { isExternalLink } from '../../../utils/util';
import * as i0 from "@angular/core";
import * as i1 from "@angular/common";
import * as i2 from "../../po-link/po-link.component";
function PoTableColumnLinkComponent_po_link_2_Template(rf, ctx) { if (rf & 1) {
    const _r4 = i0.ɵɵgetCurrentView();
    i0.ɵɵelementStart(0, "po-link", 4);
    i0.ɵɵlistener("p-action", function PoTableColumnLinkComponent_po_link_2_Template_po_link_p_action_0_listener() { i0.ɵɵrestoreView(_r4); const ctx_r3 = i0.ɵɵnextContext(); return i0.ɵɵresetView(ctx_r3.action == null ? null : ctx_r3.action(ctx_r3.value, ctx_r3.row)); });
    i0.ɵɵelementEnd();
} if (rf & 2) {
    const ctx_r0 = i0.ɵɵnextContext();
    i0.ɵɵproperty("p-label", ctx_r0.value)("p-open-new-tab", ctx_r0.openNewTab);
} }
function PoTableColumnLinkComponent_po_link_3_Template(rf, ctx) { if (rf & 1) {
    i0.ɵɵelement(0, "po-link", 5);
} if (rf & 2) {
    const ctx_r1 = i0.ɵɵnextContext();
    i0.ɵɵproperty("p-label", ctx_r1.value)("p-open-new-tab", ctx_r1.openNewTab)("p-url", ctx_r1.link);
} }
function PoTableColumnLinkComponent_po_link_4_Template(rf, ctx) { if (rf & 1) {
    i0.ɵɵelement(0, "po-link", 5);
} if (rf & 2) {
    const ctx_r2 = i0.ɵɵnextContext();
    i0.ɵɵproperty("p-label", ctx_r2.value)("p-open-new-tab", ctx_r2.openNewTab)("p-url", ctx_r2.link);
} }
const _c0 = function (a0) { return { "po-table-link-disabled": a0 }; };
/**
 * @docsPrivate
 *
 * @description
 *
 * Componente responsável por exibir link nas colunas.
 */
export class PoTableColumnLinkComponent {
    constructor() {
        this.openNewTab = false;
    }
    get type() {
        if (this.action) {
            return 'action';
        }
        return isExternalLink(this.link) ? 'externalLink' : 'internalLink';
    }
}
PoTableColumnLinkComponent.ɵfac = function PoTableColumnLinkComponent_Factory(t) { return new (t || PoTableColumnLinkComponent)(); };
PoTableColumnLinkComponent.ɵcmp = /*@__PURE__*/ i0.ɵɵdefineComponent({ type: PoTableColumnLinkComponent, selectors: [["po-table-column-link"]], inputs: { action: ["p-action", "action"], disabled: ["p-disabled", "disabled"], link: ["p-link", "link"], openNewTab: ["p-open-new-tab", "openNewTab"], row: ["p-row", "row"], value: ["p-value", "value"] }, decls: 5, vars: 7, consts: [[1, "po-table-link", 3, "ngClass"], [3, "ngSwitch"], [3, "p-label", "p-open-new-tab", "p-action", 4, "ngSwitchCase"], [3, "p-label", "p-open-new-tab", "p-url", 4, "ngSwitchCase"], [3, "p-label", "p-open-new-tab", "p-action"], [3, "p-label", "p-open-new-tab", "p-url"]], template: function PoTableColumnLinkComponent_Template(rf, ctx) { if (rf & 1) {
        i0.ɵɵelementStart(0, "div", 0);
        i0.ɵɵelementContainerStart(1, 1);
        i0.ɵɵtemplate(2, PoTableColumnLinkComponent_po_link_2_Template, 1, 2, "po-link", 2);
        i0.ɵɵtemplate(3, PoTableColumnLinkComponent_po_link_3_Template, 1, 3, "po-link", 3);
        i0.ɵɵtemplate(4, PoTableColumnLinkComponent_po_link_4_Template, 1, 3, "po-link", 3);
        i0.ɵɵelementContainerEnd();
        i0.ɵɵelementEnd();
    } if (rf & 2) {
        i0.ɵɵproperty("ngClass", i0.ɵɵpureFunction1(5, _c0, ctx.disabled));
        i0.ɵɵadvance(1);
        i0.ɵɵproperty("ngSwitch", ctx.type);
        i0.ɵɵadvance(1);
        i0.ɵɵproperty("ngSwitchCase", "action");
        i0.ɵɵadvance(1);
        i0.ɵɵproperty("ngSwitchCase", "externalLink");
        i0.ɵɵadvance(1);
        i0.ɵɵproperty("ngSwitchCase", "internalLink");
    } }, dependencies: [i1.NgClass, i1.NgSwitch, i1.NgSwitchCase, i2.PoLinkComponent], encapsulation: 2, changeDetection: 0 });
(function () { (typeof ngDevMode === "undefined" || ngDevMode) && i0.ɵsetClassMetadata(PoTableColumnLinkComponent, [{
        type: Component,
        args: [{ selector: 'po-table-column-link', changeDetection: ChangeDetectionStrategy.OnPush, template: "<div class=\"po-table-link\" [ngClass]=\"{ 'po-table-link-disabled': disabled }\">\r\n  <ng-container [ngSwitch]=\"type\">\r\n    <po-link\r\n      *ngSwitchCase=\"'action'\"\r\n      [p-label]=\"value\"\r\n      [p-open-new-tab]=\"openNewTab\"\r\n      (p-action)=\"action?.(value, row)\"\r\n    ></po-link>\r\n    <po-link *ngSwitchCase=\"'externalLink'\" [p-label]=\"value\" [p-open-new-tab]=\"openNewTab\" [p-url]=\"link\"></po-link>\r\n    <po-link *ngSwitchCase=\"'internalLink'\" [p-label]=\"value\" [p-open-new-tab]=\"openNewTab\" [p-url]=\"link\"></po-link>\r\n  </ng-container>\r\n</div>\r\n" }]
    }], null, { action: [{
            type: Input,
            args: ['p-action']
        }], disabled: [{
            type: Input,
            args: ['p-disabled']
        }], link: [{
            type: Input,
            args: ['p-link']
        }], openNewTab: [{
            type: Input,
            args: ['p-open-new-tab']
        }], row: [{
            type: Input,
            args: ['p-row']
        }], value: [{
            type: Input,
            args: ['p-value']
        }] }); })();
