// @po-ui/ng-components/esm2020/lib/components/po-grid/po-grid-head/po-grid-head.component.mjs
import { Component, Input } from '@angular/core';
import * as i0 from "@angular/core";
export class PoGridHeadComponent {
    constructor() {
        this.align = 'left';
        this.boolean = false;
        this.position = '';
        this.tabIndex = -1;
    }
    set width(value) {
        this._width = value;
    }
    get width() {
        return this._width ? `${this._width}px` : '100%';
    }
}
PoGridHeadComponent.ɵfac = function PoGridHeadComponent_Factory(t) { return new (t || PoGridHeadComponent)(); };
PoGridHeadComponent.ɵcmp = /*@__PURE__*/ i0.ɵɵdefineComponent({ type: PoGridHeadComponent, selectors: [["po-grid-head"]], inputs: { align: ["p-align", "align"], boolean: ["p-freeze", "boolean"], position: ["p-position", "position"], tabIndex: ["p-tab-index", "tabIndex"], title: ["p-title", "title"], width: ["p-width", "width"] }, decls: 4, vars: 6, consts: [[3, "id", "tabindex"]], template: function PoGridHeadComponent_Template(rf, ctx) { if (rf & 1) {
        i0.ɵɵelementStart(0, "div")(1, "div", 0)(2, "span");
        i0.ɵɵtext(3);
        i0.ɵɵelementEnd()()();
    } if (rf & 2) {
        i0.ɵɵadvance(1);
        i0.ɵɵclassMapInterpolate1("po-grid-head-content po-grid-head-", ctx.align, "");
        i0.ɵɵpropertyInterpolate("tabindex", ctx.title ? 0 : -1);
        i0.ɵɵproperty("id", ctx.position);
        i0.ɵɵadvance(2);
        i0.ɵɵtextInterpolate(ctx.title || "\u00A0");
    } }, encapsulation: 2 });
(function () { (typeof ngDevMode === "undefined" || ngDevMode) && i0.ɵsetClassMetadata(PoGridHeadComponent, [{
        type: Component,
        args: [{ selector: 'po-grid-head', template: "<div>\r\n  <div [id]=\"position\" class=\"po-grid-head-content po-grid-head-{{ align }}\" tabindex=\"{{ title ? 0 : -1 }}\">\r\n    <span>{{ title || '&nbsp;' }}</span>\r\n  </div>\r\n</div>\r\n" }]
    }], null, { align: [{
            type: Input,
            args: ['p-align']
        }], boolean: [{
            type: Input,
            args: ['p-freeze']
        }], position: [{
            type: Input,
            args: ['p-position']
        }], tabIndex: [{
            type: Input,
            args: ['p-tab-index']
        }], title: [{
            type: Input,
            args: ['p-title']
        }], width: [{
            type: Input,
            args: ['p-width']
        }] }); })();
