// @po-ui/ng-components/esm2020/lib/interceptors/po-http-request/po-http-request.module.mjs
import { HTTP_INTERCEPTORS } from '@angular/common/http';
import { NgModule } from '@angular/core';
import { PoLoadingModule } from '../../components/po-loading/po-loading.module';
import { PoComponentInjectorService } from '../../services/po-component-injector/po-component-injector.service';
import { PoHttpRequesControltService } from './po-http-request-control-service';
import { PoHttpRequestInterceptorService } from './po-http-request-interceptor.service';
import * as i0 from "@angular/core";
export class PoHttpRequestModule {
}
PoHttpRequestModule.ɵfac = function PoHttpRequestModule_Factory(t) { return new (t || PoHttpRequestModule)(); };
PoHttpRequestModule.ɵmod = /*@__PURE__*/ i0.ɵɵdefineNgModule({ type: PoHttpRequestModule });
PoHttpRequestModule.ɵinj = /*@__PURE__*/ i0.ɵɵdefineInjector({ providers: [
        PoHttpRequesControltService,
        PoHttpRequestInterceptorService,
        PoComponentInjectorService,
        {
            provide: HTTP_INTERCEPTORS,
            useClass: PoHttpRequestInterceptorService,
            multi: true
        }
    ], imports: [PoLoadingModule] });
(function () { (typeof ngDevMode === "undefined" || ngDevMode) && i0.ɵsetClassMetadata(PoHttpRequestModule, [{
        type: NgModule,
        args: [{
                imports: [PoLoadingModule],
                providers: [
                    PoHttpRequesControltService,
                    PoHttpRequestInterceptorService,
                    PoComponentInjectorService,
                    {
                        provide: HTTP_INTERCEPTORS,
                        useClass: PoHttpRequestInterceptorService,
                        multi: true
                    }
                ]
            }]
    }], null, null); })();
(function () { (typeof ngJitMode === "undefined" || ngJitMode) && i0.ɵɵsetNgModuleScope(PoHttpRequestModule, { imports: [PoLoadingModule] }); })();
