// @po-ui/ng-components/esm2020/lib/components/po-label/po-label.module.mjs
import { CommonModule } from '@angular/common';
import { NgModule } from '@angular/core';
import { PoLabelComponent } from './po-label.component';
import * as i0 from "@angular/core";
export class PoLabelModule {
}
PoLabelModule.ɵfac = function PoLabelModule_Factory(t) { return new (t || PoLabelModule)(); };
PoLabelModule.ɵmod = /*@__PURE__*/ i0.ɵɵdefineNgModule({ type: PoLabelModule });
PoLabelModule.ɵinj = /*@__PURE__*/ i0.ɵɵdefineInjector({ imports: [CommonModule] });
(function () { (typeof ngDevMode === "undefined" || ngDevMode) && i0.ɵsetClassMetadata(PoLabelModule, [{
        type: NgModule,
        args: [{
                declarations: [PoLabelComponent],
                exports: [PoLabelComponent],
                imports: [CommonModule]
            }]
    }], null, null); })();
(function () { (typeof ngJitMode === "undefined" || ngJitMode) && i0.ɵɵsetNgModuleScope(PoLabelModule, { declarations: [PoLabelComponent], imports: [CommonModule], exports: [PoLabelComponent] }); })();
