// @po-ui/ng-components/esm2020/lib/components/po-tabs/po-tab/po-tab.component.mjs
import { Component } from '@angular/core';
import { PoTabBaseComponent } from './po-tab-base.component';
import * as i0 from "@angular/core";
const _c0 = ["*"];
/**
 * @docsExtends PoTabBaseComponent
 */
export class PoTabComponent extends PoTabBaseComponent {
    constructor(elementRef) {
        super();
        this.elementRef = elementRef;
    }
    ngAfterContentInit() {
        this.setDisplayOnActive();
    }
    setDisplayOnActive() {
        this.elementRef.nativeElement.style.display = this.active ? '' : 'none';
    }
}
PoTabComponent.ɵfac = function PoTabComponent_Factory(t) { return new (t || PoTabComponent)(i0.ɵɵdirectiveInject(i0.ElementRef)); };
PoTabComponent.ɵcmp = /*@__PURE__*/ i0.ɵɵdefineComponent({ type: PoTabComponent, selectors: [["po-tab"]], features: [i0.ɵɵInheritDefinitionFeature], ngContentSelectors: _c0, decls: 1, vars: 0, template: function PoTabComponent_Template(rf, ctx) { if (rf & 1) {
        i0.ɵɵprojectionDef();
        i0.ɵɵprojection(0);
    } }, encapsulation: 2 });
(function () { (typeof ngDevMode === "undefined" || ngDevMode) && i0.ɵsetClassMetadata(PoTabComponent, [{
        type: Component,
        args: [{ selector: 'po-tab', template: "<ng-content></ng-content>\r\n" }]
    }], function () { return [{ type: i0.ElementRef }]; }, null); })();
