// @po-ui/ng-components/esm2020/lib/components/po-listbox/po-listbox.component.mjs
import { Component, ViewChild } from '@angular/core';
import { PoListBoxBaseComponent } from './po-listbox-base.component';
import { isExternalLink, isTypeof, openExternalLink } from '../../utils/util';
import * as i0 from "@angular/core";
import * as i1 from "../../services/po-language/po-language.service";
import * as i2 from "@angular/router";
import * as i3 from "@angular/common";
import * as i4 from "../po-loading/po-loading.component";
import * as i5 from "@angular/cdk/listbox";
import * as i6 from "./po-item-list/po-item-list.component";
import * as i7 from "./po-search-list/po-search-list.component";
const _c0 = ["listbox"];
const _c1 = ["listboxItemList"];
const _c2 = ["searchElement"];
function PoListBoxComponent_po_search_list_3_Template(rf, ctx) { if (rf & 1) {
    const _r7 = i0.ɵɵgetCurrentView();
    i0.ɵɵelementStart(0, "po-search-list", 6, 7);
    i0.ɵɵlistener("p-change", function PoListBoxComponent_po_search_list_3_Template_po_search_list_p_change_0_listener($event) { i0.ɵɵrestoreView(_r7); const ctx_r6 = i0.ɵɵnextContext(); return i0.ɵɵresetView(ctx_r6.callChangeSearch($event)); });
    i0.ɵɵelementEnd();
} if (rf & 2) {
    const ctx_r1 = i0.ɵɵnextContext();
    i0.ɵɵproperty("p-literals", ctx_r1.literalSearch)("p-field-value", ctx_r1.fieldValueSearch)("p-placeholder", ctx_r1.placeholderSearch);
} }
function PoListBoxComponent_ul_4_li_2_Template(rf, ctx) { if (rf & 1) {
    const _r12 = i0.ɵɵgetCurrentView();
    i0.ɵɵelementStart(0, "li", 12);
    i0.ɵɵlistener("click", function PoListBoxComponent_ul_4_li_2_Template_li_click_0_listener() { i0.ɵɵrestoreView(_r12); const ctx_r11 = i0.ɵɵnextContext(2); return i0.ɵɵresetView(ctx_r11.changeAll.emit()); })("keydown", function PoListBoxComponent_ul_4_li_2_Template_li_keydown_0_listener($event) { i0.ɵɵrestoreView(_r12); const ctx_r13 = i0.ɵɵnextContext(2); return i0.ɵɵresetView(ctx_r13.changeAllEmit($event)); });
    i0.ɵɵelement(1, "po-item-list", 13);
    i0.ɵɵelementEnd();
} if (rf & 2) {
    const ctx_r9 = i0.ɵɵnextContext(2);
    i0.ɵɵattribute("aria-checked", ctx_r9.checkboxAllValue === null ? "mixed" : ctx_r9.checkboxAllValue);
    i0.ɵɵadvance(1);
    i0.ɵɵproperty("p-selected", ctx_r9.checkboxAllValue)("p-label", ctx_r9.literals.selectAll)("p-checkbox-value", ctx_r9.checkboxAllValue);
} }
function PoListBoxComponent_ul_4_li_3_po_item_list_1_Template(rf, ctx) { if (rf & 1) {
    const _r17 = i0.ɵɵgetCurrentView();
    i0.ɵɵelementStart(0, "po-item-list", 16);
    i0.ɵɵlistener("p-selectcheckbox-item", function PoListBoxComponent_ul_4_li_3_po_item_list_1_Template_po_item_list_p_selectcheckbox_item_0_listener($event) { i0.ɵɵrestoreView(_r17); const ctx_r16 = i0.ɵɵnextContext(3); return i0.ɵɵresetView(ctx_r16.checkboxClicked($event)); });
    i0.ɵɵpipe(1, "json");
    i0.ɵɵelementEnd();
} if (rf & 2) {
    const item_r14 = i0.ɵɵnextContext().$implicit;
    const ctx_r15 = i0.ɵɵnextContext(2);
    i0.ɵɵproperty("p-disabled", ctx_r15.returnBooleanValue(item_r14, "disabled"))("p-visible", ctx_r15.returnBooleanValue(item_r14, "visible"))("p-checkbox-value", ctx_r15.isSelectedItem(item_r14))("p-label", item_r14[ctx_r15.fieldLabel])("p-value", item_r14[ctx_r15.fieldValue])("p-selected", ctx_r15.isSelectedItem(item_r14) || item_r14.selected)("p-separator", item_r14.separator)("p-danger", item_r14.danger || item_r14.type === "danger")("p-icon", item_r14.icon)("p-type", ctx_r15.type)("p-item", item_r14)("p-field-value", ctx_r15.fieldValue)("p-field-label", ctx_r15.fieldLabel)("p-multiselect-template", ctx_r15.multiselectTemplate)("p-template-context", item_r14);
    i0.ɵɵattribute("data-item-list", i0.ɵɵpipeBind1(1, 16, item_r14));
} }
function PoListBoxComponent_ul_4_li_3_Template(rf, ctx) { if (rf & 1) {
    const _r20 = i0.ɵɵgetCurrentView();
    i0.ɵɵelementStart(0, "li", 14);
    i0.ɵɵlistener("click", function PoListBoxComponent_ul_4_li_3_Template_li_click_0_listener() { const restoredCtx = i0.ɵɵrestoreView(_r20); const item_r14 = restoredCtx.$implicit; const ctx_r19 = i0.ɵɵnextContext(2); return i0.ɵɵresetView(ctx_r19.onSelectItem(item_r14)); })("keydown", function PoListBoxComponent_ul_4_li_3_Template_li_keydown_0_listener($event) { const restoredCtx = i0.ɵɵrestoreView(_r20); const item_r14 = restoredCtx.$implicit; const ctx_r21 = i0.ɵɵnextContext(2); return i0.ɵɵresetView(ctx_r21.onKeyDown(item_r14, $event)); });
    i0.ɵɵtemplate(1, PoListBoxComponent_ul_4_li_3_po_item_list_1_Template, 2, 18, "po-item-list", 15);
    i0.ɵɵelementEnd();
} if (rf & 2) {
    const item_r14 = ctx.$implicit;
    const ctx_r10 = i0.ɵɵnextContext(2);
    i0.ɵɵproperty("cdkOption", item_r14[ctx_r10.fieldValue])("cdkOptionDisabled", ctx_r10.returnBooleanValue(item_r14, "disabled") || ctx_r10.returnBooleanValue(item_r14, "visible") === false);
    i0.ɵɵattribute("aria-selected", ctx_r10.isSelectedItem(item_r14) || item_r14.selected);
    i0.ɵɵadvance(1);
    i0.ɵɵproperty("ngIf", ctx_r10.returnBooleanValue(item_r14, "visible") !== false);
} }
function PoListBoxComponent_ul_4_Template(rf, ctx) { if (rf & 1) {
    i0.ɵɵelementStart(0, "ul", 8, 9);
    i0.ɵɵtemplate(2, PoListBoxComponent_ul_4_li_2_Template, 2, 4, "li", 10);
    i0.ɵɵtemplate(3, PoListBoxComponent_ul_4_li_3_Template, 2, 4, "li", 11);
    i0.ɵɵelementEnd();
} if (rf & 2) {
    const ctx_r2 = i0.ɵɵnextContext();
    i0.ɵɵproperty("cdkListboxMultiple", ctx_r2.type === "check");
    i0.ɵɵadvance(2);
    i0.ɵɵproperty("ngIf", ctx_r2.type === "check" && ctx_r2.items.length && !(ctx_r2.searchElement == null ? null : ctx_r2.searchElement.inputValue) && !ctx_r2.hideSelectAll);
    i0.ɵɵadvance(1);
    i0.ɵɵproperty("ngForOf", ctx_r2.items);
} }
function PoListBoxComponent_div_5_Template(rf, ctx) { if (rf & 1) {
    i0.ɵɵelementStart(0, "div", 17)(1, "span");
    i0.ɵɵtext(2);
    i0.ɵɵelementEnd()();
} if (rf & 2) {
    const ctx_r3 = i0.ɵɵnextContext();
    i0.ɵɵadvance(2);
    i0.ɵɵtextInterpolate1(" ", ctx_r3.literals.noItems, "");
} }
function PoListBoxComponent_div_6_Template(rf, ctx) { if (rf & 1) {
    i0.ɵɵelementStart(0, "div", 18);
    i0.ɵɵelement(1, "po-loading");
    i0.ɵɵelementEnd();
} }
const _c3 = [[["", "p-popup-header-template", ""]]];
const _c4 = ["[p-popup-header-template]"];
export class PoListBoxComponent extends PoListBoxBaseComponent {
    constructor(renderer, languageService, router) {
        super(languageService);
        this.renderer = renderer;
        this.router = router;
    }
    ngAfterViewInit() {
        this.setListBoxMaxHeight();
        this.listboxItemList?.nativeElement.focus();
    }
    ngOnChanges(changes) {
        if (changes?.items) {
            this.setListBoxMaxHeight();
        }
    }
    onSelectItem(itemListAction) {
        const isDisabled = itemListAction.hasOwnProperty('disabled') &&
            itemListAction.disabled !== null &&
            itemListAction.disabled !== undefined
            ? this.returnBooleanValue(itemListAction, 'disabled')
            : false;
        const isVisible = itemListAction.hasOwnProperty('visible') &&
            itemListAction.visible !== null &&
            itemListAction.visible !== undefined
            ? this.returnBooleanValue(itemListAction, 'visible')
            : true;
        if (itemListAction && itemListAction.action && !isDisabled && isVisible) {
            itemListAction.action(this.param || itemListAction);
        }
        if (itemListAction && itemListAction.url && !isDisabled && isVisible) {
            return this.openUrl(itemListAction.url);
        }
    }
    onKeyDown(itemListAction, event) {
        event.preventDefault();
        if ((event && event.code === 'Enter') || event.code === 'Space') {
            if (this.type !== 'check') {
                this.onSelectItem(itemListAction);
            }
            else {
                this.onSelectCheckBoxItem(itemListAction);
            }
        }
        if (event && event.code === 'Escape') {
            this.closeEvent.emit();
        }
    }
    checkboxClicked({ option, selected }) {
        if (this.type === 'check') {
            this.change.emit({ selected, option });
        }
    }
    onSelectCheckBoxItem(option) {
        const selected = !this.isSelectedItem(option);
        this.checkboxClicked({ option, selected });
    }
    isSelectedItem(option) {
        return this.selectedOptions.some(selectedItem => selectedItem[this.fieldValue] === option[this.fieldValue]);
    }
    changeAllEmit(event) {
        if ((event && event.code === 'Enter') || event.code === 'Space') {
            this.changeAll.emit();
        }
    }
    callChangeSearch(event) {
        this.changeSearch.emit(event);
    }
    returnBooleanValue(itemListAction, property) {
        return isTypeof(itemListAction[property], 'function')
            ? itemListAction[property](this.param || itemListAction)
            : itemListAction[property];
    }
    setListBoxMaxHeight() {
        const itemsLength = this.items.length;
        if (itemsLength > 6) {
            if (this.type === 'check' && !this.hideSearch) {
                this.renderer.setStyle(this.listbox.nativeElement, 'maxHeight', `${44 * 6 - 44 / 3 + 60}px`);
            }
            else {
                this.renderer.setStyle(this.listbox.nativeElement, 'maxHeight', `${44 * 6 - 44 / 3}px`);
            }
        }
    }
    openUrl(url) {
        if (isExternalLink(url)) {
            return openExternalLink(url);
        }
        if (url) {
            return this.router.navigate([url]);
        }
    }
}
PoListBoxComponent.ɵfac = function PoListBoxComponent_Factory(t) { return new (t || PoListBoxComponent)(i0.ɵɵdirectiveInject(i0.Renderer2), i0.ɵɵdirectiveInject(i1.PoLanguageService), i0.ɵɵdirectiveInject(i2.Router)); };
PoListBoxComponent.ɵcmp = /*@__PURE__*/ i0.ɵɵdefineComponent({ type: PoListBoxComponent, selectors: [["po-listbox"]], viewQuery: function PoListBoxComponent_Query(rf, ctx) { if (rf & 1) {
        i0.ɵɵviewQuery(_c0, 7);
        i0.ɵɵviewQuery(_c1, 5);
        i0.ɵɵviewQuery(_c2, 5);
    } if (rf & 2) {
        let _t;
        i0.ɵɵqueryRefresh(_t = i0.ɵɵloadQuery()) && (ctx.listbox = _t.first);
        i0.ɵɵqueryRefresh(_t = i0.ɵɵloadQuery()) && (ctx.listboxItemList = _t.first);
        i0.ɵɵqueryRefresh(_t = i0.ɵɵloadQuery()) && (ctx.searchElement = _t.first);
    } }, features: [i0.ɵɵInheritDefinitionFeature, i0.ɵɵNgOnChangesFeature], ngContentSelectors: _c4, decls: 7, vars: 8, consts: [[1, "po-listbox", 3, "hidden"], ["listbox", ""], [3, "p-literals", "p-field-value", "p-placeholder", "p-change", 4, "ngIf"], ["cdkListbox", "", "class", "po-listbox-list", 3, "cdkListboxMultiple", 4, "ngIf"], ["class", "po-multiselect-container-no-data po-text-center", 4, "ngIf"], ["class", "po-multiselect-container-loading po-text-center", 4, "ngIf"], [3, "p-literals", "p-field-value", "p-placeholder", "p-change"], ["searchElement", ""], ["cdkListbox", "", 1, "po-listbox-list", 3, "cdkListboxMultiple"], ["listboxItemList", ""], ["cdkOption", "selectAll", 3, "click", "keydown", 4, "ngIf"], ["class", "po-listbox-item", 3, "cdkOption", "cdkOptionDisabled", "click", "keydown", 4, "ngFor", "ngForOf"], ["cdkOption", "selectAll", 3, "click", "keydown"], ["p-type", "check", "p-value", "selectAll", 3, "p-selected", "p-label", "p-checkbox-value"], [1, "po-listbox-item", 3, "cdkOption", "cdkOptionDisabled", "click", "keydown"], [3, "p-disabled", "p-visible", "p-checkbox-value", "p-label", "p-value", "p-selected", "p-separator", "p-danger", "p-icon", "p-type", "p-item", "p-field-value", "p-field-label", "p-multiselect-template", "p-template-context", "p-selectcheckbox-item", 4, "ngIf"], [3, "p-disabled", "p-visible", "p-checkbox-value", "p-label", "p-value", "p-selected", "p-separator", "p-danger", "p-icon", "p-type", "p-item", "p-field-value", "p-field-label", "p-multiselect-template", "p-template-context", "p-selectcheckbox-item"], [1, "po-multiselect-container-no-data", "po-text-center"], [1, "po-multiselect-container-loading", "po-text-center"]], template: function PoListBoxComponent_Template(rf, ctx) { if (rf & 1) {
        i0.ɵɵprojectionDef(_c3);
        i0.ɵɵelementStart(0, "div", 0, 1);
        i0.ɵɵprojection(2);
        i0.ɵɵtemplate(3, PoListBoxComponent_po_search_list_3_Template, 2, 3, "po-search-list", 2);
        i0.ɵɵtemplate(4, PoListBoxComponent_ul_4_Template, 4, 3, "ul", 3);
        i0.ɵɵtemplate(5, PoListBoxComponent_div_5_Template, 3, 1, "div", 4);
        i0.ɵɵtemplate(6, PoListBoxComponent_div_6_Template, 2, 0, "div", 5);
        i0.ɵɵelementEnd();
    } if (rf & 2) {
        i0.ɵɵclassProp("po-listbox-check", ctx.type === "check");
        i0.ɵɵproperty("hidden", ctx.visible);
        i0.ɵɵattribute("data-type", ctx.type);
        i0.ɵɵadvance(3);
        i0.ɵɵproperty("ngIf", !ctx.hideSearch && ctx.type === "check");
        i0.ɵɵadvance(1);
        i0.ɵɵproperty("ngIf", !ctx.isServerSearching);
        i0.ɵɵadvance(1);
        i0.ɵɵproperty("ngIf", !ctx.items.length && !ctx.isServerSearching && ctx.type === "check");
        i0.ɵɵadvance(1);
        i0.ɵɵproperty("ngIf", ctx.isServerSearching && ctx.type === "check");
    } }, dependencies: [i3.NgForOf, i3.NgIf, i4.PoLoadingComponent, i5.CdkListbox, i5.CdkOption, i6.PoItemListComponent, i7.PoSearchListComponent, i3.JsonPipe], encapsulation: 2 });
(function () { (typeof ngDevMode === "undefined" || ngDevMode) && i0.ɵsetClassMetadata(PoListBoxComponent, [{
        type: Component,
        args: [{ selector: 'po-listbox', template: "<div #listbox class=\"po-listbox\" [class.po-listbox-check]=\"type === 'check'\" [attr.data-type]=\"type\" [hidden]=\"visible\">\r\n  <ng-content select=\"[p-popup-header-template]\"></ng-content>\r\n\r\n  <po-search-list\r\n    #searchElement\r\n    *ngIf=\"!hideSearch && type === 'check'\"\r\n    (p-change)=\"callChangeSearch($event)\"\r\n    [p-literals]=\"literalSearch\"\r\n    [p-field-value]=\"fieldValueSearch\"\r\n    [p-placeholder]=\"placeholderSearch\"\r\n  ></po-search-list>\r\n\r\n  <ul\r\n    #listboxItemList\r\n    cdkListbox\r\n    [cdkListboxMultiple]=\"type === 'check'\"\r\n    class=\"po-listbox-list\"\r\n    *ngIf=\"!isServerSearching\"\r\n  >\r\n    <li\r\n      cdkOption=\"selectAll\"\r\n      *ngIf=\"type === 'check' && items.length && !searchElement?.inputValue && !hideSelectAll\"\r\n      [attr.aria-checked]=\"checkboxAllValue === null ? 'mixed' : checkboxAllValue\"\r\n      (click)=\"changeAll.emit()\"\r\n      (keydown)=\"changeAllEmit($event)\"\r\n    >\r\n      <po-item-list\r\n        [p-selected]=\"checkboxAllValue\"\r\n        p-type=\"check\"\r\n        [p-label]=\"literals.selectAll\"\r\n        p-value=\"selectAll\"\r\n        [p-checkbox-value]=\"checkboxAllValue\"\r\n      >\r\n      </po-item-list>\r\n    </li>\r\n    <li\r\n      class=\"po-listbox-item\"\r\n      *ngFor=\"let item of items\"\r\n      [cdkOption]=\"item[fieldValue]\"\r\n      [cdkOptionDisabled]=\"returnBooleanValue(item, 'disabled') || returnBooleanValue(item, 'visible') === false\"\r\n      [attr.aria-selected]=\"isSelectedItem(item) || item.selected\"\r\n      (click)=\"onSelectItem(item)\"\r\n      (keydown)=\"onKeyDown(item, $event)\"\r\n    >\r\n      <po-item-list\r\n        *ngIf=\"returnBooleanValue(item, 'visible') !== false\"\r\n        [p-disabled]=\"returnBooleanValue(item, 'disabled')\"\r\n        [p-visible]=\"returnBooleanValue(item, 'visible')\"\r\n        [p-checkbox-value]=\"isSelectedItem(item)\"\r\n        [attr.data-item-list]=\"item | json\"\r\n        [p-label]=\"item[fieldLabel]\"\r\n        [p-value]=\"item[fieldValue]\"\r\n        [p-selected]=\"isSelectedItem(item) || item.selected\"\r\n        [p-separator]=\"item.separator\"\r\n        [p-danger]=\"item.danger || item.type === 'danger'\"\r\n        [p-icon]=\"item.icon\"\r\n        [p-type]=\"type\"\r\n        [p-item]=\"item\"\r\n        [p-field-value]=\"fieldValue\"\r\n        [p-field-label]=\"fieldLabel\"\r\n        [p-multiselect-template]=\"multiselectTemplate\"\r\n        [p-template-context]=\"item\"\r\n        (p-selectcheckbox-item)=\"checkboxClicked($event)\"\r\n      ></po-item-list>\r\n    </li>\r\n  </ul>\r\n\r\n  <div\r\n    *ngIf=\"!items.length && !isServerSearching && type === 'check'\"\r\n    class=\"po-multiselect-container-no-data po-text-center\"\r\n  >\r\n    <span> {{ literals.noItems }}</span>\r\n  </div>\r\n\r\n  <div *ngIf=\"isServerSearching && type === 'check'\" class=\"po-multiselect-container-loading po-text-center\">\r\n    <po-loading></po-loading>\r\n  </div>\r\n</div>\r\n" }]
    }], function () { return [{ type: i0.Renderer2 }, { type: i1.PoLanguageService }, { type: i2.Router }]; }, { listbox: [{
            type: ViewChild,
            args: ['listbox', { static: true }]
        }], listboxItemList: [{
            type: ViewChild,
            args: ['listboxItemList', { static: false }]
        }], searchElement: [{
            type: ViewChild,
            args: ['searchElement']
        }] }); })();
