// @po-ui/ng-components/esm2020/lib/components/po-listbox/po-listbox.module.mjs
import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { CdkListboxModule } from '@angular/cdk/listbox';
import { PoListBoxComponent } from './po-listbox.component';
import { PoItemListComponent } from './po-item-list/po-item-list.component';
import { PoSearchListComponent } from './po-search-list/po-search-list.component';
import { PoCheckboxModule } from '../po-field/po-checkbox/po-checkbox.module';
import { PoIconModule } from '../po-icon/po-icon.module';
import { PoLoadingModule } from '../po-loading/po-loading.module';
import * as i0 from "@angular/core";
export class PoListBoxModule {
}
PoListBoxModule.ɵfac = function PoListBoxModule_Factory(t) { return new (t || PoListBoxModule)(); };
PoListBoxModule.ɵmod = /*@__PURE__*/ i0.ɵɵdefineNgModule({ type: PoListBoxModule });
PoListBoxModule.ɵinj = /*@__PURE__*/ i0.ɵɵdefineInjector({ imports: [CommonModule, PoCheckboxModule, PoIconModule, PoLoadingModule, CdkListboxModule] });
(function () { (typeof ngDevMode === "undefined" || ngDevMode) && i0.ɵsetClassMetadata(PoListBoxModule, [{
        type: NgModule,
        args: [{
                declarations: [PoListBoxComponent, PoItemListComponent, PoSearchListComponent],
                exports: [PoListBoxComponent],
                imports: [CommonModule, PoCheckboxModule, PoIconModule, PoLoadingModule, CdkListboxModule]
            }]
    }], null, null); })();
(function () { (typeof ngJitMode === "undefined" || ngJitMode) && i0.ɵɵsetNgModuleScope(PoListBoxModule, { declarations: [PoListBoxComponent, PoItemListComponent, PoSearchListComponent], imports: [CommonModule, PoCheckboxModule, PoIconModule, PoLoadingModule, CdkListboxModule], exports: [PoListBoxComponent] }); })();
