// @po-ui/ng-components/esm2020/lib/components/po-listbox/po-listbox-base.component.mjs
import { __decorate } from "tslib";
import { Directive, EventEmitter, Input, Output } from '@angular/core';
import { poLocaleDefault } from '../../services/po-language/po-language.constant';
import { InputBoolean } from '../../decorators';
import { PoItemListType } from './enums/po-item-list-type.enum';
import * as i0 from "@angular/core";
import * as i1 from "../../services/po-language/po-language.service";
export const poListBoxLiteralsDefault = {
    en: {
        noItems: 'No items found'
    },
    es: {
        noItems: 'No se encontraron artículos'
    },
    pt: {
        noItems: 'Nenhum item encontrado'
    },
    ru: {
        noItems: 'ничего не найдено'
    }
};
/**
 * @description
 * O componente `po-listbox` é uma caixa suspensa que aparece sobre a interface após ser acionado por um gatilho visível em tela, como o dropdown. Ele apoia trazendo agrupamentos de opções. O componente listbox é composto pelo componente [**PO Item List**](/documentation/po-item-list).
 */
export class PoListBoxBaseComponent {
    constructor(languageService) {
        this._items = [];
        this.language = poLocaleDefault;
        this.visible = false;
        this.selectItem = new EventEmitter();
        this.closeEvent = new EventEmitter();
        // MULTISELECT PROPERTIES
        //output para evento do checkbox
        this.change = new EventEmitter();
        //output para evento do checkbox de selecionar todos
        this.changeAll = new EventEmitter();
        // Propriedade que recebe a lista de opções selecionadas.
        this.selectedOptions = [];
        this.fieldValue = 'value';
        this.fieldLabel = 'label';
        // Evento disparado a cada tecla digitada na pesquisa.
        this.changeSearch = new EventEmitter();
        // Propriedade que recebe o valor de comparação de pesquisa
        this.fieldValueSearch = 'value';
        // Propriedade que indica se o campo de pesquisa deverá ser escondido.
        this.hideSearch = false;
        // Propriedade que indica se o campo "Selecionar todos" deverá ser escondido.
        this.hideSelectAll = false;
        this.isServerSearching = false;
        this.language = languageService.getShortLanguage();
    }
    set type(value) {
        this._type = PoItemListType[value] ?? 'action';
    }
    get type() {
        return this._type;
    }
    set items(items) {
        this._items = Array.isArray(items) ? items : [];
    }
    get items() {
        return this._items;
    }
    set literals(value) {
        if (value instanceof Object && !(value instanceof Array)) {
            this._literals = {
                ...poListBoxLiteralsDefault[poLocaleDefault],
                ...poListBoxLiteralsDefault[this.language],
                ...value
            };
        }
        else {
            this._literals = poListBoxLiteralsDefault[this.language];
        }
    }
    get literals() {
        return this._literals || poListBoxLiteralsDefault[this.language];
    }
}
PoListBoxBaseComponent.ɵfac = function PoListBoxBaseComponent_Factory(t) { return new (t || PoListBoxBaseComponent)(i0.ɵɵdirectiveInject(i1.PoLanguageService)); };
PoListBoxBaseComponent.ɵdir = /*@__PURE__*/ i0.ɵɵdefineDirective({ type: PoListBoxBaseComponent, inputs: { visible: ["p-visible", "visible"], type: ["p-type", "type"], items: ["p-items", "items"], literals: ["p-literals", "literals"], param: ["p-param", "param"], checkboxAllValue: ["p-checkboxAllValue", "checkboxAllValue"], selectedOptions: ["p-selected-options", "selectedOptions"], fieldValue: ["p-field-value", "fieldValue"], fieldLabel: ["p-field-label", "fieldLabel"], literalSearch: ["p-literal-search", "literalSearch"], fieldValueSearch: ["p-field-value-search", "fieldValueSearch"], hideSearch: ["p-hide-search", "hideSearch"], hideSelectAll: ["p-hide-select-all", "hideSelectAll"], multiselectTemplate: ["p-multiselect-template", "multiselectTemplate"], placeholderSearch: ["p-placeholder-search", "placeholderSearch"], isServerSearching: ["p-is-searching", "isServerSearching"] }, outputs: { selectItem: "p-select-item", closeEvent: "p-close", change: "p-change", changeAll: "p-change-all", changeSearch: "p-change-search" } });
__decorate([
    InputBoolean()
], PoListBoxBaseComponent.prototype, "visible", void 0);
__decorate([
    InputBoolean()
], PoListBoxBaseComponent.prototype, "isServerSearching", void 0);
(function () { (typeof ngDevMode === "undefined" || ngDevMode) && i0.ɵsetClassMetadata(PoListBoxBaseComponent, [{
        type: Directive
    }], function () { return [{ type: i1.PoLanguageService }]; }, { visible: [{
            type: Input,
            args: ['p-visible']
        }], type: [{
            type: Input,
            args: ['p-type']
        }], items: [{
            type: Input,
            args: ['p-items']
        }], literals: [{
            type: Input,
            args: ['p-literals']
        }], param: [{
            type: Input,
            args: ['p-param']
        }], selectItem: [{
            type: Output,
            args: ['p-select-item']
        }], closeEvent: [{
            type: Output,
            args: ['p-close']
        }], change: [{
            type: Output,
            args: ['p-change']
        }], changeAll: [{
            type: Output,
            args: ['p-change-all']
        }], checkboxAllValue: [{
            type: Input,
            args: ['p-checkboxAllValue']
        }], selectedOptions: [{
            type: Input,
            args: ['p-selected-options']
        }], fieldValue: [{
            type: Input,
            args: ['p-field-value']
        }], fieldLabel: [{
            type: Input,
            args: ['p-field-label']
        }], changeSearch: [{
            type: Output,
            args: ['p-change-search']
        }], literalSearch: [{
            type: Input,
            args: ['p-literal-search']
        }], fieldValueSearch: [{
            type: Input,
            args: ['p-field-value-search']
        }], hideSearch: [{
            type: Input,
            args: ['p-hide-search']
        }], hideSelectAll: [{
            type: Input,
            args: ['p-hide-select-all']
        }], multiselectTemplate: [{
            type: Input,
            args: ['p-multiselect-template']
        }], placeholderSearch: [{
            type: Input,
            args: ['p-placeholder-search']
        }], isServerSearching: [{
            type: Input,
            args: ['p-is-searching']
        }] }); })();
