// @po-ui/ng-components/esm2020/lib/components/po-logo/po-logo.component.mjs
import { Component, Input } from '@angular/core';
import { isTypeof } from '../../utils/util';
import * as i0 from "@angular/core";
import * as i1 from "../../services/po-language/po-language.service";
import * as i2 from "@angular/common";
function PoLogoComponent_a_0_Template(rf, ctx) { if (rf & 1) {
    i0.ɵɵelementStart(0, "a", 2);
    i0.ɵɵelement(1, "img", 3);
    i0.ɵɵelementEnd();
} if (rf & 2) {
    const ctx_r0 = i0.ɵɵnextContext();
    i0.ɵɵadvance(1);
    i0.ɵɵclassMap(ctx_r0.className);
    i0.ɵɵproperty("alt", ctx_r0.logoAlt)("src", ctx_r0.logo, i0.ɵɵsanitizeUrl)("title", ctx_r0.logoAlt);
} }
function PoLogoComponent_ng_template_1_Template(rf, ctx) { if (rf & 1) {
    i0.ɵɵelement(0, "img", 3);
} if (rf & 2) {
    const ctx_r2 = i0.ɵɵnextContext();
    i0.ɵɵclassMap(ctx_r2.className);
    i0.ɵɵproperty("alt", ctx_r2.logoAlt)("src", ctx_r2.logo, i0.ɵɵsanitizeUrl)("title", ctx_r2.logoAlt);
} }
const poLogoLiteralsDefault = {
    en: {
        logomarcaHome: 'Home logo'
    },
    es: {
        logomarcaHome: 'Logomarca inicio'
    },
    pt: {
        logomarcaHome: 'Logomarca início'
    },
    ru: {
        logomarcaHome: 'Дом Логомарка'
    }
};
const MAX_LENGHT = 125;
export class PoLogoComponent {
    constructor(poLanguageService) {
        this.poLanguageService = poLanguageService;
        this.literals = {
            ...poLogoLiteralsDefault[this.poLanguageService.getLanguageDefault()],
            ...poLogoLiteralsDefault[this.poLanguageService.getShortLanguage()]
        };
        this._logoAlt = this.literals.logomarcaHome;
        /**
         * Define uma classe para o elemento `img` do componente.
         *
         * > **Importante**
         * > A classe deve ser definida apenas quando houver necessidade de ajustar o componente dentro de outro componente.
         *
         */
        this.className = 'po-logo';
        /**
         * Define se o componente terá o elemento âncora para a página inicial.
         *
         * > Caso seja definido como false, o componente apenas renderizará o elemento `img`.
         * O valor inicial é `true`.
         *
         */
        this.link = true;
    }
    /**
     * Definie o caminho para a imagem, que será exibida como logomarca.
     *
     */
    set logo(value) {
        this._logo = isTypeof(value, 'string') && value.trim() ? value : undefined;
    }
    get logo() {
        return this._logo;
    }
    /**
     * Define o texto alternativo para a logomarca.
     *
     * > **Importante**
     * > Caso esta propriedade não seja definida o texto padrão será "Logomarca início".
     */
    set logoAlt(value) {
        this._logoAlt = isTypeof(value, 'string') && value.trim() ? this.maxLength(value) : this.literals.logomarcaHome;
    }
    get logoAlt() {
        return this._logoAlt;
    }
    maxLength(value) {
        return value.length > MAX_LENGHT ? value.toString().substring(0, MAX_LENGHT) : value;
    }
}
PoLogoComponent.ɵfac = function PoLogoComponent_Factory(t) { return new (t || PoLogoComponent)(i0.ɵɵdirectiveInject(i1.PoLanguageService)); };
PoLogoComponent.ɵcmp = /*@__PURE__*/ i0.ɵɵdefineComponent({ type: PoLogoComponent, selectors: [["po-logo"]], inputs: { className: ["p-class", "className"], link: ["p-link", "link"], logo: ["p-logo", "logo"], logoAlt: ["p-logo-alt", "logoAlt"] }, decls: 3, vars: 2, consts: [["href", "/", 4, "ngIf", "ngIfElse"], ["noLink", ""], ["href", "/"], [3, "alt", "src", "title"]], template: function PoLogoComponent_Template(rf, ctx) { if (rf & 1) {
        i0.ɵɵtemplate(0, PoLogoComponent_a_0_Template, 2, 5, "a", 0);
        i0.ɵɵtemplate(1, PoLogoComponent_ng_template_1_Template, 1, 5, "ng-template", null, 1, i0.ɵɵtemplateRefExtractor);
    } if (rf & 2) {
        const _r1 = i0.ɵɵreference(2);
        i0.ɵɵproperty("ngIf", ctx.link)("ngIfElse", _r1);
    } }, dependencies: [i2.NgIf], encapsulation: 2 });
(function () { (typeof ngDevMode === "undefined" || ngDevMode) && i0.ɵsetClassMetadata(PoLogoComponent, [{
        type: Component,
        args: [{ selector: 'po-logo', template: "<a *ngIf=\"link; else noLink\" href=\"/\">\r\n  <img [alt]=\"logoAlt\" [class]=\"className\" [src]=\"logo\" [title]=\"logoAlt\" />\r\n</a>\r\n\r\n<ng-template #noLink>\r\n  <img [alt]=\"logoAlt\" [class]=\"className\" [src]=\"logo\" [title]=\"logoAlt\" />\r\n</ng-template>\r\n" }]
    }], function () { return [{ type: i1.PoLanguageService }]; }, { className: [{
            type: Input,
            args: ['p-class']
        }], link: [{
            type: Input,
            args: ['p-link']
        }], logo: [{
            type: Input,
            args: ['p-logo']
        }], logoAlt: [{
            type: Input,
            args: ['p-logo-alt']
        }] }); })();
