// @po-ui/ng-components/esm2020/lib/components/po-button-group/po-button-group.module.mjs
import { CommonModule } from '@angular/common';
import { NgModule } from '@angular/core';
import { PoButtonModule } from './../po-button/index';
import { PoTooltipModule } from '../../directives/po-tooltip/index';
import { PoButtonGroupComponent } from './po-button-group.component';
import * as i0 from "@angular/core";
/**
 * @description
 *
 * Módulo do componente po-button-group.
 */
export class PoButtonGroupModule {
}
PoButtonGroupModule.ɵfac = function PoButtonGroupModule_Factory(t) { return new (t || PoButtonGroupModule)(); };
PoButtonGroupModule.ɵmod = /*@__PURE__*/ i0.ɵɵdefineNgModule({ type: PoButtonGroupModule });
PoButtonGroupModule.ɵinj = /*@__PURE__*/ i0.ɵɵdefineInjector({ imports: [CommonModule, PoButtonModule, PoTooltipModule] });
(function () { (typeof ngDevMode === "undefined" || ngDevMode) && i0.ɵsetClassMetadata(PoButtonGroupModule, [{
        type: NgModule,
        args: [{
                imports: [CommonModule, PoButtonModule, PoTooltipModule],
                declarations: [PoButtonGroupComponent],
                exports: [PoButtonGroupComponent]
            }]
    }], null, null); })();
(function () { (typeof ngJitMode === "undefined" || ngJitMode) && i0.ɵɵsetNgModuleScope(PoButtonGroupModule, { declarations: [PoButtonGroupComponent], imports: [CommonModule, PoButtonModule, PoTooltipModule], exports: [PoButtonGroupComponent] }); })();
