// @po-ui/ng-components/esm2020/lib/components/po-table/po-table-column-label/po-table-column-label.component.mjs
import { ChangeDetectionStrategy, Component, Input } from '@angular/core';
import * as i0 from "@angular/core";
import * as i1 from "@angular/common";
import * as i2 from "../../po-tag/po-tag.component";
function PoTableColumnLabelComponent_po_tag_0_Template(rf, ctx) { if (rf & 1) {
    i0.ɵɵelement(0, "po-tag", 1);
} if (rf & 2) {
    const ctx_r0 = i0.ɵɵnextContext();
    i0.ɵɵproperty("p-color", ctx_r0.value == null ? null : ctx_r0.value.color)("p-value", ctx_r0.value == null ? null : ctx_r0.value.label)("p-text-color", ctx_r0.value == null ? null : ctx_r0.value.textColor)("p-icon", ctx_r0.value == null ? null : ctx_r0.value.icon)("p-type", ctx_r0.value == null ? null : ctx_r0.value.type);
} }
/**
 * @docsPrivate
 *
 * @description
 *
 * Componente para a criação da representação da legenda, em formato de texto .
 */
export class PoTableColumnLabelComponent {
    constructor() {
        this.hasLabel = false;
    }
    ngOnInit() {
        this.checkValueHasLabel();
    }
    checkValueHasLabel() {
        this.hasLabel = this.value?.label?.trim() ? true : false;
    }
}
PoTableColumnLabelComponent.ɵfac = function PoTableColumnLabelComponent_Factory(t) { return new (t || PoTableColumnLabelComponent)(); };
PoTableColumnLabelComponent.ɵcmp = /*@__PURE__*/ i0.ɵɵdefineComponent({ type: PoTableColumnLabelComponent, selectors: [["po-table-column-label"]], inputs: { value: ["p-value", "value"] }, decls: 1, vars: 1, consts: [[3, "p-color", "p-value", "p-text-color", "p-icon", "p-type", 4, "ngIf"], [3, "p-color", "p-value", "p-text-color", "p-icon", "p-type"]], template: function PoTableColumnLabelComponent_Template(rf, ctx) { if (rf & 1) {
        i0.ɵɵtemplate(0, PoTableColumnLabelComponent_po_tag_0_Template, 1, 5, "po-tag", 0);
    } if (rf & 2) {
        i0.ɵɵproperty("ngIf", ctx.hasLabel);
    } }, dependencies: [i1.NgIf, i2.PoTagComponent], encapsulation: 2, changeDetection: 0 });
(function () { (typeof ngDevMode === "undefined" || ngDevMode) && i0.ɵsetClassMetadata(PoTableColumnLabelComponent, [{
        type: Component,
        args: [{ selector: 'po-table-column-label', changeDetection: ChangeDetectionStrategy.OnPush, template: "<po-tag\r\n  *ngIf=\"hasLabel\"\r\n  [p-color]=\"value?.color\"\r\n  [p-value]=\"value?.label\"\r\n  [p-text-color]=\"value?.textColor\"\r\n  [p-icon]=\"value?.icon\"\r\n  [p-type]=\"value?.type\"\r\n>\r\n</po-tag>\r\n" }]
    }], null, { value: [{
            type: Input,
            args: ['p-value']
        }] }); })();
