// @po-ui/ng-components/esm2020/lib/components/po-navbar/po-navbar-actions/po-navbar-action-popup/po-navbar-action-popup.component.mjs
import { Component, Input } from '@angular/core';
import * as i0 from "@angular/core";
import * as i1 from "../../../po-popup/po-popup.component";
export class PoNavbarActionPopupComponent {
    getLastIconAction() {
        if (this.iconActions && this.iconActions.length) {
            return this.iconActions[this.iconActions.length - 1].icon;
        }
    }
}
PoNavbarActionPopupComponent.ɵfac = function PoNavbarActionPopupComponent_Factory(t) { return new (t || PoNavbarActionPopupComponent)(); };
PoNavbarActionPopupComponent.ɵcmp = /*@__PURE__*/ i0.ɵɵdefineComponent({ type: PoNavbarActionPopupComponent, selectors: [["po-navbar-action-popup"]], inputs: { iconActions: ["p-icon-actions", "iconActions"] }, decls: 5, vars: 5, consts: [["tabindex", "0", 1, "po-navbar-action-popup-content", "po-clickable", 3, "click"], ["target", ""], [3, "p-actions", "p-target"], ["popup", ""]], template: function PoNavbarActionPopupComponent_Template(rf, ctx) { if (rf & 1) {
        const _r2 = i0.ɵɵgetCurrentView();
        i0.ɵɵelementStart(0, "div", 0, 1);
        i0.ɵɵlistener("click", function PoNavbarActionPopupComponent_Template_div_click_0_listener() { i0.ɵɵrestoreView(_r2); const _r1 = i0.ɵɵreference(4); return i0.ɵɵresetView(_r1.toggle()); });
        i0.ɵɵelement(2, "span");
        i0.ɵɵelementEnd();
        i0.ɵɵelement(3, "po-popup", 2, 3);
    } if (rf & 2) {
        const _r0 = i0.ɵɵreference(1);
        i0.ɵɵadvance(2);
        i0.ɵɵclassMapInterpolate1("po-icon ", ctx.getLastIconAction(), "");
        i0.ɵɵadvance(1);
        i0.ɵɵproperty("p-actions", ctx.iconActions)("p-target", _r0);
    } }, dependencies: [i1.PoPopupComponent], encapsulation: 2 });
(function () { (typeof ngDevMode === "undefined" || ngDevMode) && i0.ɵsetClassMetadata(PoNavbarActionPopupComponent, [{
        type: Component,
        args: [{ selector: 'po-navbar-action-popup', template: "<div tabindex=\"0\" #target class=\"po-navbar-action-popup-content po-clickable\" (click)=\"popup.toggle()\">\r\n  <span class=\"po-icon {{ getLastIconAction() }}\"></span>\r\n</div>\r\n\r\n<po-popup #popup [p-actions]=\"iconActions\" [p-target]=\"target\"> </po-popup>\r\n" }]
    }], null, { iconActions: [{
            type: Input,
            args: ['p-icon-actions']
        }] }); })();
