// @po-ui/ng-components/esm2020/lib/components/po-accordion/po-accordion.component.mjs
import { Component, ContentChildren } from '@angular/core';
import { PoAccordionBaseComponent } from './po-accordion-base.component';
import { PoAccordionItemComponent } from './po-accordion-item/po-accordion-item.component';
import { PoAccordionService } from './services/po-accordion.service';
import * as i0 from "@angular/core";
import * as i1 from "./services/po-accordion.service";
import * as i2 from "@angular/common";
import * as i3 from "./po-accordion-item-body/po-accordion-item-body.component";
import * as i4 from "./po-accordion-item-header/po-accordion-item-header.component";
function PoAccordionComponent_section_1_ng_container_3_Template(rf, ctx) { if (rf & 1) {
    i0.ɵɵelementContainer(0);
} }
function PoAccordionComponent_section_1_Template(rf, ctx) { if (rf & 1) {
    const _r4 = i0.ɵɵgetCurrentView();
    i0.ɵɵelementStart(0, "section", 2)(1, "po-accordion-item-header", 3);
    i0.ɵɵlistener("p-toggle", function PoAccordionComponent_section_1_Template_po_accordion_item_header_p_toggle_1_listener($event) { const restoredCtx = i0.ɵɵrestoreView(_r4); const poAccordionItem_r1 = restoredCtx.$implicit; const ctx_r3 = i0.ɵɵnextContext(); return i0.ɵɵresetView(ctx_r3.headerToggle($event, poAccordionItem_r1)); });
    i0.ɵɵelementEnd();
    i0.ɵɵelementStart(2, "po-accordion-item-body", 4);
    i0.ɵɵtemplate(3, PoAccordionComponent_section_1_ng_container_3_Template, 1, 0, "ng-container", 5);
    i0.ɵɵelementEnd()();
} if (rf & 2) {
    const poAccordionItem_r1 = ctx.$implicit;
    i0.ɵɵclassProp("po-accordion-item-active", poAccordionItem_r1.expanded);
    i0.ɵɵadvance(1);
    i0.ɵɵproperty("p-expanded", poAccordionItem_r1.expanded)("p-label", poAccordionItem_r1.label);
    i0.ɵɵadvance(1);
    i0.ɵɵproperty("p-expanded", poAccordionItem_r1.expanded);
    i0.ɵɵadvance(1);
    i0.ɵɵproperty("ngTemplateOutlet", poAccordionItem_r1.templateRef);
} }
/**
 * @docsExtends PoAccordionBaseComponent
 *
 * @example
 *
 * <example name="po-accordion-basic" title="PO Accordion Basic" >
 *  <file name="sample-po-accordion-basic/sample-po-accordion-basic.component.html"> </file>
 *  <file name="sample-po-accordion-basic/sample-po-accordion-basic.component.ts"> </file>
 *  <file name="sample-po-accordion-basic/sample-po-accordion-basic.component.e2e-spec.ts"> </file>
 *  <file name="sample-po-accordion-basic/sample-po-accordion-basic.component.po.ts"> </file>
 * </example>
 *
 * <example name="po-accordion-labs" title="PO Accordion Labs" >
 *  <file name="sample-po-accordion-labs/sample-po-accordion-labs.component.html"> </file>
 *  <file name="sample-po-accordion-labs/sample-po-accordion-labs.component.ts"> </file>
 * </example>
 *
 * <example name="po-accordion-faq" title="PO Accordion - FAQs" >
 *  <file name="sample-po-accordion-faq/sample-po-accordion-faq.component.html"> </file>
 *  <file name="sample-po-accordion-faq/sample-po-accordion-faq.component.ts"> </file>
 * </example>
 */
export class PoAccordionComponent extends PoAccordionBaseComponent {
    constructor(accordionService) {
        super();
        this.accordionService = accordionService;
        this.receiveFromChildAccordionSubscription();
    }
    ngOnDestroy() {
        this.accordionServiceSubscription.unsubscribe();
    }
    headerToggle(event, poAccordionItem) {
        poAccordionItem.expanded = event;
        this.accordionService.sendToParentAccordionItemClicked(poAccordionItem);
    }
    receiveFromChildAccordionSubscription() {
        this.accordionServiceSubscription = this.accordionService
            .receiveFromChildAccordionClicked()
            .subscribe(poAccordionItem => this.toggle(poAccordionItem));
    }
    toggle(poAccordionItem) {
        const isCurrentAccordionCollapsed = !poAccordionItem.expanded;
        if (isCurrentAccordionCollapsed) {
            this.expandedActiveAccordionItem = null;
            return;
        }
        if (this.expandedActiveAccordionItem) {
            this.expandedActiveAccordionItem.collapse();
        }
        this.expandedActiveAccordionItem = poAccordionItem;
    }
}
PoAccordionComponent.ɵfac = function PoAccordionComponent_Factory(t) { return new (t || PoAccordionComponent)(i0.ɵɵdirectiveInject(i1.PoAccordionService)); };
PoAccordionComponent.ɵcmp = /*@__PURE__*/ i0.ɵɵdefineComponent({ type: PoAccordionComponent, selectors: [["po-accordion"]], contentQueries: function PoAccordionComponent_ContentQueries(rf, ctx, dirIndex) { if (rf & 1) {
        i0.ɵɵcontentQuery(dirIndex, PoAccordionItemComponent, 4);
    } if (rf & 2) {
        let _t;
        i0.ɵɵqueryRefresh(_t = i0.ɵɵloadQuery()) && (ctx.poAccordionItems = _t);
    } }, features: [i0.ɵɵProvidersFeature([PoAccordionService]), i0.ɵɵInheritDefinitionFeature], decls: 2, vars: 1, consts: [[1, "po-accordion", "po-container", "po-container-no-padding"], ["class", "po-accordion-item", 3, "po-accordion-item-active", 4, "ngFor", "ngForOf"], [1, "po-accordion-item"], [3, "p-expanded", "p-label", "p-toggle"], [3, "p-expanded"], [4, "ngTemplateOutlet"]], template: function PoAccordionComponent_Template(rf, ctx) { if (rf & 1) {
        i0.ɵɵelementStart(0, "div", 0);
        i0.ɵɵtemplate(1, PoAccordionComponent_section_1_Template, 4, 6, "section", 1);
        i0.ɵɵelementEnd();
    } if (rf & 2) {
        i0.ɵɵadvance(1);
        i0.ɵɵproperty("ngForOf", ctx.poAccordionItems);
    } }, dependencies: [i2.NgForOf, i2.NgTemplateOutlet, i3.PoAccordionItemBodyComponent, i4.PoAccordionItemHeaderComponent], encapsulation: 2 });
(function () { (typeof ngDevMode === "undefined" || ngDevMode) && i0.ɵsetClassMetadata(PoAccordionComponent, [{
        type: Component,
        args: [{ selector: 'po-accordion', providers: [PoAccordionService], template: "<div class=\"po-accordion po-container po-container-no-padding\">\r\n  <section\r\n    *ngFor=\"let poAccordionItem of poAccordionItems\"\r\n    class=\"po-accordion-item\"\r\n    [class.po-accordion-item-active]=\"poAccordionItem.expanded\"\r\n  >\r\n    <po-accordion-item-header\r\n      [p-expanded]=\"poAccordionItem.expanded\"\r\n      [p-label]=\"poAccordionItem.label\"\r\n      (p-toggle)=\"headerToggle($event, poAccordionItem)\"\r\n    >\r\n    </po-accordion-item-header>\r\n\r\n    <po-accordion-item-body [p-expanded]=\"poAccordionItem.expanded\">\r\n      <ng-container *ngTemplateOutlet=\"poAccordionItem.templateRef\"></ng-container>\r\n    </po-accordion-item-body>\r\n  </section>\r\n</div>\r\n" }]
    }], function () { return [{ type: i1.PoAccordionService }]; }, { poAccordionItems: [{
            type: ContentChildren,
            args: [PoAccordionItemComponent]
        }] }); })();
