// @po-ui/ng-components/esm2020/lib/services/po-dialog/po-dialog.module.mjs
import { NgModule } from '@angular/core';
import { PoModalModule } from '../../components/po-modal/po-modal.module';
import { PoDialogComponent } from './po-dialog.component';
import { PoDialogService } from './po-dialog.service';
import * as i0 from "@angular/core";
export class PoDialogModule {
}
PoDialogModule.ɵfac = function PoDialogModule_Factory(t) { return new (t || PoDialogModule)(); };
PoDialogModule.ɵmod = /*@__PURE__*/ i0.ɵɵdefineNgModule({ type: PoDialogModule });
PoDialogModule.ɵinj = /*@__PURE__*/ i0.ɵɵdefineInjector({ providers: [PoDialogService], imports: [PoModalModule] });
(function () { (typeof ngDevMode === "undefined" || ngDevMode) && i0.ɵsetClassMetadata(PoDialogModule, [{
        type: NgModule,
        args: [{
                declarations: [PoDialogComponent],
                imports: [PoModalModule],
                exports: [PoDialogComponent],
                providers: [PoDialogService]
            }]
    }], null, null); })();
(function () { (typeof ngJitMode === "undefined" || ngJitMode) && i0.ɵɵsetNgModuleScope(PoDialogModule, { declarations: [PoDialogComponent], imports: [PoModalModule], exports: [PoDialogComponent] }); })();
