// @po-ui/ng-components/esm2020/lib/components/po-disclaimer-group/po-disclaimer-remove/po-disclaimer-remove.component.mjs
import { Component, EventEmitter, Input, Output } from '@angular/core';
import * as i0 from "@angular/core";
import * as i1 from "../../po-tag/po-tag.component";
/**
 * @docsPrivate
 *
 * @description
 *
 * Este é um componente interno utilizado pelo po-disclaimer-group, se comporta como um botão e recebe uma ação para remover todos os
 * po-disclaimers do grupo.
 */
export class PoDisclaimerRemoveComponent {
    constructor() {
        /** Ação para remover todos.  */
        this.removeAllAction = new EventEmitter();
    }
    // Emite a ação de remover todos.
    removeAction() {
        this.removeAllAction.emit();
    }
}
PoDisclaimerRemoveComponent.ɵfac = function PoDisclaimerRemoveComponent_Factory(t) { return new (t || PoDisclaimerRemoveComponent)(); };
PoDisclaimerRemoveComponent.ɵcmp = /*@__PURE__*/ i0.ɵɵdefineComponent({ type: PoDisclaimerRemoveComponent, selectors: [["po-disclaimer-remove"]], inputs: { label: ["p-label", "label"] }, outputs: { removeAllAction: "p-remove-all-action" }, decls: 1, vars: 2, consts: [["p-type", "danger", 1, "po-clickable", 3, "p-inverse", "p-value", "p-click"]], template: function PoDisclaimerRemoveComponent_Template(rf, ctx) { if (rf & 1) {
        i0.ɵɵelementStart(0, "po-tag", 0);
        i0.ɵɵlistener("p-click", function PoDisclaimerRemoveComponent_Template_po_tag_p_click_0_listener() { return ctx.removeAction(); });
        i0.ɵɵelementEnd();
    } if (rf & 2) {
        i0.ɵɵproperty("p-inverse", true)("p-value", ctx.label);
    } }, dependencies: [i1.PoTagComponent], encapsulation: 2 });
(function () { (typeof ngDevMode === "undefined" || ngDevMode) && i0.ɵsetClassMetadata(PoDisclaimerRemoveComponent, [{
        type: Component,
        args: [{ selector: 'po-disclaimer-remove', template: "<po-tag class=\"po-clickable\" [p-inverse]=\"true\" p-type=\"danger\" [p-value]=\"label\" (p-click)=\"removeAction()\"> </po-tag>\r\n" }]
    }], null, { label: [{
            type: Input,
            args: ['p-label']
        }], removeAllAction: [{
            type: Output,
            args: ['p-remove-all-action']
        }] }); })();
