// @po-ui/ng-components/esm2020/lib/components/po-grid/po-grid.module.mjs
import { CommonModule } from '@angular/common';
import { FormsModule } from '@angular/forms';
import { NgModule } from '@angular/core';
import { PoFieldModule } from './../../components/po-field/po-field.module';
import { PoGridCellActionComponent } from './po-grid-cell-action/po-grid-cell-action.component';
import { PoGridCellComponent } from './po-grid-cell/po-grid-cell.component';
import { PoGridComponent } from './po-grid.component';
import { PoGridHeadComponent } from './po-grid-head/po-grid-head.component';
import * as i0 from "@angular/core";
/**
 * @description
 *
 * Módulo do componente po-grid.
 */
export class PoGridModule {
}
PoGridModule.ɵfac = function PoGridModule_Factory(t) { return new (t || PoGridModule)(); };
PoGridModule.ɵmod = /*@__PURE__*/ i0.ɵɵdefineNgModule({ type: PoGridModule });
PoGridModule.ɵinj = /*@__PURE__*/ i0.ɵɵdefineInjector({ imports: [CommonModule, FormsModule, PoFieldModule] });
(function () { (typeof ngDevMode === "undefined" || ngDevMode) && i0.ɵsetClassMetadata(PoGridModule, [{
        type: NgModule,
        args: [{
                imports: [CommonModule, FormsModule, PoFieldModule],
                declarations: [PoGridCellActionComponent, PoGridCellComponent, PoGridComponent, PoGridHeadComponent],
                exports: [PoGridComponent]
            }]
    }], null, null); })();
(function () { (typeof ngJitMode === "undefined" || ngJitMode) && i0.ɵɵsetNgModuleScope(PoGridModule, { declarations: [PoGridCellActionComponent, PoGridCellComponent, PoGridComponent, PoGridHeadComponent], imports: [CommonModule, FormsModule, PoFieldModule], exports: [PoGridComponent] }); })();
